{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"mutation.ipynb","version":"0.3.2","provenance":[],"collapsed_sections":[]},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"metadata":{"id":"biQBxvOvaf3w","colab_type":"text"},"cell_type":"markdown","source":["# **Connecting with the Kaggel API to import the data direclty to IPython Runtime.**"]},{"metadata":{"id":"oE63PFTVWPus","colab_type":"code","outputId":"4992f258-f377-476d-d330-73af50ffcc89","executionInfo":{"status":"ok","timestamp":1550638371492,"user_tz":-330,"elapsed":5240,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":208}},"cell_type":"code","source":["!pip install kaggle"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Requirement already satisfied: kaggle in /usr/local/lib/python3.6/dist-packages (1.5.2)\n","Requirement already satisfied: urllib3<1.23.0,>=1.15 in /usr/local/lib/python3.6/dist-packages (from kaggle) (1.22)\n","Requirement already satisfied: six>=1.10 in /usr/local/lib/python3.6/dist-packages (from kaggle) (1.11.0)\n","Requirement already satisfied: certifi in /usr/local/lib/python3.6/dist-packages (from kaggle) (2018.11.29)\n","Requirement already satisfied: python-dateutil in /usr/local/lib/python3.6/dist-packages (from kaggle) (2.5.3)\n","Requirement already satisfied: requests in /usr/local/lib/python3.6/dist-packages (from kaggle) (2.18.4)\n","Requirement already satisfied: tqdm in /usr/local/lib/python3.6/dist-packages (from kaggle) (4.28.1)\n","Requirement already satisfied: python-slugify in /usr/local/lib/python3.6/dist-packages (from kaggle) (2.0.1)\n","Requirement already satisfied: idna<2.7,>=2.5 in /usr/local/lib/python3.6/dist-packages (from requests->kaggle) (2.6)\n","Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/local/lib/python3.6/dist-packages (from requests->kaggle) (3.0.4)\n","Requirement already satisfied: Unidecode>=0.04.16 in /usr/local/lib/python3.6/dist-packages (from python-slugify->kaggle) (1.0.23)\n"],"name":"stdout"}]},{"metadata":{"id":"D5vCdTbkWz9X","colab_type":"code","outputId":"b2a93576-ab91-42de-94f7-a2ebfb0875af","executionInfo":{"status":"ok","timestamp":1552483188285,"user_tz":-330,"elapsed":39555,"user":{"displayName":"Purushottam Gaudel","photoUrl":"https://lh5.googleusercontent.com/-upQdFAUEDjc/AAAAAAAAAAI/AAAAAAAAARU/l_uTJkB81qQ/s64/photo.jpg","userId":"04810327086193882478"}},"colab":{"resources":{"http://localhost:8080/nbextensions/google.colab/files.js":{"data":"Ly8gQ29weXJpZ2h0IDIwMTcgR29vZ2xlIExMQwovLwovLyBMaWNlbnNlZCB1bmRlciB0aGUgQXBhY2hlIExpY2Vuc2UsIFZlcnNpb24gMi4wICh0aGUgIkxpY2Vuc2UiKTsKLy8geW91IG1heSBub3QgdXNlIHRoaXMgZmlsZSBleGNlcHQgaW4gY29tcGxpYW5jZSB3aXRoIHRoZSBMaWNlbnNlLgovLyBZb3UgbWF5IG9idGFpbiBhIGNvcHkgb2YgdGhlIExpY2Vuc2UgYXQKLy8KLy8gICAgICBodHRwOi8vd3d3LmFwYWNoZS5vcmcvbGljZW5zZXMvTElDRU5TRS0yLjAKLy8KLy8gVW5sZXNzIHJlcXVpcmVkIGJ5IGFwcGxpY2FibGUgbGF3IG9yIGFncmVlZCB0byBpbiB3cml0aW5nLCBzb2Z0d2FyZQovLyBkaXN0cmlidXRlZCB1bmRlciB0aGUgTGljZW5zZSBpcyBkaXN0cmlidXRlZCBvbiBhbiAiQVMgSVMiIEJBU0lTLAovLyBXSVRIT1VUIFdBUlJBTlRJRVMgT1IgQ09ORElUSU9OUyBPRiBBTlkgS0lORCwgZWl0aGVyIGV4cHJlc3Mgb3IgaW1wbGllZC4KLy8gU2VlIHRoZSBMaWNlbnNlIGZvciB0aGUgc3BlY2lmaWMgbGFuZ3VhZ2UgZ292ZXJuaW5nIHBlcm1pc3Npb25zIGFuZAovLyBsaW1pdGF0aW9ucyB1bmRlciB0aGUgTGljZW5zZS4KCi8qKgogKiBAZmlsZW92ZXJ2aWV3IEhlbHBlcnMgZm9yIGdvb2dsZS5jb2xhYiBQeXRob24gbW9kdWxlLgogKi8KKGZ1bmN0aW9uKHNjb3BlKSB7CmZ1bmN0aW9uIHNwYW4odGV4dCwgc3R5bGVBdHRyaWJ1dGVzID0ge30pIHsKICBjb25zdCBlbGVtZW50ID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnc3BhbicpOwogIGVsZW1lbnQudGV4dENvbnRlbnQgPSB0ZXh0OwogIGZvciAoY29uc3Qga2V5IG9mIE9iamVjdC5rZXlzKHN0eWxlQXR0cmlidXRlcykpIHsKICAgIGVsZW1lbnQuc3R5bGVba2V5XSA9IHN0eWxlQXR0cmlidXRlc1trZXldOwogIH0KICByZXR1cm4gZWxlbWVudDsKfQoKLy8gTWF4IG51bWJlciBvZiBieXRlcyB3aGljaCB3aWxsIGJlIHVwbG9hZGVkIGF0IGEgdGltZS4KY29uc3QgTUFYX1BBWUxPQURfU0laRSA9IDEwMCAqIDEwMjQ7Ci8vIE1heCBhbW91bnQgb2YgdGltZSB0byBibG9jayB3YWl0aW5nIGZvciB0aGUgdXNlci4KY29uc3QgRklMRV9DSEFOR0VfVElNRU9VVF9NUyA9IDMwICogMTAwMDsKCmZ1bmN0aW9uIF91cGxvYWRGaWxlcyhpbnB1dElkLCBvdXRwdXRJZCkgewogIGNvbnN0IHN0ZXBzID0gdXBsb2FkRmlsZXNTdGVwKGlucHV0SWQsIG91dHB1dElkKTsKICBjb25zdCBvdXRwdXRFbGVtZW50ID0gZG9jdW1lbnQuZ2V0RWxlbWVudEJ5SWQob3V0cHV0SWQpOwogIC8vIENhY2hlIHN0ZXBzIG9uIHRoZSBvdXRwdXRFbGVtZW50IHRvIG1ha2UgaXQgYXZhaWxhYmxlIGZvciB0aGUgbmV4dCBjYWxsCiAgLy8gdG8gdXBsb2FkRmlsZXNDb250aW51ZSBmcm9tIFB5dGhvbi4KICBvdXRwdXRFbGVtZW50LnN0ZXBzID0gc3RlcHM7CgogIHJldHVybiBfdXBsb2FkRmlsZXNDb250aW51ZShvdXRwdXRJZCk7Cn0KCi8vIFRoaXMgaXMgcm91Z2hseSBhbiBhc3luYyBnZW5lcmF0b3IgKG5vdCBzdXBwb3J0ZWQgaW4gdGhlIGJyb3dzZXIgeWV0KSwKLy8gd2hlcmUgdGhlcmUgYXJlIG11bHRpcGxlIGFzeW5jaHJvbm91cyBzdGVwcyBhbmQgdGhlIFB5dGhvbiBzaWRlIGlzIGdvaW5nCi8vIHRvIHBvbGwgZm9yIGNvbXBsZXRpb24gb2YgZWFjaCBzdGVwLgovLyBUaGlzIHVzZXMgYSBQcm9taXNlIHRvIGJsb2NrIHRoZSBweXRob24gc2lkZSBvbiBjb21wbGV0aW9uIG9mIGVhY2ggc3RlcCwKLy8gdGhlbiBwYXNzZXMgdGhlIHJlc3VsdCBvZiB0aGUgcHJldmlvdXMgc3RlcCBhcyB0aGUgaW5wdXQgdG8gdGhlIG5leHQgc3RlcC4KZnVuY3Rpb24gX3VwbG9hZEZpbGVzQ29udGludWUob3V0cHV0SWQpIHsKICBjb25zdCBvdXRwdXRFbGVtZW50ID0gZG9jdW1lbnQuZ2V0RWxlbWVudEJ5SWQob3V0cHV0SWQpOwogIGNvbnN0IHN0ZXBzID0gb3V0cHV0RWxlbWVudC5zdGVwczsKCiAgY29uc3QgbmV4dCA9IHN0ZXBzLm5leHQob3V0cHV0RWxlbWVudC5sYXN0UHJvbWlzZVZhbHVlKTsKICByZXR1cm4gUHJvbWlzZS5yZXNvbHZlKG5leHQudmFsdWUucHJvbWlzZSkudGhlbigodmFsdWUpID0+IHsKICAgIC8vIENhY2hlIHRoZSBsYXN0IHByb21pc2UgdmFsdWUgdG8gbWFrZSBpdCBhdmFpbGFibGUgdG8gdGhlIG5leHQKICAgIC8vIHN0ZXAgb2YgdGhlIGdlbmVyYXRvci4KICAgIG91dHB1dEVsZW1lbnQubGFzdFByb21pc2VWYWx1ZSA9IHZhbHVlOwogICAgcmV0dXJuIG5leHQudmFsdWUucmVzcG9uc2U7CiAgfSk7Cn0KCi8qKgogKiBHZW5lcmF0b3IgZnVuY3Rpb24gd2hpY2ggaXMgY2FsbGVkIGJldHdlZW4gZWFjaCBhc3luYyBzdGVwIG9mIHRoZSB1cGxvYWQKICogcHJvY2Vzcy4KICogQHBhcmFtIHtzdHJpbmd9IGlucHV0SWQgRWxlbWVudCBJRCBvZiB0aGUgaW5wdXQgZmlsZSBwaWNrZXIgZWxlbWVudC4KICogQHBhcmFtIHtzdHJpbmd9IG91dHB1dElkIEVsZW1lbnQgSUQgb2YgdGhlIG91dHB1dCBkaXNwbGF5LgogKiBAcmV0dXJuIHshSXRlcmFibGU8IU9iamVjdD59IEl0ZXJhYmxlIG9mIG5leHQgc3RlcHMuCiAqLwpmdW5jdGlvbiogdXBsb2FkRmlsZXNTdGVwKGlucHV0SWQsIG91dHB1dElkKSB7CiAgY29uc3QgaW5wdXRFbGVtZW50ID0gZG9jdW1lbnQuZ2V0RWxlbWVudEJ5SWQoaW5wdXRJZCk7CiAgaW5wdXRFbGVtZW50LmRpc2FibGVkID0gZmFsc2U7CgogIGNvbnN0IG91dHB1dEVsZW1lbnQgPSBkb2N1bWVudC5nZXRFbGVtZW50QnlJZChvdXRwdXRJZCk7CiAgb3V0cHV0RWxlbWVudC5pbm5lckhUTUwgPSAnJzsKCiAgY29uc3QgcGlja2VkUHJvbWlzZSA9IG5ldyBQcm9taXNlKChyZXNvbHZlKSA9PiB7CiAgICBpbnB1dEVsZW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignY2hhbmdlJywgKGUpID0+IHsKICAgICAgcmVzb2x2ZShlLnRhcmdldC5maWxlcyk7CiAgICB9KTsKICB9KTsKCiAgY29uc3QgY2FuY2VsID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnYnV0dG9uJyk7CiAgaW5wdXRFbGVtZW50LnBhcmVudEVsZW1lbnQuYXBwZW5kQ2hpbGQoY2FuY2VsKTsKICBjYW5jZWwudGV4dENvbnRlbnQgPSAnQ2FuY2VsIHVwbG9hZCc7CiAgY29uc3QgY2FuY2VsUHJvbWlzZSA9IG5ldyBQcm9taXNlKChyZXNvbHZlKSA9PiB7CiAgICBjYW5jZWwub25jbGljayA9ICgpID0+IHsKICAgICAgcmVzb2x2ZShudWxsKTsKICAgIH07CiAgfSk7CgogIC8vIENhbmNlbCB1cGxvYWQgaWYgdXNlciBoYXNuJ3QgcGlja2VkIGFueXRoaW5nIGluIHRpbWVvdXQuCiAgY29uc3QgdGltZW91dFByb21pc2UgPSBuZXcgUHJvbWlzZSgocmVzb2x2ZSkgPT4gewogICAgc2V0VGltZW91dCgoKSA9PiB7CiAgICAgIHJlc29sdmUobnVsbCk7CiAgICB9LCBGSUxFX0NIQU5HRV9USU1FT1VUX01TKTsKICB9KTsKCiAgLy8gV2FpdCBmb3IgdGhlIHVzZXIgdG8gcGljayB0aGUgZmlsZXMuCiAgY29uc3QgZmlsZXMgPSB5aWVsZCB7CiAgICBwcm9taXNlOiBQcm9taXNlLnJhY2UoW3BpY2tlZFByb21pc2UsIHRpbWVvdXRQcm9taXNlLCBjYW5jZWxQcm9taXNlXSksCiAgICByZXNwb25zZTogewogICAgICBhY3Rpb246ICdzdGFydGluZycsCiAgICB9CiAgfTsKCiAgaWYgKCFmaWxlcykgewogICAgcmV0dXJuIHsKICAgICAgcmVzcG9uc2U6IHsKICAgICAgICBhY3Rpb246ICdjb21wbGV0ZScsCiAgICAgIH0KICAgIH07CiAgfQoKICBjYW5jZWwucmVtb3ZlKCk7CgogIC8vIERpc2FibGUgdGhlIGlucHV0IGVsZW1lbnQgc2luY2UgZnVydGhlciBwaWNrcyBhcmUgbm90IGFsbG93ZWQuCiAgaW5wdXRFbGVtZW50LmRpc2FibGVkID0gdHJ1ZTsKCiAgZm9yIChjb25zdCBmaWxlIG9mIGZpbGVzKSB7CiAgICBjb25zdCBsaSA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpJyk7CiAgICBsaS5hcHBlbmQoc3BhbihmaWxlLm5hbWUsIHtmb250V2VpZ2h0OiAnYm9sZCd9KSk7CiAgICBsaS5hcHBlbmQoc3BhbigKICAgICAgICBgKCR7ZmlsZS50eXBlIHx8ICduL2EnfSkgLSAke2ZpbGUuc2l6ZX0gYnl0ZXMsIGAgKwogICAgICAgIGBsYXN0IG1vZGlmaWVkOiAkewogICAgICAgICAgICBmaWxlLmxhc3RNb2RpZmllZERhdGUgPyBmaWxlLmxhc3RNb2RpZmllZERhdGUudG9Mb2NhbGVEYXRlU3RyaW5nKCkgOgogICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAnbi9hJ30gLSBgKSk7CiAgICBjb25zdCBwZXJjZW50ID0gc3BhbignMCUgZG9uZScpOwogICAgbGkuYXBwZW5kQ2hpbGQocGVyY2VudCk7CgogICAgb3V0cHV0RWxlbWVudC5hcHBlbmRDaGlsZChsaSk7CgogICAgY29uc3QgZmlsZURhdGFQcm9taXNlID0gbmV3IFByb21pc2UoKHJlc29sdmUpID0+IHsKICAgICAgY29uc3QgcmVhZGVyID0gbmV3IEZpbGVSZWFkZXIoKTsKICAgICAgcmVhZGVyLm9ubG9hZCA9IChlKSA9PiB7CiAgICAgICAgcmVzb2x2ZShlLnRhcmdldC5yZXN1bHQpOwogICAgICB9OwogICAgICByZWFkZXIucmVhZEFzQXJyYXlCdWZmZXIoZmlsZSk7CiAgICB9KTsKICAgIC8vIFdhaXQgZm9yIHRoZSBkYXRhIHRvIGJlIHJlYWR5LgogICAgbGV0IGZpbGVEYXRhID0geWllbGQgewogICAgICBwcm9taXNlOiBmaWxlRGF0YVByb21pc2UsCiAgICAgIHJlc3BvbnNlOiB7CiAgICAgICAgYWN0aW9uOiAnY29udGludWUnLAogICAgICB9CiAgICB9OwoKICAgIC8vIFVzZSBhIGNodW5rZWQgc2VuZGluZyB0byBhdm9pZCBtZXNzYWdlIHNpemUgbGltaXRzLiBTZWUgYi82MjExNTY2MC4KICAgIGxldCBwb3NpdGlvbiA9IDA7CiAgICB3aGlsZSAocG9zaXRpb24gPCBmaWxlRGF0YS5ieXRlTGVuZ3RoKSB7CiAgICAgIGNvbnN0IGxlbmd0aCA9IE1hdGgubWluKGZpbGVEYXRhLmJ5dGVMZW5ndGggLSBwb3NpdGlvbiwgTUFYX1BBWUxPQURfU0laRSk7CiAgICAgIGNvbnN0IGNodW5rID0gbmV3IFVpbnQ4QXJyYXkoZmlsZURhdGEsIHBvc2l0aW9uLCBsZW5ndGgpOwogICAgICBwb3NpdGlvbiArPSBsZW5ndGg7CgogICAgICBjb25zdCBiYXNlNjQgPSBidG9hKFN0cmluZy5mcm9tQ2hhckNvZGUuYXBwbHkobnVsbCwgY2h1bmspKTsKICAgICAgeWllbGQgewogICAgICAgIHJlc3BvbnNlOiB7CiAgICAgICAgICBhY3Rpb246ICdhcHBlbmQnLAogICAgICAgICAgZmlsZTogZmlsZS5uYW1lLAogICAgICAgICAgZGF0YTogYmFzZTY0LAogICAgICAgIH0sCiAgICAgIH07CiAgICAgIHBlcmNlbnQudGV4dENvbnRlbnQgPQogICAgICAgICAgYCR7TWF0aC5yb3VuZCgocG9zaXRpb24gLyBmaWxlRGF0YS5ieXRlTGVuZ3RoKSAqIDEwMCl9JSBkb25lYDsKICAgIH0KICB9CgogIC8vIEFsbCBkb25lLgogIHlpZWxkIHsKICAgIHJlc3BvbnNlOiB7CiAgICAgIGFjdGlvbjogJ2NvbXBsZXRlJywKICAgIH0KICB9Owp9CgpzY29wZS5nb29nbGUgPSBzY29wZS5nb29nbGUgfHwge307CnNjb3BlLmdvb2dsZS5jb2xhYiA9IHNjb3BlLmdvb2dsZS5jb2xhYiB8fCB7fTsKc2NvcGUuZ29vZ2xlLmNvbGFiLl9maWxlcyA9IHsKICBfdXBsb2FkRmlsZXMsCiAgX3VwbG9hZEZpbGVzQ29udGludWUsCn07Cn0pKHNlbGYpOwo=","ok":true,"headers":[["content-type","application/javascript"]],"status":200,"status_text":""}},"base_uri":"https://localhost:8080/","height":57}},"cell_type":"code","source":["from google.colab import files\n","files.upload()"],"execution_count":0,"outputs":[{"output_type":"display_data","data":{"text/html":["\n","     <input type=\"file\" id=\"files-2bc54d28-2d54-4129-a2c8-4321d0190cf4\" name=\"files[]\" multiple disabled />\n","     <output id=\"result-2bc54d28-2d54-4129-a2c8-4321d0190cf4\">\n","      Upload widget is only available when the cell has been executed in the\n","      current browser session. Please rerun this cell to enable.\n","      </output>\n","      <script src=\"/nbextensions/google.colab/files.js\"></script> "],"text/plain":["<IPython.core.display.HTML object>"]},"metadata":{"tags":[]}},{"output_type":"execute_result","data":{"text/plain":["{}"]},"metadata":{"tags":[]},"execution_count":1}]},{"metadata":{"id":"DGSoBb1xXBXp","colab_type":"code","colab":{}},"cell_type":"code","source":["!ls -lha kaggle.json"],"execution_count":0,"outputs":[]},{"metadata":{"id":"P3GW6n4tXCKK","colab_type":"code","colab":{}},"cell_type":"code","source":["!mkdir -p ~/.kaggle\n","!cp kaggle.json ~/.kaggle/\n","\n","# This permissions change avoids a warning on Kaggle tool startup.\n","!chmod 600 ~/.kaggle/kaggle.json"],"execution_count":0,"outputs":[]},{"metadata":{"id":"i0el9ovxbDt0","colab_type":"text"},"cell_type":"markdown","source":["# Downloading the data"]},{"metadata":{"id":"4qG9zaiCXg7S","colab_type":"code","outputId":"41158c3e-0220-4052-e33c-2a74d43e8852","executionInfo":{"status":"ok","timestamp":1543843368531,"user_tz":-330,"elapsed":7419,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":486}},"cell_type":"code","source":["!kaggle competitions download -c msk-redefining-cancer-treatment"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Downloading training_variants.zip to /content\n","\r  0% 0.00/24.2k [00:00<?, ?B/s]\n","\r100% 24.2k/24.2k [00:00<00:00, 14.8MB/s]\n","Downloading test_variants.zip to /content\n","\r  0% 0.00/47.5k [00:00<?, ?B/s]\n","100% 47.5k/47.5k [00:00<00:00, 39.6MB/s]\n","Downloading training_text.zip to /content\n"," 94% 57.0M/61.0M [00:00<00:00, 68.4MB/s]\n","100% 61.0M/61.0M [00:00<00:00, 104MB/s] \n","Downloading test_text.zip to /content\n"," 90% 89.0M/99.0M [00:00<00:00, 112MB/s]\n","100% 99.0M/99.0M [00:00<00:00, 126MB/s]\n","Downloading stage2_sample_submission.csv.7z to /content\n","  0% 0.00/765 [00:00<?, ?B/s]\n","100% 765/765 [00:00<00:00, 817kB/s]\n","Downloading stage2_test_variants.csv.7z to /content\n","  0% 0.00/7.25k [00:00<?, ?B/s]\n","100% 7.25k/7.25k [00:00<00:00, 4.13MB/s]\n","Downloading stage2_test_text.csv.7z to /content\n","  0% 0.00/8.88M [00:00<?, ?B/s]\n","100% 8.88M/8.88M [00:00<00:00, 115MB/s]\n","Downloading stage1_solution_filtered.csv.7z to /content\n","  0% 0.00/1.28k [00:00<?, ?B/s]\n","100% 1.28k/1.28k [00:00<00:00, 1.49MB/s]\n","Downloading stage_2_private_solution.csv.7z to /content\n","  0% 0.00/592 [00:00<?, ?B/s]\n","100% 592/592 [00:00<00:00, 665kB/s]\n"],"name":"stdout"}]},{"metadata":{"id":"sYhEqeuRXmVm","colab_type":"code","outputId":"504a0ea1-998d-4dbb-a3b0-4c19c9bfd5e0","executionInfo":{"status":"ok","timestamp":1543843378662,"user_tz":-330,"elapsed":9625,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":208}},"cell_type":"code","source":["!unzip training_variants.zip\n","!unzip training_text.zip\n","\n","!ls"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Archive:  training_variants.zip\n","  inflating: training_variants       \n","Archive:  training_text.zip\n","  inflating: training_text           \n","kaggle.json\t\t\t test_text.zip\n","sample_data\t\t\t test_variants.zip\n","stage1_solution_filtered.csv.7z  training_text\n","stage_2_private_solution.csv.7z  training_text.zip\n","stage2_sample_submission.csv.7z  training_variants\n","stage2_test_text.csv.7z\t\t training_variants.zip\n","stage2_test_variants.csv.7z\n"],"name":"stdout"}]},{"metadata":{"id":"usmExMvVbI_S","colab_type":"text"},"cell_type":"markdown","source":["# Importing the Pakages of Python Language"]},{"metadata":{"id":"rbR1HxW2ZNo2","colab_type":"code","colab":{}},"cell_type":"code","source":["import pandas as pd\n","import numpy as np\n","import nltk\n","import re\n","from nltk.corpus import stopwords\n","import matplotlib.pyplot as plt\n","from sklearn.model_selection import train_test_split\n","\n","from sklearn.preprocessing import normalize\n","from scipy.sparse import hstack\n","\n","from sklearn.calibration import CalibratedClassifierCV\n","from sklearn.linear_model import SGDClassifier\n","from sklearn.metrics import confusion_matrix\n","from sklearn.metrics.classification import accuracy_score, log_loss\n","\n","from sklearn.feature_extraction.text import CountVectorizer\n","from sklearn.feature_extraction.text import TfidfVectorizer\n","\n","from sklearn.decomposition import TruncatedSVD\n","from sklearn.linear_model import LogisticRegression\n","from sklearn.model_selection import cross_val_predict\n","from sklearn.model_selection import StratifiedKFold"],"execution_count":0,"outputs":[]},{"metadata":{"id":"cG6us1aTbKpg","colab_type":"code","outputId":"d24ea867-2cb3-477e-963e-480b04a5c4f5","executionInfo":{"status":"ok","timestamp":1543843384767,"user_tz":-330,"elapsed":794,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":258}},"cell_type":"code","source":["data = pd.read_csv('training_variants')\n","print('Number of data points : ', data.shape[0])\n","print('Number of features : ', data.shape[1])\n","print('Features : ', data.columns.values)\n","data.head()"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Number of data points :  3321\n","Number of features :  4\n","Features :  ['ID' 'Gene' 'Variation' 'Class']\n"],"name":"stdout"},{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>Gene</th>\n","      <th>Variation</th>\n","      <th>Class</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>0</td>\n","      <td>FAM58A</td>\n","      <td>Truncating Mutations</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>1</td>\n","      <td>CBL</td>\n","      <td>W802*</td>\n","      <td>2</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>2</td>\n","      <td>CBL</td>\n","      <td>Q249E</td>\n","      <td>2</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>3</td>\n","      <td>CBL</td>\n","      <td>N454D</td>\n","      <td>3</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>4</td>\n","      <td>CBL</td>\n","      <td>L399V</td>\n","      <td>4</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["   ID    Gene             Variation  Class\n","0   0  FAM58A  Truncating Mutations      1\n","1   1     CBL                 W802*      2\n","2   2     CBL                 Q249E      2\n","3   3     CBL                 N454D      3\n","4   4     CBL                 L399V      4"]},"metadata":{"tags":[]},"execution_count":11}]},{"metadata":{"id":"oJMYrma8bQDM","colab_type":"code","outputId":"0b2bc730-15a9-4d7a-8395-51c15136e1c7","executionInfo":{"status":"ok","timestamp":1543843389845,"user_tz":-330,"elapsed":2598,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":258}},"cell_type":"code","source":["# note the seprator in this file\n","data_text =pd.read_csv(\"training_text\",sep=\"\\|\\|\",engine=\"python\",names=[\"ID\",\"TEXT\"],skiprows=1)\n","print('Number of data points : ', data_text.shape[0])\n","print('Number of features : ', data_text.shape[1])\n","print('Features : ', data_text.columns.values)\n","data_text.head()"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Number of data points :  3321\n","Number of features :  2\n","Features :  ['ID' 'TEXT']\n"],"name":"stdout"},{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>TEXT</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>0</td>\n","      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>1</td>\n","      <td>Abstract Background  Non-small cell lung canc...</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>2</td>\n","      <td>Abstract Background  Non-small cell lung canc...</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>3</td>\n","      <td>Recent evidence has demonstrated that acquired...</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>4</td>\n","      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["   ID                                               TEXT\n","0   0  Cyclin-dependent kinases (CDKs) regulate a var...\n","1   1   Abstract Background  Non-small cell lung canc...\n","2   2   Abstract Background  Non-small cell lung canc...\n","3   3  Recent evidence has demonstrated that acquired...\n","4   4  Oncogenic mutations in the monomeric Casitas B..."]},"metadata":{"tags":[]},"execution_count":12}]},{"metadata":{"id":"qxRArDTf11R2","colab_type":"code","outputId":"ae10fe69-a5fa-4489-ebe1-47c45ccec0e6","executionInfo":{"status":"ok","timestamp":1543843393023,"user_tz":-330,"elapsed":1186,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":54}},"cell_type":"code","source":["print(data_text['TEXT'][3004])"],"execution_count":0,"outputs":[{"output_type":"stream","text":["c-KIT is a member of the type 3 subclass of transmembrane receptor tyrosine kinases, characterized by 5 immunoglobulin-like domains in the extracellular region, a negative regulatory juxtamembrane (JM) domain and a split adenosine triphosphate-binding and phosphotransferase tyrosine kinase domain (reviewed in Broudy1). c-KIT activation by stem cell factor (SCF) promotes dimerization and transphosphorylation at tyrosine residues, resulting in downstream signaling events leading to cell growth. Mutations of c-KIT have been associated with hematopoietic and nonhematopoietic tumors, including systemic mast cell disease (SMCD),2-4 acute myeloid leukemia (AML),5-8 and gastrointestinal stromal tumors (GISTs).9-12  The tyrosine kinase inhibitor imatinib mesylate (Gleevec; Novartis Pharma AG, Basel, Switzerland) is efficacious in the majority of patients with GIST harboring c-KIT mutations.9,13 c-KIT is most commonly activated in GIST tumors by small deletions in the JM that are thought to constitutively activate the tyrosine kinase by loss of autoinhibitory function.14 All c-KIT JM deletion mutants tested thus far are sensitive to inhibition by imatinib mesylate. These observations have suggested that imatinib mesylate might be of value in the treatment of SMCD or AML that also have activating mutations in c-KIT. However, the most common activating alleles in the context of SMCD and AML are the D816V and D816Y mutations in the c-KIT activation loop, and these mutations have been shown to be highly resistant to imatinib mesylate in vivo15 and in vitro.16-19  PKC412 is a novel staurosporine-derived tyrosine kinase inhibitor that targets protein kinase C (PKC), kinase insert domain-containing receptor (KDR), vascular endothelial growth factor receptor-2 (VEGF-R2), platelet-derived growth factor receptor α (PDGFRα), Fms-like tyrosine kinase 3 (FLT3), and c-KIT.20 PKC412 has previously been shown to be effective against the fusion protein Fip1-like 1 (FIP1L1)-PDGFRα with mutations in the kinase domain that are resistant to imatinib mesylate.21 Since inhibitors of D816V/Y c-KIT mutations would potentially have therapeutic activity in SMCD and AML, we have evaluated the effectiveness of PKC412 against a panel of c-KIT mutations identified in patients with SMCD, GIST, and AML.  Study design  Cell culture  A 2.9-kB XhoI cDNA fragment encoding human (hu)-c-KIT (lacking amino acids 510-513, GNNK) was cloned into the XhoI site of pCDNA3. c-KIT point mutations (Table 1) were generated (Transformer Site-Directed Mutagenesis Kit; Clontech, Palo Alto, CA) and subcloned into the XhoI site of MSCV-IRES-GFP and MSCV-PGK-NEO. The IL-3-dependent mouse pre-B-cell line Ba/F3 was maintained, and retrovirus was prepared, as previously described.22 Ba/F3 cells were transduced with viral supernatant, expanded for 2 passages (with 10 ng/mL rmIL-3), then selected for IL-3-independent growth. Ba/F3 cells transduced with Neo-containing vectors were first selected in IL-3 with 100 mg/mL G418. IL-3-independent Ba/F3 cells transduced with wild-type (WT) c-KIT were selected with 20 ng/mL rhSCF.  View inlineView popup Table 1. Imatinib mesylate and PKC412 IC50s of transforming c-KIT mutations Patient samples were collected following informed consent in accordance with the institutional review board (IRB)-approved Dana-Farber Cancer Institute protocol 01-206.  Growth inhibition  Cells (1 × 104) in 100 μL medium with or without IL-3 were incubated with various concentrations of imatinib mesylate or PKC412 in 96-well plates for 48 hours in triplicate. 3H-thymidine (1 μCi [0.037 MBq]) was added, and cells were incubated for 4 hours. Cells were harvested, and 3H-thymidine incorporation was determined. Growth inhibition was plotted as the ratio of the average 3H-thymidine incorporation in drug-treated wells relative to no-drug controls. IC50s for imatinib mesylate or PKC412 were calculated by sigmoidal curve fitting software (OriginLab, Northampton, MA).  Immunoprecipitations  Cell lysates, immunoprecipitations, and immunoblotting were performed as previously described.22 c-KIT was precipitated with 5 μg anti-human c-KIT antibody (Assay Design Systems, Ann Arbor, MI), separated on a 7.5% denaturing polyacrylamide gel and transferred to nitrocellulose. Blots were probed with mouse anti-phosphotyrosine antibody 4G10 (Upstate, Charlottesville, VA), stripped, and reprobed with mouse anti-human c-KIT (E1): SC-17806 (Santa Cruz Biotechnology, Santa Cruz, CA). Whole-cell lysates (1/10th volume immunoprecipitate [IP] input) were similarly separated and probed with rabbit anti-phospho-Stat5 (signal transducer and transcriptional activator 5) or rabbit anti-phospho-Stat3, stripped, and reprobed with rabbit anti-Stat5a or rabbit anti-Stat3 (Cell Signaling Technology, Beverly, MA).  To assess constitutive tyrosine kinase autophosphorylation, cells were grown in RPMI plus 10% fetal calf serum (FCS), and 4 mg cell lysate was immunoprecipitated. To assess PKC412 inhibition of c-KIT autophosphorylation, cells were washed 3 times in RPMI without FCS or IL-3, then incubated with or without IL-3 and PKC412 (0-500 nM) for 4 hours. WT Ba/F3 cells were starved of IL-3 and serum for 4 hours prior to addition of IL-3 and PKC412. Cell equivalents (1 × 107) of cell lysate were immunoprecipitated.  Results and discussion  The IL-3-dependent murine pre-B-cell line Ba/F3 was transduced with retroviruses expressing WT c-KIT or 1 of 14 c-KIT mutants and selected for IL-3-independent growth (Table 1; Figure 1A). Ten of 14 mutant forms of c-KIT transformed Ba/F3 cells to IL-3-independent growth in the absence of rhSCF. Transforming mutations mapped to the extracellular domain (delTYD (417-419) + I,8 delTYD (417-419) + RG,8 and A502Y503dup (AYdup)10), the juxtamembrane domain (V559D,11 V560G,2 and delV559V560 (delVV)11), and the kinase domain (R634W, D816Y,3 D816V,2 and N822K7). This is the first report of the kinase domain mutation R634W, identified from a patient with chronic myelomonocytic leukemia. Consistent with a report that E839K inactivates the kinase domain,12 E839K did not transform Ba/F3 cells to IL-3-independent growth. K642E has been shown to be constitutively phosphorylated in a GIST-derived cell line10,23 but did not transform Ba/F3 cells in this study. D820G4 and V530I,8 each identified in a single patient, also did not transform Ba/F3 cells.  Immunoprecipitated c-KIT was constitutively phosphorylated in all transformed cell lines and Ba/F3 cells expressing WT c-KIT grown with 20 ng/mL rhSCF (Figure 1B). We observed differential phosphorylation of 2 c-KIT bands of approximately 160 and 145 kDa, representing the fully glycosylated cell surface receptor, and incompletely processed internalized forms of c-KIT, respectively. Our findings confirm reports that Stat3 is aberrantly phosphorylated by D816V and other c-KIT-activating mutations.24,25 Stat5 was constitutively phosphorylated by all c-KIT-activating mutations grown in the absence of IL-3.  Figure 1. Download figureOpen in new tabDownload powerpoint Figure 1. Imatinib mesylate resistant c-Kit mutations are sensitive to PKC412. (A) Schematic of the c-KIT protein indicating the relative location of structural and functional domains, as well as location of 14 c-KIT mutations evaluated. The JM delVV mutation occurs at V559V560. ▪ indicates amino terminal signal peptide. Other domains of c-KIT are indicated as follows: □, extracellular domain; TM, transmembrane domain; JM, juxtamembrane domain; TK1, nucleotide binding subdomain of split kinase domain; KI, kinase insert domain; and TK2, catalytic subdomain of split kinase domain. (B) Ba/F3-transforming c-KIT mutations are constitutively phosphorylated and phosphorylate Stat3 and Stat5. Blots from top to bottom: antiphosphotyrosine Western blot of Ba/F3 lysates immunoprecipitated with anti-c-KIT; anti-c-KIT Western blot of immunoprecipitates in top blot; anti-phospho-Stat3 Western blot of whole-cell lysates (WCLs) from IP in top blot; anti-Stat3 Western blot of same WCLs; anti-phospho-Stat5 Western blot of WCLs from IP in top blot; and anti-Stat5a Western blot of the same WCLs. (C) Representative growth curves of delTYD + RG, V560G, D816Y, and D816V transduced Ba/F3 cell lines. Plotted is the percentage ± SD of 3H-thymidine incorporation of drug-treated cells relative to no-drug controls. Cells were treated with imatinib mesylate (black) or PKC412 (red). Cells were grown in the absence of IL-3 or rhSCF. (D) Dose response of PKC412 inhibition of c-KIT, Stat5, and Stat3 phosphorylation. Shown are data from delTYD + RG, V560G, D816Y, and D816V transformed Ba/F3 cells. Blots are as in panel B. Cells were grown in the absence of IL-3 and the indicated concentration of PKC412 (nM) for 4 hours. (E) PKC412 does not inhibit IL-3-induced Stat5 phosphorylation. Blots from top to bottom: anti-phospho-Stat3 Western blot of WCLs from indicated cell lines with or without IL-3 (10 ng/mL) and PKC412 (250 nM/mL), for which Ba/F3 cells were starved of IL-3 and serum for 4 hours prior to addition of IL-3 and PKC412; anti-Stat3 Western blot of WCLs in top blot; anti-phospho-Stat5 Western blot of WCLs as in top blot; and anti-Stat5a Western blot of same WCLs. (F) Dose response of Stat5 phosphorylation to PKC412 in Ba/F3 cells. Blots are as for panel E. Ba/F3 cells were starved of IL-3 and FCS for 4 hours, then incubated with or without IL-3 (10 ng/mL) in the presence of the indicated concentration of PKC412 for 4 hours. The effect of imatinib mesylate or PKC412 on cell growth of c-KIT-expressing cell lines was evaluated in a 3H-thymidine incorporation assay (Table 1; Figure 1C; data not shown). Consistent with previous reports, D816V was resistant to imatinib mesylate (IC50, > 10 000 nM).16-19 D816Y was also observed to be resistant to imatinib mesylate (IC50, > 800 nM). Four mutations, delTYD + I, AYdup, R634W, and N822K, showed IC50s for imatinib mesylate similar to Ba/F3-WT-c-KIT + 20 ng/mL rhSCF (100-300 nM). As previously reported, V560G19 was hypersensitive to imatinib mesylate, as were the mutations delTYD + RG, V559D, and delVV (IC50s < 100 nM).  PKC412 inhibited growth of all c-KIT-transformed Ba/F3 lines, with IC50s segregating into 2 groups. The imatinib mesylate-resistant D816V and D816Y mutants, as well as delTYD + RG, V560G, R634W, and N822K, were more sensitive to PKC412 (IC50s, 33-95 nM) than was the wild-type receptor stimulated with rhSCF (IC50, 138 nM). The remaining mutations had higher IC50s for PKC412 (146-265 nM).  The level of tyrosine autophosphorylation of the imatinib mesylate-resistant D816V and D816Y c-KIT mutants, as well as phosphorylation of Stat3 and Stat5, was dose dependent on PKC412 over a wide concentration range (5-500nM) and correlated with inhibition of cell growth (Table 1; Figure 1D). Similar findings were observed for representative imatinib mesylate-sensitive extracellular (del-TYD + RG) and JM (V560G) mutants. Although Stat5 phosphorylation by D816V and D816Y was inhibited by PKC412 (Figure 1D-E), an increased level of Stat5 phosphorylation induced by stimulation with IL-3 induction was not inhibited in these cells (Figure 1E). IL-3 did not rescue PKC412 inhibition of phosphorylation of Stat3 (Figure 1E) and did not inhibit IL-3-induced Stat5 phosphorylation in WT Ba/F3 Cells (Figure 1E-F).  These data indicate that among the reported c-KIT mutations in human hematologic malignancies, 10 of 14 confer factor independent growth to Ba/F3 cells. There is variable sensitivity of these mutations to inhibition by imatinib mesylate or PKC412. Notably, however, the most common mutations D816V and D816Y are highly resistant to imatinib mesylate but sensitive to inhibition by PKC412. These findings suggest that PKC412 may be an effective therapeutic agent for treatment of patients with certain imatinib mesylate-resistant c-Kit mutations. Activating mutations in either BRAF or NRAS are seen in a significant number of malignant melanomas, but their incidence appears to be dependent to ultraviolet light exposure. Thus, BRAF mutations have the highest incidence in non-chronic sun damaged (CSD), and are uncommon in acral, mucosal and CSD melanomas. More recently, activating KIT mutations have been described in rare cases of metastatic melanoma, without further reference to their clinical phenotypes. This finding is intriguing since KIT expression is downregulated in most melanomas progressing to more aggressive lesions. In this study, we investigated a group of anal melanomas for the presence of BRAF, NRAS, KIT and PDGFRA mutations. A heterozygous KIT exon 11 L576P substitution was identified in 3 of 20 cases tested. The 3 KIT mutation-carrying tumors were strongly immunopositive for KIT protein. No KIT mutations were identified in tumors with less than 4+ KIT immunostaining. NRAS mutation was identified in one tumor. No BRAF or PDGFRA mutations were identified in either KIT positive or negative anal melanomas. In vitro drug testing of stable transformant Ba/F3 KITL576P mutant cells showed sensitivity for dasatinib (previously known as BMS-354825), a dual SRC/ABL kinase inhibitor, and imatinib. However, compared to an imatinib-sensitive KIT mutant, dasatinib was potent at lower doses than imatinib in the KITL576P mutant. These results suggest that a subset of anal melanomas show activating KIT mutations, which are susceptible for therapy with specific kinase inhibitors. © 2007 Wiley-Liss, Inc. The KIT receptor tyrosine kinase plays a critical role in the normal development and function of the melanocytes during embryonic stage and in the postnatal organism.1, 2 Loss-of-function mutations in KIT and its ligand result in white spotting phenotype in mice and humans.3, 4, 5 In contrast, activating KIT mutations have been implicated in the pathogenesis of certain human malignancies, such as gastrointestinal stromal tumors, seminomas, mastocytomas, but were not reported until recently in patients with malignant melanoma. Furthermore, KIT expression, typically found in normal melanocytes, benign nevi and in situ melanoma, appears to be downregulated in invasive and metastatic melanomas.6 This progressive decrease in KIT immunopositivity is associated with increasing dermal invasiveness of the tumor. One hypothesis is that KIT signaling may be important in regulating cell differentiation and tissue morphogenesis, and therefore loss of its expression may be a required step in the course of tumor progression.7 In this context, isolated reports of activating KIT mutations in melanoma patients,8, 9 are intriguing and suggest an alternative mechanism, in which constitutive activation of KIT through oncogenic mutations positively regulate mitogenesis and induces neoplastic transformation.  The genetic alterations identified in melanomas at different sites and with different levels of sun exposure indicate that there are distinct molecular subsets of melanoma, with direct implications on therapeutic approaches and prevention strategies. BRAF mutations, for example, are most often seen in melanomas arising in skin intermittently exposed to the sun, and are rare events in CSD or in areas never exposed to the sun (acral and mucosal membranes).10NRAS mutations occur only in melanomas without BRAF mutations and do not appear to be associated with a specific clinical phenotype.10 In contrast, increased copy number of CCND1 (Cyclin D1) or CDK4, two downstream genes in the Ras/MAPK signaling pathway, were seen in CSD-melanomas and acral and mucosal melanomas, inversely correlating with BRAF mutations. This recent genetic classification of malignant melanomas, based on the relationship to ultraviolet light exposure does not yet include the group of melanomas associated with activating KIT mutations.  The purpose of this study was to investigate the incidence of KIT expression and oncogenic mutations in a group of mucosal melanomas located in the anus, and to assess the in vitro efficacy of specific kinase inhibitors against cell lines expressing these KIT mutants. Since anal mucosal melanomas are associated with a poor prognosis, the presence of oncogenic KIT mutations could provide a rationale for exploring the use of imatinib or related kinase inhibitors in the treatment of patients with such tumor.11 Materials and methods  Patient selection and clinicopathologic features  The institutional review board and human tissue utilization committee of Memorial Sloan-Kettering Cancer Center (MSKCC) approved this study. The archive of the Department of Pathology was searched for cases of anal melanoma between 1989 and 2005. Twenty-six patients (13 women and 13 men, ranging in age from 34 to 89 years) were identified and the pathologic diagnosis was confirmed by review of the histologic slides and immunohistochemical markers. A representative formalin-fixed, paraffin-embedded block with viable tumor was selected and used for immunohistochemistry as well as for the molecular studies described below (PCR and FISH analyses). The tissue samples used for the study included 20 primary and 6 metastatic anal mucosal melanomas. From 1 patient, both the primary and metastatic tumors were examined. KIT (CD117) immunohistochemistry was performed with a murine monoclonal antibody (Dakocytomation, Carpinteria, CA), at a 1:100 dilution in hipH antigen retrieval solution (Dakocytomation). Endogenous mast cells were used as internal-positive controls. The immunostaining was evaluated microscopically and graded based on the estimated amount of immunopositive tumor cells as follows: + 25%, 2++ >25%–50%, +++ >50%–75%, ++++ >75%. KIT/PDGFRA mutational analysis  Genomic DNA was isolated in 26 cases from formalin-fixed paraffin embedded tissue using the DNeasy™ Tissue Kit (Qiagen, Valencia, CA). One microgram of genomic DNA was subjected to PCR using Platinum TaqDNA Polymerase High Fidelity (Life Technologies, Gaithersburg, MD). The PCR conditions were as follows: (i) 94°C for 4 min; (ii) 94°C for 30 sec, (iii) the relevant annealing temperature for each primer set for 30 sec, (iv) 72°C for 30 sec, (35 cycles); and (v) 72°C for 3 min. The PCR products were identified by agarose gel electrophoresis using a 2% MetaPhor™ agarose gel (BioWhittaker Applications, Rockland, ME). The PCR products were purified with the QIAquick™ PCR Purification Kit (Qiagen) before sequencing. All cases were tested for KIT exons 11, 13 and 17, as well as for PDGFRA exon 12 and 18. The primer sequences and annealing temperatures are listed in Table I. In one case (Sample no. 19) with frozen tissue available, total RNA was extracted and subjected to RT-PCR, followed by direct sequencing of the full-length KIT cDNA. The RT-PCR conditions and primers used were previously published.12 Every ABI sequence was compared to a NCBI Human KIT gene nucleotide sequence and blasted using a NCBI Standard Nucleotide Blast Search to determine the presence of mutation within a particular exon.  Table I. KIT Exon 11, 13, and 17 and PDGFRA Exon 12 and 18 Primer Sequences, with the Corresponding Annealing Temperatures Exon number\tPrimers\tPrimer sequences 5′ → 3′\tTA (°C) KIT  11\tEx11-F\tCCAGAGTGCTCTAATGACTG\t53 Ex11-R\tACCCAAAAAGGTGACATGGA  13\tEx13-F\tATCAGTTTGCCAGTTGTGCT\t53 Ex13-R\tTTTATAATCTAGCATTGCC  17\tEx17-F\tTGTGAACATCATTCAAGGCGTAC\t53 Ex17-R\tCAGGACTGTCAAGCAGAGAATGG PDGFRA  12\tEx12-F\tTCCAGTCACTGTGCTGCTTC\t54 Ex12-R\tGCAAGGGAAAAGGGAGTCTT  18\tEx18F\tACCATGGATCAGCCAGTCTT\t55 Ex18R\tTGAAGGAGGATGAGCCTGACC BRAF and NRAS mutational analysis  Ligase detection reaction (LDR) is a sensitive and specific assay for the detection of individual point mutations13, 14, 15, 16 and was performed to detect BRAF13 and NRAS point mutations. Primers were designed for the 19 possible mutations described in NRAS codons 12, 13 and 61 (Table II; Integrated DNA Technologies, Coralville, IA). Exon specific PCR product was mixed with fluorescently labeled mutant specific primers and common downstream primers. Ligation was performed with Taq DNA Ligase (New England BioLabs). The reaction mixture was cycled as follows: 94°C for 1.5 min for the initial denaturation phase followed by 10 cycles of 94° C for 1 min and 65°C for 4 min (67°C for 4 min was used for codon 13 mutations). Three microliters of the final LDR product was mixed with 2 μl of loading buffer (83% formamide, 83 mmol/L EDTA, and 0.17% Blue Dextran) and denatured at 95°C for 5 min. The mixture was then placed on ice for a rapid cooling, loaded on a 12.5% polyacrylamide gel, and run on a 377 DNA Prism Sequencer (Applied Biosystems) by electrophoresis at 1,400 V for 2.5 hr. All fluorescently labeled primers were designed at varying lengths, which caused differential electrophoretic mobility shifts of the ligated products. Genescan 350 TAMRA molecular weight marker was used as size standard and GeneScan software V3.1 was utilized for data analysis.  Table II. Ligase Detection Reaction Primer Sequences Used for NRAS Mutational Analysis LDR primer\tTA (C°)\tPrimer size\tProduct length\tPrimer sequence NRAS codon 12  G12.1R\t55.8\t21\t42\t5′Fam-ATC TGG TGG TGG TTG GAG CAC-3′  G12.1C\t58.5\t19\t40\t5′Fam-CTG GTG GTG GTT GGA GCA T-3′  G12.1S\t55.8\t17\t38\t5′Fam-GGT GGT GGT TGG AGC AA-3′ Com12.1\t60.9\t21\t \t5′Phos-GTG GTG TTG GGA AAA GCG CAC-3′  G12.2A\t61.9\t22\t48\t5′Fam-ATC TGG TGG TGG TTG GAG CAG C-3′  G12.2D\t60.5\t20\t46\t5′Fam-CTG GTG GTG GTT GGA GCA GA-3′  G12.2V\t58.2\t18\t44\t5′Fam-GGT GGT GGT TGG AGC AGT-3′ Com12.2\t61.6\t26\t \t5′Phos-TGG TGT TGG GAA AAG CGC ACT TTT TT-3′ NRAS codon 13  G13.1R\t64.8\t23\t44\t5′Fam-ACT GGT GGT GGT TGG AGC AGG TC-3′  G13.1C\t58.7\t19\t40\t5′Fam-GTG GTG GTT GGA GCA GGT T-3′  G13.1S\t62.0\t21\t42\t5′Fam-TGG TGG TGG TTG GAG CAG GTA-3′ Com13.1\t59.8\t21\t \t5′Phos-GTG TTG GGA AAA GCG CAC TGA-3′  G13.2A\t67.3\t24\t50\t5′Fam-ACT GGT GGT GGT TGG AGC AGG TGC-3′  G13.2D\t64.6\t22\t48\t5′Fam-TGG TGG TGG TTG GAG CAG GTG A-3′  G13.2V\t60.8\t20\t46\t5′Fam-CTG GTG GTT GGA GCA GGT GT-3′ Com13.2\t61.4\t26\t \t5′Phos-TGT TGG GAA AAG CGC ACT GAC AAT AT-3′ NRAS codon 61  Q61.1E\t62.8\t24\t51\t5′Fam-T CTC ATG GCA CTG TAC TCT TCT TC-3′  Q61.1K\t56.2\t23\t50\t5′Fam-CTC ATG GCA CTG TAC TCT TCT TT-3′ Com61.1\t61.5\t27\t \t5′Phos-TCC AGC TGT ATC CAG TAT GTC CAA CAA-3′  Q61.2R\t57.3\t23\t47\t5′Fam-TCT CAT GGC ACT GTA CTC TTC TC-3′  Q61.2L\t57.1\t24\t48\t5′Fam-CTC TCA TGG CAC TGT ACT CTT CTA-3′  Q61.2P\t59.5\t25\t49\t5′Fam-TCT CTC ATG GCA CTG TAC TCT TCT G-3′ Com61.2\t59.6\t24\t \t5′Phos-GTC CAG CTG TAT CCA GTA TGT CCA-3′  Q61.3Ha\t59.7\t25\t52\t5′Fam-GTC TCT CAT GGC ACT GTA CTC TTC A-3′  Q61.3Hb\t62.0\t26\t53\t5′Fam-GGT CTC TCA TGG CAC TGT ACT CTT CG-3′ Com61.3\t59.6\t27\t \t5′Phos-TGT CCA GCT GTA TCC AGT ATG TCC AAC-3′ Control  Ex1 upstream\t56.8\t28\t55\t5′Fam-CTA ATC CAG AAC CAC TTT GTA GAT GAA T-3′  Ex1 downstream\t61.3\t27\t \t5′Phos-ATG ATC CCA CCA TAG AGG TGA GGC CCA-3′  Ex2 upstream\t64.5\t24\t56\t5′Fam-GAG GAC AGG CGA AGG CTT CCT CTG-3′  Ex2 downstream\t58.8\t32\t \t5′Phos-TGT ATT TGC CAT CAA TAA TAG CAA GTC ATT TG-3′ Fluorescence in situ hybridization  Fluorescence in situ hybridization (FISH) was performed on paraffin sections from 5 tumors according to standard procedures. The KIT probes used were 2 overlapping BAC clones: CTD-3180G20 and RP11-722F21 (Invitrogen, Carlsbad, CA), labeled by nick- translation with Spectrum Green (Vysis, Abbott Laboratories, IL). A chromosome 4 centromeric probe labeled with Spectrum Orange (CEP 4, Vysis) was used as reference. The probe mix, 50–80 ng of each KIT BAC and 2 μl Cot-1 DNA (Invitrogen), was ethanol precipitated, and resuspended in hybridization buffer. The KIT probe mix was denatured at 70°C for 10 min, followed by pre-annealing at 37°C for 30 min. The KIT probe was then combined with the denatured CEP 4 probe on the slide, coverslipped and incubated overnight at 37°C. After standard post-hybridization washes, the slides were stained with DAPI and mounted in antifade (Vectashield, Vector Laboratories, Burlingame, CA). Analysis was performed using a Nikon E800 epifluorescence microscope with MetaSystems Isis 3 imaging software. A minimum of 100 cells was scanned over separate regions for each slide. Western blotting  In 1 case (Sample no. 19) frozen tissue was available and whole cell lysate was prepared by grinding 1 g of snap frozen tumor tissue, using a PowerGen 700 Homogenizer (Omni International, Marietta, GA). The ground tissue was resuspended in radioimmunoprecipitation assay lysis buffer (Upstate, Lake Placid, NY) containing a cocktail of protease and phosphatase inhibitors (Sigma, St. Louis, MO), sodium fluoride, sodium orthovanadate and phenylmethylsulfonyl fluoride. Electrophoresis and immunoblotting was performed on the protein extracts using standard protocols.12 Antibodies tested included rabbit polyclonal anti-phospho-KIT Y721 (Zymed Lab, San Francisco, CA), rabbit anti-KIT (Oncogene Science, Boston, MA), anti-phospho-PDGFRalpha (Cell Signaling Technology, Danvers, MA), anti-PDGFRalpha (Santa Cruz Biotechnology, Santa Cruz, CA) and mouse anti-Actin (Santa Cruz). The secondary antibodies used included Donkey-antimouse secondary (Santa Cruz) and antirabbit (Calbiochem, La Jolla, CA). Following hybridization with the secondary antibody, the blots were incubated with Immun-Star horse radish peroxidase luminal/enhancer (BioRad, Hercules, CA) and exposed onto Kodak Biomax MR Film (Eastman Kodak, Rochester, NY). Establishing Ba/F3 KIT L576P stable transformant cells  A KIT exon 11 L576P mutation was introduced by Site Directed Mutagenesis PCR using QuickChange II XL site-directed Mutagenesis Kit (Qiagen), on a wild type KIT cDNA cloned into the retroviral expression vector pMSCV (a generous gift from Dr. Gary Gilliland, Harvard Medical School). A similar strategy was used for obtaining a KIT exon 11 V559D, an imatinib-sensitive juxtamembrane mutation, used as control for in vitro drug testing. Primers used for mutagenesis are listed in Table III. These pMSCV DNA constructs were introduced into Ba/F3 cells (German Collection of Microorganisms and Cell Cultures—Human and Animal Cell Cultures, Braunshweig, Germany) by electroporation. Ten million Ba/F3 cells were cotransfected with the pMSCV-KIT DNA constructs (25 mg) and Hygromicin resistance plasmid (1 μg) using a GenPulser (BioRad, Hercules, CA). The electroporated cells were first selected in Hygromycin for a week, and the remaining cells were sorted by flow cytometry using GFP fluorescence. All of the sorted GFP (+) positive cells were grown for more than 2 weeks and KIT expression was confirmed by flow cytometry using PE conjugated anti-CD117 antibody (BD Biosciences, San Jose, CA). Both Ba/F3 cells KITL576P and KITV559D mutant cells survived without KIT ligand (KL). However, a slower growth and increased cell death was noticed in the absence of KL. Adding KL was used to improve cell growth, in order to obtain a cell population with minimum dead cells for apoptosis assays. Our Ba/F3 KIT mutant cells showed a better growth rate in the presence of low dose of KL (20 ng/ml), most likely due to a partial KL independent growth. The concentration of KL used, 20 ng/ml, is quite low compared to the 100 ng/ml used for maintaining the growth of Ba/F3 WTKIT.  Table III. KIT Exon 11 Primers Used for Site Directed Mutagenesis PCR Mutation\tPrimer V559D  (F)\t5′-GAAGTACAGTGGAAGGATGTTGAGGAGATA-3′  (R)\t5′-TATCTCCTCAACATCCTTCCACTGTACTTC-3′ L576P  (F)\t5′-ATAGACCCAACACAACCTCCTTATGATCACAAA-3′  (R)\t5′-TTTGTGATCATAAGGAGGTTGTGTTGGGTCTAT-3′ In vitro drug testing  Dasatinib (BMS-35482, TM: Sprycel), nilotinib (AMN107, TM: Tasigna), and imatinib (STI571, TM: Gleevec) were tested in Ba/F3 KITL576P cells and the efficacy was compared to Ba/F3 KITV559D cells, which are imatinib-sensitive. Each drug was tested at 10, 100, 1,000 and 5,000 nM. Nontreated cells served as baseline control. The drug response was monitored by measuring cell proliferation and apoptosis, as well as inhibition of KIT autophosphorylation.  Cell proliferation assays.  To determine growth-inhibitory drug effects, Ba/F3 transformants containing KIT mutants were starved from growth factor for 4 hr prior to drug administration. Ten minutes before adding the drug, growth factors were added: 10 ng/ml of IL3 for the negative control BaF3 cells and 20 ng/ml of KL for Ba/F3 cell lines expressing KIT mutant isoforms. Cells were incubated with imatinib, dasatinib and nilotinib with the above indicated doses in 24-well plates at 37°C for 48 hr. After incubation with the drug, BrdU was added to the cell culture and incubated for 6 hr before harvesting. Cells were fixed and stained with the protocol recommended for the BrdU-APC kit (Pharmingen, San Diego, CA). BrdU incorporation was determined by flow cytometry. A minimum of 2 × 105 events were acquired and the data was analyzed using FlowJo software (version 5.7.2). Cell growth curves were plotted with GraphPad Prism software, version 4.03. IC50s were calculated with the same software. Apoptosis assays.  Induction of apoptosis on Ba/F3 KIT transformants was evaluated by flow cytometry using AnnexinV-PE Apoptosis Detection kit (Pharmingen). Cells at a density of 1 × 106, activated with KL, were cultured in 6-well plates and starved from growth factors for 4 hr prior to drug delivery. Ten minutes before adding the drug, growth factors were added. Cells were incubated with imatinib, dasatinib and nilotinib at the same concentrations indicated above for 48 hr. Cells were harvested after drug treatment and stained with anti-AnnexinV-PE antibody using the provider's protocol. A minimum of 10,000 events were analyzed by FACScan (Becton-Dickison) within an hour after staining. The data and results were analyzed by FlowJo. Immunoprecipitation and Western blotting.  The Ba/F3 cells expressing mutant KIT isoforms, L576P and V559D, were treated with the indicated doses of each drug. Cells were starved from serum and growth factors for 2 hr, only growth factors were added back to media 15 min prior to drug administration. Drugs were incubated with cells at 37°C in the absence of serum for 90 min. After treatment, cells were harvested and subjected to protein extraction, using a similar protocol similar as above. Whole cell lysate (200 μg) was incubated with 2 μg of anti-KIT antibody (Assay Designs, Ann Arbor, MI) for an hour, and then the mixture was incubated with Magna beads (Pierce Biotechnology, Rockford, IL) overnight at 4°C. The beads were washed and isolated using a magnetic column and resuspended in LDS sample buffer (Invitrogen). Electrophoresis and immunoblotting was performed on the protein extracts using the standard protocol. Phosphoryated KIT was detected with anti-phospho-tyrosine antibodies Tyr20 and Tyr 99 (Santa Cruz), and total KIT was detected by mouse monoclonal anti-c-KIT (Santa Cruz). The secondary antibodies used were Donkey-anti-mouse secondary (Santa Cruz). Results  The presence of KIT mutations correlated with strong and diffuse expression of KIT  By immunohistochemical analysis, 6 (23%) of the 26 anal melanomas tested showed a diffuse and strong (4+) reactivity for CD117. Three of these tumors were primary anorectal melanomas and two were metastases from known primary anal tumors. In the three primary tumors, the strong KIT expression pattern was seen in both the intraepithelial (in situ) and in the invasive melanoma component (Fig. 1a). In 1 patient the strong and homogeneous KIT immunoreactivity was seen in both the primary melanoma as well as the corresponding metastasis (Fig. 1b). Additional 3 cases showed heterogeneous and weak (1+) immunostaining. These were all metastatic lesions from primary anal melanomas.  Figure 1.  Figure 1.  Open in figure viewer Immunohistochemical staining of anal melanoma. (a) Primary melanoma with strong and diffuse KIT immunoreactivity in both the intraepidermal (in situ) and invasive component (×100). (b) The metastasis derived from this tumor is also strongly and diffusely positive for CD117 (×200). Adequate DNA for assessing activating mutations in BRAF, NRAS, KIT, and PDGFRA was obtained in 20 of the 26 cases (Table IV). A heterozygous KIT exon 11 L576P substitution was identified in 3 (13%) of the 20 cases (Fig. 2). However, 3/4 diffusely KIT immunopositive (4+) anal melanomas with adequate DNA available showed KIT mutations. KIT mutations were not identified in tumors with less than 4+ KIT immunostaining. The only tumor that had high expression of KIT immunohistochemically, but lacked activating KIT mutation by DNA PCR and direct sequencing of the hot spots, was further studied by full length KIT cDNA sequencing, which confirmed an wild-type sequence. One (5%) of the 20 anal melanomas tested showed a NRAS mutation in codon 61. No PDGFRA or BRAF mutations were identified in either KIT positive or negative anal melanomas.  Figure 2.  Figure 2.  Open in figure viewer KIT genotyping showed an identical heterozygous L576P substitution in exon 11 in 75% of anal melanomas expressing strong KIT immunoreactivity. The upper ABI depicts the KIT exon 11 forward sequence and the lower ABI sequence the KIT exon 11 reverse. The arrow indicates the position of L576P substitution in exon 11, resulting in the amino acid change of Leucine to Proline at position 576, which corresponds to a codon change of CTT to CCT. Table IV. Genotyping Results for NRAS, KRAS, BRAF, KIT and PDGRA Mutation Analysis, Correlated with KIT Expression by Mmunohistochemistry Sample\tNRAS codon 12\tNRAS codon 13\tNRAS codon 61\tKRAS codon 12/13\tBRAF 599\tKIT ex11\tKIT ex13\tKIT ex17\tPDGFRA ex 12\tPDGFRA ex 18\tKIT IHC 1FISH showed 2 copies of KIT signal. 2Western blotting showed phosphorylated KIT protein. 3FISH showed increased number of KIT signals, compared to CEP4. NA, not available; WT, wild type. 1\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 21\tWT\tWT\t61 cod/Lys\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 3\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 4\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 5\tWT\tWT\tWT\tNA\tWT\tWT\tWT\tNA\tWT\tWT\t1+ 6\tWT\tWT\tNA\tWT\tWT\tWT\tWT\tNA\tWT\tWT\tNeg 7\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNA\tWT\tWT\tNeg 8\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 9\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 10\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\t1+ 11\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 12\tNA\tNA\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 13\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\t1+ 14\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 15\tWT\tWT\tWT\tWT\tWT\tNA\tWT\tWT\tWT\tWT\tNeg 16\tNA\tNA\tNA\tNA\tNA\tL576P\tWT\tNA\tNA\tNA\t4+ 17\tWT\tWT\tWT\tWT\tWT\tL576P\tWT\tWT\tNA\tNA\t4+ 181\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tNeg 191, 2\tWt\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\tWT\t4+ 203\tWT\tWT\tWT\tWT\tWT\tL576P\tWT\tWT\tWT\tWT\t4+ Increased copy number of KIT gene is a rare event in anal melanoma  By fluoresence in situ hybridization, the status of KIT copy number changes/amplification was studied in 5 anal melanomas. Three of these 5 cases were selected from the 4+ KIT immunoreactive tumors, while 2 cases were KIT negative by immunohistochemistry. All except one case showed 2 copies of KIT signal, with a KIT to CEP4 ratio of 1:1 (Fig. 3a, Sample no.19). One of the KIT immunopositive tumors had an increased KIT copy number, with a KIT to CEP4 ratio of 5:2 (Fig. 3b, Sample no. 20). This case also showed the presence of a KIT exon 11 L576P substitution.  Figure 3.  Figure 3.  Open in figure viewer Fluorescence in situ hybridization showing: (a) 2 copies of KIT (Red) and centromere 4 (green) (Sample no. 19), and (b) increased copy number of KIT signals, with an average ratio KIT (Red) to centromere 4 (green) signals of 5:2 (Sample no. 20). KIT activation by phosphorylation is detected in KIT positive anal melanomas  The KIT phosphorylation status was investigated in one tumor with frozen tissue available (Sample no. 19), by western blotting against phospho and total KIT antibodies, and compared to a KIT exon 11 mutated GIST. Sample no. 19 belonged to the only patient showing a 4+ KIT immunopositive tumor and lacking a concurrent KIT mutation. The Western blot showed phosphorylated KIT protein, using the KIT Y721 antibody, with a similar pattern as the GIST positive control. This result was further confirmed after stripping and re-probing with total-KIT antibody (Fig. 4). However, this sample lacked phosphorylated PDGFRA (data not shown).  Figure 4.  Figure 4.  Open in figure viewer Western blotting showing KIT phosphorylation in Sample no. 19 with a similar pattern as a GIST positive control, harboring a KIT exon 11 mutation. Dasatinib potently inhibited the cell proliferation and KIT phosphorylation of Ba/F3 KITL576P cells  After establishing stable transformants, an IL3 independent selection was performed by withdrawing IL3 and providing KL of different doses (10 ng to 100 ng/ml, data not shown). The ideal KL concentration for each KIT mutant was determined by counting the trypan blue stained cells versus viable cells with phase contrast microscope. The transforming activity was determined by IL3 independent growth curve. The effects of imatinib, dasatinib and nilotinib on transformed Ba/F3 KITL576P mutant cells was analyzed and compared to Ba/F3 cells harboring an imatinib-sensitive KIT exon 11 mutation, V559D. Dasatinib potently inhibited the proliferation of cells (IC50 of 54 nM) and the phosphorylation of KIT in the Ba/F3 KITL576P cells. Imatinib and nilotinib also inhibited the cell proliferation with an IC50 of 253 nM and 185 nM, respectively (Fig. 5a, Table V). Apoptotic studies showed that dasatinib induced apoptosis at a concentration of 100 nM, while imatinib and nilotinib showed increased apoptosis at 1,000 nM (Fig. 5c). Immunoprecipitation of treated cell lysates showed that dasatinib distinctly decreases KIT phosphorylation at 100 nM and the phosphorylation is completely absent at 1,000 nM and above (Fig. 5e). In comparison, the imatinib treated cells showed a decrease of phosphorylation at 1,000 nM. Cells treated with nilotinib showed a decrease in KIT phosphorylation at 100 nM and further decrease in phosphorylation at the doses of 1,000 nM and above.  Figure 5.  Figure 5.  Open in figure viewer In vitro drug sensitivity for imatinib, dasatinib and nilotinib of Ba/F3 stable transformants expressing KITL576P and KITV559D: (a,b) proliferation inhibition, (c,d) increased apoptosis, and (e,f) decrease in KIT phosphorylation by immunoprecipitation. Table V. Growth-Inhibitory Drug Effects in Ba/F3 Transformants Containing KIT Mutants Using 3 Kinase Inhibitors, Results Presented by IC50 (nM)  \tBa/F3 KITL576P\tBa/F3 KITV559D Imatinib\t253\t27 Dasatinib\t54\t22 Nilotinib\t185\t34 In comparison, all 3 drugs potently inhibited the cell proliferation of the Ba/F3V559D cells (Fig. 5b). The IC50 for imatinib, dasatinib and nilotinib were 27 nM, 22 nM and 34 nM, respectively (Table V). Apoptotic assays showed significantly increased apoptosis at drug concentrations of 100 nM and above (Fig. 5d). A distinct decrease in KIT phosphorylation by immunoprecipitation was identified with imatinib and nilotinib treatment at 100 nM (Fig. 5f). Dasatinib treated cells showed a mild decrease in phosphorylation even at 10 nM with a marked decrease at 100 nM and above. Discussion  The KIT receptor plays an important role in normal melanocyte development and proliferation.17 KIT is expressed in the junctional component of benign melanocytic nevi and in situ malignant melanomas, but tends to be reduced or lost with tumor progression to invasive and metastatic melanoma6—an observation that suggested that KIT might act as a tumor suppressor gene. One hypothesis is that since KIT and its ligand regulate normal melanocyte proliferation, high concentration of the protein might prevent the development of vertical growth and inhibit melanoma tumor progression. As a consequence, KIT expression was not considered to have a diagnostic utility, but its decreased expression paralleling tumor progression was implicated in the development of tumorigenic melanoma. Furthermore, KIT expression is lost in most human melanoma cell lines18 and in one study their growth appeared to be inhibited by adding KIT ligand to the media.19 These studies suggest that KIT loses its responsiveness to its ligand and their in vitro proliferative activity becomes independent of KIT signaling.  However, the role of KIT in melanomagenesis is likely more complex, since a subset of melanomas have been found to maintain KIT overexpression, even in metastatic lesions, and a few isolated reports have identified activating KIT mutations in melanomas. Went et al. identified a KIT exon 11 L576P mutation in one of 2 melanomas which were immunohistochemically positive for KIT.8 This study included 39 melanomas, 35% of cases showed some degree of KIT reactivity, with 15% showing an intense staining pattern. A follow-up study from the same group tested 10 additional KIT positive uveal melanomas, but did not identify any KIT mutations.20 In a larger series, Willmore-Payne et al.9 studied 100 malignant melanomas and found a 29% rate of KIT positivity by IHC, ranging from weak to strong. Their study did not include any mucosal melanomas, but no other information as to tumor location and/or sun-exposure was specified. The junctional and the in situ component of all melanomas showed strong KIT staining, but the invasive component was consistently negative for KIT expression. Only 2 cases of metastatic melanoma had strong immunoreactivity for KIT. However, the matching primary tumors were not available to assess the status of KIT expression in the invasive component. Both of these tumors had a KITL576P substitution mutation in the juxtamembrane domain of KIT. This study suggested that KIT activating mutations have a low incidence in malignant melanomas, of about 2%. A recent immunohistochemical study on primary anorectal melanomas,21 identified KIT expression in 12 of 16 cases tested, with a diffuse staining pattern in 38% of tumors. However no mutational analysis was provided in these KIT positive primary tumors.  Our findings indicate that anal melanomas lack BRAF mutations and infrequently show NRAS mutations (5%), but instead show a higher rate of KIT activating mutations compared to non-acral cutaneous melanomas.9 The overall incidence of KIT immunoreactivity in anal melanomas was 20%, defined as strong and diffuse pattern of staining. We found a good correlation between KIT immunoreactivity and the presence of KIT gene alterations at the molecular level, with a 75% rate of KIT mutations amongst anal melanoma expressing strong KIT protein by immunohistochemistry. Although more cases need to be tested, true KIT gene amplification does not appear to be a common event in anal melanomas, however modest increased copy number of the KIT gene is seen in up to one third of the cases. KIT activating mutations and increased copy number were not mutually exclusive, as one anal melanoma showing an increased KIT copy number by FISH, in addition had a KITL576P substitution mutation. Another important observation from this study is that KIT protein phosphorylation, as detected in our case no. 19, can be seen in tumors with strong and diffuse KIT immunoexpression, in the absence of identifiable KIT/PDFGRA mutations or increased copy number changes, suggesting that kinase inhibition might be still beneficial in these cases. These findings recapitulate closely the molecular phenotype of KIT/PDGFRA wild-type GISTs, which are also managed by specific KIT receptor tyrosine kinase inhibitors.  The KITL576P mutation in the juxtamembrane domain of KIT is not unique for melanoma and has been previously described in gastrointestinal stromal tumors (GIST).22 In our experience of more than 500 GIST cases, the incidence of L576P mutation is <1%, which is typically heterozygous and lacks a specific clinical phenotype. Only 1 of these L576P mutated GIST patients were treated with imatinib for metastatic disease and developed secondary resistance to the drug after 13 months duration.12 Of interest is that the 2 melanoma cases reported by Willmore-Payne9 harboring a L576P substitution, lacked a normal allele, consistent with a homozygous deletion. The significance of this finding remains to be further elucidated, since all our KIT mutant anal melanomas had a heterozygous pattern. Also all the KIT mutations so far reported in melanoma patients were somatic, with no familial cases of melanomas or GIST being associated with this particular juxtamembrane mutation.  The in vitro drug testing revealed that KITL576P mutation is most sensitive to the dual SRC/ABL kinase inhibitor, dasatinib. Dasatinib is an orally active, small molecule ATP-competitive, which effectively inhibits the cell proliferation of nearly all imatinib-resistant BCR-ABL isoforms.23 Due to the structural homology of KIT activation loop mutants to imatinib-resistant BCR-ABL isoforms, dasatinib was shown to inhibit KIT kinase activity, including in imatinib-resistant KIT activation loop mutants.24 The KITL576P mutant was also sensitive to imatinib and nilotinib, but requiring at least a 10-fold higher dose compared to the imatinib-sensitive KITV559D mutation.  More recently a genetic classification of melanomas was put forward based on their anatomic location and ultraviolet light exposure.10 Thus, the most common type of melanoma, occurring on skin without CSD, frequently harbors either a mutation in BRAF or NRAS, suggesting a possible therapeutic intervention targeting the RAS-RAF-ERK and PI3K pathway. In contrast, melanomas on skin with CSD show frequent increases in the copy number of CCND1, and would be less likely responsive to BRAF inhibitors, such as sorafenib. The incidence of activating KIT mutations might also be dependent to the degree of sun exposure, since it varied from 2% in the study by Willmore-Payne,9 which did not include any mucosal cases, to 13% in our study of anal melanomas.  In summary, our findings show an increased prevalence of activating KIT mutations in a specific subset of mucosal melanomas, adding another layer of complexity to the genetic heterogeneity so far described in melanomas. This finding is significant for choosing the appropriate targeted therapy, and according to our findings L576P mutant is responsive to specific kinase inhibitors, particularly dasatinib. Additional studies, including other subtypes of mucosal and acral melanomas are needed to investigate if this increased incidence of KIT activation is intrinsic to anal melanoma or pertains to other melanomas with limited or no ultraviolet light exposure.  We recently identified a KITexon11mutation in an anorectal melanoma of a patient who had an excellent response to treatment with imatinib.To determine the frequency of KIT mutations across melanoma subtypes, we surveyed a large series of tumors. Experimental Design: One hundred eighty-nine melanomas were screened for mutations in KIT exons 11, 13, and 17. KIT copy number was assessed by quantitative PCR. Asubset of cases was evaluated for BRAF and NRAS mutations. Immunohistochemistry was done to assess KIT (CD117) expression. Results: KIT mutations were detected in 23% (3 of 13) of acral melanomas, 15.6% (7 of 45) of mucosal melanomas, 7.7% (1 of 13) of conjunctival melanomas, 1.7% (1 of 58) of cutaneous melanomas, and 0% (0 of 60) of choroidal melanomas. A lmost all the KIT mutations were of the type predicted to be imatinib sensitive.There was no overlap with NRAS mutations (11.1% of acral and 24.3% of mucosal tumors) or with BRAF mutations (absent in mucosal tumors). Increased KIT copy number was detected in 27.3% (3 of 11) of acral and 26.3% (10 of 38) of mucosal melanomas, but was less common among cutaneous (6.7%; 3 of 45), conjunctival (7.1%; 1 of 14), and choroidal melanomas (0 of 28). CD117 expression, present in 39% of 105 tumors representing all melanoma types, did not correlate with either KIT mutation status or KITcopy number. Conclusions: Our findings confirm that KIT mutations are most common in acral and mucosal melanomas but do not necessarily correlate with KIT copy number or CD117 expression. Screening for KIT mutations may open up new treatment options for melanoma patients. The incidence of malignant melanoma has risen dramatically over the past 50 years, yet current therapies for advanced disease provide only modest benefit (1). There has been recent progress in unraveling the molecular genetics of this cancer, with the emergence of several subtypes that are distinguished by their site of origin and tumor genotype (2). For example, Curtin and colleagues (3) observed that 81% of melanomas arising in skin without chronic sun-induced damage have an oncogenic BRAF or NRAS mutation, whereas such mutations are less common in chronic sun-induced damage melanomas, acral melanomas, and mucosal melanomas (1). In contrast, melanomas arising in chronic sun-induced damage skin and acral sites are more likely to have increased CCND1 copy number; loss of CDKN2A or gain of CDK4 are frequent alterations in mucosal and acral melanomas (4, 5). Melanomas with oncogenic mutations in KIT have been observed in several recent studies (4, 6 – 9). Although such mutations seem to be more rare than BRAF and NRAS mutations, they may reflect the important role of KIT tyrosine kinase in melanocyte development (10). KIT mutations seem to be more common in mucosal and acral melanomas compared with cutaneous melanomas and are sometimes accompanied by an increase in KIT copy number (4, 6 – 9). Importantly, most of the reported mutations are of the type predicted to be sensitive to KIT kinase inhibitors. Success in the treatment of gastrointestinal stromal tumors (GIST) with imatinib (Gleevec; Novartis Pharma) has led to speculation that KIT-mutant melanoma might also be managed by this approach. Increases in KIT copy number have also been observed in melanoma (4). In some cases, there is an accompanying mutation, but in others, none is detected. Whether a KIT inhibitor would be effective against tumors that have amplified KIT but no detectable mutation is another question of great interest. A phase II trial of imatinib for patients with unresectable acral or mucosal melanoma is ongoing at Dana-Farber Cancer Institute. Prior trials have shown that kinase inhibitors such as imatinib have little or no activity in unselected melanoma patients (11– 14). Therefore, the current trial requires that a KIT gene mutation be documented in the tumor before a patient is treated. The first patient treated on this trial, a woman with a recurrent rectal melanoma containing a KIT exon 11 insertion/ duplication, showed a partial response to 400 mg/d of Human Cancer Biology Authors’ Affiliations: 1 Oregon Cancer Institute, 2 Department of Pathology, and 3 Division of Hematology and Oncology, Oregon Health & Science University, and 4 VAMedical Center, Portland, Oregon; 5 Department of Medical Oncology, DanaFarber Cancer Institute, Boston, Massachusetts; 6 Department of Pathology, Cleveland Clinic, Cleveland, Ohio; 7 Department of Pathology and Laboratory Medicine, Emory University, Atlanta, Georgia; and 8 Department of Pathology, VA Medical Center, Palo Alto, California Received 3/5/08; revised 5/23/08; accepted 6/15/08. The costs of publication of this article were defrayed in part by the payment of page charges. This article must therefore be hereby marked advertisement in accordance with 18 U.S.C. Section 1734 solely to indicate this fact. Requests for reprints: Christopher L. Corless, Oregon Health and Science University, Department of Pathology (mailcode L471), 3181 Southwest Sam Jackson Park Road, Portland, OR 97239. Phone: 503-494-6834; Fax: 503-494- 6787; E-mail: corlessc@ohsu.edu. F 2008 American Association for Cancer Research. doi:10.1158/1078-0432.CCR-08-0575 www.aacrjournals.org 6821 Clin Cancer Res 2008;14(21) November 1, 2008 Research. Downloaded from clincancerres.aacrjournals.org on May 28, 2017. © 2008 American Association for Cancer imatinib after just 4 weeks of therapy (15). A positron emission tomography scan confirmed marked reduction in tumor metabolic activity, consistent with KIT inhibition. The excellent response in this patient with a KIT-mutant melanoma suggests that mutation screening may be critical in identifying patients who may benefit from kinase targeted therapy. As there are only limited data on KIT alterations across the major subtypes of malignant melanoma, we surveyed a large collection of cases and, for comparison, examined BRAF and NRAS, as well. Our results confirm that a significant fraction of acral and mucosal melanomas harbor KIT gene mutations and/or increased KIT copy number. We also observed differences between these two melanoma subtypes with respect to the frequency of detectable BRAF and NRAS mutations. Consistent with previous reports (3, 4, 16– 22), we observed NRAS mutations in both of these subtypes, whereas BRAF mutations were detected in acral but not mucosal melanomas. In our series, mutations in KIT, BRAF and NRAS were mutually exclusive, indicating that each genotype may represent a distinct molecular disease subtype. This observation has clear implications for the use of targeted therapeutics in the future. Materials and Methods Melanoma samples. Blocks of formalin-fixed, paraffin-embedded melanoma tissue were obtained from the Pathology department archives of Oregon Health & Science University, the University of Washington, and Emory University, in accordance with Institutional Review Board – approved protocols at each institution. BRAF and NRAS mutation status was previously reported for 44 of the cutaneous melanomas and 62 of the choroidal melanomas, but these cases had not been analyzed for KIT (23). All of the melanomas were from adult patients, who ranged from ages 22 to 86 years. Most of the tumors analyzed were primary lesions, but recurrences or metastases (lymph node, small bowel, liver) were studied in cases where the primary was unavailable. Acral and mucosal melanomas from patients enrolled in the 06-056 phase II trial at DanaFarber Cancer Institute were included in the study. A KIT mutation in one of these cases has been detailed in a prior case report (15). DNA preparation. Tumor-rich areas (>80%) were scraped from 5-Am unstained sections by comparison with a H&E-stained slide, and genomic DNA was extracted using a Qiagen mini-kit (Qiagen, Inc.) in accordance with the manufacturer’s directions. As melanin pigment can copurify with genomic DNA and inhibit PCR reactions, we incubated samples that had obvious melanin contamination, or failed to amplify on initial testing, with Chelex-100 (Bio-Rad). Specifically, purified DNA was incubated for 10 min at room temperature with an equal volume of 5% Chelex-100 suspension equilibrated in Qiagen AE buffer. The mixture was heated to 95jC for 2 min, allowed to cool, and then the Chelex-100 resin was pelleted in a microfuge and the supernatant DNA used for PCR reactions. Approximately 70% of DNA samples that initially failed to amplify were recovered by this approach. Mutation screening. Screening for mutations in KIT exons 8, 9, 11, 13, and 17, and BRAF exons 11 and 15 was done using a combination of PCR amplification and denaturing high performance liquid chromatography, as described previously (23, 24). All suspected mutations were confirmed by bidirectional sequencing on an ABI 3130 automated sequencer. Mutations in NRAS exons 1 and 2 were identified by melting curve analysis of PCR amplicons, measuring fluorescence energy transfer between an anchor probe and a wild-type reporter probe that covered the codons of interest. PCR primers and probes were prepared by Integrated DNA Technologies.9 Primers and probes for the exon 2 assay were exactly as described by Nikiforova et al. (25). Primers and probes for the exon 1 assay were as follows: NRAS 1 forward: CACTAGGGTTTTCATTTCCATTG NRAS 1 reverse: TCCTTTAATACAGAATATGGGTAAAGA NRAS 1 anchor: TGAAATGACTGAGTACAAACTGGTGGTGGTTFluorescein NRAS 1 reporter: LC640-AGCAGGTGGTGTTGGGA-C3 blocker Twenty-five microliter reactions were carried out in a Roche LC480 Lightcycler (Roche) using Roche Probe Master mixand including 80 nmol/L of the forward primer, 800 nmol/L of reverse primer, and 800 nmol/L each of the anchor and reporter probes. The PCR conditions for both assays were as follows: 95jC for 10 min, followed by 50 cycles of 95jC for 20 s/60jC for 20 s/72jC for 20 s, then 95jC for 1 min and 45jC for 2 min. Melting curve analyses were done in the same instrument. Based on dilution analyses, the sensitivity of the NRAS exon 1 and exon 2 assays was f15% mutant allele each. Melanoma samples showing aberrant peaks by melting curve analysis were sequenced bidirectionally to confirm the presence of a mutation. Reverse transcription-PCR Assay for KIT copy number. KIT gene copy number was assessed in quantitative real-time PCR reactions by comparison with coamplified GAPDH. All primers and probes were purchased from Integrated DNA Technologies. The KIT exon 17 primers were as follows: forward AAAGATTTGTGATTTTGGTCTAGC, and reverse GAAACTAAAAATCCTTTGCA. The KIT exon 17 dual-labeled hydrolysis probe was FAM-TGTGGTTAAAGGAAACGTGAGTA-IABlkFQ. The glyceraldehyde-3-phosphate dehydrogenase (GAPD) primers were as follows: forward CACTAGGCGCTCACTGTTCT, and reverse GCGAACTCACCCGTTG. The GAPDH hydrolysis probe was TxRed-TGGGGAAGGTGAAGGTCGGA-IAbRQSp. The KIT exon 7 primers were as follows: forward AGGTAGAAACTGAAAAAGACATGC, and reverse GCCTTTTAATCTCGTTAGATGAAGT. The KIT exon 7 dual-labeled hydrolysis probe was FAM-AGGCATGCTATCCACAGGTGATTG-IABlkFQ. PCR reactions were done in duplicate in a Roche LC480 Lightcycler, with a 20 AL total volume, 50 ng genomic DNA, 200 nmol/L primers 9 http://www.idtdna.com Translational Relevance Recent studies have revealed the presence of KIT gene alterations in some cases of malignant melanoma. These alterations (activating mutations and/or increased copy number) may be particularly common in melanomas arising in acral and mucosal sites. Treatment options for these types of melanomas are limited, but oncogenic mutations of KIT are good targets for clinically available tyrosine kinase inhibitors.We recently observed a dramatic response to imatinib treatment in a patient with a KITmutant rectal melanoma. This led us to survey the type and frequency of KIT alterations in 189 melanomas representing all the major subtypes. We have confirmed that KIT mutations are more common among acral (23%) and mucosal (15.6%) melanomas than among cutaneous (1.7%) and conjunctival (7.7%) tumors. Increased KIT copy number was also more common in acral (27%) and mucosal (26%) melanomas than in the other subtypes. Interestingly, not all tumors with KIT gene alterations were immunopositive for CD117. Our findings suggest that routine screening of acral and mucosal melanomas for KIT alterations may open up new treatment avenues for patients with these tumors. Human Cancer Biology Clin Cancer Res 2008;14(21) November 1, 2008 6822 www.aacrjournals.org Research. Downloaded from clincancerres.aacrjournals.org on May 28, 2017. © 2008 American Association for Cancer and probes, and 1  Probes Master Mix(Roche). The PCR conditions were 1 cycle of 95jC for 10 min, and 50 cycles of 95jC for 10 s, and 60jC for 20 s. The second derivative maximum algorithm included with the instrument software was used to determine cycle threshold (Ct) values for each target gene. Relative copy numbers were calculated using the DDCt method, where Ct is the threshold cycle for amplification. For each sample, DCt for KIT versus GAPDH was calculated as DCt = Ct(KIT) - Ct(GAPDH). The DCt value for each experimental test sample was calibrated to a reference pool of genomic DNA prepared from 9 samples of formalinfixed, paraffin-embedded normal tissue that included colon, kidney, liver, prostate, and spleen (all archived 4 y or less), using the formula DDCt = DCt(test sample) - DCt(reference pool). Relative DNA copy number was calculated using the formula 2- DDCt. Relative copy numbers were converted to absolute copy numbers by assigning a value of 2 (diploid) to the reference pool and multiplying the relative copy number of test samples by a factor of 2. Pilot reactions established that KIT Cts between 26 and 34 cycles were in the linear range of amplification. Samples that did not amplify by 34 cycles were deemed to be of insufficient quality. Increased KIT copy number was determined by comparison to nine individual formalin-fixed, paraffin-embedded normal tissue samples. The mean and SDs of the 9 samples, based on quadruplicate measurements, was 2.25 F 0.68 KIT copies relative to GAPDH. The threshold for increased KIT copy number was set relative to these normal samples using the 95% confidence level according to Chebychev’s inequality, with the formula mean F (4.47   SD). This threshold was 5.29 copies of KIT relative to GAPDH. Fluorescence in situ hybridization for KIT. To validate the results of the reverse transcription-PCR for KIT copy number, fluorescence in situ hybridization (FISH) was done on 5-Am sections of paraffin-embedded melanoma tissue using a cocktail of three probes to chromosome 4 (Vysis LSI 4q12 tricolor; Abbott Laboratories). These probes cover the genes SCFD2 (Spectrum green), LNX (Spectrum orange), and KIT (Spectrum aqua; also includes a portion of the neighboring gene PDGFRA). The protocol was as previously described (26). Briefly, slides were baked overnight at 60jC, deparaffinized, and then dehydrated. Pretreatment washes included distilled water (10 mi), 2  SSC (3 min), and 10 mmol/L citric acid buffer (pH 6.0; 80jC, 45 min). Slides were protease treated at 37jC for 10 min, dehydrated, and air dried. After addition of the probe, the slides were denatured for 5 min at 73jC and hybridized for 14 to 18 h at 37jC. After washing, the slides were then counterstained with 4¶,6-diamidino-2-phenylindole and visualized on a fluorescence microscope. CD117 immunohistochemistry. Deparaffinized 5-Am sections of tumor were heated in Dako Target Retrieval Solution (pH 6.0; Dako USA) for 10 min in a Russell Hobbs pressure cooker (model RHNPC800P) and allowed to cool. Immunostaining for CD117 (KIT) was done using the Dako polyclonal rabbit antibody (Dako A4502) at 1:400 dilution, followed by a standard avidin-biotin detection protocol using diaminobenzidine. Hematoxylin-counterstained slides were cover-slipped and examined for the intensity of staining (0, 1+, or 2+). Results KIT mutations. Among 189 melanoma cases that were fully screened for mutations in KIT exons 11, 13, and 17, the frequency of KIT mutations was 23% for acral tumors, 15.6% for mucosal tumors, 7.7% for conjunctival tumors, 1.7% for cutaneous tumors, and 0% for choroidal tumors (Table 1). Most of the 45 mucosal tumors were sinonasal (n = 29), but 7 arose in the oral cavity, 7 in the anorectal area, and 2 in the vulva/vagina. Interestingly, the mutation frequency was higher among tumors of the anorectum/vulva/vagina (4 of 9; 44.4%) than among tumors arising in the head and neck (3 of 36; 8.3%; Table 1). However, more cases will need to be collected and analyzed to see if this trend is significant. Point mutations were the most common, resulting in substitutions in exon 11 (W557R, K558N, V559A, V559D, L576P—4 cases) or exon 17 (Y823D; Table 2). However, one acral melanoma had an in-frame exon 11 deletion (EVQWKV 554-559), and we have previously observed an exon 11 insertion/duplication in a rectal melanoma (15). All of the KIT mutations identified in our cases also occur in GISTs. As KIT exon 9 is another site of mutation in GISTs (f10% of tumors), we screened 148 melanomas, including 6 acral and 27 mucosal tumors, for mutations in this exon, but none were found. We also examined KIT exon 8, in which mutations occur in acute myelogenous leukemias but did not observe any in 99 melanomas, including 3 acral and 25 mucosal melanomas. KIT copy number. KIT gene copy number (chromosome 4) was assessed by quantitative real-time PCR for exon 17 sequence and compared with coamplified GAPDH (chromosome 12) as an internal control. In pilot assays, comparable results were obtained using h-actin (ACTB, chromosome 7) as the internal control (data not shown); however, GAPDH was selected over ACTB because this chromosome is commonly overrepresented in melanoma (3, 27, 28). Differences in the amplification thresholds (Ct) of KIT and GAPDH were compared with those in a reference pool of normal genomic DNA prepared from 9 samples of formalin-fixed, paraffinembedded benign tissue. The assay was validated by several approaches. (a) Substitution of a KIT exon 7 amplicon for exon 17 in the assay had no significant effect (data not shown). (b) A KIT-mutant melanoma cell line known to have increased copies of KIT by array CGH showed elevated copy number with the PCR assay.10 (c) FISH for KIT was done on five of the melanomas in the study. FISH signals versus reverse transcription-PCR copy number for the five cases were as follows: 2 versus 2.6; 3 versus 5.5; 5 to 7 versus 5.0; 7 to 8 versus 7.8; and ‘‘too numerous to count’’ versus 28.8. Figure 1 illustrates the FISH results for two of the cases. Increased KIT copy number was identified by reverse transcription-PCR in over one quarter of the acral and mucosal melanomas (Table 3). In 2 of 3 acral tumors and 7 of 10 mucosal tumors, no accompanying KIT mutation was detected. Extra copies of KIT were much less common among cutaneous and conjunctival tumors and were not observed in any of the choroidal melanomas tested. BRAF mutations. We previously reported BRAF mutations in 36.4% (16 of 44) of cutaneous melanomas and 0% (0 of 62) uveal melanomas (23). In this study, BRAF mutations were identified in 26.7% of conjunctival melanomas and 16.7% of acral melanomas but not in any of the mucosal melanomas (Table 1). All were the common V600E substitution in exon 15, with the exception of a single conjunctival tumor with V600R. In contrast, there were no BRAF exon 15 mutations among 47 mucosal melanomas, regardless of site. Twenty-sixmucosal cases were additionally screened for mutations in BRAF exon 11 and none were found. NRAS mutations. In a previous study, we observed that mutations in NRAS exons 1 and 2 were uncommon in cutaneous melanomas (3.7%; 1 of 27) and absent in choroidal 10 F.S. Hodi, manuscript submitted. KIT Mutation and Copy Number in Melanoma Subtypes www.aacrjournals.org 6823 Clin Cancer Res 2008;14(21) November 1, 2008 Research. Downloaded from clincancerres.aacrjournals.org on May 28, 2017. © 2008 American Association for Cancer melanomas (0 of 47; ref. 23). Our expanded series of 60 cutaneous melanomas yielded a slightly higher frequency of 11.7% (Table 1). However, NRAS mutations were considerably more common (24.7%) among 37 mucosal melanomas analyzed. As expected from the literature, the affected codons were 12 (G12C, G12D, G12V), 13 (G13R), and 61 (Q61K, Q61L, Q61R). One interesting trend was that exon 1 mutations dominated among the mucosal melanomas (8 of 9 cases), whereas exon 2 mutations were more common in the cutaneous tumors (6 of 7 cases). There was one NRAS mutation among 9 acral tumors analyzed, and none in 11 conjunctival melanomas. Correlation of KIT alterations and CD117 expression. Immumunohistochemistry for CD117 (KIT) was done on subsets of the melanoma subtypes, including examples with KIT, BRAF, or NRAS mutation, and cases with increased KIT copy number. Staining was graded as negative (0), weak/moderate (1+), or strong (2+). The results, tabulated in Table 4, showed no apparent correlation between CD117 and KIT mutation status or increased KIT copy number. Occasional tumors with a BRAF or NRAS mutation had readily detectable CD117. Strong expression was also observed in many of the wild-type tumors (21.4%), particularly among the choroidal melanomas (35.7%). Thus, CD117 expression is not a reliable indicator of KIT genotype. Discussion Went and colleagues (7) were the first to discover an activating KIT mutation (L576P) in a case of melanoma—one of two CD117-positive tumors that they sequenced. WillmorePayne et al. (9, 29) followed up on this observation in two studies using high-resolution melting curve analysis to screen melanomas for mutations in KIT exons 9, 11, 13, and 17. There were 153 melanomas (nonmucosal but otherwise of unspecified origin) described in the two studies; however, mutation screening was actually limited to 35 cases that were CD117- positive. Among these, there were three cases with an L576P substitution. In 2006, Curtin and coworkers (4) surveyed 95 melanomas of various types and found that KIT mutations were Table 1. Summary of mutations in melanoma subtypes Gene Acral Mucosal This study Other studies This study Other studies KIT 23% (3/13) 11% (3/28; ref. 4) 15.6% (7/45) 15% (3/15; ref. 6) 21% (8/38; ref. 4) 22.2% (4/18; ref. 30) BRAF 16.7% (2/12) 9.5% (2/21; ref. 42) 0% (0/45) 0% (0/13; ref. 17) 15% (2/13; ref. 18) 3% (1/38; ref. 4) 15.4% (2/39; ref. 19) 5.6% (2/36; ref. 22) 21% (6/28; ref. 4) 5.9% (1/17; ref. 16) 23% (7/30; ref. 3) 9.5% (2/21; ref. 19) 33% (5/15; ref. 20) 11% (2/19; ref. 3) NRAS 11.1% (1/9) 3.6% (1/28; ref. 21) 24.3% (9/37) 5% (1/19; ref. 3) 10% (3/30; ref. 3) 13.9% (5/36; ref. 22) 47.4% (9/19; ref. 43) *Melanomas arising in chronically sun-damaged skin. cPreselected for CD117 positivity; sites not specified. Table 2. Summary of KIT mutations in melanoma Exon Mutation This study Other studies Total Imatinib sensitive? Refs 11 Del 554-5591 1 Yes Y553N 1 1 ? (4) W557R 1 1 2 Yes (30) K558N 1 1 Yes V559A 1 2 3 Yes (4) V559D 1 1 Yes N566D 1 (with K642E) 1 ? (4) V569G 1 1 ? (30) L576P 5 7 12 Yes (4, 7, 9, 29) Ins at 583 1 1 Yes (15) 13 R634W 1 1 ? K642E 9(1 with N566D) 9 Yes (4, 6, 30) 17 D816H 2 2 No Y823D 1 1 No 18 A829P 1 1 No (4) All Mutations 11 27 38 Human Cancer Biology Clin Cancer Res 2008;14(21) November 1, 2008 6824 www.aacrjournals.org Research. Downloaded from clincancerres.aacrjournals.org on May 28, 2017. © 2008 American Association for Cancer most common among the acral 11% (3 of 28) and mucosal 21% (8 of 38) tumors. Subsequent studies by Antonescu et al. (6) and Rivera et al. (30) have confirmed that KIT mutations are more common among rectal and head and neck melanomas, respectively (Table 1). In this study, we used denaturing high performance liquid chromatography to screen a total of 189 melanomas from all sites for mutations in three exons of KIT previously reported to harbor mutations (exons 11, 13, and 17). Included were 45 mucosal melanomas (36 head and neck and 9 anorectal/vulvar/vaginal tumors). Our results, summarized in Table 1, build substantially on the earlier reports. Combining our data with the previous studies, KIT mutations are present in 14.6% of acral, 19% of mucosal, 4.3% of cutaneous, and 7.7% of conjunctival melanomas, and are absent from choroidal melanomas. A subset of our tumors was also screened for mutations in KIT exon 8 and/or exon 9, Table 1. Summary of mutations in melanoma subtypes (cont’d) Conjunctival Cutaneous Choroidal This study Other studies This study Other studies This study Other studies 7.7% (1/13) No reports 1.7% (1/58) 0% (0/50; ref. 31) 0% (0/60) 0% (0/10; ref. 41) 16.7% (3/18; ref. 4)* 16.7% (1/6; ref. 29)c 6.9% (2/29; ref. 9)c 26.7% (4/15) 14.3% (3/21; ref. 43) 22.7% (5/22; ref. 44) 40% (2/5; ref. 45) 0% (0/11) No reports Fig. 1. FISH for KIT. A, acral melanoma with KIT V559D showing increased FISH signals (fused) for probes to 4q12. Signals ranged from 6 to 8 per nucleus; the copy number was 7.8 by reverse transcription-PCR. B, a sinonasal melanoma that was wild-type for KIT showed a large increase in KIT signals, but not for the nearby loci LNX (orange signal) and SCFD2 (green signal), suggesting a localized amplification. In one of the nuclei, only one KIT allele showed amplification, whereas a neighboring nucleus displayed two clusters of KIT signals. By reverse transcription-PCR this case had the highest KIT copy number (28.8) in the study. KIT Mutation and Copy Number in Melanoma Subtypes www.aacrjournals.org 6825 Clin Cancer Res 2008;14(21) November 1, 2008 Research. Downloaded from clincancerres.aacrjournals.org on May 28, 2017. © 2008 American Association for Cancer but no mutations were found. Of course, it remains possible that there are mutations in other exons of the KIT gene. We observed that BRAF mutations were relatively more common than KIT mutations in conjunctival tumors, although the number of tumors available for study was limited. The frequency of BRAF mutations that we observed is similar to other studies and parallels the well-established dominance of BRAF mutations in cutaneous melanomas (1 – 3). Among the cutaneous melanomas in our series, there was only one KIT mutation. Woenckhaus et al. (31) reported none among 50 cutaneous tumors. On the other hand, Curtin et al. (4) observed KIT mutations at a frequency of 16.7% among melanomas arising in chronically sun-damaged skin, in which BRAF mutations are comparatively uncommon (3). Although the relationship between sun damage and the likelihood of a KIT mutation in cutaneous melanomas deserves further investigation, the cutaneous tumors in our study were not selected on this basis. Rather, they are simply representative of cases referred to a tertiary care center, including patients with advanced disease. Targeting KIT is a proven strategy in the management of advanced GIST (32– 34). Several observations suggest that KIT kinase inhibitors might also serve to control the growth of selected melanomas. First, KIT kinase activity is necessary for the development of melanocytes (10), just as it is for the development of the interstitial cells of Cajal, to which GISTs are thought to be related. As in GIST, oncogenic activation of KIT may serve as an early signal for neoplastic growth. Second, Antonescu and colleagues (6) observed KIT phosphorylation on an immunoblot of a rectal melanoma extract. Third, the great majority of the KIT mutations observed in melanoma are of the type known to be sensitive to imatinib and other KIT inhibitors (Table 3). For example, Antonescu et al. (6) showed that the V559D and L576P mutant forms of KIT are inhibited both by imatinib and dasatinib when expressed in Ba/F3 cells. Fourth, a melanoma cell line harboring a KIT exon 11 mutation was highly sensitive to growth inhibition by imatinib, whereas nonmutant cell lines were not.11 Finally, we recently reported the dramatic response to imatinib therapy of a mucosal melanoma patient with a KIT exon 11 insertion/duplication (15). Of note, in the current study, we found that the tumor in that case also had an increased KIT copy number (12.6). Imatinib has been essentially ineffective in trials of unselected melanoma patients (11, 13, 14, 35). Likewise, another KIT inhibitor, PKC412, showed little activity in melanoma (12). However, Becker et al. (36) observed in a recent editorial that their imatinib trial was underpowered to detect activity in the subtypes of melanomas most likely to harbor a KIT mutation. It is also notable that in the imatinib trial reported by Eton et al. (35), there was an acral melanoma patient who had a nearly complete response. As noted above, the statistical chance of a KIT mutation in this patient’s tumor was 14.6%. Thus, the responding patient that we have reported may not be unique and routine screening of KIT may become important for patients with advanced melanoma. As KIT mutations are much less common in melanoma than in GIST, the question arises as to what would be the most effective screening approach. After the logic that KIT should be highly expressed in KIT-mutant tumors, Willmore-Payne and colleagues (9, 29) focused their screening only on CD117- positive cases. Correspondingly, Antonescu et al. (6) observed that all three of their KIT-mutant melanomas were 4+ positive for CD117, whereas nearly all of the other tumors in their series were either 1+ or negative. There is, however, evidence from the GIST literature that CD117 expression and KIT genotype do not always correlate. For example, KIT mutations are detected in 16% of CD117-negative GISTs (37), yet such tumors are still responsive to imatinib. Chirieac and coworkers (38) confirmed that there is no relationship between CD117 expression and imatinib response among GIST tumors. It is perhaps not surprising then that several of our KIT-mutant melanomas lacked detectable expression of CD117 (Table 4). Using the identical antibody, the same was true of 1 of 4 KIT-mutant melanomas studied by Rivera et al. (30), and of several KITmutant melanomas examined by Curtin and colleagues (4). The latter group indicated that most of their cases became CD117 ‘‘positive’’ when a 10-fold higher concentration of the antibody was used, but false positive staining at such high concentrations of this antibody is well-documented in the literature (7, 39). Like Rivera et al. and Curtin et al., we observed strong CD117 staining in some melanomas that were KIT wild-type (Table 4). We also observed it in one case with 11F.S. Hodi et al., manuscript submitted. Table 3. Summary of KIT gene copy number % increased KIT number Increased KIT number with KIT mutation Increased KIT number without KIT mutation Acral 27.3% (3/11) 1 2 Mucosal 26.3% (10/38) 3 7 Cutaneous 6.7% (3/45) 1 2 Conjunctival 7.1% (1/14) 0 1 Choroidal 0% (0/28) 0 0 Table 4. Genotype versus CD117 expression in nonchoroidal melanomas Genotype CD117 staining 0 1+ 2+ Total KIT mutation 3 1 2 6 Increased KIT copy number 6 2 1 9 Normal KIT copy number 40 16 14 70 BRAF mutation 10 2 1 13 NRAS mutation 5 2 0 7 No mutation: Nonchoroidal 18 4 6 28 Choroidal 18 915 42 Human Cancer Biology Clin Cancer Res 2008;14(21) November 1, 2008 6826 www.aacrjournals.org Research. Downloaded from clincancerres.aacrjournals.org on May 28, 2017. © 2008 American Association for Cancer a BRAF mutation. We conclude that CD117 immunohistochemistry is not a reliable indicator of KIT genotype, and that restricting mutation screening to CD117-positive melanomas would risk missing mutant cases. In analyzing three KIT-mutant melanomas, Willmore-Payne et al. (9) found that one tumor had a modest increase in KIT copy number, as assessed by FISH (2.88-fold). The remaining tumors were homozygous mutants based on comparison of FISH and DNA sequence. Similarly, Antonescu and coworkers (6) observed increased KIT copy number (2.5-fold) in 1 of 2 mutant cases examined by FISH. Selective overrepresentation of chromosome 4q12, which includes KIT, was measured by array CGH in a subset of cutaneous melanomas evaluated by Bastian et al. (27), and in acral, mucosal, and cutaneous melanomas (from chronically sun-damaged skin) studied by Curtin et al. (4). To simplify the assessment of KIT copy number, we developed a quantitative real-time PCR assay for use on formalin-fixed, paraffin-derived tumor DNA. This assay has the advantage that it can be done on the same material used to screen for KIT mutations; however, it may be less sensitive than other approaches because the copy number of the reference gene (GAPDH) may itself vary in some cases. With this in mind, we identified a number of melanomas with increases in KIT, in one case up to 28.8 copies. Included were cases with KIT mutations, but most were wild-type tumors. It is intriguing that the mucosal melanoma in the patient that we previously reported as responding dramatically to imatinib therapy had both a KIT mutation and increased KIT copy number. The usefulness of our assay will be examined in ongoing trials of KIT kinase inhibitors (imatinib or sunitinib) for patients with acral or mucosal melanoma. Although the significance of increased KIT copy number with regard to the treatment of melanoma remains to be determined, the great majority of KIT mutations that have been identified in this cancer are potentially sensitive to imatinib and other KIT inhibitors. Table 2 summarizes all of the mutations reported to date, including the current study. The two most common mutations (L576P and K642E) together account for 55% of the cases; both are inhibited by imatinib in vitro and predict for good imatinib response in GIST (32– 34). The only mutations likely to be insensitive to available inhibitors are those affecting the activation loop encoded by exons 17 and 18. Fortunately, only 4 of 38 cases (10.5%) reported thus far had a mutation in this domain (Table 2). An alternative treatment strategy for such cases might be a MEK1 inhibitor. BRAF-mutant melanoma cell lines are highly sensitive to MEK1 inhibition (40) but some RAS-mutant and FGFR1-mutant melanoma cell lines show similar sensitivities (28). Molecular correlation studies in the ongoing trials with BRAF and MEK1 inhibitors will prove very interesting in this regard. In summary, we have conducted the largest survey to date of KIT genotypes across various melanoma subtypes. Our findings confirm that acral and mucosal melanomas are the subtypes most likely to harbor KIT mutations and/or increased KIT copy number. However, our results also indicate that this oncogene may play a role in occasional cases of conjunctival and cutaneous melanoma. The mutations do not overlap with mutations in BRAF or NRAS, but they are sometimes associated with increased KIT copy number. We also observed that NRAS mutations are even more common in mucosal melanomas than KIT mutations, and this should be borne in mind in future efforts to match patient tumors with targeted therapeutics  The pathogenesis of most gastrointestinal stromal tumors (GISTs) results from activating mutations of KIT or of platelet-derived growth factor receptor α (PDGFRA). More than 80% of GISTs express mutated, constitutively active KIT, and another 5% to 7% express mutated PDGFRA; 10% to 15% of tumors have no associated mutations in these kinases.1-3  Imatinib mesylate, a selective inhibitor of KIT and PDGFRA (and of platelet-derived growth factor receptor β [PDGFRB] and BCR-ABL kinase), has revolutionized the treatment of GIST; however, up to 14% of GISTs exhibit primary resistance to imatinib (defined as progression within 3 to 6 months of initiating therapy),4-6 and another 40% to 50% develop resistance within 2 years of beginning therapy (ie, secondary resistance).5,6 Sunitinib malate (SUTENT; Pfizer, New York, NY), another small-molecule tyrosine kinase inhibitor (TKI) with selectivity for KIT and PDGFRA (and for PDGFRB, all three isoforms of vascular endothelial growth factor receptor [VEGFR], FMS-like tyrosine kinase 3 [FLT3], colony-stimulating factor 1 receptor [CSF-1R], and glial cell line-derived neurotrophic factor receptor [rearranged during transfection; RET; Pfizer, New York, NY; data on file]),7-11 has demonstrated clinical benefit in phase I to phase III trials of patients with imatinib-resistant or -intolerant GIST.12,13 Sunitinib has been approved multinationally for the treatment of patients with GIST for whom prior imatinib therapy failed because of disease progression or drug intolerance.  GIST responsiveness to imatinib varies by primary KIT genotype; exon 11-mutant GISTs are more sensitive than exon 9-mutant or wild-type GISTs (ie, those that lack KIT or PDGFRA mutations).3,14,15 Exons 11 and 9 are the most common sites of KIT mutation in GIST (approximately 70% and 15% of tumors, respectively).3,14 Secondary kinase mutations are common in GISTs that exhibit secondary resistance but not in those that exhibit primary resistance.16,17 Secondary point mutations associated with imatinib resistance usually are located in the drug/adenosine triphosphate (ATP) binding pocket of the receptor (encoded by exons 13 and 14) or in the activation loop (encoded by exon 17).16-28 Two recent studies that used cell-based assays reported that sunitinib inhibited the kinase activity of KIT receptors that contained mutations in the drug/ATP binding pocket that confer resistance to imatinib.29,30 Because these mutations (ie, T670I and V654A [substitutions of isoleucine for threonine at position 670 and alanine for valine at position 654, respectively]) are commonly found in patients with GIST who have secondary imatinib resistance, the results provide a possible basis for sunitinib antitumor activity in patients with imatinib-refractory GIST.  To further explore the relationship between primary and secondary GIST kinase mutations and the response to sunitinib, we determined primary and secondary KIT or PDGFRA mutations in biopsied tissue from patients with imatinib-refractory GIST who received sunitinib as part of a phase I/II trial,12 and we correlated the presence of these mutations with clinical benefit. In addition, in vitro studies assessed the sensitivity of KIT and PDGFRA mutants to sunitinib and imatinib directly.  PATIENTS AND METHODS  Biopsies for genotype analyses were obtained from patients enrolled on a sunitinib phase I/II trial that was described in an earlier report of efficacy/safety results from the study.12 Patients were adults who had histologically confirmed metastatic/unresectable GIST and documented failure of imatinib caused by resistance or intolerance. Most patients (55 of 97) received sunitinib 50 mg/d in 6-week cycles that comprised 4 weeks on, followed by 2 weeks off, treatment. Additional information about methods is listed in the Appendix (online only).  RESULTS  Primary Tumor Genotype and Efficacy Tissue for pre-imatinib genotype analysis was available for 78 of 97 patients on the trial. These patients overall had bulky metastatic disease and had received a median of 78 weeks of prior imatinib therapy (Table 1). Primary KIT mutations were identified in 83% of tumors, whereas 5% had PDGFRA mutations, and 12% contained wild-type KIT and PDGFRA (Appendix Table A1, online only). The most KIT mutations (69%) were located in exon 11, then in exon 9 (30% of KIT mutations), and then in exon 13 (2% of KIT mutations). PDGFRA mutations were located in exon 12 in one patient's tumor and in exon 18 in the tumors of three patients.  Clinical benefit (partial response [PR] or stable disease [SD] for ≥ 6 months) was observed for the three most common GIST genotypes (Table 2). The clinical benefit rate was 58% for tumors with primary KIT exon 9 mutations, 34% for those with exon 11 mutations, and 56% for those with wild-type KIT and PDGFRA before imatinib therapy. Objective responses (ie, PRs) were significantly more common in patients with KIT exon 9 than exon 11 mutant GISTs (37% v 5%; P = .002). Of the four patients with PDGFRA mutations, none experienced clinical benefit. Among patients classified as imatinib-intolerant (n = 4), tumor genotyping revealed a primary KIT exon 9 mutation in one (who achieved a PR) and a wild-type genotype in the other three patients (who achieved SD, two for > 6 months).  Median progression-free survival (PFS) was significantly longer for patients with primary KIT exon 9 mutations (19.4 months; 95% CI, 11.1 to not yet attained [NA]; P = .0005) or a wild-type genotype (19.0 months; 95% CI, 3.9 to NA; P = .0356) than for those with KIT exon 11 mutations (5.1 months; 95% CI, 4.5 to 7.8; Fig 1A). PFS did not differ significantly between patients with exon 9 mutations and a wild-type genotype. Median overall survival (OS) was also significantly longer for patients with exon 9 mutations (26.9 months; 95% CI, 12.2 to NA; P = .012) or a wild-type genotype (30.5 months; 95% CI, 19.8 to NA; P = .0132) than for those with exon 11 mutations (12.3 months; 95% CI, 8.8 to 19.6; Fig 1B). OS did not differ significantly between patients with exon 9 mutations or a wild-type genotype.  Secondary Tumor Genotype and Efficacy A total of 109 post-imatinib biopsy specimens were available from 67 patients, and secondary KIT mutations were identified in 33 patients (Appendix Table A1). Consistent with prior reports, the mutation distribution was nonrandom, and clusters occurred in exons 13 and 14 that encode the drug/ATP binding pocket of the receptor and exon 17 that encodes the kinase activation loop (Fig 2A). The most commonly identified secondary mutation was V654A in exon 13. Two tumors had secondary KIT exon 18 mutations. One patient had different secondary mutations (exon 13 V654A and exon 17 D816H) in different lesions. Secondary kinase mutations were significantly more common in GISTs with primary KIT exon 11 mutations than in those with exon 9 mutations (73% v 19%; P = .0003). Of the four samples with primary PDGFRA mutations, one had a secondary mutation in exon 18 (primary mutation in exon 12), two lacked secondary mutations (both had primary exon 18 D842V mutations), and the fourth lacked a post-imatinib sample. No secondary mutations were found in the eight post-imatinib samples that lacked primary KIT or PDGFRA mutations.  Among all patients with KIT mutations, the median PFS with sunitinib was significantly longer for the 18 patients who had secondary KIT exon 13 or 14 mutations (7.8 months; 95% CI, 4.5 to 10.1) than for the 13 patients who had exon 17 or 18 mutations (2.3 months; 95% CI, 1.0 to 5.1; P = .0157; Fig 2B). Likewise, median OS was numerically longer in the former than the latter group (13.0 months [95% CI, 8.9 to 22.4] v 4.0 months [95% CI, 2.2 to 19.6]; P = .160; Fig 2C), and clinical benefit rates were higher (61% v 15%; P = .011; Table 2). Nearly identical results were obtained when only patients with primary exon 11 mutations were considered. For patients with primary exon 11 mutations, there were no significant differences in PFS or OS between those patients with or without secondary mutations.  In Vitro Measures of Activity With Specific Mutants Sunitinib potently inhibited the activity of ligand-activated wild-type KIT, and the KIT exon 11 V560D and exon 9 AY insertion mutants: 50% inhibitory concentration (IC50) values were less than 100 nmol/L for all three kinases (Table 3; Fig 3A). By comparison, the corresponding IC50 values for imatinib were approximately 1,000 nmol/L for wild-type KIT, 100 nmol/L for the V560D mutant, and 1,000 nmol/L for the exon 9 AY mutant. Sunitinib also potently inhibited the phosphorylation of KIT double mutants, in which the second mutation occurred in the drug/ATP binding site of the receptor, such as V560D + V654A (exons 11 + 13) and V560D + T670I (exons 11 + 14). These double mutants were resistant to inhibition by imatinib in vitro. Conversely, KIT double mutants, in which the second mutation occurred in the activation loop (V560D + D816H, V560D + D820G, V560D + N822K, and V560D + Y823D), were resistant to inhibition by sunitinib or imatinib, with sunitinib IC50 values of 1,000 nmol/L or higher. Notably, the V560D + A829P double mutant had an imatinib IC50 that was only two- to three-fold higher than that of V560D alone. In contrast, V560D + A829P was resistant to sunitinib at doses of up to 1,000 nmol/L. The rarity of A829P as a secondary mutation could be caused by its relatively preserved imatinib sensitivity. Similar results to those obtained when exon 11 V560D was used as the primary mutation were obtained when the exon 9 AY insertion was used instead (Table 3; Fig 3A).  To confirm these findings, we tested the relative potency of imatinib or sunitinib at inhibiting KIT kinase activity in GIST cell lines obtained from imatinib-resistant tumors (Fig 3B). The GIST48 cell line is homozygous for a primary KIT exon 11 V560D mutation and is heterozygous for a secondary exon 17 D820A mutation.17 Concentrations of imatinib greater than 1,000 nmol/L were insufficient to completely inhibit KIT activation in this cell line. (This concentration is 10-fold higher than that necessary to block KIT exon 11-mutant isoforms in GIST cell lines in other studies.24,31,32) Sunitinib was less potent than imatinib at inhibiting KIT autophosphorylation in GIST48 cells. Notably, low doses (100 nmol/L) of either imatinib or sunitinib had a partial inhibitory effect on KIT phosphorylation, presumably because of inhibition of a minority population of V560D homodimers. The GIST430 cell line is heterozygous for a KIT exon 11 deletion mutation and an exon 13 V654A substitution (both on the same allele).17 Sunitinib had significantly greater potency than imatinib for inhibition of KIT autophosphorylation in GIST430 cells (IC50, 1,000 nmol/L for imatinib v < 100 nmol/L for sunitinib).  We also tested the potency of sunitinib at inhibiting the phosphorylation of wild-type PDGFRA or the V561D point mutant: the IC50 values were less than 100 nmol/L for both (Table 3; Fig 3C). V561D, located in the receptor juxtamembrane domain encoded by exon 12, is a relatively common primary PDGFRA mutation in patients with GIST.1 Conversely, D842V, which is the most common PDGFRA mutation in GISTs, which resides in the activation loop encoded by exon 18, and which confers imatinib resistance both as a primary or a secondary mutation,1 conferred resistance to sunitinib in these in vitro experiments (Table 3; Fig 3C). In the clinical study, D842V was detected as a primary mutation in two patients and as a secondary mutation in one patient.  DISCUSSION  These results extend previously reported findings from this study that showed a correlation between sunitinib activity and GIST kinase genotype in patients who have metastatic/unresectable GIST and have experienced imatinib failure.33 Data on the relative responsiveness of different molecular subgroups of imatinib-resistant GIST may help to optimize treatment of patients with GIST and may help to better understand the basis of sunitinib activity in these patients. Such studies may also advance understanding of the mechanisms of resistance and may facilitate development of strategies to circumvent it.  The analyses reported here assessed the effect of tumor kinase genotype on sunitinib activity by using clinical study data complemented by in vitro cellular assays. Although sunitinib demonstrated clinical activity against GISTs of the three most common primary genotypes, both datasets indicated that primary and secondary mutations in the pathogenic kinase strongly influence sunitinib activity. Both the clinical benefit and the objective response rates with sunitinib were higher in patients with primary KIT exon 9 mutations than with exon 11 mutations (clinical benefit rates: 58% v 34%; objective response rates: 37% v 5%; P = .002). Similarly, PFS and OS were significantly longer in patients with primary KIT exon 9 mutations or a wild-type genotype than in those with KIT exon 11 mutations. These results are the converse of those reported for imatinib, in which objective response rates were higher and PFS and OS were longer in patients with GIST who harbored exon 11 mutations than in those who had exon 9 mutations or a wild-type genotype.3,14,15 Notably, the potency of sunitinib against wild-type and exon 9-mutant KIT was superior to that of imatinib in vitro, whereas both drugs exhibited similar potency against KIT exon 11 mutant kinases. A possible explanation is that these mutational sites have different structural effects on KIT, with different consequences for interaction with the two TKIs. Indeed, exon 9 mutations were recently reported to have structural consequences similar to ligand-mediated receptor dimerization.34 This mechanism of kinase activation appears distinct from that caused by mutation of the intracellular juxtamembrane domain encoded by exon 11.35 Others have also observed the impact of mutational site on TKI potency in vitro: by using an isogenic BaF3 model, the imatinib IC50 in cells that expressed exon 9 mutations was found to be approximately eight-fold higher than that obtained in cells that expressed the exon 11 V559D mutation.36 These results suggest that the greater clinical benefit seen for sunitinib-treated patients with exon 9-mutant or wild-type imatinib-resistant GISTs may be related to the greater potency of sunitinib against these kinases. They also suggest that genotypically defined subsets of patients may experience different clinical outcomes when treated with first-line imatinib than with sunitinib. Sunitinib is currently approved only as second-line therapy for GIST, but studies are being planned to evaluate its efficacy and safety as first-line treatment. On the other hand, sunitinib has yet to be tested in imatinib-naïve patients, and the majority of patients in this study with primary KIT exon 11 mutations had acquired secondary KIT mutations that confer imatinib resistance. Studies in imatinib-naïve patients will be required to definitely assess the effect of a primary exon 11 mutation alone on sunitinib activity in vivo.  This study also showed that secondary kinase mutations were significantly more common in GISTs with primary KIT exon 11 than exon 9 mutations and that they did not occur in GISTs with a wild-type genotype, which is consistent with previous reports that secondary kinase mutations are common in GISTs that exhibit secondary imatinib resistance but not in those that exhibit primary resistance.16,17 Moreover, the frequency of secondary mutations is likely to have been underestimated in this analysis, because only one patient in our analysis was found to have different secondary mutations in different lesions, and intra- and interlesion heterogeneity of secondary mutations in GISTs has been documented by others.20,25 Only a limited number of small-needle biopsy specimens were available per patient in our study (mean, 1.4 biopsy specimens per patient; range, 0-3). In particular, it is probable that further sampling would have revealed secondary mutations in those tumors with primary KIT exon 11 mutations that appeared to lack them. Because exon 11 mutants are strongly inhibited by imatinib, secondary resistance is more likely to require the selection and subsequent expansion of clones expressing a second, resistance-conferring mutation than GISTs with exon 9 mutations or a wild-type genotype, which are more likely to be intrinsically resistant to imatinib. Consistent with this, the median duration of prior imatinib treatment for patients who had primary exon 11 mutations was 22.8 months, compared with 12.5 and 10.5 months for patients who had exon 9 mutations or a wild-type genotype, respectively (Table 2). However, it is worth noting that, although the duration of imatinib treatment was a significant prognostic factor for PFS and OS in a univariate analysis, it was not a significant factor in a multivariate analysis (data not shown). Although multivariate analyses performed on such a small sample must be interpreted with caution, they confirmed that primary and secondary KIT genotype were significant prognostic factors for PFS and were marginally significant prognostic factors for OS.  Consistent with previous studies,16,18-28 secondary KIT mutations in patients with imatinib-resistant GIST enrolled on the current study tended to cluster in exons 13 and 14, which encode the drug/ATP binding pocket of the receptor, or in exon 17, which encodes the kinase activation loop. Of note, our in vitro studies showed that sunitinib potently inhibited the kinase activity of KIT receptors that contained secondary mutations in the drug/ATP binding pocket and that are resistant to imatinib, such as V654A (exon 13) and T670I (exon 14). These secondary mutations were coexpressed with a common primary mutation (V560D), which recreated the situation often observed in GISTs that exhibit secondary imatinib resistance. Previous ex vivo studies have also shown that sunitinib inhibits imatinib-resistant KIT receptors that contain mutations in the drug/ATP binding pocket.29,30 However, the in vitro studies performed here also showed that sunitinib was relatively ineffective at inhibiting KIT receptors that contained secondary mutations localized to the activation loop. Consistent with these in vitro findings, PFS and OS were longer and the clinical benefit rate was higher for patients in the clinical trial who had secondary KIT exon 13 or 14 (ie, ATP-binding-pocket) mutations than those with secondary KIT exon 17 or 18 (ie, activation-loop) mutations.  The results of this study provide one explanation for the activity of sunitinib in patients with imatinib-refractory GIST that has been seen in this and other trials.13 However, antiangiogenic effects of sunitinib treatment also may contribute to its effectiveness. In addition to KIT and PDGFRA activity, sunitinib also selectively inhibits PDGFRB and all three isotypes of VEGFR, whereas imatinib inhibits PDGFRB but not VEGFRs. Studies in animal models indicate that dual inhibition of PDGFR and VEGFR produces greater antiangiogenic effects than inhibition of only one or the other,37-39 which suggests that sunitinib may produce greater antiangiogenic effects than imatinib and that these effects may contribute to its activity against imatinib-refractory GISTs.  Of note is our observation that secondary KIT mutants that involve the activation loop are insensitive to both sunitinib and imatinib. Given that different tumor clones in one individual may acquire imatinib resistance because of different secondary mutations, including those involving the KIT activation loop,20,25 not all imatinib-resistant tumors may respond well to sunitinib therapy. Conversely, some GISTs with secondary KIT activation-loop mutations may still be susceptible to sunitinib because of its potent antiangiogenic effects. Additional research of this issue is warranted.  AUTHORS’ DISCLOSURES OF POTENTIAL CONFLICTS OF INTEREST  Although all authors completed the disclosure declaration, the following author(s) indicated a financial or other interest that is relevant to the subject matter under consideration in this article. Certain relationships marked with a “U” are those for which no compensation was received; those relationships marked with a “C” were compensated. For a detailed description of the disclosure categories, or for more information about ASCO's conflict of interest policy, please refer to the Author Disclosure Declaration and the Disclosures of Potential Conflicts of Interest section in Information for Contributors.  Employment or Leadership Position: Xin Huang, Pfizer Inc, (C); Darrel P. Cohen, Pfizer Inc (C); Charles M. Baum, Pfizer Inc (C) Consultant or Advisory Role: Michael C. Heinrich, Novartis (C), Pfizer Inc (U); Christopher L. Corless, Pfizer Inc (C); George D. Demetri, Novartis (C), Pfizer Inc (C), Infinity (C) Stock Ownership: Michael C. Heinrich, Molecular MD; Xin Huang, Pfizer Inc; Darrel P. Cohen, Pfizer Inc; Charles M. Baum, Pfizer Inc Honoraria: Michael C. Heinrich, Novartis; Robert G. Maki, Pfizer Inc; Christopher L. Corless, Pfizer Inc; George D. Demetri, Novartis, Pfizer Inc Research Funding: Michael C. Heinrich, Novartis, Pfizer Inc; Robert G. Maki, Pfizer Inc; George D. Demetri, Novartis, Pfizer Inc, Infinity Expert Testimony: George D. Demetri, Novartis (U), Pfizer Inc (U), Infinity (U) Other Remuneration: None  AUTHOR CONTRIBUTIONS  Conception and design: Michael C. Heinrich, Wen-Bin Ou, Jonathan A. Fletcher, Darrel P. Cohen, George D. Demetri  Financial support: Michael C. Heinrich, Jonathan A. Fletcher, Charles M. Baum, George D. Demetri  Administrative support: Michael C. Heinrich, Diana Griffith, George D. Demetri  Provision of study materials or patients: Michael C. Heinrich, Robert G. Maki, Jonathan A. Fletcher, George D. Demetri  Collection and assembly of data: Michael C. Heinrich, Robert G. Maki, Cristina R. Antonescu, Amy Harlow, Diana Griffith, Ajia Town, Arin McKinley, Wen-Bin Ou, Jonathan A. Fletcher, Christopher D.M. Fletcher, Xin Huang, Charles M. Baum, George D. Demetri  Data analysis and interpretation: Michael C. Heinrich, Christopher L. Corless, Cristina R. Antonescu, Amy Harlow, Diana Griffith, Ajia Town, Arin McKinley, Wen-Bin Ou, Jonathan A. Fletcher, Xin Huang, Charles M. Baum, George D. Demetri  Manuscript writing: Michael C. Heinrich, Christopher L. Corless, Jonathan A. Fletcher, Darrel P. Cohen, Charles M. Baum, George D. Demetri  Final approval of manuscript: Michael C. Heinrich, Robert G. Maki, Christopher L. Corless, Cristina R. Antonescu, Amy Harlow, Diana Griffith, Ajia Town, Arin McKinley, Wen-Bin Ou, Jonathan A. Fletcher, Christopher D.M. Fletcher, Xin Huang, Darrel P. Cohen, Charles M. Baum, George D. Demetri  Appendix  Methods: mutational analysis. Archival tumor biopsy specimens collected before initiation of imatinib therapy were obtained with written patient consent, as were additional specimens collected on days 1 and 11 of the current study before sunitinib dosing. Genomic DNA was extracted from formalin-fixed, paraffin-embedded specimens, was amplified by using polymerase chain reaction, and was analyzed for KIT or platelet-derived growth factor receptor α (PDGFRA) mutations, as previously described (Heinrich MC, Corless CL, Duensing A, et al: Science 299:708-710, 2003; Corless CL, McGreevey L, Haley A, et al: Am J Pathol 160:1567-1572, 2002; Rader AE, Avery A, Wait CL, et al: Cancer 93:269-275, 2001; Choy YS, Dabora SL, Hall F, et al: Ann Hum Genet 63:383-391, 1999) by using primer pairs and denaturing high-performance liquid chromatography conditions reported elsewhere (Heinrich MC, Corless CL, Blanke CD, et al: J Clin Oncol 24:4764-4774, 2006).  Methods: tumor responses. Tumor responses were assessed at baseline and the end of every other cycle on the basis of the Response Evaluation Criteria in Solid Tumors (Therasse P, Arbuck SG, Eisenhauer EA, et al: J Natl Cancer Inst 92:205-216, 2000). Clinical benefit was defined as a partial response or stable disease that lasted 6 months or more. Additional efficacy measures included progression-free survival and overall survival. The current analyses include follow-up through March 2006.  Methods: sensitivity of kinase mutants to tyrosine kinase inhibitors in vitro. Chinese hamster ovary cells were transiently transfected with mutated KIT or PDGFRA cDNA constructs and treated with various concentrations of sunitinib or imatinib, as previously described (Heinrich MC, Corless CL, Demetri GD, et al: J Clin Oncol 21:4342-4349, 2003; Heinrich MC, Corless CL, Blanke CD, et al: J Clin Oncol 24:4764-4774, 2006). The common exon 11 mutation V560D was selected as prototypic primary KIT mutation, and several representative exon 13, 14, and 17 mutations were selected as secondary mutations, for these constructs. Experiments that involved recombinant DNA were performed under biosafety level 2 conditions in accordance with published guidelines (National Institutes of Health; available at: http://www4.od.nih.gov/oba/rac/guidelines/guidelines.html). The gastrointestinal stromal tumor GIST48 and GIST430 cell lines were established from imatinib-resistant GISTs, as previously described (Heinrich MC, Corless CL, Blanke CD et al: J Clin Oncol 24:4764-4774, 2006). Protein lysates from transfected Chinese hamster ovary cells or GIST cell lines were prepared and were subjected to immunoprecipitation by using anti-KIT or anti-PDGFRA antibodies followed by sequential immunoblotting for phospho-KIT or total KIT, or for phosphotyrosine or total PDGFRA, respectively, as previously reported (Heinrich MC, Corless CL, Duensing A, et al: Science 299:708-710, 2003; Heinrich MC, Corless CL, Demetri GD, et al: J Clin Oncol 21:4342-4349, 2003).  Methods: statistical analyses. Descriptive statistics were used to summarize clinical responses and mutation status, and the two-sided Fisher's exact test was used to evaluate differences in response by tumor genotype. Kaplan-Meier methods were used to assess progression-free and overall survival by KIT genotype. Multivariate analyses were performed with Cox proportional hazard models. Gastrointestinal stromal tumors (GISTs) commonly harbor oncogenic mutations of the KIT or platelet-derived growth factor alpha (PDGFRA) kinases, which are targets for imatinib. In clinical studies, 75% to 90% of patients with advanced GISTs experience clinical benefit from imatinib. However, imatinib resistance is an increasing clinical problem. Patients and Methods One hundred forty-seven patients with advanced, unresectable GISTs were enrolled onto a randomized, phase II clinical study of imatinib. Specimens from pretreatment and/or imatinibresistant tumors were analyzed to identify molecular correlates of imatinib resistance. Secondary kinase mutations of KIT or PDGFRA that were identified in imatinib-resistant GISTs were biochemically profiled for imatinib sensitivity. Results Molecular studies were performed using specimens from 10 patients with primary and 33 patients with secondary resistance. Imatinib-resistant tumors had levels of activated KIT that were similar to or greater than those typically found in untreated GISTs. Secondary kinase mutations were rare in GISTs with primary resistance but frequently found in GISTs with secondary resistance (10% v 67%; P   .002). Evidence for clonal evolution and/or polyclonal secondary kinase mutations was seen in three (18.8%) of 16 patients. Secondary kinase mutations were nonrandomly distributed and were associated with decreased imatinib sensitivity compared with typical KIT exon 11 mutations. Using RNAi technology, we demonstrated that imatinib-resistant GIST cells remain dependent on KIT kinase activity for activation of critical downstream signaling pathways. Conclusion Different molecular mechanisms are responsible for primary and secondary imatinib resistance in GISTs. These findings have implications for future approaches to the growing problem of imatinib resistance in patients with advanced GISTs. J Clin Oncol 24:4764-4774. INTRODUCTION Gastrointestinal stromal tumors (GISTs) are the most common type of sarcoma arising in the digestive tract and are generally distinguished from other abdominal sarcomas by the expression of KIT receptortyrosine kinase . This kinase is important not only as a diagnostic markerfor GISTs, but serves as a primary oncogene in approximately 80% of these tumors, as evidenced by activating mutations of the KIT gene. Imatinib (Gleevec; Novartis Pharmaceuticals, Basel, Switzerland) inhibits the enzymatic activity of several oncogenic tyrosine kinases, including KIT and platelet-derived growth factor alpha (PDGFRA). In clinical studies, 75% to 90% of patients with advanced GISTs treated with imatinib experienced a clinical benefit (stable disease  complete/ partial objective response).1-3 These imatinibinduced responses correlated with tumor kinase mutational status.4,5 Patients with KIT exon 11- mutant GIST have a higher response rate and a significantly longer median survival compared with patients with exon 9-mutant GISTs, and those whose GISTs lack KIT or PDGFRA mutations.4 Although imatinib has revolutionized the treatment of advanced GISTs, clinical resistance to this drug has proved to be a significant problem with more prolonged follow-up. Secondary kinase mutations acquired during imatinib treatment have been reported in several small series.6-15 In this study we evaluated molecular From the Division of Hematology/ Oncology, Department of Pathology, Oregon Health & Science University Cancer Institute, Oregon Health & Science University; Portland Veterans Affairs Medical Center, Portland, OR; Dana-Farber Cancer Institute; Harvard Cancer Center; Department of Pathology, Brigham & Women’s Hospital, Boston, MA; Division of Oncologic Surgery, Hitchcock-Dartmouth Medical Center, Lebanon, NH; Division of Medical Oncology, Fox-Chase Cancer Center, Philadelphia, PA; Novartis Pharmaceuticals Corporation, Hanover, NJ; University of Helsinki, Helsinki; and the Department of Surgery, University of Turku, Turku, Finland. Submitted February 17, 2006; accepted June 23, 2006; published online ahead of print at www.jco.org on September 5, 2006. Supported in part by Novartis Pharmaceuticals, VA Merit Review Grant (M.C.H.), GIST Cancer Research Fund (M.C.H.), B.P., Lester and Regina John Foundation (M.C.H.). Terms in blue are defined in the glossary, found at the end of this article and online at www.jco.org. Authors’ disclosures of potential con- flicts of interest and author contributions are found at the end of this article. Address reprint requests to Michael C. Heinrich, MD, R&D-19 3710 U.S. Veterans Hospital Rd, Portland, OR 97239; e-mail: Heinrich@ohsu.edu. 0732-183X/06/2429-4764/$20.00 DOI: 10.1200/JCO.2006.06.2265 JOURNAL OF CLINICAL ONCOLOGY ORIGINAL REPORT VOLUME 24   NUMBER 29   OCTOBER 10 2006 4764 Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. markers of imatinib resistance in a series of well-documented GIST samples from a phase II study of imatinib. PATIENTS AND METHODS Patient Materials Tumor samples were obtained from patients enrolled in the CSTI571 B2222 phase II trial ofimatinib (sponsored by Novartis, Basel, Switzerland)for the treatment of advanced GIST.2,4 Of 147 original patients, 92 had documented disease-related treatment failure as of May, 2005. Samples were obtained from 43 progressive-disease patients who consented to analysis of their tumors. Per the clinical protocol, tumor progression was defined according to traditional Southwest Oncology Group response criteria.16 Specifically, progressive disease was defined as either a 50% or greater increase in the of sum of the products of all measurable lesions over the smallest sum observed (or baseline if no decrease was observed); clear worsening from previous examinations of any assessable disease; reappearance of any lesion that had disappeared; appearance of a new lesion; or the failure to return for evaluation due to death or deteriorating clinical condition. The appearance of a nodule within a mass on follow-up imaging studies was not considered to meet the protocolspecified definition of progression. The study was approved by the local institutional reviewboardofeachparticipatinginstitution,andwritteninformedconsent was obtained from each patient. In addition, informed consent for the analysis of tumor-associated genetic alterations was obtained independently of patient consent for participation in the clinical study. Imatinib response correlates were obtained using paired GIST biopsies taken during the week before initiation of imatinib and biopsies taken during the first week of therapy. Imatinib resistance correlates were evaluated in biopsies taken at the time of GIST progression while patients were still receiving imatinib. Routine pathology review, including KIT immunostaining, was performed on all biopsies. Reagents and Cell Lines The GIST430 and GIST48 cell lines were established from imatinibresistant GISTs. Site-directedmutagenesiswas used to generate relevant single, double, or triple mutations of KIT or PDGFRA cDNA.4,17,18 Experiments involving recombinant DNAwere performed using biosafety level 2 conditionsin accordance with National Institutes of Health Guidelines for Research Involving Recombinant DNAMolecules.19 Imatinibmesylatewas purchasedfrom the Oregon Health & Science University Hospital pharmacy (Portland, OR). KIT Short Hairpin RNA Studies AKITlentiviralshort hairpin RNA (shRNA)was obtainedfromWilliam Hahn, MD, PhD (Dana-Farber Cancer Institute, Boston, MA, and Broad Institute RNAi consortium, Cambridge, MA). This shRNA was assembled by ligating KIT forward 5-CCGGCCATAAGGTTTCGTTTCTGTACTCGAGT ACAGAAACGAAACCTTATGGTTTTTG-3 and reverse 5-AATTCAAAAA CCATAAGGTTTCGTTTCTGTACTCGAGTACAGAAACGAAACCTTAT GG-3 oligomers into the AgeI and EcoRI sites of a pLKO.1puro lentiviral vector. Lentiviral preps were produced by cotransfecting pLKO.1puro empty vector or pLKO.1puro-KIT (shRNA), pCMV R8.91, and pMD.G helper virus packaging plasmids (at a 10:10:1 ratio) into 293T cells. These transfections were performed using lipofectamine and PLUS reagent (Invitrogen, Carlsbad, CA), and lentivirus supernatants were harvested at 24, 36, 48, and 60 hours. Viral titers were determined in GIST882 cells, according to a protocol from Invitrogen. GIST882 or GIST430 cells were infected, in the presence of 8  g/mL of polybrene, and were then lysed for western blot analysis at 72 hours postinfection. Immunoblotting Studies Whole cell lysates were prepared from GIST biopsies or GIST cell lines.20,21The quality of eachlysatewas determined byimmunoblottingfor the GIST markers KIT and PKC, as described previously.20,21 Immunoblotting was performed as described previously. In addition, hamster monoclonal anti–bcl-2 (the kind gift of Stan Korsmeyer, MD, Dana-Farber Cancer Institute) was utilized in the current report. Protein expression studies of mutant KIT or PDGFRA isoforms were performed as previously described by transient transfection of Chinese Hamster ovary cells. The transfected cells were treated with various concentrations of imatinib before preparation of protein lystates. KIT protein was assayed for activation status (phosphorylation) by Table 1. PCR Primers Used for Detection of Mutations of KIT or PDGFRA Exon Forward Primer Reverse Primer D-HPLC Temperatures K8-18 TTGGTATTTTTGTCCAGGAACTGAGC AGAGAACAGCTCCCAAAGAAAAATCC NA K9 ATGCTCTGCTTCTGTACTGCC CAGAGCCTAAACATCCCCTTA 50.0 °C K11 CCAGAGTGCTCTAATGACTG ACCCAAAAAGGTGACATGGA 50.0 °C/56.2 °C K11-13 CCCTTTCTCCCCACAGAAACCCATG ACACGGCTTTACCTCCAATG NA K11-13 ACACGGCTTTACCTCCAATG AGACAATAAAAGGCAGCTTGGACGAC NA K12 CTGCACAAATGGTCCTTCAA CAAAAAGCACAACTGGCAAAC 60.5 °C K13 CATCAGTTTGCCAGTTGTGC ACACGGCTTTACCTCCAATG 59.5 °C K14 CTCACCTTCTTTCTAACCTTTTCTT CCCATGAACTGCCTGTCAAC 57.2 °C K15 GACCCATGAGTGCCCTTCT TCTCTGTAAAGTTACTCTTGGTTG 59.1 °C K16 AGTGATCTGCCTGCAAGTTCACATTAG GCTCTAAAATGCTCTGTTCTC 60.8 °C K17 TGTATTCACAGAGACTTGGC GGATTTACATTATGAAAGTCACAGG 58.0 °C K18 TTCTGTTCAATTTTGTTGAGCTTC GCAGGACACCAATGAAACTT 58.7 °C K19 GCAAACTGTGTCTCAGGAAGC CCCTCAACATCTGGGTTTCT 56.9 °C K20 AAAACAAGCTGAGGGCATTG TGGGAGAAGGGGGATTCTAT 59.3 °C K21 TGTTCGTTGTAGGGACTGCT AACCATCATGGAAGCCAAAG 63.0 °C K11-13 CCCTTTCTCCCCACAGAAACCCATG ACACGGCTTTACCTCCAATG NA K11-13 CCAGAGTGCTCTAATGACTG AGACAATAAAAGGCAGCTTGGACGAC NA P10 GGCCCTATACTTAGGCCCTTT TCAGCTGATGAGTTGTCCTGA 50.0 °C/62.0 °C P12 TCCAGTCACTGTGCTGCTTC GCAAGGGAAAAGGGAGTCTT 50.0 °C/59.7 °C P14 TGGTAGCTCAGCTGGACTGAT GGGATGGAGAGTGGAGGATT 59.1 °C P15 CCATCTCCTAACGGCTTTTG CAGGACATGGGTCTTTCCAT 57.9 °C P17 ACCTGATGATTTCCTGCTGC CGTCCACACTCCACTCACTG 58.2 °C P18 ACCATGGATCAGCCAGTCTT TGAAGGAGGATGAGCCTGAC 50 °C/61.6 °C P19 TGCTGTGGATCATCAGTGAGT TTCCTTCCAGTGGTGTTTCA 60.1 °C P20 CAATGCACTGAGCGTTTGTT CTCCCCCTAGACCCACAGAC 59.8 °C Abbreviations: PDGFRA, platelet-derived growth factor alpha; D-HPLC, denaturing high-performance liquid chromotography; K, KIT exon; NA, not available; P, PDGFRA exon. Imatinib Resistance in GISTs www.jco.org 4765 Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. immunoprecipitation using an anti-KIT antibody, followed by sequential immunoblotting for phosphoKIT (using antiphosphotyrosine antibody) or total KIT (anti-KITmonoclonal antibody). In the case of PDGFRAmutations, anidenticalmethodologywas used exceptfor substitution of an anti-PDGFRA antibody for the immunoprecipitation and total PDGFRA immunodetection steps. IC50 (50% inhibitory concentration) was measured by densitometry of the phosphoKIT bands and normalization using total KIT expression (to correct for variations in loading of KIT protein in the various lanes).4,17,18 Mutation Detection Methods Mutational analyses were performed on genomic DNA extracted from paraffin embedded or fresh frozen tumor tissue using a combination of polymerase chain reaction (PCR) amplification, denaturing high-performance liquid chromatography (D-HPLC) screening, and automated sequencing, as described previously.4,17,22 PCR primer pairs and D-HPLC conditions are listed in Table 1. PCR Evaluations of the Allelic Relationships of Primary and Secondary Mutations Allelic relationships of primary and secondary KIT mutations were evaluated by reverse transcriptase-PCR (exons 8-18) or genomic PCR (exons 11 to 13) using primer pairs listed in Table 1. The resultant PCR products were cloned and individual bacterial colonies were bidirectionally sequenced.23 RESULTS Description of Patient Population The study population consisted of 147 patients treated in a randomized, phase II study of imatinib for patients with advanced GISTs.2,24As ofMay 2005, themedian time to treatmentfailurefor the entire phase II study population was 19.3 months and the median survival was estimated to be 57 months. Late treatment failure was primarily due to disease progression rather than discontinuation of therapy related to medication intolerance. Treatmentfailuresweredividedinto twogroups.Patientswhonever achieved a partial remission and showed continued tumor growth during the first 180 days of treatment were defined as having primary imatinib resistance. Disease progression after a minimum of 6 months of partial remission or stable disease was defined as secondary imatinib resistance. This study is based on comparative analyses of tumor samples obtained from both these groups before and after treatment failure. KIT Kinase Is Inhibited by Imatinib in Responding GISTs, but Not in Tumors With Primary Resistance Activation of KIT was evaluated in fresh frozen metastatic GIST lesions from two patients (11 and 34) who underwent biopsy in the week before starting imatinib therapy and again after 5 or 7 days of treatment (Fig 1 and Table 2). Both patients had GISTs withKITexon 11 mutations. KIT phosphorylation was evident in the preimatinib biopsies, but became nearly undetectable during the first few days of imatinib therapy. KIT inhibition was accompanied by dramatic decreases in the phosphorylated forms of mTOR, AKT, and mitogenactivated protein kinase (MAPK). Imatinib therapy also decreased expression of a proliferation marker (PCNA) and a critical antiapoptotic protein (bcl-2). We analyzed the activation status of KIT-dependent signaling pathways in biopsies obtained from two patients whose GISTs had a KIT exon 9 mutation and showed primary imatinib resistance (Fig 2A and Table 3). Frozen tumor samples were not available from either patient pretreatment; nevertheless, on-treatment biopsies contained phosphorylated KIT protein in quantities equivalent to that typically present in untreated GIST specimens.21 The pattern of intracellular signaling was similar to that seen in untreated GISTs (Fig 1 and Duensing et al21), indicating that primary resistance is associated with persistent KIT phosphorylation and activation of downstream AKT and MAPK pathways. Reactivation of KIT and KIT-Dependent Signaling in Secondary Resistance Two patients who had objective responses to imatinib underwent tumor biopsy after documented progression at 18 and 32 months. Matched pretreatment biopsies were available for both patients (Fig 2B and Table 2). Patient 1 had a primary KIT exon 11 deletion. Patient 14 had two activating KIT mutations in multiple pretreatment specimens (K642E and N822K). This is the only such example of double mutations in our series of more than 1,000 imatinib-naı¨ve patients with GISTs.17,25 We speculate that one mutation was the initiating mutation and the other developed during malignant progression. The imatinib-resistant tumors contained abundant amounts of activated KIT. Notably, the ratio of phosphorylated-AKT (P-AKT) to AKT was markedly increased in the progression samples Fig 1. Imatinib inhibits KIT phosphorylation and activation of KIT-dependent signaling in responding gastrointestinal stromal tumors. Paired biopsy specimens were obtained before initiation of imatinib (baseline) and within the first week of therapy. The on-treatment samples have marked decreases in the levels of phosphorylated (P-) forms of KIT, AKT, mitogen-activated protein kinase (MAPK), and mTOR. Heinrich et al 4766 JOURNAL OF CLINICAL ONCOLOGY Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. compared with matched pretreatment samples. In contrast, MAPK was only significantly activated in one of the progression samples (patient 14). Secondary Kinase Mutations Are Common in Secondary but Not Primary Imatinib Resistance To determine whether imatinib resistance is associated with acquired (secondary) kinase mutations, we screened the cytoplasmic domain exons of KIT and PDGFRA in DNA prepared from 79 tumor samples obtained from 43 patients with primary or secondary drug resistance (Tables 2 and 3). In 90.7% of patients (39 of 43), the pretreatment kinase genotype had been previously determined. In 10 patients with primary resistance the median time to treatmentfailure was 3.6 months (range, 0.8 to 5.8; Table 3). No secondary mutations of KIT or PDGFRA kinase domain were found in specimens obtained from these patients at the time of progression. Patient 39 had primary progression on a dose of 400 mg of imatinib. No secondary kinase mutations were found at the time of progression on 400mg, and cross-over to 600mg led to disease stablilization. Twentythree months later, the patient had tumor progression and ultimately underwent surgical debulking. Specimens from this procedure had multiple secondary kinase mutations (Table 3 and Fig 3). In thirty-three patients with secondary resistance, the median time to treatment failure was 20.2 months (range, 7.2 to 52.7). Sixtyone progression specimens obtained from these 33 patients were judged to contain GISTs. The originalmutation,whichwas knownfor 29 of 33 patients based on analysis of preimatinib specimens, was confirmed in all progression biopsies (54 of 54 samples). Overall, 22 (67%) of 33 patients with secondary resistance had one or more secondary kinase mutations (21 KIT and 1 PDGFRA; Table 2 and Fig 3). This frequency is distinctly higher than that in patients with primary resistance (10% v 67%; P   .002). Notably, all secondary KIT kinase mutations were found in GISTs with an Table 2. Patients With Secondary Imatinib Resistance Patient Age/Sex Initial Imatinib Dose (mg) Primary Site Best Response Time From Diagnosis to Imatinib (days) TTF (days) Primary Mutation Secondary Mutation (mutations/samples) Status 1 64/F 400 SI PR 664 930 KIT exon 11 Del WKVVE557-561 KIT exon 17 N822K (4/4) AWD 2 53/F 400 Stomach PR 2,921 883 KIT exon 11 Del WK557-558 KIT exon 17 Y823D (1/1) DOD 3 50/M 400 SI PR 348 717 KIT exon 11 Del PMYE551-554 KIT exon 13 V654A (3/3) AWD 4 25/M 600 Stomach SD 403 508 None None (0/1) AWD 5 39/M 400 SI PR 591 504 KIT exon 11 V560D  KIT exon 13 insertion KIT exon 13 insertion & V654A (3/4) & KIT exon 17 D816H (1/4) DOD 6 54/M 600 SI PR 1,037 512 KIT exon 11 W557G KIT exon 17 Y823D (1/3) DOD 7 43/M 400 IAB PR 1,252 497 PDGFRA V561D PDGFRA D842V (1/1) DOD 8 49/M 600 Stomach PR 2,120 1,580 KIT exon 11 Del PMYE551-554 None (0/2) DOD 9 58/F 600 Stomach PR 2,329 379 KIT exon 11 Del 554-558EVQWK None (0/1) AWD 10 49/M 600 SI PR 643 1,564 KIT exon 11 Del KV558-559N KIT exon 17 N822K (1/1) AWD 11 50/F 400 SI PR 3,615 349 KIT exon 11 Del VV559-560 None (0/1) DOD 12 62/F 600 SI PR 1,609 584 KIT exon 11 Del WKVV557-560C KIT exon 17 Y823D (1/4) DOD 13 46/M 600 SI PR 58 1,297 KIT exon 11 Del MYEVQW552-557 KIT exon 13 V654A (1/1) AWD 14 63/M 600 Stomach PR 380 500 KIT exon 13 K642E  exon 17 N822H KIT exon 17 C809G (2/2) DOD 15 67/M 400 IAB SD 155 215 None None (0/1) AWD 16 52/M 400 Stomach PR 27 246 KIT exon 11 Del KPMYEVQWK550-558 None (0/2) DOD 17 54/F 400 IAB PR 778 766 KIT exon 11 V560G KIT exon 13 V654A (3/3) AWD 18 67/M 400 IAB PR 104 498 KIT exon 11 V559D None (0/2) DOD 19 59/M 400 IAB PR 125 587 KIT exon 11 V560D KIT exon 17 D820A (2/2) DOD 20 59/F 600 SI PR 613 417 KIT exon 9 KIT exon 17 D820G (1/2) DOD 21 54/M 400 Stomach PR 952 366 KIT exon 11 Del KPMYEVQWK550-558 KIT exon 13 V654A (1/1) DOD 22 65/M 400 IAB PR 164 701 KIT exon 11 DelYEVQWK553-558 KIT exon 13 V654A (2/2) DOD 23 70/F 400 Stomach PR 2,285 846 KIT exon 11 Del EVQWK554-558 KIT exon 17 Y823D (1/1) AWD 24 45/F 600 Stomach PR 60 384 KIT exon 11 DelKPMYEVQWK550-558 KIT exon 14 T6701 (2/3) DOD 25 44/F 400 Stomach PR 411 1,028 KIT exon 11 Del WKVV557-560F KIT exon 17 Y823D (1/1) AWD 26 37/M 600 SI PR 141 499 KIT exon 11 Del EVQWKVVEEINGNNYVYI554-571 None (0/1) DOD 27 53/F 600 SI PR 1,267 766 KIT exon 9 None (0/1) AWD 28 63/M 600 SI PR 167 462 KIT exon 9 KIT exon 13 V654A (2/2) DOD 29 60/M 400 Stomach PR 122 237 KIT exon 11 insertion K558NP None (0/1) DOD 30 63/M 400 Stomach PR 138 505 KIT exon 11 Del KV558-559 None (0/1) DOD 31 71/M 600 IAB PR 507 747 KIT exon 11 L576P KIT exon 13 V654A (1/1) DOD 32 38/M 400 SI SD 33 416 KIT exon 11 Del KPMYEVQW550-557FL KIT exon 17 D820G (1/1) DOD 33 60/M 400 Stomach PR 275 252 KIT exon 13 K642E KIT exon 17 D816H (2/4) DOD 34 77/M 400 Stomach PR 448 881 KIT exon 11 V560D NP AWD Abbreviations: TTF, time to treatment failure; F, female; SI, small intestine; PR, partial response; Del, deletion; AWD, alive with disease; DOD, dead of disease; M, male; SD, stable disease; IAB, intra-abdominal; NP, not performed. Imatinib Resistance in GISTs www.jco.org 4767 Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. underlying primary KIT mutation and these secondary mutations were not present in pretreatment specimens. No secondary mutations were identified in one GIST lacking a primary KIT or PDGFRA mutation. In addition, the only secondary PDGFRA mutation identified was in a GIST with a primary PDGFRA V561D mutation. The secondary KIT kinase mutations were nonrandom and involved either the adenosine triphosphate (ATP) binding pocket of the kinase domain (V654A, T670I) or the kinase activation loop (C809G, D816H, D820A/E/G, N822K/Y, Y823D). Figure 3 summarizes the spectrum and frequency of secondary KIT mutations in this series and other published reports. In contrast to primary resistance, two of threeKITexon 9mutant GISTs werefound to have additionalKITkinasemutations in biopsies obtained after the development of secondary imatinib resistance. However, the frequency of secondary KIT mutations in KIT exon 9 mutant GISTs was not obviously different between cases with primary resistance versus secondary resistance (1 of 7 v 2 of 3; P   .18). Evidence for Clonal Evolution and/or Polyclonal Secondary Resistance Mutations in Individual Patients Multiple biopsy and/or resection specimens were available from 16 patients with progressive disease on imatinib. In three of these Fig 2. KIT phosphorylation and activation of KIT-dependent signaling pathways in gastrointestinal stromal tumors with (A) primary and (B) secondary imatinib resistance. Two parts of the same minimally progressing metastasis were evaluated at month 3 in patient 40 (lanes 1 and 2). Two different progressing metastases were evaluated at month 6 in patient 42 (lanes 3 and 4). Paired tumor specimens were obtained before treatment (baseline) and at the time of secondary imatinib resistance. P-, phosphorylated; MAPK, mitogenactivated protein kinase. Table 3. Patients With Primary Imatinib Resistance Patient Age/Sex Initial Imatinib Dose (mg) Primary Site Best Response Time From Diagnosis to Imatinib (days) TTF (days) Primary Mutation Secondary Mutation Status 35 52/M 600 LI NE 829 NE  PDGFRA D842V None (0/2) DOD 36 48/F 600 IAB SD 294 156 KIT exon 9 None (0/1) DOD 37 45/M 400 SI PD 620 23 KIT exon 9 None (0/1) DOD 38 33/M 600 SI SD 2,415 166 KIT exon 9 None (0/1) DOD 39 36/F 400 SI PD 638 74 KIT exon 9 N822K† (1/6) AWD N822Y† (1/6) D820E† (2/6) 40 41/F 600 SI SD 849 107 KIT exon 9 None (0/3) AWD 41 65/M 400 IAB PD 1,609 84 KIT N822K None (0/1) DOD 42 52/M 600 SI SD 1,378 173 KIT exon 9 None (0/1) DOD 43 51/M 400 SI SD 1,145 173 KIT exon 9 None (0/1) AWD 44 54/M 600 Stomach PD 1,590 89 PDGFRA D842V None (0/1) DOD Abbreviations: TTF, time to treatment failure; M, male; LI, large intestine; NE, not assessable; DOD, dead of disease; F, female; IAB, intra-abdominal; SD, stable disease; SI, small intestine; PD, progressive disease; AWD, alive with disease.   TTF could not be defined due to use of different imaging modalities in staging exams. †This patient had no detectable secondary mutation at the time of initial progression. However, secondary kinase mutations were found in specimens obtained more than 1 year after initial progression. Heinrich et al 4768 JOURNAL OF CLINICAL ONCOLOGY Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. patients (3 of 16; 18.8%), we found evidence of clonal evolution and/or polyclonal secondary kinase mutations (Tables 2 and 4). Patient 5 had a baselineKITV560D substitution on one allele and a single adenine base insertion in KIT exon 13 of the other allele, resulting in a frameshift and protein truncation. Thus, this patient’s tumorwasfunctionally homozygous due to loss of thewild-type allele. In specimens obtained at the time of imatinib resistance, a V654A mutation was present in two biopsy specimens and D816H in a third specimen. All three specimens had the original preimatinib KIT exon 11 and 13 mutations. Allelic Distribution of Secondary Kinase Mutations Most KIT mutant GISTs (  90%) are heterozygous for the original activating mutation and therefore have one wild-type and one mutant KIT allele. Without exception, all secondary mutations were on the same allele as the primary mutation (Table 4). As noted above, patient 14 (Tables 2 and 4) had a GIST with two monoallelic preimatinib KIT mutations (K642E and N822H). An acquired C809G substitutionfound at the time ofimatinib resistancewas on the same allele as the K642E and N822H mutations. Secondary Kinase Mutations Lead to Imatinib Resistance In Vitro To determine if secondary kinase mutations were the cause of clinically observed imatinib resistance,we tested the in vitro sensitivity of selected single, double, or triplemutant kinases toimatinib. Because GISTs manifest a wide range of mutations in exon 11 (Table 2), we chose a representative KIT exon 11 point mutation, V560D, as the standard for comparing imatinib sensitivity of single and double mutant KIT kinases.25 Secondary mutations of interest (for example, V654A) were then engineered into the V560D isoform as well as into a wild-type cDNA, allowing comparative analyses of the effects of secondary mutations on imatinib sensitivity. Several of the secondary KIT mutations identified in our series have been previously identified in human cancers: D816H, D820A, N822H/K, and Y823D.4,17,18,26-30 In contrast, there are no reports of V654A, T670I, or C809G mutations occurring as a primary mutation. The isolated KIT V560D mutation was very sensitive to imatinib (IC50  100 nmol/L; Fig 4). Both V654A and T670I resulted in strong imatinib resistance, with IC50s of 5 and 10  mol/L, respectively— either in isolation or when coexpressed with KIT V560D. Similar results were obtained usingKITexon 9 AY as the primary mutation (Fig 4). A number of amino acids located in or near the KIT activation loopweremutatedinimatinib-resistant tumors (Fig 3 and Table 2). In vitro profiling of isolated activation loop mutations revealed a spectrum of imatinib sensitivity rangingfrom relatively sensitive (IC50 of 100-200 nmol/L, N882K, Y823D) to highly resistant (IC50   5,000 nmol/L, D816H). When coexpressed with V560D, all of the double mutant kinases were extremely resistant to imatinib, with the exception of V560D  N822K which was only moderately imatinib resistant (Fig 5A). Similar results were obtained using KIT exon 9 AY-insertion or K642E as the primary mutation (Fig 5B). Notably, both primary mutations were intrinsically imatinib sensitive, but Fig 3. Secondary KIT mutations in imatinib-resistant gastrointestinal stromal tumors. Data from the current report (21 patients) and other published reports (43 patients) are indicated with red and black symbols, respectively.6-14 Patients with different secondary mutations in geographically separate metastases are indicated by numbered symbols. KIT codons are listed in the shaded boxes. Table 4. Alleleic Distribution Studies Patient Primary Mutation Secondary Mutation Allelic Distribution of Mutations 5 KIT V560D V654A Cis 5 KIT V560D D816H Cis 5 KIT V560D Exon 13 insertion leading to premature stop codon Trans 14 KIT K642E  N822H C809 All Cis 19 KIT V560D D820A Cis 23 KIT Del554-558EVQWK Y823D Cis 28 KIT exon 9 V654A Cis 33 KIT K642E D816H Cis Abbreviations: Cis, same allele as primary mutation; Trans, opposite allele from primary mutation. Imatinib Resistance in GISTs www.jco.org 4769 Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. coexpression ofD820GwithKITexon 9 resultedinmoderateimatinib resistance. Coexpression of the imatinib-sensitive K642E mutation with D816H resulted in extreme imatinib resistance. A doubly mutant (K642E  N822H) kinase modeled on patient 14wasmoderately resistant toimatinib (IC50of 2 mol/L).Notably, patient 14 received a dose of 600 mg daily of imatinib and experienced an objective partial response that lasted 500 days. The C809G mutation found at the time of progression proved resistant in vitro (IC50   5  mol/L), both in isolation and when coexpressed with KIT K642E  N822H. In patient 7, a primary PDGFRA V561D mutation (known to be sensitive to imatinib) was accompanied by a PDGFRA D842V mutation at the onset of progression. The doubly mutant V561D  D842V kinase proved imatinib-resistant with an IC50 similar to the isolated D842V isoform (Fig 6).17 GIST Cell Lines With Secondary Kinase Mutations Have In Vitro Imatinib Resistance To further validate our biochemical studies, we generated cell lines from two patients with secondary imatinib resistance (neither patient was part of the phase II study). The GIST48 cell line has a combination of mutations (homozygous V560D  heterozygous D820A) similar to that documented in samples from patient 19. Concentrations of imatinib higher than 1  mol/L were required for complete inhibition of KIT activation in this cell line (Fig 7A). This concentration is 10-fold greater than that necessary to block exon 11 mutant isoforms of KIT in GIST cells.8,21,31 Activation of AKT was partially but not completely inhibited by imatinib doses of 0.1 to 5  mol/L. The GIST430 cell line is heterozygous for a KIT exon 11 deletion mutation and the V654A substitution (both on the same allele). Eight samples from imatinib-resistant tumors in our study had a similar combination of KIT exon 11 and V654A mutations. The doubly mutantKITisoform expressed by GIST430 cellswasimatinib resistant (IC50of 2.5 mol/L; IC90of 5 mol/L).AKT activationin this cell line was only partially inhibited by imatinib doses higher than 2.5  mol/L (Fig 7A). We used RNAi to determine whether KIT expression was still required for activation of signaling pathways in imatinibresistant GIST cells. shRNA knockdown of total KIT expression in the previously described imatinib-sensitive GIST882 cell line (homozygous KIT K642E mutation)31 resulted in parallel decreases in phospho-KIT, phospho-AKT, phospho-p70S6K, and the proliferation marker cyclin A expression (Fig 7B). In contrast, there was no change in the expression of p70S6K, AKT, or PI3K. Comparable findings were obtained after KIT shRNA knockdown in GIST430 cells, demonstrating that activation of proliferation/survival signaling pathways remains KIT dependent in this imatinib-resistant cell line.KITknockdown in the cell lines also induced flow-cytometric evidence for G1 block, decreased S phase, and markedly increased apoptosis (data not shown). DISCUSSION We analyzed genomic mechanisms of imatinib resistance in a cohort of patients that were part of a randomized phase II study of imatinib for treatment of advanced, unresectable GIST. Forty-three (46.7%) of 92 patients with disease-related treatment failures consented to studies of their tumor samples. The resulting collection of 83 Fig 4. Biochemical characterization of secondary mutations involving the adenosine triphosphate (ATP) binding pocket of KIT. Secondary mutations involving the ATP binding pocket were tested for in vitro sensitivity to imatinib when expressed in isolation or when coexpressed with KIT exon 11 V560D or KIT exon 9 mutations. P-, phosphorylated. Heinrich et al 4770 JOURNAL OF CLINICAL ONCOLOGY Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. specimens from well-monitored patients receiving standardized imatinib treatment is the largest examined to date for molecular mechanisms related to drug resistance. Imatinib-resistant GISTs show activation of the same signaling pathways that are operative in untreated GISTs. Interestingly, primary imatinib resistance is infrequently associated with secondary kinase mutations (1 of 10; 10%). PDGFRA D842V and KIT exon 9 mutations were significantly over-represented in this group compared with secondary imatinib resistance. GISTs with these genotypes are less responsive to imatinib therapy than GISTs with an associated KIT exon 11 mutation.4,5 The PDGFRA D842V mutation has moderate to high level in vitro resistance to imatinib explaining why this mutation was responsible for primary imatinib resistance in the two patients with a primary PDGFRA D842V mutation.4,8,32 This conclusion is strengthened by our Fig 5. Biochemical characterization of secondary mutations involving the KIT activation loop. Secondary mutations involving the KIT activation loop were tested for in vitro sensitivity to imatinib when expressed in (A) isolation or when coexpressed with the KIT exon 11 V560D or (B) in isolation or coexpressed with KIT exon 13 K642E or KIT exon 9 mutations. P-, phosphorylated. Fig 6. Platelet-derived growth factor alpha (PDGFRA) D842V mutation is associated with primary and secondary imatinib resistance. Imatinib potently inhibits the phosphorylation of the primary PDGFRA exon 12 V561D mutation (patient 17, pretreatment), but not PDGFRA D842V—either expressed in isolation (patients 1 and 10) or when coexpressed with V561D (patient 17, progression specimen). Imatinib Resistance in GISTs www.jco.org 4771 Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. observation that the PDGFRA D842V mutation was associated with secondary imatinib resistance in a GIST harboring a primary PDGFRA V561D mutation. Debiec-Rychter et al8 also reported a case of imatinib-resistant GIST associated with acquisition of a PDGFRA D842V mutation in a GIST with a primary KIT exon 11 mutation. Themolecularmechanisms underlying primary resistancein GISTs with KIT exon 9 mutations are as yet unidentified, but may be related to differencesin the underlying biology of this subset ofGISTs.We speculate that some KIT exon 9 mutant GISTs have an alternative mechanism of KIT activation that does not require the enzymatic activity of KIT. Notably, imatinib only binds to the inactive form of KIT.33,34 Therefore, Fig 7. Biochemical characterization of secondary KIT mutations expressed in imatinib-resistant gastrointestinal stromal tumor (GIST) cell lines. GIST48 is an imatinib-resistant cell line with a homozygous KIT V560D and a heterozygous D820A mutation. GIST430 is an imatinib-resistant cell line with a heterozygous KIT exon 11 deletion mutation and V654A mutation (same allele). GIST882 is an imatinib-sensitive GIST cell line expressing a homozygous K642E mutation. (A) Effects of imatinib treatment. (B) Effects of RNAi inhibition of KIT. P-KIT, phosphorylated KIT; P-AKT, phosphorylated AKT; shRNA, short hairpin RNA. Heinrich et al 4772 JOURNAL OF CLINICAL ONCOLOGY Downloaded from ascopubs.org by Memorial Sloan Kettering Cancer Center on May 25, 2017 from 140.163.000.005 Copyright © 2017 American Society of Clinical Oncology. All rights reserved. KIT-independent mechanisms of KIT phosphorylation could stabilize the protein in an active, imatinib-resistant conformation. GISTs with KIT exon 11 mutations commonly become imatinib resistant due to acquisition of secondary kinase mutations located on the same allele as the original gain of function KIT mutation. These secondary kinase mutations appear to be the predominant mechanism for late imatinib resistance in KIT exon 11 mutant GISTs. Fluorescence in situ hybridization assays (data not shown) revealed low-level KIT amplification (3 to 4 copies per cell), accompanied by KIT protein overexpression, in only two patients, suggesting that genomic KIT amplification has only a minor role in GIST imatinib resistance.We believe that the differencein thefrequency of secondary mutations between GISTs with underlying KIT exon 11 versus KIT exon 9 mutations is reflective of the underlying biology of these tumors—namely, that KIT exon 11 mutant GISTs are more addicted to KIT signaling thanKITexon 9 mutant GISTs. In this model,KITexon 9 mutant GISTs are able to use alternative signaling mechanisms to drive cellular proliferation. In contrast,KITexon 11mutant GISTs are more restricted in their signaling requirements and themost common mechanism of resistance is expansion of a clone with an imatinibresistant kinase mutation. Using RNAi KIT knockdown, we demonstrated that imatinibresistant GIST cell lines with secondary kinase mutations remain dependent onKIT signalingfor growth and survival. These findings have implications for future approaches to the growing problem of imatinib resistance in patients with advanced GISTs.  Activating a mutation in KIT or PDGFR (platelet-derived growth factor receptor) is considered to be the main molecular driver of most GI stromal tumors (GISTs). Almost all KIT mutations occur in exon 9, 11, 13, or 17, and PDGFRA mutations in exons 12, 14, or 18.1 Ten percent to 15% of GISTs that lack KIT and PDGFRA mutations are often referred to as wild-type GISTs, though they may have mutations in other genes.2,3  Most GISTs are localized when detected, but approximately 40% eventually recur despite macroscopically complete surgery.4 Adjuvant imatinib reduces the risk, provided that GIST harbors an imatinib-sensitive mutation.5–7 In general, KIT exon 11 mutations are most sensitive to imatinib, whereas the PDGFRA exon 18 mutation leading to Asp842Val (D842V) is considered imatinib-resistant.8,9 Administering adjuvant imatinib for at least 3 years after surgery is recommended when the risk of recurrence is considered high.10,11  Tumor-mutation analysis is important to identify imatinib-resistant mutations and KIT exon 9 mutations that may require a high-dose of imatinib (> 400 mg/day),12 but whether mutation analysis confers prognostic information beyond the standard prognostic factors identified is controversial.13–15 Commonly used risk-stratification schemes, such as the Armed Forces Institute of Pathology (AFIP) scheme and the National Institutes of Health (NIH) scheme and its modification, are based on tumor size, site, and mitotic count and do not include tumor mutation status in risk stratification.16–18 Yet, certain mutations have been repeatedly linked with survival and GIST clinical features. For example, Asp842Val (D842V) is associated with gastric location and favorable survival,14,19 whereas KIT exon 11 mutations affecting the codons 557 and 558 are associated with unfavorable outcome.13,20 GISTs with KIT exon 9 mutation usually arise in the small bowel and are associated with frequent recurrence.21,22  Outcome data related to single mutations are few, likely because of the presence of a large number of different mutations, particularly in KIT exon 11, and the rarity of GIST. Moreover, adjuvant imatinib modifies the risk for recurrence owing to varying sensitivity of different mutations to the drug.23 We investigated the influence of KIT and PDGFRA mutations on recurrence-free survival (RFS) in a GIST patient population treated with macroscopically complete surgery. To obtain a large and relatively unselected series to study the influence of the most common single mutations on survival, we first identified population-based series from the literature, and then pooled the individual patients' data.4  PATIENTS AND METHODS  Patients We identified population-based GIST patient series published between January 1, 2000, and January 1, 2010, from electronic medical literature databases. To reduce a risk for a selection bias, we required that each series consist of virtually all GISTs diagnosed within a defined geographical area and time period, include at least 50 patients, and provide data about the main prognostic factors. We identified 10 series24–32 that included a total of 2,560 patients, of whom 1,625 had RFS information available. The study was approved by an institutional review committee.4  All patients had operable GIST with no detectable metastases at the time of the diagnosis.24–32 Patients with multiple or recurrent GISTs were excluded. All patients underwent macroscopically complete surgery, and none received adjuvant or neoadjuvant systemic therapy. The tumor was required to have morphology compatible with GIST and to stain for KIT at immunohistochemistry. The median follow-up time of the patients alive was 4.0 years (range, 0.01 to 45.8 years; for RFS, 0.01 to 26.9 years).  A mutation analysis had been done in 1,091 (42.6%) of the 2,560 GISTs. Of these, only data about the mutated gene and exon were available for 144 patients (13.2%). The 1,091 patients with mutational data available had similar distributions of age, sex, tumor size, mitotic count, and site as did patients who did not have mutational data available (n = 1,469; P > .05 for each comparison; Appendix Table A1[online-only]). Mutation analysis was usually done for KIT exons 9, 11, 13, and 17 and PDGFRA exons 12 and 18, but the numbers of exons analyzed in each case were not available.  MolecGIST Series The MolecGIST series is a prospectively collected population-based series consisting of GIST patients diagnosed in France between May 2006 and June 2008.33 The series was not included in the Global Pooled Series, because the study was not fully reported before January 1, 2010, and because some patients had received adjuvant or neoadjuvant imatinib. The MolecGIST mutational data were included in our current study, but the survival data were excluded to avoid a potential bias related to selected use of adjuvant imatinib for higher risk patients (Fig 1).  figure Fig 1. CONSORT diagram. GIST, GI stromal tumors; RFS, recurrence-free survival.  The clinical records and tumor histology were reviewed, and the diagnosis of GIST was based on histology and immunostaining for KIT or DOG-1.33 Of the 596 patients in the series, 507 patients had a local GIST excised surgically; and, of these, 414 patients (81.7%) had mutation analysis of KIT (exons 9, 11, 13, and 17) and PDGFRA (exons 12, 14, and 18) performed. The 93 tumors that were not analyzed for mutations had similar features as the ones that were analyzed (Appendix Table A2). The study was approved by an ethics committee.33  Statistical Analysis RFS was calculated from the date of surgery or the date of making the histologic diagnosis to the date of first tumor recurrence. Patients who were alive were censored at the time of data collection, and patients who died were censored on their dates of death. Survival between groups was compared using the Kaplan-Meier life-table method and the log-rank test. Hazard ratios and their CIs were calculated using a univariable Cox model, and prognostic factors were compared using a Cox proportional hazards model. Frequency tables were analyzed using the χ2 test. Continuous distributions between groups were compared with the Mann-Whitney U test or Kruskall-Wallis test. The P values are two-sided and are not adjusted for multiple testing. The data were analyzed using Statistical Package for Social Sciences version 17.0 (SPSS, Chicago, IL).  RESULTS  Patients Mutation analysis data were available for 1,505 (49.1%) of 3,067 patients (Fig 1). The median age of the patients with mutation analyses available was 64 years at the time of GIST detection (range, 9 to 106 years), and 778 patients (51.7%) were male. Only six patients (0.4%) were younger than 20 years.  PDGFRA Patients with PDGFRA mutations and those with wild-type GIST had a smaller risk for recurrence than patients with KIT mutations (PDGFRA: hazard ratio [HR], 0.34; 95% CI, 0.16 to 0.73; P = .004; wild-type GIST: 0.52; 95% CI, 0.35 to 0.76; P < .001; Fig 2A). There was no significant difference between patients with PDGFRA mutations and those with KIT mutations in their age at diagnosis, sex, or tumor size (each: P > .10). But tumors with PDGFRA mutations had lower mitotic counts (median, two v four per 50 high power fields; P < .001) and they were more often gastric (88.0% v 54.7%; P < .001; Appendix Table A3). Patients with PDGFRA exon 12 and those with exon 18 mutations had similar RFS (Fig 2B). A total of 33 different mutations were identified in PDGFRA. The most frequent changes were Asp842Val (D842V) and Val561Asp (V561D), which made up 84 (54.9%) and 14 (9.2%) of the 153 defined PDGFRA alterations, respectively.  figure Fig 2. Associations between KIT and platelet-derived growth factor alpha gene (PDGFRA) mutations and recurrence-free survival after surgery for GI stromal tumors (GISTs). (A) GISTs with KIT mutation, PDGFRA mutation, or no mutation in either gene (one patient had mutations in both KIT and PDGFRA); (B) GISTs with either PDGFRA exon 12 or 18 mutation; (C) GISTs with KIT exon 9, 11, 13, or 17 mutation. wt, wild type.  KIT Exons Mutations in different KIT exons were associated with similar RFS, though patients with exon 9 mutations tended to have inferior RFS compared with patients with exon 11 mutations (P = .074; Fig 2C). Approximately 80% of exon 9 mutated GISTs arose in the small bowel (Appendix Table A3). Most GISTs with KIT exon 13 or exon 17 mutations were also nongastric (exon 13: 14 of 16; exon 17: six of 10), whereas 583 (59.4%) of the 982 exon 11–mutated GISTs with the site of origin available were gastric (P < .001). There was no significant difference between GISTs with exon 9, 13, or 17 mutations and those with exon 11 mutations in tumor size or mitotic count (P > .05 for each pair-wise comparison), but exon 13 mutations tended to be associated with smaller mitotic counts (P = .064).  KIT Mutation Types A total of 301 different single KIT mutations were identified. Most (n = 293) were found in exon 11; whereas 97.6% (83 of 85) exon 9 mutations led to Ala502_Tyr503 duplication, 93.3% (14 of 15) exon 13 mutations to Lys642Glu (K642E), and 66.7% (six of nine) exon 17 mutations to Asn822Lys (N822K).  Most frequent of the 981 KIT exon 11 mutations with the mutation type available were deletions (43.8%), followed by substitution mutations (30.6%) and insertion/deletion (indel) mutations (15.8%; Table 1). Deletion of a single codon was found in 71 tumors (7.2%), and occurred only at nine codons (553, 557 to 560, 568, 575, 576, or 579). Most (n = 101; 78.3%) of the 129 deletions of two codons were deletions of 557 and 558. Most exon 11 deletion mutations involved codon 557 and/or 558 (242 of 419; 57.8%), and 317 mutations (75.7%) involved one or more of the codons 557 to 560. The duplication mutations occurred at 3′ of codons 557 and 558 in all cases except one tumor, where a DNA segment located at 5′ of codon 557 was duplicated, and had a median length of ten codons (range, two to 22).  Table\t Table 1. Associations Between the Type of KIT Exon 11 Mutation and GIST Site, Size, and Mitotic Count   Table 1. Associations Between the Type of KIT Exon 11 Mutation and GIST Site, Size, and Mitotic Count  Type of KIT Exon 11 Mutation\tTotal Patients Different Mutations GIST Site Tumor Size (cm) Mitotic Count (per 50 HPFs) Esophagus Stomach Small Intestine Colon or Rectum Non-GI Tract* Site NA (No.) No. of Patients\t%\tNo.\t%\tNo. of Patients\t%\tNo. of Patients\t%\tNo. of Patients\t%\tNo. of Patients\t%\tNo. of Patients\t%\tMedian\tRange\tMedian\tRange Deletion\t430\t43.8\t111\t37.9\t5\t55.6\t254\t44.2\t132\t42.7\t18\t45.0\t15\t42.9\t6\t6.0\t0.1-33.0\t5\t0-250     1 codon\t71\t7.2\t9\t3.1\t1\t11.1\t45\t7.8\t21\t6.8\t2\t5.0\t2\t5.7\t0\t5.2\t0.9-22.0\t3\t0-60     ≥ 2 codons\t359\t36.6\t102\t34.8\t4\t44.4\t209\t36.3\t111\t35.9\t16\t40.0\t13\t37.1\t6\t7.0\t0.1-33.0\t5\t0-250 Insertion-deletion\t155\t15.8\t96\t32.7\t3\t33.3\t73\t12.7\t60\t19.4\t15\t37.5\t2\t5.7\t2\t6.4\t0.7-26.0\t5\t0-250 Duplication\t52\t5.3\t37\t12.6\t0\t0\t45\t7.8\t7\t2.3\t0\t0\t0\t0\t0\t4.5\t1.5-19.0\t3\t0-45 Insertion\t27\t2.8\t13\t4.4\t1\t11.1\t16\t2.8\t4\t1.3\t0\t0\t3\t8.6\t3\t8.0\t0.5-45.0\t4\t0-34 Substitution\t300\t30.6\t19\t6.5\t0\t0\t176\t30.6\t103\t33.3\t7\t17.5\t12\t34.3\t2\t5.5\t0.3-30.0\t3\t0-150 Other†\t17\t1.7\t17\t5.8\t0\t0\t11\t1.9\t3\t1.0\t0\t0\t3\t8.6\t0\t4.5\t0.6-20.0\t2.5\t0-13 Total\t981\t100.0\t293\t100.0\t9\t100.0\t575\t100.0\t309\t100.0\t40\t100.0\t35\t100.0\t13\t6.0\t0.1-45.0\t4\t0-250 NOTE. The type of mutation was not available in 15 (1.5%) of 996 patients with KIT exon 11 mutation. Of the 981 patients with the type of mutation available, data were not available for the tumor site in 13 patients (1.3%), exact (noncategorized) tumor size in 32 patients (3.3%), and exact (noncategorized) mitotic count in 192 patients (19.6%).  Abbreviations: GIST, GI stromal tumor; HPF, high power field of the microscope; NA, not available.  *Includes mesentery (n = 8), peritoneum (n = 3), retroperitoneum (n = 8), and other (n = 10) intra-abdominal sites, and 6 patients classified as non-GI tract GIST.  †Includes six tumors with two substitution mutations, six with a duplication mutation and an insertion mutation, four with a deletion mutation and a substitution mutation, and one with an insertion-deletion mutation and a substitution mutation.  Insertion-deletion mutations were more frequent in nongastric than gastric GISTs (20.4% v 12.7%, respectively; P = .001), whereas exon 11 duplication mutations were more common in gastric than nongastric GISTs (7.8% v 1.8%, respectively; P < .001; Table 1). GISTs with an exon 11 duplication mutation and those with a substitution mutation were smaller than GISTs with an exon 11 deletion mutation (P = .012 and P = .017, respectively), and GISTs with a substitution mutation also had lower mitotic counts (P = .004). GISTs with deletions of only one codon were smaller than those with deletions of two or more codons (P = .045) and had lower mitotic counts (P = .004).  Only one (2.9%) of the 35 patients with KIT exon 11 duplication mutations and RFS data available had GIST recurrence, whereas GISTs with a deletion mutation or an insertion-deletion mutation recurred often (Fig 3A). Patients with only one deleted exon 11 codon had more favorable RFS than those with a longer deletion or an insertion-deletion mutation (Fig 3B). Mutations that involved codons 557 or 558 were associated with unfavorable RFS (Fig 3C).  figure Fig 3. Associations between KIT and PDGFRA mutations and recurrence-free survival after surgery for GI stromal tumors. (A) KIT exon 11 mutation types; (B) KIT exon 11 deletion types; (C) mutations involving KIT exon 11 codons 557 and/or 558; (D) the three most common KIT alterations; (E) common KIT substitution mutations; and (F) PDGFRA mutations.  Common Single Mutations Deletion of Trp557_Lys558 (KIT exon 11), substitution of Asp842Val (PDGFRA exon 18), and duplication of Ala502_Tyr503 (KIT exon 9) were the most common single alterations (Table 2). Besides GISTs with Ala502_Tyr503dup, GISTs with Lys642Glu also frequently arose from the small intestine. GISTs with Trp557_Lys558del were usually large when detected. Tumors with Asp842Val, Leu576Pro, Trp557Arg, or Lys642Glu, and those with Val560del had small median mitotic counts.  Table\t Table 2. Frequent KIT or PDGFRA Alterations   Table 2. Frequent KIT or PDGFRA Alterations  Protein Alteration\tGene and Exon\tGISTs (n = 1,505) GIST Site* Tumor Size (cm) Mitotic Count (per 50 HPFs) Stomach Small Intestine Colon or Rectum† Non-GI Tract Site NA (No.) No. of Tumors\t%\tNo. of Patients\t%\tNo. of Patients\t%\tNo. of Patients\t%\tNo. of Patients\t%\tMedian\tRange\tMedian\tRange Trp557_Lys558del\tKIT, exon 11\t101\t6.7\t65\t65.6\t24\t24.2\t7\t7.1\t3\t3.0\t2\t8.0\t0.6-30.0\t6\t0-200 Asp842Val (D842V)‡\tPDGFRA, exon 18\t84\t5.6\t69\t88.5\t2\t2.6\t2\t2.6\t5\t6.4\t6\t5.0\t0.6-30.0\t2\t0-200 Ala502_Tyr503dup\tKIT, exon 9\t83\t5.5\t9\t10.8\t68\t81.9\t3\t3.6\t3\t3.6\t0\t6.0\t0.8-32.0\t3\t0-115 Val559Asp (V559D)\tKIT, exon 11\t70\t4.7\t50\t72.5\t18\t26.1\t0\t0\t1\t1.4\t1\t5.3\t0.8-30.0\t4\t0-150 Val560Asp (V560D)§\tKIT, exon 11\t51\t3.4\t35\t70.0\t13\t26.0\t2\t4.0\t0\t2.0\t1\t6.0\t0.7-25.0\t4\t0-33 Trp557Arg (W557R)\tKIT, exon 11\t50\t3.3\t32\t64.0\t13\t26.0\t1\t2.0\t4\t8.0\t0\t4.5\t0.4-13.0\t2\t0-70 Val560del (equal to Val559del)\tKIT, exon 11\t41\t2.7\t26\t63.4\t13\t31.7\t1\t2.4\t1\t2.4\t0\t5.0\t1.0-22.0\t1\t0-60 Leu576Pro (L576P)\tKIT, exon 11\t28\t1.9\t12\t42.9\t14\t50.0\t0\t0\t2\t7.1\t0\t4.5\t0.3-19.0\t0\t0-14 Val559Ala (V559A)§\tKIT, exon 11\t20\t1.3\t10\t50.0\t9\t45.0\t0\t0\t1\t5.0\t0\t6.0\t1.5-25.0\t4\t1-21 Trp557Gly (W557G)‖\tKIT, exon 11\t19\t1.3\t9\t47.4\t8\t42.1\t1\t5.3\t1\t5.3\t0\t7.3\t0.5-22.0\t5\t0-150 Lys642Glu (K642E)\tKIT, exon 13\t14\t0.9\t2\t14.3\t12\t85.7\t0\t0\t0\t0\t0\t5.25\t1.2-15.0\t1\t0-15 Abbreviations: GIST, GI stromal tumor; HPF, high power field of the microscope; NA, not available; PDGFRA, platelet-derived growth factor alpha gene.  *None of the GISTs presented in the esophagus.  †All but five GISTs were rectal (the five colonic GISTs had either Trp557_Lys558del, Asp842Val, Trp557Arg, Val560del, or Trp557Gly).  ‡Two additional patients had Asp842Tyr.  §Fifteen additional patients had Val559Gly, nine had Val560Gly, nine had Val560Glu, and one had Val560Gln.  ‖Three additional patients had Trp557Lys.  The three most common single alterations (Trp557_Lys558del, Ala502_Tyr503dup, and Val559Asp) were associated with similar RFS (Fig 3D). Of the exon 11 substitution alterations, Trp557Arg, Leu576Pro, and Val559Ala tended to have the best RFS, whereas Val559Asp and Trp557Gly the worst (Fig 3E). Mutations causing substitution alterations at Asp842 had RFS similar to other PDGFRA mutations (Fig 3F).  Different substitution mutations at the same codon were not necessarily associated with similar clinical features. GISTs with Val559Asp were more often gastric than those with Val559Ala or Val559Gly (72.5% v 42.9%; P = .003), and patients with Val559Ala tended to have more favorable RFS than those with Val559Asp (P = .098). Patients with Trp557Arg had lower mitotic counts than those with Trp557Gly (median, two v five, respectively; P = .016) and better RFS (P = .016; Fig 3E; Table 2).  Multivariable Survival Analyses When the mutated gene (KIT v PDGFRA or wild type) was included as a covariable in a Cox multivariable hazards model, with tumor size (as a continuous variable), site (stomach v another site in the GI tract v non-GI tract location), and mitotic count (as a continuous variable), the presence of KIT mutations had an independent adverse influence on RFS (P = .011; HR, 1.89; 95% CI, 1.15 to 3.08); but mitotic count, size, and gastric site were each stronger prognostic factors (mitotic count: HR, 1.018; 95% CI, 1.015 to 1.022; size: HR, 1.10; 95% CI, 1.07 to 1.12; gastric site: HR, 0.45; 95% CI, 0.33 to 0.62; each P < .001). When the analysis was restricted to KIT-mutated GISTs and the mutated exon (9, 11, 13, or 17) was entered together with tumor size, site, and mitotic count into the model, the mutated exon was not independently associated with RFS (Appendix Table A4).  Patients with an identical mutation had greatly variable outcomes depending on the standard prognostic factors, notably, tumor mitotic count (Fig 4). When tumor size and mitotic count were entered as continuous covariables in a Cox multivariable model in the subset of patients with KIT exon 11 mutation leading to Trp557-Lys558del, both were independently associated with RFS (HR, 1.10; 95% CI, 1.04 to 1.17; P = .001; and HR, 1.02; 95% CI, 1.003 to 1.03; respectively; P = .012,). Similarly, tumor size was an independent prognostic factor for RFS in the subset of patients with Val559Asp and mitotic count in the subset with Ala502-Tyr503dup (data not shown).  figure Fig 4. Influence of tumor mitotic count on recurrence-free survival in subgroups defined by tumor mutation analysis. (A) GI stromal tumors with a KIT exon 11 deletion mutation; (B) Trp557_Lys558del; (C) a KIT exon 11 substitution mutation; (D) Val559Asp; (E) Ala502_Tyr503dup; or (F) PDGFRA mutation.  AFIP and NIH Schemes Finally, we investigated how the AFIP and the modified NIH schemes stratify the risk of recurrence in the subsets of patients with a mutation type associated with favorable RFS (Table 3). Approximately 30% of such patients were stratified into a high risk category and an additional 20% into an intermediate risk category, suggesting that these risk-stratification schemes may overestimate the risk of GIST recurrence when the mutational type is favorable. When the presence of favorable mutation (Table 3) was entered as a covariable into a Cox multivariable hazards model (favorable v other mutation) with either the AFIP scheme or the modified NIH scheme, the mutation type and the AFIP scheme (Appendix Table A5), and the mutation type and the modified NIH scheme (Appendix Table A6), each independently influenced RFS. After stratifying the GISTs using the prognostic heat maps,4 8.3% to 23.3% of the patients with one of the favorable mutations had 21% to 40% estimated risk for recurrence within the first 10 years after surgery, and 8.3% to 22.4% had a greater than 40% risk (Appendix Table A7).  Table\t Table 3. Risk Groups of Patents With Prognostically Favorable KIT or PDGFRA Mutation   Table 3. Risk Groups of Patents With Prognostically Favorable KIT or PDGFRA Mutation  Risk Group\tKIT Exon 11, Duplication Mutation (5-year RFS, 96.9%) KIT Exon 11, Deletion of 1 Codon (5-year RFS, 82.2%) KIT Exon 11 Mutation, Leading to Trp557Arg, Val559Ala, or Leu576Pro (5-year RFS, 82.5% to 95.2%) PDGFRA Mutation (any; 5-year RFS, 82.2%) Total No. of Patients\t%\tNo. of Patients\t%\tNo. of Patients\t%\tNo. of Patients\t%\tNo. of Patients\t% AFIP risk group\t\t\t\t\t\t\t\t\t\t     1 or 2 (low risk)\t20\t54.1\t20\t46.5\t31\t56.4\t35\t45.5\t106\t50.1     3a (intermediate risk)\t9\t24.3\t12\t27.9\t11\t20.0\t18\t23.4\t50\t23.6     3b, 4, 5, 6a, or 6b (high risk)\t8\t21.6\t11\t25.6\t13\t23.6\t24\t31.1\t56\t26.4 Modified NIH risk group\t\t\t\t\t\t\t\t\t\t     Very low or low\t20\t54.1\t20\t44.4\t31\t47.0\t36\t46.7\t107\t49.5     Intermediate\t9\t24.3\t6\t13.3\t7\t10.6\t17\t21.8\t39\t18.1     High\t8\t21.6\t19\t42.2\t18\t27.3\t25\t32.1\t70\t32.4 Abbreviations: AFIP, the Armed Forces Institute of Pathology; NIH, the National Institutes of Health; PDGFRA, platelet-derived growth factor alpha gene; RFS, recurrence-free survival.  DISCUSSION  Mutation analysis of KIT and PDGFRA provided potentially important prognostic information. As suggested by others,34 patients with PDGFRA mutations had substantially better RFS than those with KIT mutations, and tumor KIT mutations were associated with an increased risk of GIST recurrence in a multivariable analysis. Besides PDGFRA mutations, we identified other mutational subgroups with favorable RFS. Patients with exon 11 duplication mutations had a low risk for recurrence, suggesting that these patients are generally not candidates for adjuvant treatments. Other mutational subgroups with favorable RFS (who were not identified earlier) included patients with one codon deletion mutation at KIT exon 11 and substitution mutations leading to Trp557Arg, Val559Ala, or Leu576Pro. The most commonly used risk-stratification schemes overestimated the risk of GIST recurrence in patients with one of these generally favorable mutation types, and presence of a favorable mutation had independent influence on RFS beyond that provided by the AFIP or the modified NIH schemes in multivariable analyses.  The standard prognostic factors, the mitotic count in particular, generally predict the risk of recurrence more accurately than GIST mutation status.7 We demonstrate that the standard factors, such as the mitotic count, have a great impact on outcome despite an identical KIT or PDGFRA mutation. Of note, patients with a generally unfavorable mutation, such as KIT exon 9 mutation leading to Ala502_Tyr503 duplication or exon 11 deletion mutation causing Try557_Lys558 deletion, were still at low risk for GIST recurrence, provided that the mitotic count was very low. In contrast, patients with PDGFRA mutations had a high risk of recurrence when tumor mitotic count was high. Therefore, tumor mutation status should not be interpreted in isolation from other prognostic factors. This observation is well in line with studies on micro-GISTs, which are common and clinically benign and consist of less than 10 millimeter lesions with few or no mitotic figures, but yet often contain KIT or PDGFRA mutations.35,36  The relative influence of the common Asp842Val alteration and other PDGFRA alterations on RFS has been unknown. Our current data suggest that the risk of recurrence is generally low regardless of the type of PDGFRA mutation. Similarly, there are few prior data about the risk associated with KIT exon 13 and 17 mutations. In keeping with a joint analysis of 54 patients with such mutations,37 most GISTs with KIT exon 13 and 17 mutation are intestinal but, despite this, the outcome of the patients was not inferior to that of the patients with other KIT mutations. Approximately 70% of KIT exon 17 mutations lead to Asn822Lys,37 which is considered only moderately imatinib sensitive.38 It may no longer be reasonable to group all KIT exon 11 mutations together in studies on prognosis, since exon 11 mutations are associated with variable outcomes and clinical features. Our present findings suggest that even mutations at a single codon leading to substitutions with different amino acid residues may be associated with different risks of recurrence.  Our study has limitations. We collected population-based data in an attempt to avoid a selection bias, but there is some heterogeneity in the prognostic factors between the series,4 and tumor-mutation analysis was available in only 49.1% of 3,067 patients. We found no evidence of a selection bias in the main prognostic factors but cannot exclude this. Some mutations may have been missed, and exons in which mutations occur rarely were not sequenced, which may have inflated the numbers of GISTs with no mutations found. Tumor-rupture data were frequently unavailable and were not assessed. The mitotic counts were available per one microscope high power field, which vary in size depending on the microscope. To our knowledge, our current series is the largest on mutation analyses in GISTs reported, but despite this the numbers of patients remained relatively small in some subgroup analyses, often because of a lack of RFS information.  We conclude that GIST mutation analyses provide clinically useful prognostic information in a patient population treated with surgery alone, but the standard factors are generally stronger for prognosis and are associated with RFS even in subsets of patients with an identical KIT or PDGFRA mutation. Yet, the mutational data are important to consider when a prognostically favorable mutation is found, such as a duplication mutation in KIT exon 11. Most gastrointestinal stromal tumors (GIST) have an activating mutation in either KITor PDGFRA. Imatinib is a selective tyrosine kinase inhibitor and achieves a partial response or stable disease in about 80% of patients with metastatic GIST. It is now clear that some patients with GISTdevelop resistance to imatinib during chronic therapy.To identify the mechanism of resistance, we studied 31patients with GISTwho were treated withimatinib and then underwent surgical resection.There were13 patients whowere nonresistant toimatinib, 3 withprimary resistance, and15 with acquired resistance afterinitial benefit from the drug.There were no secondary mutations in KITor PDGFRA in the nonresistant or primary resistance groups. In contrast, secondary mutations were found in 7 of15 (46%) patients with acquired resistance, each of whomhad a primary mutationinKITexon11. Most secondary mutations were located in KITexon 17. KIT phosphorylation was heterogeneous and did not correlate with clinical response to imatinib or mutation status.That acquired resistance to imatinib in GISTcommonly occurs via secondary gene mutation in the KIT kinase domain has implications for strategies to delay or prevent imatinib resistance and to employ newer targeted therapies. Gastrointestinal stromal tumor (GIST) is the most common mesenchymal tumor of the intestinal tract. After surgical removal of primary GIST, tumor recurrence is frequent, and previously, about 50% of patients died within 5 years of initial diagnosis (1). In 1998, Kitamura and colleagues reported that GIST is commonly associated with activating mutations in the KIT receptor tyrosine kinase (2). More recently, we showed that a knock-in KIT activation mutation, KITV558D/+, is sufficient to induce GIST in mice (3). Imatinib mesylate (STI571, Gleevec, Novartis Pharmaceuticals, Basel, Switzerland) is an oral agent that specifically inhibits the BCR-ABL and ABL tyrosine kinases as well as the KIT and PDGFR receptor tyrosine kinases (4). Initially, imatinib was applied to the treatment of chronic myelogenous leukemia (CML), where it achieves a 90% complete response rate (5, 6). Subsequently, imatinib was tested in metastatic or unresectable GIST and found to induce a partial response or stable disease in >80% of patients (7, 8). Primary resistance to imatinib, in which GIST continues to progress despite the institution of therapy, occurs in about 15% of patients. Remarkably, the 2-year survival of patients with metastatic GIST is now f70% (9). In up to 90% of cases, GISTs have activating mutations in either the KIT or PDGFRA receptor tyrosine kinases (10–13). The most common site of KIT mutation is in the 5V end of the exon (11), which encodes the juxtamembrane domain, and usually deletions or substitutions of codons 550 to 560 occur. KIT exon 9 mutation occurs in 10% to 15% of patients. It defines a distinct subset of GISTs that are often located in the small bowel and have an aggressive clinical behavior (11, 14). Infrequently, a mutation is identified in KIT exon 13 or 17 (15, 16). Another 5% of patients with GIST have a PDGFRA mutation that typically involves exon 12 or 18 (12, 13). About 10% of patients do not have a detectable mutation in either KIT or PDGFRA. In particular, GISTs that occur in pediatric patients are nearly always wild-type for both genes (11A). Recent data suggest a possible correlation between imatinib response and the type of mutation, as tumors with an exon 9 mutation or wild-type KIT are less likely to respond to imatinib (17, 18). Although most patients with advanced GIST benefit from imatinib treatment, it is now clear that many patients subsequently develop resistance to the agent. The median time to progression is about 24 months (9). The mechanism of acquired resistance to imatinib in GIST has not been well defined. In CML, second site mutation in BCR-ABL is the predominant mechanism of imatinib resistance (19, 20). Therefore, we postulated that acquired resistance to imatinib in GIST is due to secondary site mutation in the KIT or PDGFRA genes. We did molecular analysis of 65 tumor nodules from 31 patients who were treated with imatinib. Clin Cancer Res 2005;11(11) June 1, 2005 4182 www.aacrjournals.org Authors’ Affiliations: Departments of 1 Pathology, 2 Surgery, and 3 Medicine, 4 Memorial Sloan-Kettering Cancer Center and Developmental Biology, 5 Cell Biology, and 6 Structural Biology and Programs, Sloan-Kettering Institute, New York, NewYork Received 11/4/04; revised 1/16/05; accepted 3/4/05. Grant support: ACS MRSG CCE-106841 (C.R. Antonescu); CA102774 and HL/ DK55748 (P. Besmer); PO1 CA 47179 (M.F. Brennan); CA94503 and CA102613 (R.P. DeMatteo). The costs of publication of this article were defrayed in part by the payment of page charges.This article must therefore be hereby marked advertisement in accordance with 18 U.S.C. Section 1734 solely to indicate this fact. Requests for reprints: Cristina R. Antonescu, Department of Pathology, Memorial Sloan-Kettering Cancer Center, 1275 York Avenue, New York, NY 10021. Phone: 212-639-5721; Fax: 212-717-3203; E-mail: antonesc@mskcc.org. F2005 American Association for Cancer Research. Cancer Therapy: Preclinical Research. Downloaded from clincancerres.aacrjournals.org on May 29, 2017. © 2005 American Association for Cancer Three patients showed primary resistance to imatinib, whereas 15 patients acquired resistance to imatinib during therapy. Materials and Methods Clinicopathologic analysis. Patients with the diagnosis of GIST who were treated with imatinib and underwent surgical resection of their tumor at Memorial Sloan-Kettering Cancer Center were identified from our prospective sarcoma database. There were 31 patients who had adequate tumor tissue for molecular analyses following treatment. Patient, tumor, and treatment information was obtained from the database and by reviewing medical charts. Primary resistance to imatinib was defined as continued growth of any tumor despite the institution of imatinib therapy. Acquired resistance was defined as new tumor growth that occurred subsequent to an initial period during 2 to 3 months on imatinib in which the patient had stable or responding disease. Tumor size was determined by computed tomography scan or magnetic resonance imaging. This study was approved by the Institutional Review Board. Pathologic material was examined and the diagnosis was confirmed using standard H&E staining and CD117 immunohistochemistry on formalin-fixed, paraffin-embedded tissue as previously described (11). Histologic response to imatinib was based on gross and microscopic findings of necrosis and fibrosis and was scored for each tumor nodule as: minimal (<10% response), low (10-50% response), moderate (50- 90% response), or high (z90% response) degree of response. KIT/PDGFRA genotyping. Mutation analysis was done as described previously (11). Genomic DNA was isolated from snap-frozen tumor tissue samples stored at  70jC, using a standard phenol-chloroform organic extraction protocol. Adequate DNA for mutational analysis was obtained in 65 tumor nodules from 31 patients. All cases were tested for the known sites of KIT (exons 9, 11, 13, 14, and 17) and PDGFRA (exons 12 and 18) mutations. One microgram of genomic DNA was subjected to PCR using Platinum TaqDNA Polymerase High Fidelity (Life Technologies, Inc., Gaithersburg, MD). Primer sequences and annealing temperatures were as described (11), with the addition of primers for KIT exon 14 (GTCTGATCCACTGAAGCTG at 50jC and ACCCCATGAACTGCCTGTC at 51jC), PDGFRA exon 12 (TCCAGTCACTGTGCTGCTTC and GCAAGGGAAAAGGGAGTCTT at 54jC), and PDGFRA exon 18 (ACCATGGATCAGCCAGTCTT and TGAAGGAGGATGAGCCTGACC at 55jC). Direct sequencing of PCR products was done for all exons tested and each ABI sequence was compared with the National Center for Biotechnology Information human KIT and PDGFRA gene sequences. In four patients, including three with acquired resistance, there was adequate tumor tissue available for mutation analysis from a surgical resection that occurred prior to initiation of imatinib. In each case, including one patient who developed a second site mutation, we confirmed the same primary mutation as in the recurrence. cDNA sequencing and cloning. For tumors with secondary KIT mutations by genomic DNA analysis, we then amplified and sequenced the KIT cDNA from exons 10 to 18. In this way, we confirmed our initial genotype results and also determined whether the secondary mutation occurred on the same or opposite allele as the primary mutation. Adequate RNA was obtained in 32 tumor nodules from 18 patients by using the RNA Wiz reagent (Ambion, Inc., Austin, TX) and the guanidinium isothiocynate-phenol chloroform method. Five micrograms of RNA were transcribed using reverse transcriptase superscript II (Invitrogen, Carlsbad, CA). The cDNA was subjected to PCR using primers in KIT exon 10 (GCCGGATCCTTTCGTAATCGTAGCT) containing a BamHI linker and exon 18 (GGCGAATTCTGTATACACAGTTGAAAATG) with an EcoRI linker. Amplification of the KIT insert was done separately with Platinum Taq (Invitrogen) and Pfu ultra high-fidelity DNA polymerase (Stratagene, La Jolla, CA) in order to exclude false-positive mutations due to errors in polymerase proofreading. For cloning, the KIT insert and the chloramphenicolresistant pBC KS+ vector (Stratagene) were digested with BamHI and EcoRI and then ligated using the rapid DNA ligation kit (Roche Applied Science, Penzburg, Germany). Plasmid transformation of XL-1 blue supercompetent cells (Stratagene) and colony selection on chloramphenicol Luria-Bertani agar plates were done using standard methods. For each sample, 10 distinct clones were picked and expanded overnight in a shaking incubator. Colony PCR was done using Taq polymerase and the same KIT exon 10 and 18 linker primers. Clones containing the insert were selected and DNA was isolated with a DNA Miniprep kit (Qiagen, Inc., Valencia, CA) and then directly sequenced. Western blotting. Adequate tissue for protein extraction was available in 24 patients (43 tumor nodules). We also included for comparison 8 untreated GISTs with KIT exon 11 mutations, 4 of which were in-frame deletions, and 4 were substitutions. A CD117-negative and desmin-positive high-grade gastric leiomyosarcoma was included as a negative control. For preparation of whole lysates, 1 g of snapfrozen tumor was ground to powder in liquid nitrogen using a PowerGen 700 Homogenizer (Omni International, Marietta, GA), and resuspended in radioimmunoprecipitation assay lysis buffer (Upstate, Lake Placid, NY) containing a cocktail of protease and phosphatase inhibitors (Sigma, St. Louis, MO). Protein concentrations were determined with the Bio-Rad Protein Assay (Bio-Rad Laboratories, Hercules, CA). Protein extracts were subjected to electrophoresis and immunoblotting using a standard protocol. Antibodies included mouse anti-phospho-Tyr20 and Tyr99 (Santa Cruz Biotechnology, Santa Cruz, CA), rabbit polyclonal anti-phospho-KIT Y721 (Zymed Lab, Inc., San Francisco, CA), rabbit anti-KIT (Oncogene Science, Boston, MA), mouse anti-actin (Santa Cruz), donkey anti-mouse secondary antibody (Santa Cruz), and anti-rabbit secondary antibody (Calbiochem, La Jolla, CA). Blots were incubated with Immun-Star horseradish peroxidase luminol/ enhancer (Bio-Rad) and exposed to Kodak BioMax MR Film (Eastman Kodak Company, Rochester, NY). Results with phospho-KIT were scored as: 1, strong; 2, weak; and 3, negative. Fluorescence in situ hybridization analysis. Touch preps of frozen tumor tissue were done in nine nodules from eight patients. The slides were fixed in 3:1 methanol/acetic acid then stored at  20jC. Fluorescence in situ hybridization (FISH) was done according to standard procedures. Briefly, the slides were pretreated with pepsinHCl (0.007 mol/L HCl, 8 Ag/mL pepsin) at 37jC for 3 to 5 minutes, rinsed in PBS, fixated in 1% formaldehyde for 10 minutes, then rinsed, dehydrated, and air-dried. The slides were then denatured in 70% formamide at 68jC for 2 to 4 minutes, quenched, dehydrated, and airdried. The KIT probes used were two overlapping BAC clones: CTD- 3180G20 and RP11-722F21 (Invitrogen), labeled by nick-translation with Spectrum Green (Vysis, Abbott Laboratories, IL). A chromosome 4 centromeric probe labeled with Spectrum Orange (CEP 4, Vysis) was used as reference. The probe mix, 50 to 80 ng of each KIT BAC and 2 AL Cot-1 DNA (Invitrogen), was ethanol-precipitated, and resuspended in hybridization buffer. The KIT probe mix was denatured at 70jC for 10 minutes, followed by pre-annealing at 37jC for 30 minutes. The KIT probe was then combined with the denatured CEP 4 probe on the slide, coverslipped and incubated overnight at 37jC. After standard posthybridization washes, the slides were stained with 4V,6-diamidino-2- phenylindole and mounted in antifade (Vectashield, Vector Laboratories). Analysis was done using a Nikon E800 epifluorescence microscope with MetaSystems Isis 3 imaging software. A minimum of 100 cells was scanned over separate regions for each slide. Results Imatinib resistance may develop during chronic imatinib therapy. We identified 31 patients with GIST who were treated with imatinib and subsequently underwent surgical resection. There were 8 females and 23 males, and the median www.aacrjournals.org Clin Cancer Res 2005;11(11) June 1, 2005 4183 Acquired Imatinib Resistance in GIST Research. Downloaded from clincancerres.aacrjournals.org on May 29, 2017. © 2005 American Association for Cancer age at initial diagnosis was 56 (range 31-84) years. The extent of disease at the original diagnosis of GIST was localized in 19 patients and metastatic in 12. Disease at the start of imatinib therapy consisted of a primary tumor alone in 6 patients, a primary GIST with liver and/or peritoneal metastasis in 7 patients, and recurrent liver and/or peritoneal disease in the remaining 18 patients. Imatinib was administered at a starting dose of 400 to 600 mg per day. Based on the responsiveness of their disease to imatinib at the time of surgery, we categorized patients into ‘‘nonresistant,’’ ‘‘primary resistance,’’ or ‘‘acquired resistance.’’ Elective removal of residual GIST following either a partial response or stable disease on imatinib was done in the 14 nonresistant patients (Table 1). Their median duration of imatinib therapy was 8 (1-22) months. Surgery was done in 3 patients with primary resistance to imatinib. Each had progression of disease detected on computed tomography within 3 months of starting therapy (Table 2). There were 15 patients who developed acquired resistance. Each had either a partial response (n = 10) or stable disease (n = 5) in response to imatinib and then subsequently acquired resistance and radiologic progression of disease. The median length of imatinib treatment in patients with acquired resistance was 18 (8-32) months (Table 2). One of these patients (#14) first had an operation to remove stable disease and later underwent another resection to remove resistant disease. Two patients required surgery because of a complication during imatinib therapy, involving either bleeding or perforation. Histologic response correlates with clinical response to imatinib. The primary tumor was located in the stomach in 8 patients, small bowel in 17, rectum in 5, and intra-abdominal in 1 patient. The median size of the primary GIST was 9.5 (2.6-24) cm. There were 13 tumors larger than 10 cm. The morphology of the primary tumor was spindle-shaped in 25 cases, epithelioid in 4, and mixed in 2 tumors. Histologic response was graded based on the combined extent of tumor necrosis and fibrosis, because both are usually absent in untreated GIST, although central necrosis may be present in large tumors. Patients with primary resistance uniformly lacked any histologic evidence of response to imatinib (patients 29-31, Table 2). Six of 14 patients in the nonresistant group had a histologic response of >90%. Moderate histologic responses (50-90%) occurred in four others. Only three nonresistant patients had a minimal histologic response despite a clinical partial response or stable disease. In contrast, in the acquired resistant group, all but two patients had had at least one nodule lacking any response (Table 2). Secondary KIT mutation occurs in acquired imatinib resistance. In the nonresistant group, there were 11 patients with a KIT mutation (3 in exon 9, 7 in exon 11, and 1 in exon 13), 1 patient with a PDGFRA mutation (exon 18), and 2 were wildtype. There were no identifiable secondary mutations in the nonresistant group. However, there was one patient (#2) who had a homozygous exon 11 deletion in 6 of 15 nodules tested. The significance of this finding is unclear as the patient had stable disease and currently has no evaluable tumor. As expected, the 3 patients with primary resistance to imatinib also had only a single mutation, which involved KIT exon 9 or 11 or PDGFRA exon 18. Of the 15 patients with acquired resistance to imatinib, 14 had a common primary KIT mutation (11 in exon 11 and 3 in exon 9) and 1 patient was wild-type. Secondary mutations Clin Cancer Res 2005;11(11) June 1, 2005 4184 www.aacrjournals.org Table 1. Patients with nonresistant GIST Primary tumor location Histology Disease at imatinib start Time on imatinib* (months) Best clinical response Histologic response Primary mutation Secondary mutation PhosphoKIT Y721 1 SB S P, L22 PR 2, 3 Ex 11VYIDPTQL569-576del absent (0 of 2) 2, 3 2Ac SB S P, PE, L15 PR 2, 3 Ex 11P551HMY552-554del absent (0 of 6) LOH (3 of 6) 1 2B L20 PR 1, 2 Ex 11P551HMY552-554del absent (0 of 3) LOH (3 of 3) 3 3A SB S PE 12 SD 1 Ex 9 INS502AY absent (0 of 1) NA 3Bc PE 18 SD 1 Ex 9 INS502AY absent (0 of 3) 1, 3 4c SB S P, L3 SD 1 Ex 9 INS502AY absent (0 of 3) 1 5 rectum S P 9 PR 4 WT absent (0 of 1) NA 6 SB S P 4 SD 2 Ex 9 INS502AY absent (0 of 1) 1 7 rectum S P 6 PR 4 Ex 11VQWKV555-559del absent (0 of 1) 3 8 SB S P 2 PR 1 Ex 11K558N, INS559Q absent (0 of 1) NA 9 SB S PE, L1 PR 3, 4 WT absent (0 of 1) NA 10 stomach S P 8 PR 4 Ex 11WK557-558del absent (0 of 1) NA 11 stomach E PE 5 SD 2, 3 PDGFRA D842V absent (0 of 2) 1 12 SB S P, PE, L9 PR 4 Ex 11INGNNYVYIDPTQL563-576del absent (0 of 2) 3 13 rectum S P 8 PR 3 Ex 13 K642E absent (0 of 1) 2 14A SB S PE 3 PR 4 Ex 11VYIDPTQL569-576del absent (0 of 1) 2 Abbreviations: SB, small bowel; S, spindle; E, epithelioid; P, primary tumor; PE, peritoneal; L, liver; PR, partial response; SD, stable disease; del, deletion; LOH, loss of heterozygosity; INS, insertion; NA, not available. *Time from the start of imatinib to surgical resection. c Tested for KITamplification by FISH. Cancer Therapy: Preclinical Research. Downloaded from clincancerres.aacrjournals.org on May 29, 2017. © 2005 American Association for Cancer were detected in 10 tumor nodules samples from seven (46%) patients and involved five different residues. The majority were KIT exon 17 mutations and occurred in six of seven patients with secondary mutations. All involved substitutions and three were N822K, two were D820Y, and one was Y823D (Table 2; Fig. 1). The other two secondary mutations included substitutions in exon 13 (V654A) and exon 14 (T670I). We again identified loss of the wild-type allele (loss of heterozygosity), WK557-8del in one patient (#28, one of three nodules tested) as we had in one nonresistant patient (#2). In 10 of 15 patients with acquired resistance, more than one nodule was available for molecular analysis. In two of them an identical secondary mutation was identified in both nodules tested. In three patients, the secondary mutation was present on one but not all nodules. Interestingly, in one other patient (#24) a nodule showed an exon 13 V654A mutation, whereas a subsequent nodule removed 7 months later showed an exon 17 www.aacrjournals.org Clin Cancer Res 2005;11(11) June 1, 2005 4185 Table 2. Patients with either acquired or primary resistance to imatinib Primary tumor location Histology Disease at imatinib start Time on imatinib* (months) Best clinical response Histologic response Primary mutation Secondary mutation PhosphoKIT Y721 14Bc SB S PE 29 PR 1 Ex 11VYIDPTQL569-576del D820Y (2 of 2) 2 15 rectum S + E CW 28 SD 1 Ex 11KVVEE558-562del-LOH Y823D (1of 1) 3 16c SB S PE 13 PR 1 Ex 11L576P absent (0 of 1) 1 17c rectum S PE, L15 PR 1 Ex 9 INS502AY absent (0 of 1) 1 18 SB S PE 26 SD 1, 2 Ex 9 INS502AY absent (0 of 4) 1, 2 19 stomach S PE, L27 PR 1, 4 Ex 11WK557-8del T670I (2 of 2) 1, 2 20 I-abd S PE, L13 PR 1 Ex 11V559D absent (0 of 2) 2 21 stomach E PE 8 SD 1, 2 WT absent (0 of 1) 2 22c stomach S + E PE 25 PR 2, 4 Ex 11K558N, INS559P N822K (1of 3) 2 23 SB S L14 SD 1 Ex 9 INS502AY absent (0 of 1) NA 24A SB S PE 9 PR 1, 2 Ex 11V560del V654A (1of 2) 2 24B PE 16 PR 1 Ex 11V560del N822K (1of 2) 1 25 SB S L32 SD 2, 4 Ex 11V559D D820Y (1of 2) 3 26 stomach E L20 PR 1 Ex 11INS574TQLPYD absent (0 of 2) NA 27 SB S P, L16 PR 1, 3 Ex 11NGNNYVYIDPTQLPY564- 578del N822K (1of 2) 1 28 stomach S P, PE, L25 PR 1, 4 Ex 11WK557-558del absent (0 of 3) LOH (1of 3) 2 29 stomach E L12 PrimR 1 PDGFRA D842V absent (0 of 1) NA 30 SB S P, L3 PrimR 1 Ex 11PY577-558del absent (0 of 1) 1 31c SB S PE, L8 PrimR 1 Ex 9 INS502AY absent (0 of 2) 1 Abbreviations: SB, small bowel; S, spindle; E, epithelioid; P, primary tumor; PE, peritoneal; L, liver; PR, partial response; SD, stable disease; del, deletion; LOH, loss of heterozygosity; INS, insertion; NA, not available; I-abd, intra-abdominal; PrimR, primary resistance; CW, chest wall. *Time from the start of imatinib to surgical resection. cTested for KITamplification by FISH. Fig. 1. Primary and secondary KIT mutations in seven patients who acquired resistance to imatinib. All the primary mutations occurred in exon 11 and they tended to be deletions. Secondary mutations were detected predominantly in exon 17 but also in exons 13 and 14.The wild-type sequence is shown for comparison. Numbers in the first column refer to the individual patients. Acquired Imatinib Resistance in GIST Research. Downloaded from clincancerres.aacrjournals.org on May 29, 2017. © 2005 American Association for Cancer N822K substitution. The remaining four patients had only a primary mutation identified. To confirm the secondary mutations found at the genomic level and to exclude the possibility of polyclonal resistance and/ or the presence of additional KIT mutations that were missed, amplification of the KIT cDNA from exons 10 to 18 was done followed by subcloning and direct sequencing. A total number of 32 samples from 18 patients were analyzed, including 2 preimatinib samples, 13 nonresistant nodules from 6 patients, 14 samples from 9 patients with acquired resistance and 3 nodules from 2 primary resistance patients. This analysis included 10 nodules from 5 patients with acquired resistance and second site mutations. Secondary mutations were confirmed in 7 of 7 nodules and the absence of mutation in the 3 other nodules. No additional KIT mutations were detected. Notably, the primary and secondary mutations were all located on the same allele. KIT activation is variable regardless of response to imatinib. KIT activation as measured by phosphorylation was heterogeneous and did not consistently correlate with histologic response or response to imatinib (Table 1; Fig. 2A). Surprisingly, from the 11 operations in 10 nonresistant patients tested, only three specimens lacked evidence of KIT activation. Meanwhile, five nonresistant patients had strong KIT activation (Fig. 2A) and three had moderate expression. From 14 operations in 13 patients in the acquired resistance group, 6 patients had strong KIT activation, 5 had moderate expression, and 2 had none. Notably, KIT activation was also variable in the subset of patients with a second mutation, as there was strong activation in three, moderate expression in three, and no detectable phospho-KIT in two instances. In fact, the three patients (#22, 24, 27) with an identical N822K secondary mutation had either weak (n = 1) or strong (n = 2) phosphoKIT expression. There was also intraindividual variability, because in patient #19, carrying an identical T670I secondary mutation in the two nodules tested, the phospho-KIT was strong and weak (Fig. 2B). Furthermore, the patient (#24) with distinct secondary KIT mutations in two metachronous nodules had strong KIT activation in one and weak activation in the other (Fig. 2B). KIT activation variability was also noted within the resistant nodules without secondary mutations (Fig. 2C). Both patients with primary resistance did have strong phosphoKIT staining (Fig. 2C; Table 2). Inconsistent KIT activation was also observed in the untreated control group. Although, phospho-KIT was detected in all eight tumors tested, it had a weak pattern in five, including all four GISTs with KIT exon 11 in-frame deletions, and was strong in three (Fig. 2D). All three untreated GISTs with identical 557-8 WK deletion showed a weak expression only of the mature isoform. KIT amplification is not a frequent event in imatinib resistance. Tissue for FISH analysis was available in nine nodules from eight patients: three nonresistant (#2, 3, 4), four with acquired resistance (three with no secondary mutations #16, 17, 22, and one carrying a D820Y, #14B) and one patient with primary resistance (two nodules, #31). Copy number ratio for KIT: CEP 4 in the nine nodules tested did not differ from the normal control liver sample and no evidence of KIT amplification was identified in any of the tumors tested (Fig. 3). Discussion We attempted to determine the mechanism of acquired resistance to imatinib in patients with GIST in order to identify strategies to prevent or delay its development. We did a molecular analysis of KIT and PDGFRA mutations in a cohort of patients who were treated with imatinib and then underwent surgical resection. We found second site KIT mutations to be Clin Cancer Res 2005;11(11) June 1, 2005 4186 www.aacrjournals.org Fig. 2. Western blot analysis of activated and total KIT. A, nonresistant group; B, acquired resistance patients with secondary mutations; C, acquired resistance patients without secondary mutations (16 and 17) and primary resistance; and D, untreated GISTs with KIT exon11mutations compared with the negative control of a leiomyosarcoma. Cancer Therapy: Preclinical Research. Downloaded from clincancerres.aacrjournals.org on May 29, 2017. © 2005 American Association for Cancer common in acquired resistance to imatinib. Of the 15 patients with responsive or stable disease who then experienced disease progression, 7 had an identifiable second mutation. In each case, both the first and the second mutation were located in the KIT gene. In fact, they were located on the same allele. In contrast, we have not identified two simultaneous KIT or PDGFRA mutations in >300 untreated GIST tumors that we have analyzed. All resistant tumors that had a second mutation had a primary KIT exon 11 mutation. These primary mutations were mainly in-frame deletions but there was one insertion and one substitution mutation (Fig. 1). The eight second site mutations were all substitutions that involved five different residues. By comparison, the incidence of point mutations in nontreated GIST is at most 15%. The five residues derived from either the first (exons 13 or 14) or second (exon 17) KIT kinase domain. In contrast, the incidence of exons 13, 14, or 17 mutations in untreated GIST is about 1% (11, 15). Six of the secondary mutations were located in exon 17 between amino acids 820 and 823, the most common being N822K, which was seen in three cases. Although N822K and D820Y have been previously reported as a primary sporadic mutation (10) and germ line mutation in a familial GIST syndrome (21), respectively, we believe that in these resistant nodules, N822K and D820Y substitutions represent a secondary mutation because they were associated with an exon 11 mutation and only one of the multiple nodules analyzed from the same patient carried this mutation. Similar KIT kinase mutations, N822K and Y823D, have been described recently in a subset of seminomas (22). To date, all primary mutations reported in KIT exon 13 have been a K642E substitution (10, 15–17). However, we identified a V654A mutation in exon 13. Interestingly, a recent study by Chen et al. (23) reported a V654A mutation in all six resistant nodules taken from five GIST patients. Similarly, this second mutation in their study was also found on the same allele as the primary mutation. However, there are several differences between our findings and those of Chen et al. (23) including: (a) the incidence of second mutations in acquired resistance (100% versus 46%), (b) the location of second mutations (exon 13 versus different areas of the KIT kinase domain), and (c) the type of primary mutation of tumors developing second mutation (exon 9 and 11 versus only exon 11 mutations). Some of these discrepancies might be due to a limited number of cases in both series. Although no primary mutations have been previously described in KIT exon 14, a recent report has shown a T670I substitution in a patient with acquired resistance (24). One of our seven patients with a second mutation had a similar T670I mutation. New activating kinase mutations were also reported in seven of nine (78%) GIST patients with a unique pattern of resistance, defined radiologically as a ‘‘nodule within a mass’’, and thought to represent the progression of a clone resistant to imatinib (25). Recently, the crystal structures of the autoinhibited inactive and active conformations of the KIT kinase have been determined, as well as the structure of the KIT kinase in complex with imatinib (26). These structures provide insight into the mechanism of the normal regulation of KIT kinase function but moreover they provide explanations for the basis of constitutive activation of mutations found in neoplasms such as GIST. A critical feature in KIT kinase function is the role of the juxtamembrane domain of the KIT receptor in regulating kinase activity. In the inactive autoinhibited state, the juxtamembrane domain of KIT inserts into the kinase active site and thus disrupts the formation of the active conformation. Critical residues in these interactions are WK557-558 and VV559-560. Mutation of these residues disrupts these inhibitory interactions and destabilizes the inactive autoinhibited conformation of the KIT kinase. Thus, a diversity of juxtamembrane domain mutations represents the majority of primary oncogenic mutations found in GIST. The activation loop (A-loop) of the KIT kinase, which includes Y823, a pseudosubstrate of the KIT kinase, is another major site of oncogenic mutation. During KIT kinase activation, Y823 becomes phosphorylated and this seems to stabilize the open active conformation of the A-loop presumably by strong negative electrostatic interactions of the phosphate residue. Oncogenic activation loop mutations such as Y823D, which mimic Y823 phosphorylation, thus stabilizing the active conformation of the A-loop. The structure of the KIT-imatinib complex revealed that, similar to BCR-ABL, imatinib binds the inactive conformation of the kinase although the KITimatinib complex deviates somewhat from the autoinhibited inactive KIT kinase conformation. It is therefore not surprising that A-loop mutations are generally not inhibited by imatinib, although there seem to be exceptions. Therefore, there are two possible mechanisms of how resistance to imatinib therapy may develop. First, second site mutations may stabilize the active conformation of the KIT kinase which prevents imatinib binding. Alternatively, second site mutations may specifically interfere with imatinib binding without affecting the overall KIT kinase conformation. Our findings are in agreement with these predictions. First, five of the second site mutations in this study are located in the A-loop (26). Tyr823 was found to be substituted by aspartic acid in one of the resistant tumors and other mutations included D820Y and N822K. Although the Y823D mutation introduces a tyrosine-phosphate mimic (Fig. 4), the others may destabilize the inactive conformation by introduction of a positively charged side chain into a positively charged pocket formed on the COOH-terminal lobe of the kinase. Second, one mutation seems to block imatinib binding to KIT. In T670I, the gatekeeper residue Thr670 is replaced by an isoleucine residue. This mutation disrupts an important H-bond between imatinib and the kinase and the www.aacrjournals.org 4187 Clin Cancer Res 2005;11(11) June 1, 2005 Fig. 3. FISH analysis on a patient with acquired resistance. Patient #18 who had no secondary mutation showed a normal chromosome copy number for both KIT (green) and reference centromeric chromosome 4 (red). Acquired Imatinib Resistance in GIST Research. Downloaded from clincancerres.aacrjournals.org on May 29, 2017. © 2005 American Association for Cancer isoleucine methyl group protrudes into the imatinib binding site precluding proper imatinib binding to the KIT kinase (Fig. 4). Investigation of CML revealed that the predominant mechanism of acquired resistance to imatinib is second site mutation in BCR-ABL (19, 20). The BCR-ABL mutations described to date in resistant CML either alter the imatinib binding site or the residues responsible for the unique conformational change that permits imatinib binding and prevents the inactive conformation necessary for imatinib binding (20). BCR-ABL kinase domain substitutions occurred in 29 of 32 (90%) resistant tumors, whereas genomic amplification of BCR-ABL transcript was infrequent. Similarly, we found secondary KIT mutation in patients with acquired resistance. However, there are several distinct features between acquired resistance in CML and GIST (Table 3). First, preexisting BCR-ABL kinase mutations have been found in 2 of 4 patients (20) and in 5 of 24 patients (27) who did not respond to imatinib. Furthermore, BCR-ABL kinase mutations were also found in 4 of 13 patients with chronic phase CML who had stable disease on imatinib and this predicted subsequent clinical relapse (20). We did not find secondary mutations in any of the 14 GIST patients with nonresistant disease. Shah et al. (20) found polyclonal resistance, in which two to four clones harbored distinct mutant BCR-ABL kinase domains mutations in 12 of 32 CML patients. We did not detect polyclonal resistance in 10 samples from five patients with acquired resistance, because all cDNA subclones from one tumor nodule revealed only a single type of secondary mutation. Although we did not find coexisting, multiple second mutations from patients with acquired resistance, one patient had two different secondary mutations in resistant nodules at two different time points. Thus, tumor nodules within a patient may develop independent means of imatinib resistance. The mechanism for the development of a second mutation in GIST is unclear. The long duration of imatinib therapy (median of 27 months) in patients with acquired resistance who developed a second mutation makes it less likely that a preexisting clone is responsible for acquired resistance (Fig. 5). In CML, the presence of BCR-ABL kinase domain mutations in pre-imatinib samples of patients with primary resistance, the relapse of patients in the blast phase is within 1 to 3 months, and the finding of multiple mutations in resistant patients suggest clonal selection of preexisting mutations that confer imatinib resistance. In GIST, the possibility of preexisting mutations remains to be evaluated carefully. Our inability to find secondary mutations in eight resistant GIST patients suggests that other gene mutations, or altogether Clin Cancer Res 2005;11(11) June 1, 2005 4188 www.aacrjournals.org Fig. 4. Interaction of imatinib and the KIT kinase. The activation loop (A-loop; 809-831) is highlighted in red and the autoinhibitory domain (547-579) is highlighted in purple. The sites of mutation at Thr670 and Tyr823 are shown in space-filling representation in yellow. A, active form with ADP bound (ADP is shown as ball-and-stick representation). B, inactive form with imatinib bound (imatinib shown as ball-and-stick representation). Figures generated with the programs MOLSCRIPT (35) and RASTER3D (36). Coordinates correspond to those reported by Mol et al. (26). Table 3. Comparison of imatinib resistance in CML and GIST CML* GISTc Acquired resistance Kinase domain mutations 29 of 32 7 of 15 Median duration of imatinib therapy 20 monthsb 27 months Gene amplification in acquired resistance 3 of 11 0 of 4 Activation of transcribed protein 11of 11 heterogeneous Primary resistance Kinase domain mutations pre-imatinib 7 of 28 0 of 4 Nonresistant disease Kinase domain mutations 4 of 13 0 of 14 NOTE: Results are listed as number of patients with feature over number tested. *Based on refs. (19, 20, 27). cCurrent report. bChronic phase CML. Cancer Therapy: Preclinical Research. Downloaded from clincancerres.aacrjournals.org on May 29, 2017. © 2005 American Association for Cancer different mechanisms, may be involved. Fletcher et al. (28) suggested additional mechanisms of resistance besides secondary KIT mutations in GIST, including KIT genomic amplification and activation of an alternative receptor tyrosine kinase protein in the absence of KIT expression. We have found no evidence for KIT genomic amplification in our five resistant patients. Based on experimental data, other possibilities are increased serum acid glycoprotein levels and increased multidrug resistance gene expression (29, 30). More recently, Van Oosterom and colleagues (17, 31) found preliminary evidence that bioavailability of imatinib may decrease during chronic therapy, possibly due to up-regulation of hepatic enzymes responsible for drug clearance. It is now well documented that cessation of imatinib during stable measurable disease results in tumor proliferation (32) and therefore subtherapeutic imatinib levels may also result in tumor progression. Our analysis of phosphorylated KIT protein yielded heterogeneous results, signifying the complexity of KIT activation in treated GIST. Strikingly, the majority of nonresistant patients still had activated KIT. Most of the patients in the acquired resistance group also had phosphoKIT and almost half had strong expression. This finding suggests that GIST progression may still depend on KIT and alternative inhibitors of KIT, or its downstream pathway, may be of therapeutic benefit. Nevertheless, there were three patients with acquired resistance whose tumors lacked detectable KIT activation. There was no discernible pattern of KIT activation based on histologic response, the type of primary mutation, or the presence of a second KIT mutation. There was not even a correlation between phosphorylated KIT and total KIT. Our findings are consistent with those of Duensing et al. (33) who found that both phosphorylated and total KIT varied substantially among tumors, even within identical KIT genotype. Our data with KIT phosphorylation may be confounded by the fact that we stopped imatinib generally 2 to 5 days prior to surgery. Currently, there are no established guidelines for the perioperative use of imatinib but our approach seems to be safe. Secondary mutations during chronic imatinib use has several implications for the clinical management of patients with advanced GIST. Our recommendations are based on the supposition that the risk of developing resistance is proportional to the amount of residual viable tumor in patients on imatinib. Therefore, we consider extirpation of all gross disease whenever possible. Indeed, that was the rationale for operating on most of the patients in this report who were nonresistant. The fact that several nonresistant patients had only minor histologic responses along with activated KIT underscores the potential for viable tumor to progress. An argument may even be made for debulking (removing as much gross disease as possible) in patients with extensive residual disease during imatinib therapy. Alternatively, optimal management may include the combined use of imatinib with other newer molecular agents. Although we did not detect any second mutations in nonresistant patients, Shah et al. (20) found second mutations in 4 of 13 patients with stable CML and that predicted subsequent relapse. Patients with stable GIST who may be found to have a second mutation (such as by biopsy) should certainly be considered for additional surgery or other molecular agents due to the likelihood of impending progression. Of course, serial tumor assessment is not practical in GIST as it is in CML. A variety of other molecular inhibitors are currently under investigation for advanced GIST that is refractory to imatinib. The furthest along is SU11248 (Pfizer Inc., New York, NY), an inhibitor of multiple tyrosine kinases. It seems to be effective in imatinib-resistant GIST, especially in patients with exon 9 mutations (34). In summary, we report a clinical and molecular study of acquired resistance to imatinib in GIST. We found that secondary mutations are common in imatinib resistance. The mutations tend to be single amino acid substitutions in the KIT kinase domains and occur particularly in exon 17. Secondary mutations were not seen in the pre-imatinib, nonresistant, or primary resistant tumors. There was considerable heterogeneity between KIT activation and responsiveness of GIST to imatinib. Our findings have implications for strategies to treat or avert imatinib resistance and might be useful in the design of secondgeneration kinase inhibitors.  Purpose: Resistance is commonly acquired in patients with metastatic gastrointestinal stromal tumor who are treated with imatinib mesylate, often due to the development of secondary mutations in the KIT kinase domain. We sought to investigate the efficacy of second-line tyrosine kinase inhibitors, such as sorafenib, dasatinib, and nilotinib, against the commonly observed imatinib-resistant KIT mutations (KITV654A, KITT670I, KITD820Y, and KITN822K) expressed in the Ba/F3 cellular system.  Experimental Design:In vitro drug screening of stable Ba/F3 KIT mutants recapitulating the genotype of imatinib-resistant patients harboring primary and secondary KIT mutations was investigated. Comparison was made to imatinib-sensitive Ba/F3 KIT mutant cells as well as Ba/F3 cells expressing only secondary KIT mutations. The efficacy of drug treatment was evaluated by proliferation and apoptosis assays, in addition to biochemical inhibition of KIT activation.  Results: Sorafenib was potent against all imatinib-resistant Ba/F3 KIT double mutants tested, including the gatekeeper secondary mutation KITWK557-8del/T670I, which was resistant to other kinase inhibitors. Although all three drugs tested decreased cell proliferation and inhibited KIT activation against exon 13 (KITV560del/V654A) and exon 17 (KITV559D/D820Y) double mutants, nilotinib did so at lower concentrations.  Conclusions: Our results emphasize the need for tailored salvage therapy in imatinib-refractory gastrointestinal stromal tumors according to individual molecular mechanisms of resistance. The Ba/F3 KITWK557-8del/T670I cells were sensitive only to sorafenib inhibition, whereas nilotinib was more potent on imatinib-resistant KITV560del/V654A and KITV559D/D820Y mutant cells than dasatinib and sorafenib.  Gastrointestinal stromal tumor (GIST) is the most common mesenchymal tumor of the gastrointestinal tract. Constitutive activation through oncogenic mutations of the KIT or platelet-derived growth factor receptor A (PDGFRA) receptor tyrosine kinase plays an important role in the pathogenesis of GISTs (1, 2). Imatinib mesylate (Gleevec, Novartis Pharmaceuticals) potently inhibits BCR-ABL, PDGFRA, PDGFRB, and KIT and has been the frontline therapy in chronic myelogenous leukemia and advanced GISTs since its regulatory approval. Imatinib mesylate achieves a clinical response in ∼80% of GIST patients (3, 4), which is directly dependent on the presence and genomic location of KIT/PDGFRA mutations. Patients with KIT exon 11 mutations have a partial response rate of 84%, whereas tumors with an exon 9 mutation or wild-type KIT are less likely to respond (5, 6). Although imatinib has improved survival in GIST, complete response is rare. Furthermore, it is now clear that the majority of patients who initially benefit from tyrosine kinase inhibitors eventually become resistant, with a median time to progression on imatinib mesylate of 2 years (7). After an initial response, 46% to 67% of patients will develop imatinib resistance through acquisition of a secondary mutation in the KIT kinase domain (8–10). These secondary mutations tend to be single amino acid substitutions in exon 17 most often, but they also occur in exons 13 and 14 (8). The mechanism for the development of second-site KIT mutations remains unclear, but resistant patients with identifiable secondary mutations have been treated with imatinib longer than resistant patients lacking secondary mutations (median, 27 versus 14.5 months; ref. 8). These findings suggest that clonal selection of existing mutations before imatinib therapy is unlikely to explain acquired resistance. The frequency of secondary mutations is also determined by the location of the primary KIT mutations, with GISTs harboring KIT exon 11 mutations more commonly becoming imatinib resistant due to acquisition of secondary mutations compared with KIT exon 9–mutated GIST. One possible explanation for this difference is that KIT exon 9 mutant GISTs are able to use alternative mechanisms, whereas KIT exon 11 GISTs are more dependent on KIT signaling (10). Polyclonal acquired resistance as a result of multiple secondary KIT mutations in geographically separate metastases has been reported in up to 18% (10–12). Consistent with secondary clonal evolution, the primary mutation is always detectable in each tumor and there is never more than one new mutation in the same sample. These findings suggest that long-term imatinib therapy leads to clonal selection of resistant tumor subclones. RNA interference KIT knockdown experiments revealed that imatinib-resistant cell lines carrying secondary KIT mutations remain dependent on KIT signaling for growth and survival (10). Alternative mechanisms, such as low-level genomic KIT amplifications, play only a minor role in imatinib resistance in GIST (8, 11).  The initial success of imatinib as a targeted therapeutic drug and the better understanding of drug-kinase interaction based on crystallographic structure have led to the development of several novel small-molecule compounds that have a variable spectrum of kinase inhibition. As the median survival is only 15 months once imatinib resistance develops (13), it is important to validate novel therapeutic strategies within the specific KIT/PDGFRA genotype that confer imatinib resistance. Some of the alternative agents to imatinib being tested in clinical trials include nilotinib, sorafenib, and dasatinib. The initial results of these trials are still being analyzed, but the efficacy of these drugs might not be clear due to the heterogeneity of imatinib-resistant patients enrolled. Thus, a more systematic in vitro approach of validating individual drug efficacy vis-à-vis specific resistant genotypes might better guide therapy and clinical trial design. These small molecules function as KIT kinase inhibitors, and some in addition inhibit other downstream targets. Nilotinib (AMN107) was designed based on the crystal structures of imatinib and the ABL kinase complex. This multitargeted kinase inhibitor features higher affinity to the ATP-binding site of ABL kinase to override imatinib resistance caused by BCR-ABL mutation (14). In addition, nilotinib selectively inhibits the KIT and PDGFR tyrosine kinases (14). Another small molecule, which showed great promise in the chronic myelogenous leukemia patients that are resistant or intolerant to imatinib, is dasatinib (BMS-354825). Dasatinib is an ATP competitor, with dual SRC/ABL activity, which inhibits most BCR-ABL mutants by binding to the active form of the kinase (15). Furthermore, dasatinib potently inhibits KIT, both wild-type and juxtamembrane domain mutant KIT (16).  Sorafenib (BAY 43-9006, Nexavar) is a novel biaryl urea compound that was initially developed as a specific inhibitor of serine-threonine kinase RAF. In addition, sorafenib has activity with few receptor tyrosine kinases, including vascular endothelial growth factor receptor-2, vascular endothelial growth factor receptor-3, PDGFRB, and KIT (17). Sorafenib has shown broad-spectrum, dose-dependent antitumor activity against xenograft models of human colon, lung, breast, ovarian, and pancreatic cancers and melanoma (18).  Based on our hypothesis that individual KIT mutant oncoproteins will affect on drug sensitivity, we investigated the efficacy of sorafenib, nilotinib, and dasatinib on Ba/F3 cells expressing various KIT mutants, which recapitulate imatinib-resistant GIST genotypes.  Materials and Methods  Ba/F3 KIT mutant transformants.KIT mutations recapitulating the genotype found in four imatinib-resistant GIST patients were introduced by site-directed mutagenesis PCR, using QuikChange II XL Site-Directed Mutagenesis kit (Qiagen, Inc.), on a retroviral expression vector containing wild-type human KIT cDNA (GNNK isoform), pMSCV-WTKIT-IRES-GFP (a generous gift from Dr. Gary Gilliland, Harvard Medical School, Boston, MA). KIT double mutant isoforms were generated hosting a primary KIT exon 11 mutation and a secondary mutation in KIT exon 13, 14, or 17 as follows: KITV560del/V654A, KITWK557-8del/T670I, KITV559D/D820Y, and KITV560del/N822K. Single KIT mutants carrying the secondary mutation alone were also generated for comparison, including KITT670I, KITD820Y, and KITN822K. The imatinib-sensitive exon 11 single mutants KITV559D and KITWK557-8del as well as the imatinib less-sensitive exon 9 KITAY502-3ins were used as controls.  The parental cell line Ba/F3 (German Collection of Microorganisms and Cell Cultures, Human and Animal Cell Cultures Collection) is an interleukin-3 (IL-3)-dependent murine pro-B-cell line lacking intrinsic KIT expression. Retroviral vector plasmids containing mutant KIT cDNA isoforms (25 μg of each construct) and the hygromycin selection marker were cotransfected into Ba/F3 cells at the ratio of 25:1 via electroporation using GenePulser (Bio-Rad). Stable transfectants were established by triple selection. The electroporated cells were first selected with hygromycin for 7 to 10 days, and the remaining cells were sorted according to green fluorescent protein fluorescence. The selected green fluorescent protein–positive cells were grown for 2 weeks in the presence of hygromycin, and finally, cells were grown in the absence of IL-3. The cell surface KIT expression was confirmed by flow cytometry using PE-conjugated anti-CD117 (BD Biosciences) antibody. Total KIT expression was confirmed by Western blotting using anti-KIT (Oncogene) antibody.  Cell proliferation assays. To determine the growth inhibition drug effects, Ba/F3 KIT mutant cells were starved from growth factor for 4 h. Fifteen minutes before drug administration, growth factors were added: 10 ng/mL IL-3 for the IL-3–dependent Ba/F3 KIT mutants (KITD820Y and KITV560del/N822K) and 20 ng/mL KL for the remaining KL-dependent Ba/F3 KIT mutant isoforms. Cells were incubated with 10 to 10,000 nmol/L of sorafenib, dasatinib, nilotinib, and imatinib at 37°C for 48 h. The concentrations of drug used in these studies are in line with the concentrations achievable in human plasma based on pharmacokinetic studies of the kinase inhibitors. Dasatinib and nilotinib were synthesized and kindly provided by Dr. Bayard Clarkson (Sloan-Kettering Institute, New York, NY). Imatinib and sorafenib were purchased commercially. Bromodeoxyuridine was then added and incubated for 6 h before harvesting. Cells were fixed and stained using the standard protocol of the BrdU-APC kit (PharMingen). Bromodeoxyuridine incorporation was determined by flow cytometry. All experiments were done in triplicate. A minimum of 2 × 105 events was acquired and the data were analyzed using FlowJo software (version 5.7.2). Cell growth inhibition curves and IC50 values were plotted and calculated using GraphPad Prism software, version 4.03.  Apoptosis assays. Induction of apoptosis on Ba/F3 KIT transformants was evaluated by flow cytometry using the Annexin V-PE Apoptosis Detection kit (PharMingen) in duplicate experiments. Cells at a density of 1 × 106, activated with KL, were cultured in 24-well plates and starved from growth factors for 4 h before drug delivery. Fifteen minutes before adding the drug, growth factors were added. After drug incubation for 48 h using the concentrations indicated above, cells were harvested and stained with anti-Annexin V-PE antibody using the provider's protocol. A minimum of 10,000 events was analyzed by FACScan (Becton Dickinson) within an hour after staining. The data and results were analyzed by FlowJo.  Immunoprecipitation and Western blotting. The Ba/F3 cells expressing single and double KIT mutants were treated with the indicated doses of each drug. Cells were starved from serum and growth factors for 2 h, and only growth factors were added back to medium 15 min before drug administration. Drugs were incubated at 37°C in the absence of serum for 90 min. After treatment, cells were harvested and subjected to protein extraction. Whole-cell lysate (200 μg) was incubated with 2 μg of anti-KIT antibody (Assay Designs, Inc.) for an hour, and then the mixture was incubated with Magna beads (Pierce Biotechnology) overnight at 4°C. The beads were washed and isolated using a magnetic column and resuspended in LDS sample buffer (Invitrogen). Electrophoresis and immunoblotting were done on the protein extracts using the standard protocol. Phosphorylated KIT was detected with anti-phosphotyrosine antibodies PY20 and PY99 (Santa Cruz Biotechnology), and total KIT was detected by mouse monoclonal anti-c-KIT (Santa Cruz Biotechnology). The secondary antibodies used were donkey anti-mouse secondary (Santa Cruz Biotechnology).  Results  To test existing second-line inhibitors on imatinib-resistant KIT oncoproteins, we developed stable Ba/F3 transfectants expressing single and double mutant isoforms and selected by IL-3–independent growth. Eight of the 10 KIT mutant forms induced transformation by conferring factor-independent growth to Ba/F3 cells. KITD820Y and KITV560del/N822K Ba/F3 mutants did not achieve IL-3 independence. Ba/F3 transfectants expressing double KIT mutants, KITV560del/V654A, KITWK557-8del/T670I, KITV559D/D820Y, and KITV560del/N822K, and single KIT mutant isoforms, KITT670I and KITN822K, were tested with sorafenib, dasatinib, nilotinib, and imatinib. The efficacy of each drug was compared with imatinib-sensitive KIT juxtamembrane and extracellular domain mutations, Ba/F3 KITV559D, KITWK557-8del, and KITAY502-3ins.  Dasatinib is more potent than other kinase inhibitors on imatinib-sensitive Ba/F3 KIT mutants. The in vitro cell proliferation assay showed that dasatinib inhibited cell growth of imatinib-sensitive Ba/F3 KITV559D with an IC50 of 27 nmol/L, whereas imatinib, nilotinib, and sorafenib had an IC50 of 63, 44, and 66 nmol/L, respectively. Dasatinib results are summarized in Table 1 . Dasatinib induced apoptosis in >20% of cells at 10 nmol/L, whereas the other drugs induced programmed cell death at 100 nmol/L. The biochemical assays showed consistent KITV559D phosphorylation inhibition by all four inhibitors tested, with dasatinib being the most potent. KIT kinase activity was distinctly inhibited using <10 nmol/L dasatinib, whereas imatinib, nilotinib, and sorafenib showed the same results at 100 nmol/L.  View inlineView popup Table 1. Comparison of in vitro cell proliferation inhibition of imatinib-resistant KIT single and double mutants expressed in the Ba/F3 cell system Similarly, the growth of KITWK557-8del cells was inhibited by dasatinib with an IC50 of 58 nmol/L, whereas imatinib, nilotinib, and sorafenib had an IC50 of 460, 83, and 211 nmol/L, respectively (Fig. 1 ). Furthermore, significant induction of apoptosis was noted at 100 nmol/L of dasatinib and nilotinib.  Fig. 1. Download figureOpen in new tabDownload powerpoint Fig. 1. Imatinib-sensitive juxtamembrane mutation KITWK557-8del showing sensitivity for all drugs tested; however, dasatinib and nilotinib inhibited cell growth (A) at an IC50 of 58 and 83 nmol/L, respectively, and induced significant apoptosis at 100 nmol/L (B). 7-AAD, 7-aminoactinomycin D. Dasatinib inhibited the imatinib less-sensitive Ba/F3 KITAY502-3ins with an IC50 of 74 nmol/L, whereas imatinib, nilotinib, and sorafenib had an IC50 of 509, 671, and 400 nmol/L, respectively. Dasatinib induced significant (>90%) programmed cell death of KITAY502-3ins cells at 100 nmol/L, whereas nilotinib induced ∼65% and sorafenib induced 35% of programmed cell death at the same concentration. Imatinib did not induce overt apoptosis until 1,000 nmol/L. In keeping with the cell proliferation inhibition and apoptosis results, imatinib did not inhibit KIT phosphorylation at 100 nmol/L but showed distinct inhibition on KIT activity at 1,000 nmol/L.  Imatinib-resistant gatekeeper single KITT670I and double KITWK557-8del/T670I mutants are responsive to sorafenib but resistant to dasatinib and nilotinib. Sorafenib was the only drug that inhibited the cell proliferation of the Ba/F3 KITWK557-8del/T670I transfectant with an IC50 of 1,063 nmol/L and induction of significant programmed cell death between 1,000 and 5,000 nmol/L (Fig. 2 ). Sorafenib also showed distinct inhibition of the kinase activity of the KL-dependent double mutant KITWK557-8del/T670I isoform at ∼1,000 nmol/L. In contrast, imatinib, dasatinib, and nilotinib did not inhibit its phosphorylation even up to 10,000 nmol/L. Similar experiments were done on the Ba/F3 cells expressing the gatekeeper single mutant KITT670I, which showed a comparable drug sensitivity profile. Sorafenib significantly inhibited the kinase activity of this single mutant at 1,000 nmol/L, inhibited cell growth with an IC50 of 918 nmol/L, and induced significant apoptosis between 1,000 and 5,000 nmol/L, whereas imatinib, nilotinib, and dasatinib did not show effects on both biochemical and biological assays (Fig. 2).  Fig. 2. Download figureOpen in new tabDownload powerpoint Fig. 2. Sorafenib potently inhibited cell proliferation and kinase activity and induced overt apoptosis of the Ba/F3 transfectant expressing KITWK557-8del/T670I mutant. Imatinib, dasatinib, and nilotinib showed no effect on cell growth or KIT kinase activity. A, Ba/F3 cells stably expressing KITWK557-8del/T670I were treated with escalating doses of sorafenib in the presence of 20 ng/mL KL for 48 h. Cellular proliferation as determined by bromodeoxyuridine (BrdU) incorporation assays was inhibited by sorafenib at an IC50 of 1,063 nmol/L. Other drugs did not show growth inhibition (data not shown) up to 10,000 nmol/L. B, apoptosis assays, measured by Annexin V and 7-aminoactinomycin D staining, showed that sorafenib induced significant apoptosis at 1,000 nmol/L. Imatinib, nilotinib, and dasatinib did not induce overt apoptosis up to 5,000 nmol/L. C, immunoblotting for phosphorylated (PY20 + PY99) and total forms of KIT was done to evaluate the inhibitory effect of the drugs on KIT activation. Sorafenib showed marked KIT activation inhibition at 1,000 nmol/L, whereas the other three drugs did not show an inhibitory effect up to 10,000 nmol/L. Nilotinib inhibited cell growth and induced apoptosis of imatinib-resistant KITV560del/V654A mutant cells at lower doses than dasatinib and sorafenib. Nilotinib potently inhibited the proliferation of Ba/F3 KITV560del/V654A cells with an IC50 of 192 nmol/L. This double mutant was also sensitive, but at higher concentrations, to dasatinib and sorafenib inhibition at an IC50 of 585 and 1,074 nmol/L, respectively (Fig. 3A ). The apoptosis assays showed similar results for these three inhibitors. Interestingly, imatinib showed induction of apoptosis in 50% of cells at 1,000 nmol/L, whereas at 5,000 nmol/L 90% of cells entered programmed cell death (Fig. 3B). Biochemically, nilotinib, dasatinib, and sorafenib inhibited KIT phosphorylation at 1,000 nmol/L. Imatinib partially inhibited the phosphorylation of KITV560del/V654A oncoprotein at 1,000 nmol/L, which was completely inhibited at 5,000 nmol/L (Fig. 3C).  Fig. 3. Download figureOpen in new tabDownload powerpoint Fig. 3. A, sorafenib, dasatinib, and nilotinib showed significant inhibition on cell growth of Ba/F3 cells expressing KITV560del/V654A at 1,000 nmol/L or lower. Imatinib showed moderate effects between 1,000 and 5,000 nmol/L. B, nilotinib induced significant apoptosis between 100 and 1,000 nmol/L, whereas dasatinib, sorafenib, and imatinib did between 1,000 and 5,000 nmol/L. C, nilotinib, dasatinib, and sorafenib showed significant inhibition on KIT activation at 1,000 nmol/L, whereas imatinib at 5,000 nmol/L. Imatinib-resistant KITV559D/D820Y was sensitive to nilotinib, dasatinib, and sorafenib inhibition. All three inhibitors tested decreased cell proliferation of the imatinib-resistant Ba/F3 KITV559D/D820Y cells (see Table 1). Nilotinib inhibited cell growth at an IC50 of 297 nmol/L, whereas sorafenib and dasatinib had similar effects at a slightly higher dose, IC50 of 944 and 432 nmol/L, respectively. Dasatinib induced significant apoptosis at 1,000 nmol/L, whereas nilotinib and sorafenib showed moderate effect on induction of apoptosis between 1,000 and 5,000 nmol/L. Biochemically, all three drugs showed distinct inhibition of KIT phosphorylation at 1,000 nmol/L.  Phosphorylation of imatinib-resistant KITV560del/N822K is inhibited by nilotinib at lower concentrations than dasatinib and sorafenib. The Ba/F3 KITV560del/N822K cells did not achieve IL-3 independence. Biochemically, nilotinib completely inhibited KIT phosphorylation at 100 nmol/L concentration, whereas dasatinib and sorafenib at 1,000 nmol/L (data not shown). Proliferation and apoptosis studies done on these IL-3–dependent cells did not show an effect with any of the drugs tested. This result might be related to other IL-3–dependent signaling pathways not targeted by the drugs tested.  Discussion  Although most GIST patients respond to imatinib initially, they often develop progressing lesions at multiple metastatic sites within 2 to 5 years after starting therapy. In most imatinib-resistant GISTs, KIT is reactivated and the downstream signaling pathways remain KIT dependent (8, 11). In about half of imatinib-resistant patients, the development of a second-site KIT mutation is the cause for KIT reactivation (8, 11, 12), whereas in the remaining cases it is largely unknown. The second-site mutations occur without exception on the same allele as the primary mutation, affect either the first or second kinase domains (exons 13, 14, and 17), and are mostly single amino acid substitutions (8, 10). Crystallographic studies showed that imatinib binds only to the inactive form of KIT (19). Secondary KIT mutations in the kinase domains stabilize the active form of KIT and thus weaken or prevent its drug binding, leading to clinical resistance. As the imatinib alternatives presently tested in clinical trials show a limited survival benefit (13), there is an urgent need to validate novel therapeutic strategies, tailored to the mechanism of resistance in individual patients.  As shown by several groups, the location of the primary KIT or PDGFRA mutation directly predicts the clinical response to imatinib in GIST patients. Tumors with an underlying primary KIT mutation in exon 11 represent the subgroup with better response rates than other mutational subtypes (5, 6). Furthermore, this group is also at higher risk to develop secondary mutations compared with KIT exon 9–mutated GISTs (8, 11, 12). This observation further supports that the probability of developing a secondary mutation increases with duration of imatinib treatment, which often is longer in GISTs harboring exon 11 mutation than in those with exon 9 mutation or wild-type GISTs. However, the presence of secondary KIT mutations per se within the acquired imatinib resistance setting is not an indicator of a worse prognosis compared with imatinib-refractory tumors without secondary mutations. The latter group escapes the inhibitory efficacy of imatinib through yet unresolved molecular mechanisms (12).  Another level of complexity relies on the fact that long-term imatinib therapy leads to clonal selection of distinct resistant tumor subclones, the so-called “polyclonal acquired resistance” (10–12). Thus, each tumor nodule under progression may undergo individual clonal evolution, resulting in multiple secondary mutations developed at different metastatic sites within the same patient. As such, designing salvage strategies in these imatinib-resistant settings should address inhibition of all the known genomic activating mutations in the oncoprotein.  Most of the prior in vitro drug screening studies on imatinib-resistant KIT mutants investigate only one drug at a time (9, 20; see Table 2 ). Thus, comparing results between different laboratories, using different cell systems (primary GIST cells, Ba/F3, HEK, etc.), is quite difficult. Our study undertakes a detailed investigation of three imatinib alternative small molecules on a large panel of Ba/F3 KIT single and double mutant cells using biological and biochemical methods of assessing drug response.  View inlineView popup Table 2. Imatinib, dasatinib, and nilotinib drug sensitivity in previously reported imatinib-sensitive and imatinib-resistant KIT mutant oncoproteins Previous reports (12) suggested that GISTs with secondary KIT exon 14 gatekeeper mutations (T670I) show a more aggressive behavior with earlier metastasis and shorter progression-free survival. Of interest is that KITT670I mutation has not been identified as a primary mutation in GIST and only occurs under the selective pressure of imatinib therapy. The gatekeeper amino acid T670 of KIT was identified as one of the key hydrogen bonds for imatinib binding (19, 21); thus, mutations at this position have a profound effect on drug binding. The bulky isoleucine substituting the threonine residue abolishes the affinity not only for imatinib but also for dasatinib (22, 23). Similarly, the gatekeeper mutation in BCR-ABL, T315I, is resistant to nilotinib at up to 10,000 nmol/L (14). In keeping with these findings, the Ba/F3 KIT cells expressing a T670I single or double mutant were insensitive to all drugs except sorafenib. Sorafenib was initially designed to inhibit cell proliferation by targeting the RAF/mitogen-activated protein kinase/extracellular signal-regulated kinase kinase/extracellular signal-regulated kinase pathway and tumor angiogenesis by targeting vascular endothelial growth factor receptor-2/PDGFRB (17). It was recently approved by the Food and Drug Administration for the treatment of advanced renal cell carcinoma. Sorafenib was also shown to interact with KIT, inhibiting its KIT kinase activity in in vitro kinase assays (18). These data support the ongoing phase II National Cancer Institute–sponsored multicenter investigation of sorafenib in imatinib- and sunitinib-resistant GIST (Clinicaltrials.gov identifier NCT00265798). However, the biochemical or biological consequences of sorafenib inhibition in the context of KIT mutant cells have not been yet investigated. Importantly, our results show that sorafenib inhibits the KITT670I gatekeeper mutation either expressed as a single or double mutant oncoprotein in the Ba/F3 cellular system. Thus, sorafenib inhibited the KIT kinase activity and cell growth, as well as induced apoptosis at 1,000 nmol/L concentration, in both Ba/F3 KITWK557-8del/T670I double mutant and Ba/F3 KITT670I single mutant cells. In fact, sorafenib was the only drug potent against all different secondary mutations tested, including KIT exons 13 and 17.  Dasatinib is a potent, orally bioavailable inhibitor of several kinases, including BCR-ABL and SRC family of kinases (24). It was recently approved by the Food and Drug Administration for the treatment of chronic myelogenous leukemia and Philadelphia chromosome–positive ALL with resistance or intolerance to previous therapy, including imatinib (25). Dasatinib was previously shown to be more potent than imatinib in inhibiting KITV559D (23), KITV560G (16), and wild-type KIT (15). Our results also indicate that dasatinib is more effective than other kinase inhibitors on imatinib-sensitive exon 11 mutant Ba/F3 KITV559D and KITWK557-8del cells.  Schittenhelm et al. (16) reported that dasatinib inhibits the kinase activity of KITV560G/D816V double mutant cells in a dose-dependent manner. Screening the sensitivity of different activating tyrosine kinase oncoproteins carrying the gatekeeper mutation, Carter et al. (23) showed that dasatinib has no effect on KITV559D/T670I double mutant. Their results also indicate that similar gatekeeper mutations in BCR-ABL (T315I) and EGFR (T670M) oncoproteins are resistant to dasatinib therapy. Our data support that KITT670I expressed either as a single or double mutant oncoprotein in the Ba/F3 cells confers resistance to dasatinib. In contrast, Ba/F3 cells expressing KIT exon 13 or exon 17 single or double mutants were sensitive to dasatinib inhibition. Dasatinib was recently examined in a phase I study in which GIST patients were enrolled (Clinicaltrials.gov identifier NCT00099606); final data are not yet published.  Nilotinib was specifically designed as a more potent inhibitor of mutant and wild-type BCR-ABL compared with imatinib (14). In addition, Weisberg et al. (14) showed that nilotinib is as effective as imatinib in inhibiting exon 11 KITV560del and exon 13 KITK642E mutant Ba/F3 cells. In a different study, nilotinib showed potent inhibition against KITV560G mutant cells (26). Not previously reported, our results indicate nilotinib efficacy on the KIT double mutants involving exon 11 and exons 13 or 17. Specifically, nilotinib induced cell growth inhibition in imatinib-resistant KITV560del/V654A and KITV559D/D820Y mutant cells at lower concentrations than dasatinib and sorafenib. In contrast, nilotinib was not effective against the gatekeeper KIT exon 14 single (KITT670I) or double (KITWK557-8del/T670I) mutant. Similarly, Weisberg et al. (14) showed that, although nilotinib inhibited all the other secondary mutations of BCR-ABL, it had no effect in inhibiting the T315I gatekeeper mutation, which is analogous to the T670I mutation in KIT. Nilotinib was investigated in combination with imatinib in patients who are imatinib resistant (Clinicaltrials.gov identifier NCT00135005); final results of the study are pending.  Previous preclinical data using the Ba/F3 cellular system showed that imatinib-resistant KIT V654A and T670I mutants are sensitive to sunitinib (SU11248; ref. 20). The growth of Ba/F3 cells expressing KITWK557-8del/V654A or KITWK557-8del/T670I was completely inhibited by sunitinib at concentrations below 500 nmol/L (IC50, 50-100 nmol/L; Table 2). Although other imatinib-resistant KIT mutants, such as the more common second kinase domain KIT mutations, were not investigated, Ba/F3 cells expressing PDGFRA-D842V were less sensitive to sunitinib, and their growth proliferation was inhibited at 5 μmol/L.  Anecdotal evidence based mainly on molecular modeling of mutated KIT receptors suggests that some secondary mutations acquired in the imatinib resistance setting might be weaker than others (27). Although most second-site KIT mutations affect the imatinib-binding site, the extent of structural changes conferred varies with the individual amino acid substitution; thus, T670I substantially modifies the binding pocket, whereas V654A induces only relatively confined changes (8, 27). When compared with the biochemical activity of KIT kinase, Tamborini et al. (27) showed that, in both substitutions, the phosphorylation of the receptor persisted in the presence of 3 mmol/L imatinib, in keeping with an activated status of the KIT receptors. However, the single and double mutant KITV654A oncoprotein expressed in COS cells seemed to be unphosphorylated at 6 mmol/L imatinib, whereas the single and double mutant KITT670I did not. These results suggest that the effects of V654A mutation might be subverted by a dose escalation of the imatinib and it would be expected that a clinical response would be restored. Our results are in support of these findings, showing that imatinib partially inhibited the phosphorylation of KITV560del/V654A oncoprotein at 1,000 nmol/L, which was completely inhibited at 5,000 nmol/L. Similarly, induction of apoptosis with imatinib was evident at 1,000 nmol/L in 50% of cells, whereas at 5,000 nmol/L most cells entered programmed cell death.  Alternative salvage options other than inhibiting KIT signaling pathway are presently being explored. These strategies promise a therapeutic solution to the challenge of heterogeneous imatinib resistance. Bauer et al. (28) investigated the activity of heat shock protein 90 inhibitor 17-allylamino-18-demethoxy-geldanamycin (17-AAG) in enhancing cellular degradation of constitutively activated KIT oncoprotein. Thus, 17-AAG was effective in inhibiting KIT-dependent imatinib-sensitive and imatinib-resistant GIST cell lines but not KIT-independent GIST cells, suggesting that its effects result mainly from inactivation of KIT oncoprotein. Taking a different approach, Sambol et al. (29) showed the efficacy of flavopiridol, a cyclin-dependent kinase inhibitor, on a primary GIST cell line carrying an imatinib-sensitive KITK642E exon 13 mutation, by induction of apoptosis and transcriptional down-regulation of KIT. A possible combination of targeted kinase inhibitors with drugs such as flavopiridol or 17-AAG may be more effective in the setting of polyclonal acquired resistance.  In conclusion, second-line tyrosine kinase inhibitors are selectively potent against imatinib-resistant KIT mutations and might provide an effective alternative to imatinib in the management of these patients. Specifically, sorafenib was found to be effective against all the KIT double mutants tested, including the secondary mutation in the “gatekeeper” residue (KITT670I), which is resistant to the other kinase inhibitors. In contrast, imatinib-resistant KITV560del/V654A and KITV559D/D820Y mutant cells were inhibited by all three inhibitors tested, with nilotinib showing a lower IC50 than dasatinib and sorafenib. These in vitro data set the stage for designing future clinical trials, where selection of the second-generation kinase inhibitor will be tailored based on the individual mechanism of imatinib resistance. The presence of a KITT670I gatekeeper mutation for example in the resistant disease may indicate sorafenib or sunitinib as the next therapeutic option, whereas other secondary mutations may be more susceptible to nilotinib.  Gastrointestinal stromal tumors (GISTs) are the most common mesenchymal tumors in the human digestive tract, but their molecular etiology and cellular origin are unknown. Sequencing of c-kit complementary DNA, which encodes a proto-oncogenic receptor tyrosine kinase (KIT), from five GISTs revealed mutations in the region between the transmembrane and tyrosine kinase domains. All of the corresponding mutant KIT proteins were constitutively activated without the KIT ligand, stem cell factor (SCF). Stable transfection of the mutant c-kitcomplementary DNAs induced malignant transformation of Ba/F3 murine lymphoid cells, suggesting that the mutations contribute to tumor development. GISTs may originate from the interstitial cells of Cajal (ICCs) because the development of ICCs is dependent on the SCF-KIT interaction and because, like GISTs, these cells express both KIT and CD34. The c-kit proto-oncogene encodes a type III receptor tyrosine kinase (KIT) (1), the ligand of which is SCF (2). SCF-KIT interaction is essential for development of melanocytes, erythrocytes, germ cells, mast cells and ICCs (3, 4). Gain-of-function mutations of the c-kit gene have been found in several tumor mast cell lines of rodents and humans (5, 6) and in mast cell tumors of humans (7). Here we investigate the mutational status of c-kit in mesenchymal tumors of the human gastrointestinal (GI) tract.  We collected 58 mesenchymal tumors that developed in the GI wall (4 in the esophagus, 36 in the stomach, 14 in the small intestine, and 4 in the large intestine). KIT expression was examined by immunohistochemistry (8). Eight authentic leiomyomas and an authentic schwannoma did not express KIT. The remaining 49 mesenchymal tumors were diagnosed as gastrointestinal stromal tumors (GISTs), and 94% (46/49) of these expressed KIT. Examination of these tumors for expression of CD34, which is a reliable marker for GISTs (9), revealed that 82% (40/49) were CD34-positive, and 78% (38/49) were positive for both KIT and CD34 (Fig.1, A to I). Three of five KIT-negative GISTs were also CD34-negative.    Download high-res image  Open in new tab  Download Powerpoint Figure 1 Coexpression of KIT and CD34 in human GISTs. (A to C) Serial sections of a tumor that developed in the muscle layers of the stomach. (D toF) Higher magnification of (A) to (C). (A) and (D) are stained with hematoxylin and eosin. The boundary of the tumor is indicated by arrows in (A). (B) and (E) are stained with anti-KIT and (C) and (F) with anti-CD34 (8). Arrowheads in (A) to (C) indicate that endothelial cells of blood vessels express CD34 but not KIT. lm; longitudinal muscle layer. (Gto I) Coexpression of KIT and CD34 in another GIST, also from the stomach, demonstrated by confocal laser scanning microscopy (8). (G) Binding of rabbit anti-KIT to tumor cells demonstrated by FITC-labeled anti-rabbit IgG (green). (H) Binding of mouse anti-CD34 to tumor cells demonstrated by RPE-labeled anti-mouse IgG (red). (I) Merged confocal image of (G) and (H) showing coexpression of KIT and CD34 in tumor cells (yellow). Bars in (A) to (C), 50 μm; in (D) to (I), 100 μm.  We compared the immunohistochemical characteristics of GISTs with those of ICCs, cells that regulate autonomous contraction of the GI tract (4). ICCs are located in and near the circular muscle layer of the stomach (10), small intestine (11), and large intestine (12). Because ICCs of the small intestine surrounding myenteric ganglion cells are easily identified by their specific localization, we examined the immunohistochemical characteristics of these cells and found that they were double-positive for KIT and CD34 (Fig. 2, A to G). ICCs in the circular muscle layer of the stomach and small intestine and ICCs in the myenteric plexus region and circular muscle layer of the large intestine were also double-positive for KIT and CD34.    Download high-res image  Open in new tab  Download Powerpoint Figure 2 Coexpression of KIT and CD34 in ICCs surrounding myenteric ganglion cells in normal human small intestine. (A and B) Serial sections of the normal human small intestine. (C and D) Higher magnification of (A) and (B), respectively. (A) and (C) are stained with anti-KIT and (B) and (D) are stained with anti-CD34 (8). The localization of KIT and CD34 double-positive cells is consistent with that of ICCs, which are present between the circular muscle layer (cm) and the longitudinal muscle layer (lm), and surrounding the myenteric ganglion cells (gc) (11). (E to G) Demonstration by confocal laser scanning microscopy (8). (E) Binding of rabbit anti-KIT. (F) Binding of mouse anti-CD34. (G) Merged confocal image of (E) and (F). Arrows in (F) and (G) show endothelial cells and fibroblast-like cells that express CD34 but not KIT. Bars in (A) to (G), 50 μm.  We obtained the complete coding region of c-kit cDNA from six GISTs and control tissues using the reverse transcriptase–polymerase chain reaction (RT-PCR) (13). Ten independent c-kit clones were obtained from each sample. In 5 of the 6 GISTs (GIST 1 to 5), 4 to 6 clones out of 10 examined showed mutations in the region between the transmembrane and tyrosine kinase domains (hereafter called juxtamembrane domain) (Fig. 3). These mutations were located within an 11–amino acid stretch (Lys-550 to Val-560), but at nonidentical sites. No mutations were detectable in other domains of c-kit cDNA. Because ∼50% of the cDNA clones from each GIST did not show any mutations, we conclude that only one of the two c-kit alleles was mutated in each case. Direct sequencing of the PCR products confirmed the mutations in the juxtamembrane domain of c-kit cDNA in the five GISTs.    Download high-res image  Open in new tab  Download Powerpoint Figure 3 Mutations of c-kitin GISTs. GIST 1 showed an in-frame deletion of 6 base pairs (bp). GIST 2 showed an in-frame deletion of 15 bp. GIST 3 showed the same in-frame deletion as observed in GIST 2 and an additional point mutation at codon 550 (AAA to ATA) that resulted in a Lys550→Ile substitution. GIST 4 showed a point mutation at codon 559 (GTT to GAT) that resulted in a Val559→Asp substitution. GIST 5 showed an in-frame deletion of 27 bp. The c-kit mutations in the juxtamembrane domain of the HMC-1 human mast cell leukemia cell line (5) and the FMA3 murine mastocytoma cell line (6) are shown for comparison. Deleted amino acids are shown by dashes (–) and mutated amino acids by boxes. Murine and human KIT are of different lengths (1), so the amino acid numbering in the FMA3 KIT is different. Abbreviations used are as follows: SP, signal peptide; EC, extracellular domain; TM, transmembrane domain; JM, juxtamembrane domain; TK1 and TK2, tyrosine kinase domains; and KI, kinase insert. Abbreviations for the amino acid residues are as follows: D, Asp; E, Glu; G, Gly; H, His; I, Ile; K, Lys; L, Leu; M, Met; N, Asn; P, Pro; Q, Gln; T, Thr; V, Val; W, Trp; and Y, Tyr.  We next examined whether the c-kit mutations found in the GISTs resulted in constitutive activation of the c-kit receptor tyrosine kinase by transient introduction of the mutant c-kit cDNAs into the 293T human embryonic kidney (HEK) cell line (5, 14, 15). The wild-type c-kit cDNA was introduced as a negative control, and the tyrosine kinase domain mutant and the juxtamembrane domain mutant found in the HMC-1 human mast cell leukemia cell line (5) were introduced as positive controls. Wild-type KIT was phosphorylated on tyrosine only when recombinant human (rh) SCF was added to the culture medium (Fig.4A). In contrast, the gain-of-function KIT mutants found in HMC-1 cells were phosphorylated on tyrosine without the addition of rhSCF, as reported previously (5). The magnitude of the constitutive tyrosine phosphorylation was greater in the tyrosine kinase domain mutant than in the juxtamembrane domain mutant. The c-kitmutants found in GISTs also showed the constitutive tyrosine phosphorylation in 293T cells without rhSCF (Fig. 4A). The constitutive tyrosine phosphorylation of the juxtamembrane mutant of HMC-1 cells was of similar magnitude to that of the juxtamembrane mutants of GISTs. In in vitro kinase assays (16), the c-kit mutants found in the GISTs exhibited constitutive kinase activation that was similar in magnitude to that of the juxtamembrane domain mutant of HMC-1 cells (5) (Fig. 4B).    Download figure  Open in new tab  Download Powerpoint   Download figure  Open in new tab  Download Powerpoint Figure 4 Constitutive activation of the mutant KIT found in GISTs. The mutant c-kit cDNAs found in five GISTs were transfected into 293T cells. Human wild-type c-kit cDNA and mutant c-kit cDNAs found in the HMC-1 cells (6, 21) were also transfected as controls. (A) Tyrosine phosphorylation was examined with or without rhSCF stimulation (15). (B) Immune complex kinase assay carried out without rhSCF stimulation (16).  To investigate the biological consequences of the mutant c-kit, we introduced the c-kit mutations found in the GISTs into the mouse c-kit cDNA (17) and then stably transfected the cDNA into the interleukin 3 (IL-3)–dependent Ba/F3 murine lymphoid cell line (18). As a control, mouse wild-type c-kit cDNA was also transfected into Ba/F3 cells. We estimated Ba/F3 cell proliferation using an MTT colorimetric assay (19). Ba/F3 cells with the wild-type murine c-kitgrew in the presence of either recombinant mouse (rm) IL-3 or rmSCF; Ba/F3 cells with the mutated murine c-kit grew autonomously without rmIL-3 and rmSCF (Fig. 5, A and B). Ba/F3 cells with the mutated murine c-kit also grew autonomously in nude mice (Fig. 5C) (20). The constitutive kinase activation of all KIT mutants found in the five GISTs was confirmed in Ba/F3 cells (Fig. 5D) (5, 21).    Download figure  Open in new tab  Download Powerpoint   Download figure  Open in new tab  Download Powerpoint Figure 5 Autonomous proliferation of Ba/F3 cells transfected with mutated murine c-kit in culture and in nude mice. (A and B) MTT colorimetric assay (19) in the presence of rmIL-3 (A) or rmSCF (B). Data are expressed as the mean of four wells. Untransfected Ba/F3 cells (□), Ba/F3 cells with the murine wild-type c-kit (○), and Ba/F3 cells with the murine mutated c-kit [GIST 1 (▴), GIST 2 (•), GIST 3 (▾), GIST 4 (▪), and GIST 5 (x). (C) Development of tumors in nude mice (20) after injection of the Ba/F3 cells with each mutated c-kit. The original Ba/F3 cells and Ba/F3 cells with murine wild-type c-kit did not form tumors. Data are expressed as the mean of results in five mice. (D) Immune complex kinase assay (16). Ba/F3 cells with various c-kit cDNAs were cultured without rmSCF.  Although various cells including hematopoietic stem cells express both KIT and CD34 (22), ICCs are the only cells that are double-positive for KIT and CD34 in normal GI wall of humans. This strongly suggests that KIT and CD34 double-positive GISTs might originate from ICCs, although we cannot exclude the possibility that ICCs and GISTs simply show common undifferentiated characteristics such as those observed in multipotential hematopoietic stem cells.  The mechanism by which KIT becomes constitutively activated appears to be different for the tyrosine kinase domain mutant and the juxtamembrane domain mutant (6, 21). The former is constitutively activated without forming dimers (21), whereas the latter constitutively dimerizes without binding SCF (6, 21). The tyrosine kinase domain mutation of KIT has been found only in mast cell neoplasms (7) and its juxtamembrane domain mutation only in GISTs. The mechanisms by which these different mutations cause malignant transformation of different cell types remain to be investigated.\n"],"name":"stdout"}]},{"metadata":{"id":"5bIjbRB3ddqT","colab_type":"code","outputId":"9068e53f-7093-467c-f2bc-f2bfd608f107","executionInfo":{"status":"ok","timestamp":1543843396703,"user_tz":-330,"elapsed":1013,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":69}},"cell_type":"code","source":["import nltk\n","nltk.download('stopwords')"],"execution_count":0,"outputs":[{"output_type":"stream","text":["[nltk_data] Downloading package stopwords to /root/nltk_data...\n","[nltk_data]   Unzipping corpora/stopwords.zip.\n"],"name":"stdout"},{"output_type":"execute_result","data":{"text/plain":["True"]},"metadata":{"tags":[]},"execution_count":14}]},{"metadata":{"id":"QfHQCG7r7V2C","colab_type":"code","outputId":"30edc95f-2c60-4f30-9a05-66fbc3a62ae4","executionInfo":{"status":"ok","timestamp":1543843399886,"user_tz":-330,"elapsed":607,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":54}},"cell_type":"code","source":["words = set(stopwords.words('english'))\n","print(words)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["{'doesn', 'who', 'whom', 'are', 'aren', 'there', 'from', 'then', 'have', 'which', 'too', 'shan', \"wouldn't\", 'up', 'some', 'very', 'any', 'now', 'll', 'after', 'by', 'my', 'needn', 'nor', 'and', \"it's\", 'were', 'only', 'ours', 'they', 'out', 'haven', 'being', 'with', 're', 'before', 'it', 'each', \"weren't\", 'couldn', \"you'd\", \"couldn't\", 'its', 'ma', \"that'll\", \"don't\", 'other', 've', 'did', 'such', 'ourselves', 'm', 'to', 's', 'when', \"you're\", 'is', 'i', 'a', 'under', \"should've\", 'his', 'own', 'over', 'isn', 'not', 'against', 'once', 'all', 'how', 'no', 'can', 'about', 'themselves', 'been', \"haven't\", 'having', 'this', 'yourself', 'below', 'myself', 'should', 'most', 'himself', 'if', 'd', 'him', 'yourselves', \"you'll\", \"you've\", 'for', 'will', 'mustn', 'in', 'didn', 'their', 'than', \"needn't\", 'wouldn', 'itself', 'an', 'your', 'had', 'during', 'where', 'while', 'me', 'be', \"didn't\", \"mightn't\", 'wasn', 'why', 'those', 'but', 'same', 'won', 'few', 'we', 'yours', 'her', \"doesn't\", 'here', 'he', \"shouldn't\", 'just', \"aren't\", 'hadn', \"she's\", 'further', 'because', 'hers', 'am', 'has', 'until', 'she', 'off', 'so', 'that', 'herself', 'down', 'does', 'as', \"won't\", 'do', \"shan't\", 'again', 'both', 't', 'you', 'or', \"isn't\", \"wasn't\", 'the', 'hasn', 'through', \"hasn't\", 'mightn', 'more', 'shouldn', 'o', 'into', 'theirs', \"mustn't\", 'don', 'what', 'on', 'ain', \"hadn't\", 'of', 'was', 'at', 'y', 'between', 'our', 'these', 'weren', 'them', 'above', 'doing'}\n"],"name":"stdout"}]},{"metadata":{"id":"CspB1XDubyT_","colab_type":"text"},"cell_type":"markdown","source":["# **NLP Preprocessing on the Text data**"]},{"metadata":{"id":"QbO4d0lebW24","colab_type":"code","outputId":"44b1d801-48a7-4fec-dc36-fcb951ba0004","executionInfo":{"status":"ok","timestamp":1543843639493,"user_tz":-330,"elapsed":235639,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":104}},"cell_type":"code","source":["# loading stop words from nltk library\n","stop_words = set(stopwords.words('english'))\n","\n","\n","def nlp_preprocessing(total_text, index, column):\n","    if type(total_text) is not int:\n","        string = \"\"\n","        total_text = re.sub(\"[^a-zA-Z0-9\\n]\",\" \", str(total_text))\n","        total_text = re.sub(\"\\s+\",\" \", total_text)\n","        total_text = total_text.lower()\n","        \n","        l = ' '.join([w for w in total_text.split() if w not in stop_words])\n","        #removing special character, multiple space and stopwords from text\n","        data_text[column][index] = l\n","\n","for index, row in data_text.iterrows():\n","    nlp_preprocessing(row['TEXT'], index, 'TEXT')"],"execution_count":0,"outputs":[{"output_type":"stream","text":["/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:13: SettingWithCopyWarning: \n","A value is trying to be set on a copy of a slice from a DataFrame\n","\n","See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n","  del sys.path[0]\n"],"name":"stderr"}]},{"metadata":{"id":"yAod3WtQ9U0S","colab_type":"code","outputId":"2f2170ca-9fd5-4efe-9b20-082f8fdea1f7","executionInfo":{"status":"ok","timestamp":1543845936867,"user_tz":-330,"elapsed":2680,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":54}},"cell_type":"code","source":["print(data_text['TEXT'][3004])"],"execution_count":0,"outputs":[{"output_type":"stream","text":["c kit member type 3 subclass transmembrane receptor tyrosine kinases characterized 5 immunoglobulin like domains extracellular region negative regulatory juxtamembrane jm domain split adenosine triphosphate binding phosphotransferase tyrosine kinase domain reviewed broudy1 c kit activation stem cell factor scf promotes dimerization transphosphorylation tyrosine residues resulting downstream signaling events leading cell growth mutations c kit associated hematopoietic nonhematopoietic tumors including systemic mast cell disease smcd 2 4 acute myeloid leukemia aml 5 8 gastrointestinal stromal tumors gists 9 12 tyrosine kinase inhibitor imatinib mesylate gleevec novartis pharma ag basel switzerland efficacious majority patients gist harboring c kit mutations 9 13 c kit commonly activated gist tumors small deletions jm thought constitutively activate tyrosine kinase loss autoinhibitory function 14 c kit jm deletion mutants tested thus far sensitive inhibition imatinib mesylate observations suggested imatinib mesylate might value treatment smcd aml also activating mutations c kit however common activating alleles context smcd aml d816v d816y mutations c kit activation loop mutations shown highly resistant imatinib mesylate vivo15 vitro 16 19 pkc412 novel staurosporine derived tyrosine kinase inhibitor targets protein kinase c pkc kinase insert domain containing receptor kdr vascular endothelial growth factor receptor 2 vegf r2 platelet derived growth factor receptor pdgfr fms like tyrosine kinase 3 flt3 c kit 20 pkc412 previously shown effective fusion protein fip1 like 1 fip1l1 pdgfr mutations kinase domain resistant imatinib mesylate 21 since inhibitors d816v c kit mutations would potentially therapeutic activity smcd aml evaluated effectiveness pkc412 panel c kit mutations identified patients smcd gist aml study design cell culture 2 9 kb xhoi cdna fragment encoding human hu c kit lacking amino acids 510 513 gnnk cloned xhoi site pcdna3 c kit point mutations table 1 generated transformer site directed mutagenesis kit clontech palo alto ca subcloned xhoi site mscv ires gfp mscv pgk neo il 3 dependent mouse pre b cell line ba f3 maintained retrovirus prepared previously described 22 ba f3 cells transduced viral supernatant expanded 2 passages 10 ng ml rmil 3 selected il 3 independent growth ba f3 cells transduced neo containing vectors first selected il 3 100 mg ml g418 il 3 independent ba f3 cells transduced wild type wt c kit selected 20 ng ml rhscf view inlineview popup table 1 imatinib mesylate pkc412 ic50s transforming c kit mutations patient samples collected following informed consent accordance institutional review board irb approved dana farber cancer institute protocol 01 206 growth inhibition cells 1 104 100 l medium without il 3 incubated various concentrations imatinib mesylate pkc412 96 well plates 48 hours triplicate 3h thymidine 1 ci 0 037 mbq added cells incubated 4 hours cells harvested 3h thymidine incorporation determined growth inhibition plotted ratio average 3h thymidine incorporation drug treated wells relative drug controls ic50s imatinib mesylate pkc412 calculated sigmoidal curve fitting software originlab northampton immunoprecipitations cell lysates immunoprecipitations immunoblotting performed previously described 22 c kit precipitated 5 g anti human c kit antibody assay design systems ann arbor mi separated 7 5 denaturing polyacrylamide gel transferred nitrocellulose blots probed mouse anti phosphotyrosine antibody 4g10 upstate charlottesville va stripped reprobed mouse anti human c kit e1 sc 17806 santa cruz biotechnology santa cruz ca whole cell lysates 1 10th volume immunoprecipitate ip input similarly separated probed rabbit anti phospho stat5 signal transducer transcriptional activator 5 rabbit anti phospho stat3 stripped reprobed rabbit anti stat5a rabbit anti stat3 cell signaling technology beverly assess constitutive tyrosine kinase autophosphorylation cells grown rpmi plus 10 fetal calf serum fcs 4 mg cell lysate immunoprecipitated assess pkc412 inhibition c kit autophosphorylation cells washed 3 times rpmi without fcs il 3 incubated without il 3 pkc412 0 500 nm 4 hours wt ba f3 cells starved il 3 serum 4 hours prior addition il 3 pkc412 cell equivalents 1 107 cell lysate immunoprecipitated results discussion il 3 dependent murine pre b cell line ba f3 transduced retroviruses expressing wt c kit 1 14 c kit mutants selected il 3 independent growth table 1 figure 1a ten 14 mutant forms c kit transformed ba f3 cells il 3 independent growth absence rhscf transforming mutations mapped extracellular domain deltyd 417 419 8 deltyd 417 419 rg 8 a502y503dup aydup 10 juxtamembrane domain v559d 11 v560g 2 delv559v560 delvv 11 kinase domain r634w d816y 3 d816v 2 n822k7 first report kinase domain mutation r634w identified patient chronic myelomonocytic leukemia consistent report e839k inactivates kinase domain 12 e839k transform ba f3 cells il 3 independent growth k642e shown constitutively phosphorylated gist derived cell line10 23 transform ba f3 cells study d820g4 v530i 8 identified single patient also transform ba f3 cells immunoprecipitated c kit constitutively phosphorylated transformed cell lines ba f3 cells expressing wt c kit grown 20 ng ml rhscf figure 1b observed differential phosphorylation 2 c kit bands approximately 160 145 kda representing fully glycosylated cell surface receptor incompletely processed internalized forms c kit respectively findings confirm reports stat3 aberrantly phosphorylated d816v c kit activating mutations 24 25 stat5 constitutively phosphorylated c kit activating mutations grown absence il 3 figure 1 download figureopen new tabdownload powerpoint figure 1 imatinib mesylate resistant c kit mutations sensitive pkc412 schematic c kit protein indicating relative location structural functional domains well location 14 c kit mutations evaluated jm delvv mutation occurs v559v560 indicates amino terminal signal peptide domains c kit indicated follows extracellular domain tm transmembrane domain jm juxtamembrane domain tk1 nucleotide binding subdomain split kinase domain ki kinase insert domain tk2 catalytic subdomain split kinase domain b ba f3 transforming c kit mutations constitutively phosphorylated phosphorylate stat3 stat5 blots top bottom antiphosphotyrosine western blot ba f3 lysates immunoprecipitated anti c kit anti c kit western blot immunoprecipitates top blot anti phospho stat3 western blot whole cell lysates wcls ip top blot anti stat3 western blot wcls anti phospho stat5 western blot wcls ip top blot anti stat5a western blot wcls c representative growth curves deltyd rg v560g d816y d816v transduced ba f3 cell lines plotted percentage sd 3h thymidine incorporation drug treated cells relative drug controls cells treated imatinib mesylate black pkc412 red cells grown absence il 3 rhscf dose response pkc412 inhibition c kit stat5 stat3 phosphorylation shown data deltyd rg v560g d816y d816v transformed ba f3 cells blots panel b cells grown absence il 3 indicated concentration pkc412 nm 4 hours e pkc412 inhibit il 3 induced stat5 phosphorylation blots top bottom anti phospho stat3 western blot wcls indicated cell lines without il 3 10 ng ml pkc412 250 nm ml ba f3 cells starved il 3 serum 4 hours prior addition il 3 pkc412 anti stat3 western blot wcls top blot anti phospho stat5 western blot wcls top blot anti stat5a western blot wcls f dose response stat5 phosphorylation pkc412 ba f3 cells blots panel e ba f3 cells starved il 3 fcs 4 hours incubated without il 3 10 ng ml presence indicated concentration pkc412 4 hours effect imatinib mesylate pkc412 cell growth c kit expressing cell lines evaluated 3h thymidine incorporation assay table 1 figure 1c data shown consistent previous reports d816v resistant imatinib mesylate ic50 10 000 nm 16 19 d816y also observed resistant imatinib mesylate ic50 800 nm four mutations deltyd aydup r634w n822k showed ic50s imatinib mesylate similar ba f3 wt c kit 20 ng ml rhscf 100 300 nm previously reported v560g19 hypersensitive imatinib mesylate mutations deltyd rg v559d delvv ic50s 100 nm pkc412 inhibited growth c kit transformed ba f3 lines ic50s segregating 2 groups imatinib mesylate resistant d816v d816y mutants well deltyd rg v560g r634w n822k sensitive pkc412 ic50s 33 95 nm wild type receptor stimulated rhscf ic50 138 nm remaining mutations higher ic50s pkc412 146 265 nm level tyrosine autophosphorylation imatinib mesylate resistant d816v d816y c kit mutants well phosphorylation stat3 stat5 dose dependent pkc412 wide concentration range 5 500nm correlated inhibition cell growth table 1 figure 1d similar findings observed representative imatinib mesylate sensitive extracellular del tyd rg jm v560g mutants although stat5 phosphorylation d816v d816y inhibited pkc412 figure 1d e increased level stat5 phosphorylation induced stimulation il 3 induction inhibited cells figure 1e il 3 rescue pkc412 inhibition phosphorylation stat3 figure 1e inhibit il 3 induced stat5 phosphorylation wt ba f3 cells figure 1e f data indicate among reported c kit mutations human hematologic malignancies 10 14 confer factor independent growth ba f3 cells variable sensitivity mutations inhibition imatinib mesylate pkc412 notably however common mutations d816v d816y highly resistant imatinib mesylate sensitive inhibition pkc412 findings suggest pkc412 may effective therapeutic agent treatment patients certain imatinib mesylate resistant c kit mutations activating mutations either braf nras seen significant number malignant melanomas incidence appears dependent ultraviolet light exposure thus braf mutations highest incidence non chronic sun damaged csd uncommon acral mucosal csd melanomas recently activating kit mutations described rare cases metastatic melanoma without reference clinical phenotypes finding intriguing since kit expression downregulated melanomas progressing aggressive lesions study investigated group anal melanomas presence braf nras kit pdgfra mutations heterozygous kit exon 11 l576p substitution identified 3 20 cases tested 3 kit mutation carrying tumors strongly immunopositive kit protein kit mutations identified tumors less 4 kit immunostaining nras mutation identified one tumor braf pdgfra mutations identified either kit positive negative anal melanomas vitro drug testing stable transformant ba f3 kitl576p mutant cells showed sensitivity dasatinib previously known bms 354825 dual src abl kinase inhibitor imatinib however compared imatinib sensitive kit mutant dasatinib potent lower doses imatinib kitl576p mutant results suggest subset anal melanomas show activating kit mutations susceptible therapy specific kinase inhibitors 2007 wiley liss inc kit receptor tyrosine kinase plays critical role normal development function melanocytes embryonic stage postnatal organism 1 2 loss function mutations kit ligand result white spotting phenotype mice humans 3 4 5 contrast activating kit mutations implicated pathogenesis certain human malignancies gastrointestinal stromal tumors seminomas mastocytomas reported recently patients malignant melanoma furthermore kit expression typically found normal melanocytes benign nevi situ melanoma appears downregulated invasive metastatic melanomas 6 progressive decrease kit immunopositivity associated increasing dermal invasiveness tumor one hypothesis kit signaling may important regulating cell differentiation tissue morphogenesis therefore loss expression may required step course tumor progression 7 context isolated reports activating kit mutations melanoma patients 8 9 intriguing suggest alternative mechanism constitutive activation kit oncogenic mutations positively regulate mitogenesis induces neoplastic transformation genetic alterations identified melanomas different sites different levels sun exposure indicate distinct molecular subsets melanoma direct implications therapeutic approaches prevention strategies braf mutations example often seen melanomas arising skin intermittently exposed sun rare events csd areas never exposed sun acral mucosal membranes 10nras mutations occur melanomas without braf mutations appear associated specific clinical phenotype 10 contrast increased copy number ccnd1 cyclin d1 cdk4 two downstream genes ras mapk signaling pathway seen csd melanomas acral mucosal melanomas inversely correlating braf mutations recent genetic classification malignant melanomas based relationship ultraviolet light exposure yet include group melanomas associated activating kit mutations purpose study investigate incidence kit expression oncogenic mutations group mucosal melanomas located anus assess vitro efficacy specific kinase inhibitors cell lines expressing kit mutants since anal mucosal melanomas associated poor prognosis presence oncogenic kit mutations could provide rationale exploring use imatinib related kinase inhibitors treatment patients tumor 11 materials methods patient selection clinicopathologic features institutional review board human tissue utilization committee memorial sloan kettering cancer center mskcc approved study archive department pathology searched cases anal melanoma 1989 2005 twenty six patients 13 women 13 men ranging age 34 89 years identified pathologic diagnosis confirmed review histologic slides immunohistochemical markers representative formalin fixed paraffin embedded block viable tumor selected used immunohistochemistry well molecular studies described pcr fish analyses tissue samples used study included 20 primary 6 metastatic anal mucosal melanomas 1 patient primary metastatic tumors examined kit cd117 immunohistochemistry performed murine monoclonal antibody dakocytomation carpinteria ca 1 100 dilution hiph antigen retrieval solution dakocytomation endogenous mast cells used internal positive controls immunostaining evaluated microscopically graded based estimated amount immunopositive tumor cells follows 25 2 25 50 50 75 75 kit pdgfra mutational analysis genomic dna isolated 26 cases formalin fixed paraffin embedded tissue using dneasy tissue kit qiagen valencia ca one microgram genomic dna subjected pcr using platinum taqdna polymerase high fidelity life technologies gaithersburg md pcr conditions follows 94 c 4 min ii 94 c 30 sec iii relevant annealing temperature primer set 30 sec iv 72 c 30 sec 35 cycles v 72 c 3 min pcr products identified agarose gel electrophoresis using 2 metaphor agarose gel biowhittaker applications rockland pcr products purified qiaquick pcr purification kit qiagen sequencing cases tested kit exons 11 13 17 well pdgfra exon 12 18 primer sequences annealing temperatures listed table one case sample 19 frozen tissue available total rna extracted subjected rt pcr followed direct sequencing full length kit cdna rt pcr conditions primers used previously published 12 every abi sequence compared ncbi human kit gene nucleotide sequence blasted using ncbi standard nucleotide blast search determine presence mutation within particular exon table kit exon 11 13 17 pdgfra exon 12 18 primer sequences corresponding annealing temperatures exon number primers primer sequences 5 3 ta c kit 11 ex11 f ccagagtgctctaatgactg 53 ex11 r acccaaaaaggtgacatgga 13 ex13 f atcagtttgccagttgtgct 53 ex13 r tttataatctagcattgcc 17 ex17 f tgtgaacatcattcaaggcgtac 53 ex17 r caggactgtcaagcagagaatgg pdgfra 12 ex12 f tccagtcactgtgctgcttc 54 ex12 r gcaagggaaaagggagtctt 18 ex18f accatggatcagccagtctt 55 ex18r tgaaggaggatgagcctgacc braf nras mutational analysis ligase detection reaction ldr sensitive specific assay detection individual point mutations13 14 15 16 performed detect braf13 nras point mutations primers designed 19 possible mutations described nras codons 12 13 61 table ii integrated dna technologies coralville ia exon specific pcr product mixed fluorescently labeled mutant specific primers common downstream primers ligation performed taq dna ligase new england biolabs reaction mixture cycled follows 94 c 1 5 min initial denaturation phase followed 10 cycles 94 c 1 min 65 c 4 min 67 c 4 min used codon 13 mutations three microliters final ldr product mixed 2 l loading buffer 83 formamide 83 mmol l edta 0 17 blue dextran denatured 95 c 5 min mixture placed ice rapid cooling loaded 12 5 polyacrylamide gel run 377 dna prism sequencer applied biosystems electrophoresis 1 400 v 2 5 hr fluorescently labeled primers designed varying lengths caused differential electrophoretic mobility shifts ligated products genescan 350 tamra molecular weight marker used size standard genescan software v3 1 utilized data analysis table ii ligase detection reaction primer sequences used nras mutational analysis ldr primer ta c primer size product length primer sequence nras codon 12 g12 1r 55 8 21 42 5 fam atc tgg tgg tgg ttg gag cac 3 g12 1c 58 5 19 40 5 fam ctg gtg gtg gtt gga gca 3 g12 1s 55 8 17 38 5 fam ggt ggt ggt tgg agc aa 3 com12 1 60 9 21 5 phos gtg gtg ttg gga aaa gcg cac 3 g12 2a 61 9 22 48 5 fam atc tgg tgg tgg ttg gag cag c 3 g12 2d 60 5 20 46 5 fam ctg gtg gtg gtt gga gca ga 3 g12 2v 58 2 18 44 5 fam ggt ggt ggt tgg agc agt 3 com12 2 61 6 26 5 phos tgg tgt tgg gaa aag cgc act ttt tt 3 nras codon 13 g13 1r 64 8 23 44 5 fam act ggt ggt ggt tgg agc agg tc 3 g13 1c 58 7 19 40 5 fam gtg gtg gtt gga gca ggt 3 g13 1s 62 0 21 42 5 fam tgg tgg tgg ttg gag cag gta 3 com13 1 59 8 21 5 phos gtg ttg gga aaa gcg cac tga 3 g13 2a 67 3 24 50 5 fam act ggt ggt ggt tgg agc agg tgc 3 g13 2d 64 6 22 48 5 fam tgg tgg tgg ttg gag cag gtg 3 g13 2v 60 8 20 46 5 fam ctg gtg gtt gga gca ggt gt 3 com13 2 61 4 26 5 phos tgt tgg gaa aag cgc act gac aat 3 nras codon 61 q61 1e 62 8 24 51 5 fam ctc atg gca ctg tac tct tct tc 3 q61 1k 56 2 23 50 5 fam ctc atg gca ctg tac tct tct tt 3 com61 1 61 5 27 5 phos tcc agc tgt atc cag tat gtc caa caa 3 q61 2r 57 3 23 47 5 fam tct cat ggc act gta ctc ttc tc 3 q61 2l 57 1 24 48 5 fam ctc tca tgg cac tgt act ctt cta 3 q61 2p 59 5 25 49 5 fam tct ctc atg gca ctg tac tct tct g 3 com61 2 59 6 24 5 phos gtc cag ctg tat cca gta tgt cca 3 q61 3ha 59 7 25 52 5 fam gtc tct cat ggc act gta ctc ttc 3 q61 3hb 62 0 26 53 5 fam ggt ctc tca tgg cac tgt act ctt cg 3 com61 3 59 6 27 5 phos tgt cca gct gta tcc agt atg tcc aac 3 control ex1 upstream 56 8 28 55 5 fam cta atc cag aac cac ttt gta gat gaa 3 ex1 downstream 61 3 27 5 phos atg atc cca cca tag agg tga ggc cca 3 ex2 upstream 64 5 24 56 5 fam gag gac agg cga agg ctt cct ctg 3 ex2 downstream 58 8 32 5 phos tgt att tgc cat caa taa tag caa gtc att tg 3 fluorescence situ hybridization fluorescence situ hybridization fish performed paraffin sections 5 tumors according standard procedures kit probes used 2 overlapping bac clones ctd 3180g20 rp11 722f21 invitrogen carlsbad ca labeled nick translation spectrum green vysis abbott laboratories il chromosome 4 centromeric probe labeled spectrum orange cep 4 vysis used reference probe mix 50 80 ng kit bac 2 l cot 1 dna invitrogen ethanol precipitated resuspended hybridization buffer kit probe mix denatured 70 c 10 min followed pre annealing 37 c 30 min kit probe combined denatured cep 4 probe slide coverslipped incubated overnight 37 c standard post hybridization washes slides stained dapi mounted antifade vectashield vector laboratories burlingame ca analysis performed using nikon e800 epifluorescence microscope metasystems isis 3 imaging software minimum 100 cells scanned separate regions slide western blotting 1 case sample 19 frozen tissue available whole cell lysate prepared grinding 1 g snap frozen tumor tissue using powergen 700 homogenizer omni international marietta ga ground tissue resuspended radioimmunoprecipitation assay lysis buffer upstate lake placid ny containing cocktail protease phosphatase inhibitors sigma st louis mo sodium fluoride sodium orthovanadate phenylmethylsulfonyl fluoride electrophoresis immunoblotting performed protein extracts using standard protocols 12 antibodies tested included rabbit polyclonal anti phospho kit y721 zymed lab san francisco ca rabbit anti kit oncogene science boston anti phospho pdgfralpha cell signaling technology danvers anti pdgfralpha santa cruz biotechnology santa cruz ca mouse anti actin santa cruz secondary antibodies used included donkey antimouse secondary santa cruz antirabbit calbiochem la jolla ca following hybridization secondary antibody blots incubated immun star horse radish peroxidase luminal enhancer biorad hercules ca exposed onto kodak biomax mr film eastman kodak rochester ny establishing ba f3 kit l576p stable transformant cells kit exon 11 l576p mutation introduced site directed mutagenesis pcr using quickchange ii xl site directed mutagenesis kit qiagen wild type kit cdna cloned retroviral expression vector pmscv generous gift dr gary gilliland harvard medical school similar strategy used obtaining kit exon 11 v559d imatinib sensitive juxtamembrane mutation used control vitro drug testing primers used mutagenesis listed table iii pmscv dna constructs introduced ba f3 cells german collection microorganisms cell cultures human animal cell cultures braunshweig germany electroporation ten million ba f3 cells cotransfected pmscv kit dna constructs 25 mg hygromicin resistance plasmid 1 g using genpulser biorad hercules ca electroporated cells first selected hygromycin week remaining cells sorted flow cytometry using gfp fluorescence sorted gfp positive cells grown 2 weeks kit expression confirmed flow cytometry using pe conjugated anti cd117 antibody bd biosciences san jose ca ba f3 cells kitl576p kitv559d mutant cells survived without kit ligand kl however slower growth increased cell death noticed absence kl adding kl used improve cell growth order obtain cell population minimum dead cells apoptosis assays ba f3 kit mutant cells showed better growth rate presence low dose kl 20 ng ml likely due partial kl independent growth concentration kl used 20 ng ml quite low compared 100 ng ml used maintaining growth ba f3 wtkit table iii kit exon 11 primers used site directed mutagenesis pcr mutation primer v559d f 5 gaagtacagtggaaggatgttgaggagata 3 r 5 tatctcctcaacatccttccactgtacttc 3 l576p f 5 atagacccaacacaacctccttatgatcacaaa 3 r 5 tttgtgatcataaggaggttgtgttgggtctat 3 vitro drug testing dasatinib bms 35482 tm sprycel nilotinib amn107 tm tasigna imatinib sti571 tm gleevec tested ba f3 kitl576p cells efficacy compared ba f3 kitv559d cells imatinib sensitive drug tested 10 100 1 000 5 000 nm nontreated cells served baseline control drug response monitored measuring cell proliferation apoptosis well inhibition kit autophosphorylation cell proliferation assays determine growth inhibitory drug effects ba f3 transformants containing kit mutants starved growth factor 4 hr prior drug administration ten minutes adding drug growth factors added 10 ng ml il3 negative control baf3 cells 20 ng ml kl ba f3 cell lines expressing kit mutant isoforms cells incubated imatinib dasatinib nilotinib indicated doses 24 well plates 37 c 48 hr incubation drug brdu added cell culture incubated 6 hr harvesting cells fixed stained protocol recommended brdu apc kit pharmingen san diego ca brdu incorporation determined flow cytometry minimum 2 105 events acquired data analyzed using flowjo software version 5 7 2 cell growth curves plotted graphpad prism software version 4 03 ic50s calculated software apoptosis assays induction apoptosis ba f3 kit transformants evaluated flow cytometry using annexinv pe apoptosis detection kit pharmingen cells density 1 106 activated kl cultured 6 well plates starved growth factors 4 hr prior drug delivery ten minutes adding drug growth factors added cells incubated imatinib dasatinib nilotinib concentrations indicated 48 hr cells harvested drug treatment stained anti annexinv pe antibody using provider protocol minimum 10 000 events analyzed facscan becton dickison within hour staining data results analyzed flowjo immunoprecipitation western blotting ba f3 cells expressing mutant kit isoforms l576p v559d treated indicated doses drug cells starved serum growth factors 2 hr growth factors added back media 15 min prior drug administration drugs incubated cells 37 c absence serum 90 min treatment cells harvested subjected protein extraction using similar protocol similar whole cell lysate 200 g incubated 2 g anti kit antibody assay designs ann arbor mi hour mixture incubated magna beads pierce biotechnology rockford il overnight 4 c beads washed isolated using magnetic column resuspended lds sample buffer invitrogen electrophoresis immunoblotting performed protein extracts using standard protocol phosphoryated kit detected anti phospho tyrosine antibodies tyr20 tyr 99 santa cruz total kit detected mouse monoclonal anti c kit santa cruz secondary antibodies used donkey anti mouse secondary santa cruz results presence kit mutations correlated strong diffuse expression kit immunohistochemical analysis 6 23 26 anal melanomas tested showed diffuse strong 4 reactivity cd117 three tumors primary anorectal melanomas two metastases known primary anal tumors three primary tumors strong kit expression pattern seen intraepithelial situ invasive melanoma component fig 1a 1 patient strong homogeneous kit immunoreactivity seen primary melanoma well corresponding metastasis fig 1b additional 3 cases showed heterogeneous weak 1 immunostaining metastatic lesions primary anal melanomas figure 1 figure 1 open figure viewer immunohistochemical staining anal melanoma primary melanoma strong diffuse kit immunoreactivity intraepidermal situ invasive component 100 b metastasis derived tumor also strongly diffusely positive cd117 200 adequate dna assessing activating mutations braf nras kit pdgfra obtained 20 26 cases table iv heterozygous kit exon 11 l576p substitution identified 3 13 20 cases fig 2 however 3 4 diffusely kit immunopositive 4 anal melanomas adequate dna available showed kit mutations kit mutations identified tumors less 4 kit immunostaining tumor high expression kit immunohistochemically lacked activating kit mutation dna pcr direct sequencing hot spots studied full length kit cdna sequencing confirmed wild type sequence one 5 20 anal melanomas tested showed nras mutation codon 61 pdgfra braf mutations identified either kit positive negative anal melanomas figure 2 figure 2 open figure viewer kit genotyping showed identical heterozygous l576p substitution exon 11 75 anal melanomas expressing strong kit immunoreactivity upper abi depicts kit exon 11 forward sequence lower abi sequence kit exon 11 reverse arrow indicates position l576p substitution exon 11 resulting amino acid change leucine proline position 576 corresponds codon change ctt cct table iv genotyping results nras kras braf kit pdgra mutation analysis correlated kit expression mmunohistochemistry sample nras codon 12 nras codon 13 nras codon 61 kras codon 12 13 braf 599 kit ex11 kit ex13 kit ex17 pdgfra ex 12 pdgfra ex 18 kit ihc 1fish showed 2 copies kit signal 2western blotting showed phosphorylated kit protein 3fish showed increased number kit signals compared cep4 na available wt wild type 1 wt wt wt wt wt wt wt wt wt wt neg 21 wt wt 61 cod lys wt wt wt wt wt wt wt neg 3 wt wt wt wt wt wt wt wt wt wt neg 4 wt wt wt wt wt wt wt wt wt wt neg 5 wt wt wt na wt wt wt na wt wt 1 6 wt wt na wt wt wt wt na wt wt neg 7 wt wt wt wt wt wt wt na wt wt neg 8 wt wt wt wt wt wt wt wt wt wt neg 9 wt wt wt wt wt wt wt wt wt wt neg 10 wt wt wt wt wt wt wt wt wt wt 1 11 wt wt wt wt wt wt wt wt wt wt neg 12 na na wt wt wt wt wt wt wt wt neg 13 wt wt wt wt wt wt wt wt wt wt 1 14 wt wt wt wt wt wt wt wt wt wt neg 15 wt wt wt wt wt na wt wt wt wt neg 16 na na na na na l576p wt na na na 4 17 wt wt wt wt wt l576p wt wt na na 4 181 wt wt wt wt wt wt wt wt wt wt neg 191 2 wt wt wt wt wt wt wt wt wt wt 4 203 wt wt wt wt wt l576p wt wt wt wt 4 increased copy number kit gene rare event anal melanoma fluoresence situ hybridization status kit copy number changes amplification studied 5 anal melanomas three 5 cases selected 4 kit immunoreactive tumors 2 cases kit negative immunohistochemistry except one case showed 2 copies kit signal kit cep4 ratio 1 1 fig 3a sample 19 one kit immunopositive tumors increased kit copy number kit cep4 ratio 5 2 fig 3b sample 20 case also showed presence kit exon 11 l576p substitution figure 3 figure 3 open figure viewer fluorescence situ hybridization showing 2 copies kit red centromere 4 green sample 19 b increased copy number kit signals average ratio kit red centromere 4 green signals 5 2 sample 20 kit activation phosphorylation detected kit positive anal melanomas kit phosphorylation status investigated one tumor frozen tissue available sample 19 western blotting phospho total kit antibodies compared kit exon 11 mutated gist sample 19 belonged patient showing 4 kit immunopositive tumor lacking concurrent kit mutation western blot showed phosphorylated kit protein using kit y721 antibody similar pattern gist positive control result confirmed stripping probing total kit antibody fig 4 however sample lacked phosphorylated pdgfra data shown figure 4 figure 4 open figure viewer western blotting showing kit phosphorylation sample 19 similar pattern gist positive control harboring kit exon 11 mutation dasatinib potently inhibited cell proliferation kit phosphorylation ba f3 kitl576p cells establishing stable transformants il3 independent selection performed withdrawing il3 providing kl different doses 10 ng 100 ng ml data shown ideal kl concentration kit mutant determined counting trypan blue stained cells versus viable cells phase contrast microscope transforming activity determined il3 independent growth curve effects imatinib dasatinib nilotinib transformed ba f3 kitl576p mutant cells analyzed compared ba f3 cells harboring imatinib sensitive kit exon 11 mutation v559d dasatinib potently inhibited proliferation cells ic50 54 nm phosphorylation kit ba f3 kitl576p cells imatinib nilotinib also inhibited cell proliferation ic50 253 nm 185 nm respectively fig 5a table v apoptotic studies showed dasatinib induced apoptosis concentration 100 nm imatinib nilotinib showed increased apoptosis 1 000 nm fig 5c immunoprecipitation treated cell lysates showed dasatinib distinctly decreases kit phosphorylation 100 nm phosphorylation completely absent 1 000 nm fig 5e comparison imatinib treated cells showed decrease phosphorylation 1 000 nm cells treated nilotinib showed decrease kit phosphorylation 100 nm decrease phosphorylation doses 1 000 nm figure 5 figure 5 open figure viewer vitro drug sensitivity imatinib dasatinib nilotinib ba f3 stable transformants expressing kitl576p kitv559d b proliferation inhibition c increased apoptosis e f decrease kit phosphorylation immunoprecipitation table v growth inhibitory drug effects ba f3 transformants containing kit mutants using 3 kinase inhibitors results presented ic50 nm ba f3 kitl576p ba f3 kitv559d imatinib 253 27 dasatinib 54 22 nilotinib 185 34 comparison 3 drugs potently inhibited cell proliferation ba f3v559d cells fig 5b ic50 imatinib dasatinib nilotinib 27 nm 22 nm 34 nm respectively table v apoptotic assays showed significantly increased apoptosis drug concentrations 100 nm fig 5d distinct decrease kit phosphorylation immunoprecipitation identified imatinib nilotinib treatment 100 nm fig 5f dasatinib treated cells showed mild decrease phosphorylation even 10 nm marked decrease 100 nm discussion kit receptor plays important role normal melanocyte development proliferation 17 kit expressed junctional component benign melanocytic nevi situ malignant melanomas tends reduced lost tumor progression invasive metastatic melanoma6 observation suggested kit might act tumor suppressor gene one hypothesis since kit ligand regulate normal melanocyte proliferation high concentration protein might prevent development vertical growth inhibit melanoma tumor progression consequence kit expression considered diagnostic utility decreased expression paralleling tumor progression implicated development tumorigenic melanoma furthermore kit expression lost human melanoma cell lines18 one study growth appeared inhibited adding kit ligand media 19 studies suggest kit loses responsiveness ligand vitro proliferative activity becomes independent kit signaling however role kit melanomagenesis likely complex since subset melanomas found maintain kit overexpression even metastatic lesions isolated reports identified activating kit mutations melanomas went et al identified kit exon 11 l576p mutation one 2 melanomas immunohistochemically positive kit 8 study included 39 melanomas 35 cases showed degree kit reactivity 15 showing intense staining pattern follow study group tested 10 additional kit positive uveal melanomas identify kit mutations 20 larger series willmore payne et al 9 studied 100 malignant melanomas found 29 rate kit positivity ihc ranging weak strong study include mucosal melanomas information tumor location sun exposure specified junctional situ component melanomas showed strong kit staining invasive component consistently negative kit expression 2 cases metastatic melanoma strong immunoreactivity kit however matching primary tumors available assess status kit expression invasive component tumors kitl576p substitution mutation juxtamembrane domain kit study suggested kit activating mutations low incidence malignant melanomas 2 recent immunohistochemical study primary anorectal melanomas 21 identified kit expression 12 16 cases tested diffuse staining pattern 38 tumors however mutational analysis provided kit positive primary tumors findings indicate anal melanomas lack braf mutations infrequently show nras mutations 5 instead show higher rate kit activating mutations compared non acral cutaneous melanomas 9 overall incidence kit immunoreactivity anal melanomas 20 defined strong diffuse pattern staining found good correlation kit immunoreactivity presence kit gene alterations molecular level 75 rate kit mutations amongst anal melanoma expressing strong kit protein immunohistochemistry although cases need tested true kit gene amplification appear common event anal melanomas however modest increased copy number kit gene seen one third cases kit activating mutations increased copy number mutually exclusive one anal melanoma showing increased kit copy number fish addition kitl576p substitution mutation another important observation study kit protein phosphorylation detected case 19 seen tumors strong diffuse kit immunoexpression absence identifiable kit pdfgra mutations increased copy number changes suggesting kinase inhibition might still beneficial cases findings recapitulate closely molecular phenotype kit pdgfra wild type gists also managed specific kit receptor tyrosine kinase inhibitors kitl576p mutation juxtamembrane domain kit unique melanoma previously described gastrointestinal stromal tumors gist 22 experience 500 gist cases incidence l576p mutation 1 typically heterozygous lacks specific clinical phenotype 1 l576p mutated gist patients treated imatinib metastatic disease developed secondary resistance drug 13 months duration 12 interest 2 melanoma cases reported willmore payne9 harboring l576p substitution lacked normal allele consistent homozygous deletion significance finding remains elucidated since kit mutant anal melanomas heterozygous pattern also kit mutations far reported melanoma patients somatic familial cases melanomas gist associated particular juxtamembrane mutation vitro drug testing revealed kitl576p mutation sensitive dual src abl kinase inhibitor dasatinib dasatinib orally active small molecule atp competitive effectively inhibits cell proliferation nearly imatinib resistant bcr abl isoforms 23 due structural homology kit activation loop mutants imatinib resistant bcr abl isoforms dasatinib shown inhibit kit kinase activity including imatinib resistant kit activation loop mutants 24 kitl576p mutant also sensitive imatinib nilotinib requiring least 10 fold higher dose compared imatinib sensitive kitv559d mutation recently genetic classification melanomas put forward based anatomic location ultraviolet light exposure 10 thus common type melanoma occurring skin without csd frequently harbors either mutation braf nras suggesting possible therapeutic intervention targeting ras raf erk pi3k pathway contrast melanomas skin csd show frequent increases copy number ccnd1 would less likely responsive braf inhibitors sorafenib incidence activating kit mutations might also dependent degree sun exposure since varied 2 study willmore payne 9 include mucosal cases 13 study anal melanomas summary findings show increased prevalence activating kit mutations specific subset mucosal melanomas adding another layer complexity genetic heterogeneity far described melanomas finding significant choosing appropriate targeted therapy according findings l576p mutant responsive specific kinase inhibitors particularly dasatinib additional studies including subtypes mucosal acral melanomas needed investigate increased incidence kit activation intrinsic anal melanoma pertains melanomas limited ultraviolet light exposure recently identified kitexon11mutation anorectal melanoma patient excellent response treatment imatinib determine frequency kit mutations across melanoma subtypes surveyed large series tumors experimental design one hundred eighty nine melanomas screened mutations kit exons 11 13 17 kit copy number assessed quantitative pcr asubset cases evaluated braf nras mutations immunohistochemistry done assess kit cd117 expression results kit mutations detected 23 3 13 acral melanomas 15 6 7 45 mucosal melanomas 7 7 1 13 conjunctival melanomas 1 7 1 58 cutaneous melanomas 0 0 60 choroidal melanomas lmost kit mutations type predicted imatinib sensitive overlap nras mutations 11 1 acral 24 3 mucosal tumors braf mutations absent mucosal tumors increased kit copy number detected 27 3 3 11 acral 26 3 10 38 mucosal melanomas less common among cutaneous 6 7 3 45 conjunctival 7 1 1 14 choroidal melanomas 0 28 cd117 expression present 39 105 tumors representing melanoma types correlate either kit mutation status kitcopy number conclusions findings confirm kit mutations common acral mucosal melanomas necessarily correlate kit copy number cd117 expression screening kit mutations may open new treatment options melanoma patients incidence malignant melanoma risen dramatically past 50 years yet current therapies advanced disease provide modest benefit 1 recent progress unraveling molecular genetics cancer emergence several subtypes distinguished site origin tumor genotype 2 example curtin colleagues 3 observed 81 melanomas arising skin without chronic sun induced damage oncogenic braf nras mutation whereas mutations less common chronic sun induced damage melanomas acral melanomas mucosal melanomas 1 contrast melanomas arising chronic sun induced damage skin acral sites likely increased ccnd1 copy number loss cdkn2a gain cdk4 frequent alterations mucosal acral melanomas 4 5 melanomas oncogenic mutations kit observed several recent studies 4 6 9 although mutations seem rare braf nras mutations may reflect important role kit tyrosine kinase melanocyte development 10 kit mutations seem common mucosal acral melanomas compared cutaneous melanomas sometimes accompanied increase kit copy number 4 6 9 importantly reported mutations type predicted sensitive kit kinase inhibitors success treatment gastrointestinal stromal tumors gist imatinib gleevec novartis pharma led speculation kit mutant melanoma might also managed approach increases kit copy number also observed melanoma 4 cases accompanying mutation others none detected whether kit inhibitor would effective tumors amplified kit detectable mutation another question great interest phase ii trial imatinib patients unresectable acral mucosal melanoma ongoing dana farber cancer institute prior trials shown kinase inhibitors imatinib little activity unselected melanoma patients 11 14 therefore current trial requires kit gene mutation documented tumor patient treated first patient treated trial woman recurrent rectal melanoma containing kit exon 11 insertion duplication showed partial response 400 mg human cancer biology authors affiliations 1 oregon cancer institute 2 department pathology 3 division hematology oncology oregon health science university 4 vamedical center portland oregon 5 department medical oncology danafarber cancer institute boston massachusetts 6 department pathology cleveland clinic cleveland ohio 7 department pathology laboratory medicine emory university atlanta georgia 8 department pathology va medical center palo alto california received 3 5 08 revised 5 23 08 accepted 6 15 08 costs publication article defrayed part payment page charges article must therefore hereby marked advertisement accordance 18 u c section 1734 solely indicate fact requests reprints christopher l corless oregon health science university department pathology mailcode l471 3181 southwest sam jackson park road portland 97239 phone 503 494 6834 fax 503 494 6787 e mail corlessc ohsu edu f 2008 american association cancer research doi 10 1158 1078 0432 ccr 08 0575 www aacrjournals org 6821 clin cancer res 2008 14 21 november 1 2008 research downloaded clincancerres aacrjournals org may 28 2017 2008 american association cancer imatinib 4 weeks therapy 15 positron emission tomography scan confirmed marked reduction tumor metabolic activity consistent kit inhibition excellent response patient kit mutant melanoma suggests mutation screening may critical identifying patients may benefit kinase targeted therapy limited data kit alterations across major subtypes malignant melanoma surveyed large collection cases comparison examined braf nras well results confirm significant fraction acral mucosal melanomas harbor kit gene mutations increased kit copy number also observed differences two melanoma subtypes respect frequency detectable braf nras mutations consistent previous reports 3 4 16 22 observed nras mutations subtypes whereas braf mutations detected acral mucosal melanomas series mutations kit braf nras mutually exclusive indicating genotype may represent distinct molecular disease subtype observation clear implications use targeted therapeutics future materials methods melanoma samples blocks formalin fixed paraffin embedded melanoma tissue obtained pathology department archives oregon health science university university washington emory university accordance institutional review board approved protocols institution braf nras mutation status previously reported 44 cutaneous melanomas 62 choroidal melanomas cases analyzed kit 23 melanomas adult patients ranged ages 22 86 years tumors analyzed primary lesions recurrences metastases lymph node small bowel liver studied cases primary unavailable acral mucosal melanomas patients enrolled 06 056 phase ii trial danafarber cancer institute included study kit mutation one cases detailed prior case report 15 dna preparation tumor rich areas 80 scraped 5 unstained sections comparison h e stained slide genomic dna extracted using qiagen mini kit qiagen inc accordance manufacturer directions melanin pigment copurify genomic dna inhibit pcr reactions incubated samples obvious melanin contamination failed amplify initial testing chelex 100 bio rad specifically purified dna incubated 10 min room temperature equal volume 5 chelex 100 suspension equilibrated qiagen ae buffer mixture heated 95jc 2 min allowed cool chelex 100 resin pelleted microfuge supernatant dna used pcr reactions approximately 70 dna samples initially failed amplify recovered approach mutation screening screening mutations kit exons 8 9 11 13 17 braf exons 11 15 done using combination pcr amplification denaturing high performance liquid chromatography described previously 23 24 suspected mutations confirmed bidirectional sequencing abi 3130 automated sequencer mutations nras exons 1 2 identified melting curve analysis pcr amplicons measuring fluorescence energy transfer anchor probe wild type reporter probe covered codons interest pcr primers probes prepared integrated dna technologies 9 primers probes exon 2 assay exactly described nikiforova et al 25 primers probes exon 1 assay follows nras 1 forward cactagggttttcatttccattg nras 1 reverse tcctttaatacagaatatgggtaaaga nras 1 anchor tgaaatgactgagtacaaactggtggtggttfluorescein nras 1 reporter lc640 agcaggtggtgttggga c3 blocker twenty five microliter reactions carried roche lc480 lightcycler roche using roche probe master mixand including 80 nmol l forward primer 800 nmol l reverse primer 800 nmol l anchor reporter probes pcr conditions assays follows 95jc 10 min followed 50 cycles 95jc 20 60jc 20 72jc 20 95jc 1 min 45jc 2 min melting curve analyses done instrument based dilution analyses sensitivity nras exon 1 exon 2 assays f15 mutant allele melanoma samples showing aberrant peaks melting curve analysis sequenced bidirectionally confirm presence mutation reverse transcription pcr assay kit copy number kit gene copy number assessed quantitative real time pcr reactions comparison coamplified gapdh primers probes purchased integrated dna technologies kit exon 17 primers follows forward aaagatttgtgattttggtctagc reverse gaaactaaaaatcctttgca kit exon 17 dual labeled hydrolysis probe fam tgtggttaaaggaaacgtgagta iablkfq glyceraldehyde 3 phosphate dehydrogenase gapd primers follows forward cactaggcgctcactgttct reverse gcgaactcacccgttg gapdh hydrolysis probe txred tggggaaggtgaaggtcgga iabrqsp kit exon 7 primers follows forward aggtagaaactgaaaaagacatgc reverse gccttttaatctcgttagatgaagt kit exon 7 dual labeled hydrolysis probe fam aggcatgctatccacaggtgattg iablkfq pcr reactions done duplicate roche lc480 lightcycler 20 al total volume 50 ng genomic dna 200 nmol l primers 9 http www idtdna com translational relevance recent studies revealed presence kit gene alterations cases malignant melanoma alterations activating mutations increased copy number may particularly common melanomas arising acral mucosal sites treatment options types melanomas limited oncogenic mutations kit good targets clinically available tyrosine kinase inhibitors recently observed dramatic response imatinib treatment patient kitmutant rectal melanoma led us survey type frequency kit alterations 189 melanomas representing major subtypes confirmed kit mutations common among acral 23 mucosal 15 6 melanomas among cutaneous 1 7 conjunctival 7 7 tumors increased kit copy number also common acral 27 mucosal 26 melanomas subtypes interestingly tumors kit gene alterations immunopositive cd117 findings suggest routine screening acral mucosal melanomas kit alterations may open new treatment avenues patients tumors human cancer biology clin cancer res 2008 14 21 november 1 2008 6822 www aacrjournals org research downloaded clincancerres aacrjournals org may 28 2017 2008 american association cancer probes 1 probes master mix roche pcr conditions 1 cycle 95jc 10 min 50 cycles 95jc 10 60jc 20 second derivative maximum algorithm included instrument software used determine cycle threshold ct values target gene relative copy numbers calculated using ddct method ct threshold cycle amplification sample dct kit versus gapdh calculated dct ct kit ct gapdh dct value experimental test sample calibrated reference pool genomic dna prepared 9 samples formalinfixed paraffin embedded normal tissue included colon kidney liver prostate spleen archived 4 less using formula ddct dct test sample dct reference pool relative dna copy number calculated using formula 2 ddct relative copy numbers converted absolute copy numbers assigning value 2 diploid reference pool multiplying relative copy number test samples factor 2 pilot reactions established kit cts 26 34 cycles linear range amplification samples amplify 34 cycles deemed insufficient quality increased kit copy number determined comparison nine individual formalin fixed paraffin embedded normal tissue samples mean sds 9 samples based quadruplicate measurements 2 25 f 0 68 kit copies relative gapdh threshold increased kit copy number set relative normal samples using 95 confidence level according chebychev inequality formula mean f 4 47 sd threshold 5 29 copies kit relative gapdh fluorescence situ hybridization kit validate results reverse transcription pcr kit copy number fluorescence situ hybridization fish done 5 sections paraffin embedded melanoma tissue using cocktail three probes chromosome 4 vysis lsi 4q12 tricolor abbott laboratories probes cover genes scfd2 spectrum green lnx spectrum orange kit spectrum aqua also includes portion neighboring gene pdgfra protocol previously described 26 briefly slides baked overnight 60jc deparaffinized dehydrated pretreatment washes included distilled water 10 mi 2 ssc 3 min 10 mmol l citric acid buffer ph 6 0 80jc 45 min slides protease treated 37jc 10 min dehydrated air dried addition probe slides denatured 5 min 73jc hybridized 14 18 h 37jc washing slides counterstained 4 6 diamidino 2 phenylindole visualized fluorescence microscope cd117 immunohistochemistry deparaffinized 5 sections tumor heated dako target retrieval solution ph 6 0 dako usa 10 min russell hobbs pressure cooker model rhnpc800p allowed cool immunostaining cd117 kit done using dako polyclonal rabbit antibody dako a4502 1 400 dilution followed standard avidin biotin detection protocol using diaminobenzidine hematoxylin counterstained slides cover slipped examined intensity staining 0 1 2 results kit mutations among 189 melanoma cases fully screened mutations kit exons 11 13 17 frequency kit mutations 23 acral tumors 15 6 mucosal tumors 7 7 conjunctival tumors 1 7 cutaneous tumors 0 choroidal tumors table 1 45 mucosal tumors sinonasal n 29 7 arose oral cavity 7 anorectal area 2 vulva vagina interestingly mutation frequency higher among tumors anorectum vulva vagina 4 9 44 4 among tumors arising head neck 3 36 8 3 table 1 however cases need collected analyzed see trend significant point mutations common resulting substitutions exon 11 w557r k558n v559a v559d l576p 4 cases exon 17 y823d table 2 however one acral melanoma frame exon 11 deletion evqwkv 554 559 previously observed exon 11 insertion duplication rectal melanoma 15 kit mutations identified cases also occur gists kit exon 9 another site mutation gists f10 tumors screened 148 melanomas including 6 acral 27 mucosal tumors mutations exon none found also examined kit exon 8 mutations occur acute myelogenous leukemias observe 99 melanomas including 3 acral 25 mucosal melanomas kit copy number kit gene copy number chromosome 4 assessed quantitative real time pcr exon 17 sequence compared coamplified gapdh chromosome 12 internal control pilot assays comparable results obtained using h actin actb chromosome 7 internal control data shown however gapdh selected actb chromosome commonly overrepresented melanoma 3 27 28 differences amplification thresholds ct kit gapdh compared reference pool normal genomic dna prepared 9 samples formalin fixed paraffinembedded benign tissue assay validated several approaches substitution kit exon 7 amplicon exon 17 assay significant effect data shown b kit mutant melanoma cell line known increased copies kit array cgh showed elevated copy number pcr assay 10 c fish kit done five melanomas study fish signals versus reverse transcription pcr copy number five cases follows 2 versus 2 6 3 versus 5 5 5 7 versus 5 0 7 8 versus 7 8 numerous count versus 28 8 figure 1 illustrates fish results two cases increased kit copy number identified reverse transcription pcr one quarter acral mucosal melanomas table 3 2 3 acral tumors 7 10 mucosal tumors accompanying kit mutation detected extra copies kit much less common among cutaneous conjunctival tumors observed choroidal melanomas tested braf mutations previously reported braf mutations 36 4 16 44 cutaneous melanomas 0 0 62 uveal melanomas 23 study braf mutations identified 26 7 conjunctival melanomas 16 7 acral melanomas mucosal melanomas table 1 common v600e substitution exon 15 exception single conjunctival tumor v600r contrast braf exon 15 mutations among 47 mucosal melanomas regardless site twenty sixmucosal cases additionally screened mutations braf exon 11 none found nras mutations previous study observed mutations nras exons 1 2 uncommon cutaneous melanomas 3 7 1 27 absent choroidal 10 f hodi manuscript submitted kit mutation copy number melanoma subtypes www aacrjournals org 6823 clin cancer res 2008 14 21 november 1 2008 research downloaded clincancerres aacrjournals org may 28 2017 2008 american association cancer melanomas 0 47 ref 23 expanded series 60 cutaneous melanomas yielded slightly higher frequency 11 7 table 1 however nras mutations considerably common 24 7 among 37 mucosal melanomas analyzed expected literature affected codons 12 g12c g12d g12v 13 g13r 61 q61k q61l q61r one interesting trend exon 1 mutations dominated among mucosal melanomas 8 9 cases whereas exon 2 mutations common cutaneous tumors 6 7 cases one nras mutation among 9 acral tumors analyzed none 11 conjunctival melanomas correlation kit alterations cd117 expression immumunohistochemistry cd117 kit done subsets melanoma subtypes including examples kit braf nras mutation cases increased kit copy number staining graded negative 0 weak moderate 1 strong 2 results tabulated table 4 showed apparent correlation cd117 kit mutation status increased kit copy number occasional tumors braf nras mutation readily detectable cd117 strong expression also observed many wild type tumors 21 4 particularly among choroidal melanomas 35 7 thus cd117 expression reliable indicator kit genotype discussion went colleagues 7 first discover activating kit mutation l576p case melanoma one two cd117 positive tumors sequenced willmorepayne et al 9 29 followed observation two studies using high resolution melting curve analysis screen melanomas mutations kit exons 9 11 13 17 153 melanomas nonmucosal otherwise unspecified origin described two studies however mutation screening actually limited 35 cases cd117 positive among three cases l576p substitution 2006 curtin coworkers 4 surveyed 95 melanomas various types found kit mutations table 1 summary mutations melanoma subtypes gene acral mucosal study studies study studies kit 23 3 13 11 3 28 ref 4 15 6 7 45 15 3 15 ref 6 21 8 38 ref 4 22 2 4 18 ref 30 braf 16 7 2 12 9 5 2 21 ref 42 0 0 45 0 0 13 ref 17 15 2 13 ref 18 3 1 38 ref 4 15 4 2 39 ref 19 5 6 2 36 ref 22 21 6 28 ref 4 5 9 1 17 ref 16 23 7 30 ref 3 9 5 2 21 ref 19 33 5 15 ref 20 11 2 19 ref 3 nras 11 1 1 9 3 6 1 28 ref 21 24 3 9 37 5 1 19 ref 3 10 3 30 ref 3 13 9 5 36 ref 22 47 4 9 19 ref 43 melanomas arising chronically sun damaged skin cpreselected cd117 positivity sites specified table 2 summary kit mutations melanoma exon mutation study studies total imatinib sensitive refs 11 del 554 5591 1 yes y553n 1 1 4 w557r 1 1 2 yes 30 k558n 1 1 yes v559a 1 2 3 yes 4 v559d 1 1 yes n566d 1 k642e 1 4 v569g 1 1 30 l576p 5 7 12 yes 4 7 9 29 ins 583 1 1 yes 15 13 r634w 1 1 k642e 9 1 n566d 9 yes 4 6 30 17 d816h 2 2 y823d 1 1 18 a829p 1 1 4 mutations 11 27 38 human cancer biology clin cancer res 2008 14 21 november 1 2008 6824 www aacrjournals org research downloaded clincancerres aacrjournals org may 28 2017 2008 american association cancer common among acral 11 3 28 mucosal 21 8 38 tumors subsequent studies antonescu et al 6 rivera et al 30 confirmed kit mutations common among rectal head neck melanomas respectively table 1 study used denaturing high performance liquid chromatography screen total 189 melanomas sites mutations three exons kit previously reported harbor mutations exons 11 13 17 included 45 mucosal melanomas 36 head neck 9 anorectal vulvar vaginal tumors results summarized table 1 build substantially earlier reports combining data previous studies kit mutations present 14 6 acral 19 mucosal 4 3 cutaneous 7 7 conjunctival melanomas absent choroidal melanomas subset tumors also screened mutations kit exon 8 exon 9 table 1 summary mutations melanoma subtypes cont conjunctival cutaneous choroidal study studies study studies study studies 7 7 1 13 reports 1 7 1 58 0 0 50 ref 31 0 0 60 0 0 10 ref 41 16 7 3 18 ref 4 16 7 1 6 ref 29 c 6 9 2 29 ref 9 c 26 7 4 15 14 3 3 21 ref 43 22 7 5 22 ref 44 40 2 5 ref 45 0 0 11 reports fig 1 fish kit acral melanoma kit v559d showing increased fish signals fused probes 4q12 signals ranged 6 8 per nucleus copy number 7 8 reverse transcription pcr b sinonasal melanoma wild type kit showed large increase kit signals nearby loci lnx orange signal scfd2 green signal suggesting localized amplification one nuclei one kit allele showed amplification whereas neighboring nucleus displayed two clusters kit signals reverse transcription pcr case highest kit copy number 28 8 study kit mutation copy number melanoma subtypes www aacrjournals org 6825 clin cancer res 2008 14 21 november 1 2008 research downloaded clincancerres aacrjournals org may 28 2017 2008 american association cancer mutations found course remains possible mutations exons kit gene observed braf mutations relatively common kit mutations conjunctival tumors although number tumors available study limited frequency braf mutations observed similar studies parallels well established dominance braf mutations cutaneous melanomas 1 3 among cutaneous melanomas series one kit mutation woenckhaus et al 31 reported none among 50 cutaneous tumors hand curtin et al 4 observed kit mutations frequency 16 7 among melanomas arising chronically sun damaged skin braf mutations comparatively uncommon 3 although relationship sun damage likelihood kit mutation cutaneous melanomas deserves investigation cutaneous tumors study selected basis rather simply representative cases referred tertiary care center including patients advanced disease targeting kit proven strategy management advanced gist 32 34 several observations suggest kit kinase inhibitors might also serve control growth selected melanomas first kit kinase activity necessary development melanocytes 10 development interstitial cells cajal gists thought related gist oncogenic activation kit may serve early signal neoplastic growth second antonescu colleagues 6 observed kit phosphorylation immunoblot rectal melanoma extract third great majority kit mutations observed melanoma type known sensitive imatinib kit inhibitors table 3 example antonescu et al 6 showed v559d l576p mutant forms kit inhibited imatinib dasatinib expressed ba f3 cells fourth melanoma cell line harboring kit exon 11 mutation highly sensitive growth inhibition imatinib whereas nonmutant cell lines 11 finally recently reported dramatic response imatinib therapy mucosal melanoma patient kit exon 11 insertion duplication 15 note current study found tumor case also increased kit copy number 12 6 imatinib essentially ineffective trials unselected melanoma patients 11 13 14 35 likewise another kit inhibitor pkc412 showed little activity melanoma 12 however becker et al 36 observed recent editorial imatinib trial underpowered detect activity subtypes melanomas likely harbor kit mutation also notable imatinib trial reported eton et al 35 acral melanoma patient nearly complete response noted statistical chance kit mutation patient tumor 14 6 thus responding patient reported may unique routine screening kit may become important patients advanced melanoma kit mutations much less common melanoma gist question arises would effective screening approach logic kit highly expressed kit mutant tumors willmore payne colleagues 9 29 focused screening cd117 positive cases correspondingly antonescu et al 6 observed three kit mutant melanomas 4 positive cd117 whereas nearly tumors series either 1 negative however evidence gist literature cd117 expression kit genotype always correlate example kit mutations detected 16 cd117 negative gists 37 yet tumors still responsive imatinib chirieac coworkers 38 confirmed relationship cd117 expression imatinib response among gist tumors perhaps surprising several kit mutant melanomas lacked detectable expression cd117 table 4 using identical antibody true 1 4 kit mutant melanomas studied rivera et al 30 several kitmutant melanomas examined curtin colleagues 4 latter group indicated cases became cd117 positive 10 fold higher concentration antibody used false positive staining high concentrations antibody well documented literature 7 39 like rivera et al curtin et al observed strong cd117 staining melanomas kit wild type table 4 also observed one case 11f hodi et al manuscript submitted table 3 summary kit gene copy number increased kit number increased kit number kit mutation increased kit number without kit mutation acral 27 3 3 11 1 2 mucosal 26 3 10 38 3 7 cutaneous 6 7 3 45 1 2 conjunctival 7 1 1 14 0 1 choroidal 0 0 28 0 0 table 4 genotype versus cd117 expression nonchoroidal melanomas genotype cd117 staining 0 1 2 total kit mutation 3 1 2 6 increased kit copy number 6 2 1 9 normal kit copy number 40 16 14 70 braf mutation 10 2 1 13 nras mutation 5 2 0 7 mutation nonchoroidal 18 4 6 28 choroidal 18 915 42 human cancer biology clin cancer res 2008 14 21 november 1 2008 6826 www aacrjournals org research downloaded clincancerres aacrjournals org may 28 2017 2008 american association cancer braf mutation conclude cd117 immunohistochemistry reliable indicator kit genotype restricting mutation screening cd117 positive melanomas would risk missing mutant cases analyzing three kit mutant melanomas willmore payne et al 9 found one tumor modest increase kit copy number assessed fish 2 88 fold remaining tumors homozygous mutants based comparison fish dna sequence similarly antonescu coworkers 6 observed increased kit copy number 2 5 fold 1 2 mutant cases examined fish selective overrepresentation chromosome 4q12 includes kit measured array cgh subset cutaneous melanomas evaluated bastian et al 27 acral mucosal cutaneous melanomas chronically sun damaged skin studied curtin et al 4 simplify assessment kit copy number developed quantitative real time pcr assay use formalin fixed paraffin derived tumor dna assay advantage done material used screen kit mutations however may less sensitive approaches copy number reference gene gapdh may vary cases mind identified number melanomas increases kit one case 28 8 copies included cases kit mutations wild type tumors intriguing mucosal melanoma patient previously reported responding dramatically imatinib therapy kit mutation increased kit copy number usefulness assay examined ongoing trials kit kinase inhibitors imatinib sunitinib patients acral mucosal melanoma although significance increased kit copy number regard treatment melanoma remains determined great majority kit mutations identified cancer potentially sensitive imatinib kit inhibitors table 2 summarizes mutations reported date including current study two common mutations l576p k642e together account 55 cases inhibited imatinib vitro predict good imatinib response gist 32 34 mutations likely insensitive available inhibitors affecting activation loop encoded exons 17 18 fortunately 4 38 cases 10 5 reported thus far mutation domain table 2 alternative treatment strategy cases might mek1 inhibitor braf mutant melanoma cell lines highly sensitive mek1 inhibition 40 ras mutant fgfr1 mutant melanoma cell lines show similar sensitivities 28 molecular correlation studies ongoing trials braf mek1 inhibitors prove interesting regard summary conducted largest survey date kit genotypes across various melanoma subtypes findings confirm acral mucosal melanomas subtypes likely harbor kit mutations increased kit copy number however results also indicate oncogene may play role occasional cases conjunctival cutaneous melanoma mutations overlap mutations braf nras sometimes associated increased kit copy number also observed nras mutations even common mucosal melanomas kit mutations borne mind future efforts match patient tumors targeted therapeutics pathogenesis gastrointestinal stromal tumors gists results activating mutations kit platelet derived growth factor receptor pdgfra 80 gists express mutated constitutively active kit another 5 7 express mutated pdgfra 10 15 tumors associated mutations kinases 1 3 imatinib mesylate selective inhibitor kit pdgfra platelet derived growth factor receptor pdgfrb bcr abl kinase revolutionized treatment gist however 14 gists exhibit primary resistance imatinib defined progression within 3 6 months initiating therapy 4 6 another 40 50 develop resistance within 2 years beginning therapy ie secondary resistance 5 6 sunitinib malate sutent pfizer new york ny another small molecule tyrosine kinase inhibitor tki selectivity kit pdgfra pdgfrb three isoforms vascular endothelial growth factor receptor vegfr fms like tyrosine kinase 3 flt3 colony stimulating factor 1 receptor csf 1r glial cell line derived neurotrophic factor receptor rearranged transfection ret pfizer new york ny data file 7 11 demonstrated clinical benefit phase phase iii trials patients imatinib resistant intolerant gist 12 13 sunitinib approved multinationally treatment patients gist prior imatinib therapy failed disease progression drug intolerance gist responsiveness imatinib varies primary kit genotype exon 11 mutant gists sensitive exon 9 mutant wild type gists ie lack kit pdgfra mutations 3 14 15 exons 11 9 common sites kit mutation gist approximately 70 15 tumors respectively 3 14 secondary kinase mutations common gists exhibit secondary resistance exhibit primary resistance 16 17 secondary point mutations associated imatinib resistance usually located drug adenosine triphosphate atp binding pocket receptor encoded exons 13 14 activation loop encoded exon 17 16 28 two recent studies used cell based assays reported sunitinib inhibited kinase activity kit receptors contained mutations drug atp binding pocket confer resistance imatinib 29 30 mutations ie t670i v654a substitutions isoleucine threonine position 670 alanine valine position 654 respectively commonly found patients gist secondary imatinib resistance results provide possible basis sunitinib antitumor activity patients imatinib refractory gist explore relationship primary secondary gist kinase mutations response sunitinib determined primary secondary kit pdgfra mutations biopsied tissue patients imatinib refractory gist received sunitinib part phase ii trial 12 correlated presence mutations clinical benefit addition vitro studies assessed sensitivity kit pdgfra mutants sunitinib imatinib directly patients methods biopsies genotype analyses obtained patients enrolled sunitinib phase ii trial described earlier report efficacy safety results study 12 patients adults histologically confirmed metastatic unresectable gist documented failure imatinib caused resistance intolerance patients 55 97 received sunitinib 50 mg 6 week cycles comprised 4 weeks followed 2 weeks treatment additional information methods listed appendix online results primary tumor genotype efficacy tissue pre imatinib genotype analysis available 78 97 patients trial patients overall bulky metastatic disease received median 78 weeks prior imatinib therapy table 1 primary kit mutations identified 83 tumors whereas 5 pdgfra mutations 12 contained wild type kit pdgfra appendix table a1 online kit mutations 69 located exon 11 exon 9 30 kit mutations exon 13 2 kit mutations pdgfra mutations located exon 12 one patient tumor exon 18 tumors three patients clinical benefit partial response pr stable disease sd 6 months observed three common gist genotypes table 2 clinical benefit rate 58 tumors primary kit exon 9 mutations 34 exon 11 mutations 56 wild type kit pdgfra imatinib therapy objective responses ie prs significantly common patients kit exon 9 exon 11 mutant gists 37 v 5 p 002 four patients pdgfra mutations none experienced clinical benefit among patients classified imatinib intolerant n 4 tumor genotyping revealed primary kit exon 9 mutation one achieved pr wild type genotype three patients achieved sd two 6 months median progression free survival pfs significantly longer patients primary kit exon 9 mutations 19 4 months 95 ci 11 1 yet attained na p 0005 wild type genotype 19 0 months 95 ci 3 9 na p 0356 kit exon 11 mutations 5 1 months 95 ci 4 5 7 8 fig 1a pfs differ significantly patients exon 9 mutations wild type genotype median overall survival os also significantly longer patients exon 9 mutations 26 9 months 95 ci 12 2 na p 012 wild type genotype 30 5 months 95 ci 19 8 na p 0132 exon 11 mutations 12 3 months 95 ci 8 8 19 6 fig 1b os differ significantly patients exon 9 mutations wild type genotype secondary tumor genotype efficacy total 109 post imatinib biopsy specimens available 67 patients secondary kit mutations identified 33 patients appendix table a1 consistent prior reports mutation distribution nonrandom clusters occurred exons 13 14 encode drug atp binding pocket receptor exon 17 encodes kinase activation loop fig 2a commonly identified secondary mutation v654a exon 13 two tumors secondary kit exon 18 mutations one patient different secondary mutations exon 13 v654a exon 17 d816h different lesions secondary kinase mutations significantly common gists primary kit exon 11 mutations exon 9 mutations 73 v 19 p 0003 four samples primary pdgfra mutations one secondary mutation exon 18 primary mutation exon 12 two lacked secondary mutations primary exon 18 d842v mutations fourth lacked post imatinib sample secondary mutations found eight post imatinib samples lacked primary kit pdgfra mutations among patients kit mutations median pfs sunitinib significantly longer 18 patients secondary kit exon 13 14 mutations 7 8 months 95 ci 4 5 10 1 13 patients exon 17 18 mutations 2 3 months 95 ci 1 0 5 1 p 0157 fig 2b likewise median os numerically longer former latter group 13 0 months 95 ci 8 9 22 4 v 4 0 months 95 ci 2 2 19 6 p 160 fig 2c clinical benefit rates higher 61 v 15 p 011 table 2 nearly identical results obtained patients primary exon 11 mutations considered patients primary exon 11 mutations significant differences pfs os patients without secondary mutations vitro measures activity specific mutants sunitinib potently inhibited activity ligand activated wild type kit kit exon 11 v560d exon 9 ay insertion mutants 50 inhibitory concentration ic50 values less 100 nmol l three kinases table 3 fig 3a comparison corresponding ic50 values imatinib approximately 1 000 nmol l wild type kit 100 nmol l v560d mutant 1 000 nmol l exon 9 ay mutant sunitinib also potently inhibited phosphorylation kit double mutants second mutation occurred drug atp binding site receptor v560d v654a exons 11 13 v560d t670i exons 11 14 double mutants resistant inhibition imatinib vitro conversely kit double mutants second mutation occurred activation loop v560d d816h v560d d820g v560d n822k v560d y823d resistant inhibition sunitinib imatinib sunitinib ic50 values 1 000 nmol l higher notably v560d a829p double mutant imatinib ic50 two three fold higher v560d alone contrast v560d a829p resistant sunitinib doses 1 000 nmol l rarity a829p secondary mutation could caused relatively preserved imatinib sensitivity similar results obtained exon 11 v560d used primary mutation obtained exon 9 ay insertion used instead table 3 fig 3a confirm findings tested relative potency imatinib sunitinib inhibiting kit kinase activity gist cell lines obtained imatinib resistant tumors fig 3b gist48 cell line homozygous primary kit exon 11 v560d mutation heterozygous secondary exon 17 d820a mutation 17 concentrations imatinib greater 1 000 nmol l insufficient completely inhibit kit activation cell line concentration 10 fold higher necessary block kit exon 11 mutant isoforms gist cell lines studies 24 31 32 sunitinib less potent imatinib inhibiting kit autophosphorylation gist48 cells notably low doses 100 nmol l either imatinib sunitinib partial inhibitory effect kit phosphorylation presumably inhibition minority population v560d homodimers gist430 cell line heterozygous kit exon 11 deletion mutation exon 13 v654a substitution allele 17 sunitinib significantly greater potency imatinib inhibition kit autophosphorylation gist430 cells ic50 1 000 nmol l imatinib v 100 nmol l sunitinib also tested potency sunitinib inhibiting phosphorylation wild type pdgfra v561d point mutant ic50 values less 100 nmol l table 3 fig 3c v561d located receptor juxtamembrane domain encoded exon 12 relatively common primary pdgfra mutation patients gist 1 conversely d842v common pdgfra mutation gists resides activation loop encoded exon 18 confers imatinib resistance primary secondary mutation 1 conferred resistance sunitinib vitro experiments table 3 fig 3c clinical study d842v detected primary mutation two patients secondary mutation one patient discussion results extend previously reported findings study showed correlation sunitinib activity gist kinase genotype patients metastatic unresectable gist experienced imatinib failure 33 data relative responsiveness different molecular subgroups imatinib resistant gist may help optimize treatment patients gist may help better understand basis sunitinib activity patients studies may also advance understanding mechanisms resistance may facilitate development strategies circumvent analyses reported assessed effect tumor kinase genotype sunitinib activity using clinical study data complemented vitro cellular assays although sunitinib demonstrated clinical activity gists three common primary genotypes datasets indicated primary secondary mutations pathogenic kinase strongly influence sunitinib activity clinical benefit objective response rates sunitinib higher patients primary kit exon 9 mutations exon 11 mutations clinical benefit rates 58 v 34 objective response rates 37 v 5 p 002 similarly pfs os significantly longer patients primary kit exon 9 mutations wild type genotype kit exon 11 mutations results converse reported imatinib objective response rates higher pfs os longer patients gist harbored exon 11 mutations exon 9 mutations wild type genotype 3 14 15 notably potency sunitinib wild type exon 9 mutant kit superior imatinib vitro whereas drugs exhibited similar potency kit exon 11 mutant kinases possible explanation mutational sites different structural effects kit different consequences interaction two tkis indeed exon 9 mutations recently reported structural consequences similar ligand mediated receptor dimerization 34 mechanism kinase activation appears distinct caused mutation intracellular juxtamembrane domain encoded exon 11 35 others also observed impact mutational site tki potency vitro using isogenic baf3 model imatinib ic50 cells expressed exon 9 mutations found approximately eight fold higher obtained cells expressed exon 11 v559d mutation 36 results suggest greater clinical benefit seen sunitinib treated patients exon 9 mutant wild type imatinib resistant gists may related greater potency sunitinib kinases also suggest genotypically defined subsets patients may experience different clinical outcomes treated first line imatinib sunitinib sunitinib currently approved second line therapy gist studies planned evaluate efficacy safety first line treatment hand sunitinib yet tested imatinib na patients majority patients study primary kit exon 11 mutations acquired secondary kit mutations confer imatinib resistance studies imatinib na patients required definitely assess effect primary exon 11 mutation alone sunitinib activity vivo study also showed secondary kinase mutations significantly common gists primary kit exon 11 exon 9 mutations occur gists wild type genotype consistent previous reports secondary kinase mutations common gists exhibit secondary imatinib resistance exhibit primary resistance 16 17 moreover frequency secondary mutations likely underestimated analysis one patient analysis found different secondary mutations different lesions intra interlesion heterogeneity secondary mutations gists documented others 20 25 limited number small needle biopsy specimens available per patient study mean 1 4 biopsy specimens per patient range 0 3 particular probable sampling would revealed secondary mutations tumors primary kit exon 11 mutations appeared lack exon 11 mutants strongly inhibited imatinib secondary resistance likely require selection subsequent expansion clones expressing second resistance conferring mutation gists exon 9 mutations wild type genotype likely intrinsically resistant imatinib consistent median duration prior imatinib treatment patients primary exon 11 mutations 22 8 months compared 12 5 10 5 months patients exon 9 mutations wild type genotype respectively table 2 however worth noting although duration imatinib treatment significant prognostic factor pfs os univariate analysis significant factor multivariate analysis data shown although multivariate analyses performed small sample must interpreted caution confirmed primary secondary kit genotype significant prognostic factors pfs marginally significant prognostic factors os consistent previous studies 16 18 28 secondary kit mutations patients imatinib resistant gist enrolled current study tended cluster exons 13 14 encode drug atp binding pocket receptor exon 17 encodes kinase activation loop note vitro studies showed sunitinib potently inhibited kinase activity kit receptors contained secondary mutations drug atp binding pocket resistant imatinib v654a exon 13 t670i exon 14 secondary mutations coexpressed common primary mutation v560d recreated situation often observed gists exhibit secondary imatinib resistance previous ex vivo studies also shown sunitinib inhibits imatinib resistant kit receptors contain mutations drug atp binding pocket 29 30 however vitro studies performed also showed sunitinib relatively ineffective inhibiting kit receptors contained secondary mutations localized activation loop consistent vitro findings pfs os longer clinical benefit rate higher patients clinical trial secondary kit exon 13 14 ie atp binding pocket mutations secondary kit exon 17 18 ie activation loop mutations results study provide one explanation activity sunitinib patients imatinib refractory gist seen trials 13 however antiangiogenic effects sunitinib treatment also may contribute effectiveness addition kit pdgfra activity sunitinib also selectively inhibits pdgfrb three isotypes vegfr whereas imatinib inhibits pdgfrb vegfrs studies animal models indicate dual inhibition pdgfr vegfr produces greater antiangiogenic effects inhibition one 37 39 suggests sunitinib may produce greater antiangiogenic effects imatinib effects may contribute activity imatinib refractory gists note observation secondary kit mutants involve activation loop insensitive sunitinib imatinib given different tumor clones one individual may acquire imatinib resistance different secondary mutations including involving kit activation loop 20 25 imatinib resistant tumors may respond well sunitinib therapy conversely gists secondary kit activation loop mutations may still susceptible sunitinib potent antiangiogenic effects additional research issue warranted authors disclosures potential conflicts interest although authors completed disclosure declaration following author indicated financial interest relevant subject matter consideration article certain relationships marked u compensation received relationships marked c compensated detailed description disclosure categories information asco conflict interest policy please refer author disclosure declaration disclosures potential conflicts interest section information contributors employment leadership position xin huang pfizer inc c darrel p cohen pfizer inc c charles baum pfizer inc c consultant advisory role michael c heinrich novartis c pfizer inc u christopher l corless pfizer inc c george demetri novartis c pfizer inc c infinity c stock ownership michael c heinrich molecular md xin huang pfizer inc darrel p cohen pfizer inc charles baum pfizer inc honoraria michael c heinrich novartis robert g maki pfizer inc christopher l corless pfizer inc george demetri novartis pfizer inc research funding michael c heinrich novartis pfizer inc robert g maki pfizer inc george demetri novartis pfizer inc infinity expert testimony george demetri novartis u pfizer inc u infinity u remuneration none author contributions conception design michael c heinrich wen bin ou jonathan fletcher darrel p cohen george demetri financial support michael c heinrich jonathan fletcher charles baum george demetri administrative support michael c heinrich diana griffith george demetri provision study materials patients michael c heinrich robert g maki jonathan fletcher george demetri collection assembly data michael c heinrich robert g maki cristina r antonescu amy harlow diana griffith ajia town arin mckinley wen bin ou jonathan fletcher christopher fletcher xin huang charles baum george demetri data analysis interpretation michael c heinrich christopher l corless cristina r antonescu amy harlow diana griffith ajia town arin mckinley wen bin ou jonathan fletcher xin huang charles baum george demetri manuscript writing michael c heinrich christopher l corless jonathan fletcher darrel p cohen charles baum george demetri final approval manuscript michael c heinrich robert g maki christopher l corless cristina r antonescu amy harlow diana griffith ajia town arin mckinley wen bin ou jonathan fletcher christopher fletcher xin huang darrel p cohen charles baum george demetri appendix methods mutational analysis archival tumor biopsy specimens collected initiation imatinib therapy obtained written patient consent additional specimens collected days 1 11 current study sunitinib dosing genomic dna extracted formalin fixed paraffin embedded specimens amplified using polymerase chain reaction analyzed kit platelet derived growth factor receptor pdgfra mutations previously described heinrich mc corless cl duensing et al science 299 708 710 2003 corless cl mcgreevey l haley et al j pathol 160 1567 1572 2002 rader ae avery wait cl et al cancer 93 269 275 2001 choy ys dabora sl hall f et al ann hum genet 63 383 391 1999 using primer pairs denaturing high performance liquid chromatography conditions reported elsewhere heinrich mc corless cl blanke cd et al j clin oncol 24 4764 4774 2006 methods tumor responses tumor responses assessed baseline end every cycle basis response evaluation criteria solid tumors therasse p arbuck sg eisenhauer ea et al j natl cancer inst 92 205 216 2000 clinical benefit defined partial response stable disease lasted 6 months additional efficacy measures included progression free survival overall survival current analyses include follow march 2006 methods sensitivity kinase mutants tyrosine kinase inhibitors vitro chinese hamster ovary cells transiently transfected mutated kit pdgfra cdna constructs treated various concentrations sunitinib imatinib previously described heinrich mc corless cl demetri gd et al j clin oncol 21 4342 4349 2003 heinrich mc corless cl blanke cd et al j clin oncol 24 4764 4774 2006 common exon 11 mutation v560d selected prototypic primary kit mutation several representative exon 13 14 17 mutations selected secondary mutations constructs experiments involved recombinant dna performed biosafety level 2 conditions accordance published guidelines national institutes health available http www4 od nih gov oba rac guidelines guidelines html gastrointestinal stromal tumor gist48 gist430 cell lines established imatinib resistant gists previously described heinrich mc corless cl blanke cd et al j clin oncol 24 4764 4774 2006 protein lysates transfected chinese hamster ovary cells gist cell lines prepared subjected immunoprecipitation using anti kit anti pdgfra antibodies followed sequential immunoblotting phospho kit total kit phosphotyrosine total pdgfra respectively previously reported heinrich mc corless cl duensing et al science 299 708 710 2003 heinrich mc corless cl demetri gd et al j clin oncol 21 4342 4349 2003 methods statistical analyses descriptive statistics used summarize clinical responses mutation status two sided fisher exact test used evaluate differences response tumor genotype kaplan meier methods used assess progression free overall survival kit genotype multivariate analyses performed cox proportional hazard models gastrointestinal stromal tumors gists commonly harbor oncogenic mutations kit platelet derived growth factor alpha pdgfra kinases targets imatinib clinical studies 75 90 patients advanced gists experience clinical benefit imatinib however imatinib resistance increasing clinical problem patients methods one hundred forty seven patients advanced unresectable gists enrolled onto randomized phase ii clinical study imatinib specimens pretreatment imatinibresistant tumors analyzed identify molecular correlates imatinib resistance secondary kinase mutations kit pdgfra identified imatinib resistant gists biochemically profiled imatinib sensitivity results molecular studies performed using specimens 10 patients primary 33 patients secondary resistance imatinib resistant tumors levels activated kit similar greater typically found untreated gists secondary kinase mutations rare gists primary resistance frequently found gists secondary resistance 10 v 67 p 002 evidence clonal evolution polyclonal secondary kinase mutations seen three 18 8 16 patients secondary kinase mutations nonrandomly distributed associated decreased imatinib sensitivity compared typical kit exon 11 mutations using rnai technology demonstrated imatinib resistant gist cells remain dependent kit kinase activity activation critical downstream signaling pathways conclusion different molecular mechanisms responsible primary secondary imatinib resistance gists findings implications future approaches growing problem imatinib resistance patients advanced gists j clin oncol 24 4764 4774 introduction gastrointestinal stromal tumors gists common type sarcoma arising digestive tract generally distinguished abdominal sarcomas expression kit receptortyrosine kinase kinase important diagnostic markerfor gists serves primary oncogene approximately 80 tumors evidenced activating mutations kit gene imatinib gleevec novartis pharmaceuticals basel switzerland inhibits enzymatic activity several oncogenic tyrosine kinases including kit platelet derived growth factor alpha pdgfra clinical studies 75 90 patients advanced gists treated imatinib experienced clinical benefit stable disease complete partial objective response 1 3 imatinibinduced responses correlated tumor kinase mutational status 4 5 patients kit exon 11 mutant gist higher response rate significantly longer median survival compared patients exon 9 mutant gists whose gists lack kit pdgfra mutations 4 although imatinib revolutionized treatment advanced gists clinical resistance drug proved significant problem prolonged follow secondary kinase mutations acquired imatinib treatment reported several small series 6 15 study evaluated molecular division hematology oncology department pathology oregon health science university cancer institute oregon health science university portland veterans affairs medical center portland dana farber cancer institute harvard cancer center department pathology brigham women hospital boston division oncologic surgery hitchcock dartmouth medical center lebanon nh division medical oncology fox chase cancer center philadelphia pa novartis pharmaceuticals corporation hanover nj university helsinki helsinki department surgery university turku turku finland submitted february 17 2006 accepted june 23 2006 published online ahead print www jco org september 5 2006 supported part novartis pharmaceuticals va merit review grant c h gist cancer research fund c h b p lester regina john foundation c h terms blue defined glossary found end article online www jco org authors disclosures potential con flicts interest author contributions found end article address reprint requests michael c heinrich md r 19 3710 u veterans hospital rd portland 97239 e mail heinrich ohsu edu 0732 183x 06 2429 4764 20 00 doi 10 1200 jco 2006 06 2265 journal clinical oncology original report volume 24 number 29 october 10 2006 4764 downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved markers imatinib resistance series well documented gist samples phase ii study imatinib patients methods patient materials tumor samples obtained patients enrolled csti571 b2222 phase ii trial ofimatinib sponsored novartis basel switzerland treatment advanced gist 2 4 147 original patients 92 documented disease related treatment failure may 2005 samples obtained 43 progressive disease patients consented analysis tumors per clinical protocol tumor progression defined according traditional southwest oncology group response criteria 16 specifically progressive disease defined either 50 greater increase sum products measurable lesions smallest sum observed baseline decrease observed clear worsening previous examinations assessable disease reappearance lesion disappeared appearance new lesion failure return evaluation due death deteriorating clinical condition appearance nodule within mass follow imaging studies considered meet protocolspecified definition progression study approved local institutional reviewboardofeachparticipatinginstitution andwritteninformedconsent obtained patient addition informed consent analysis tumor associated genetic alterations obtained independently patient consent participation clinical study imatinib response correlates obtained using paired gist biopsies taken week initiation imatinib biopsies taken first week therapy imatinib resistance correlates evaluated biopsies taken time gist progression patients still receiving imatinib routine pathology review including kit immunostaining performed biopsies reagents cell lines gist430 gist48 cell lines established imatinibresistant gists site directedmutagenesiswas used generate relevant single double triple mutations kit pdgfra cdna 4 17 18 experiments involving recombinant dnawere performed using biosafety level 2 conditionsin accordance national institutes health guidelines research involving recombinant dnamolecules 19 imatinibmesylatewas purchasedfrom oregon health science university hospital pharmacy portland kit short hairpin rna studies akitlentiviralshort hairpin rna shrna obtainedfromwilliam hahn md phd dana farber cancer institute boston broad institute rnai consortium cambridge shrna assembled ligating kit forward 5 ccggccataaggtttcgtttctgtactcgagt acagaaacgaaaccttatggtttttg 3 reverse 5 aattcaaaaa ccataaggtttcgtttctgtactcgagtacagaaacgaaaccttat gg 3 oligomers agei ecori sites plko 1puro lentiviral vector lentiviral preps produced cotransfecting plko 1puro empty vector plko 1puro kit shrna pcmv r8 91 pmd g helper virus packaging plasmids 10 10 1 ratio 293t cells transfections performed using lipofectamine plus reagent invitrogen carlsbad ca lentivirus supernatants harvested 24 36 48 60 hours viral titers determined gist882 cells according protocol invitrogen gist882 gist430 cells infected presence 8 g ml polybrene lysed western blot analysis 72 hours postinfection immunoblotting studies whole cell lysates prepared gist biopsies gist cell lines 20 21the quality eachlysatewas determined byimmunoblottingfor gist markers kit pkc described previously 20 21 immunoblotting performed described previously addition hamster monoclonal anti bcl 2 kind gift stan korsmeyer md dana farber cancer institute utilized current report protein expression studies mutant kit pdgfra isoforms performed previously described transient transfection chinese hamster ovary cells transfected cells treated various concentrations imatinib preparation protein lystates kit protein assayed activation status phosphorylation table 1 pcr primers used detection mutations kit pdgfra exon forward primer reverse primer hplc temperatures k8 18 ttggtatttttgtccaggaactgagc agagaacagctcccaaagaaaaatcc na k9 atgctctgcttctgtactgcc cagagcctaaacatcccctta 50 0 c k11 ccagagtgctctaatgactg acccaaaaaggtgacatgga 50 0 c 56 2 c k11 13 ccctttctccccacagaaacccatg acacggctttacctccaatg na k11 13 acacggctttacctccaatg agacaataaaaggcagcttggacgac na k12 ctgcacaaatggtccttcaa caaaaagcacaactggcaaac 60 5 c k13 catcagtttgccagttgtgc acacggctttacctccaatg 59 5 c k14 ctcaccttctttctaaccttttctt cccatgaactgcctgtcaac 57 2 c k15 gacccatgagtgcccttct tctctgtaaagttactcttggttg 59 1 c k16 agtgatctgcctgcaagttcacattag gctctaaaatgctctgttctc 60 8 c k17 tgtattcacagagacttggc ggatttacattatgaaagtcacagg 58 0 c k18 ttctgttcaattttgttgagcttc gcaggacaccaatgaaactt 58 7 c k19 gcaaactgtgtctcaggaagc ccctcaacatctgggtttct 56 9 c k20 aaaacaagctgagggcattg tgggagaagggggattctat 59 3 c k21 tgttcgttgtagggactgct aaccatcatggaagccaaag 63 0 c k11 13 ccctttctccccacagaaacccatg acacggctttacctccaatg na k11 13 ccagagtgctctaatgactg agacaataaaaggcagcttggacgac na p10 ggccctatacttaggcccttt tcagctgatgagttgtcctga 50 0 c 62 0 c p12 tccagtcactgtgctgcttc gcaagggaaaagggagtctt 50 0 c 59 7 c p14 tggtagctcagctggactgat gggatggagagtggaggatt 59 1 c p15 ccatctcctaacggcttttg caggacatgggtctttccat 57 9 c p17 acctgatgatttcctgctgc cgtccacactccactcactg 58 2 c p18 accatggatcagccagtctt tgaaggaggatgagcctgac 50 c 61 6 c p19 tgctgtggatcatcagtgagt ttccttccagtggtgtttca 60 1 c p20 caatgcactgagcgtttgtt ctccccctagacccacagac 59 8 c abbreviations pdgfra platelet derived growth factor alpha hplc denaturing high performance liquid chromotography k kit exon na available p pdgfra exon imatinib resistance gists www jco org 4765 downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved immunoprecipitation using anti kit antibody followed sequential immunoblotting phosphokit using antiphosphotyrosine antibody total kit anti kitmonoclonal antibody case pdgframutations anidenticalmethodologywas used exceptfor substitution anti pdgfra antibody immunoprecipitation total pdgfra immunodetection steps ic50 50 inhibitory concentration measured densitometry phosphokit bands normalization using total kit expression correct variations loading kit protein various lanes 4 17 18 mutation detection methods mutational analyses performed genomic dna extracted paraffin embedded fresh frozen tumor tissue using combination polymerase chain reaction pcr amplification denaturing high performance liquid chromatography hplc screening automated sequencing described previously 4 17 22 pcr primer pairs hplc conditions listed table 1 pcr evaluations allelic relationships primary secondary mutations allelic relationships primary secondary kit mutations evaluated reverse transcriptase pcr exons 8 18 genomic pcr exons 11 13 using primer pairs listed table 1 resultant pcr products cloned individual bacterial colonies bidirectionally sequenced 23 results description patient population study population consisted 147 patients treated randomized phase ii study imatinib patients advanced gists 2 24as ofmay 2005 themedian time treatmentfailurefor entire phase ii study population 19 3 months median survival estimated 57 months late treatment failure primarily due disease progression rather discontinuation therapy related medication intolerance treatmentfailuresweredividedinto twogroups patientswhonever achieved partial remission showed continued tumor growth first 180 days treatment defined primary imatinib resistance disease progression minimum 6 months partial remission stable disease defined secondary imatinib resistance study based comparative analyses tumor samples obtained groups treatment failure kit kinase inhibited imatinib responding gists tumors primary resistance activation kit evaluated fresh frozen metastatic gist lesions two patients 11 34 underwent biopsy week starting imatinib therapy 5 7 days treatment fig 1 table 2 patients gists withkitexon 11 mutations kit phosphorylation evident preimatinib biopsies became nearly undetectable first days imatinib therapy kit inhibition accompanied dramatic decreases phosphorylated forms mtor akt mitogenactivated protein kinase mapk imatinib therapy also decreased expression proliferation marker pcna critical antiapoptotic protein bcl 2 analyzed activation status kit dependent signaling pathways biopsies obtained two patients whose gists kit exon 9 mutation showed primary imatinib resistance fig 2a table 3 frozen tumor samples available either patient pretreatment nevertheless treatment biopsies contained phosphorylated kit protein quantities equivalent typically present untreated gist specimens 21 pattern intracellular signaling similar seen untreated gists fig 1 duensing et al21 indicating primary resistance associated persistent kit phosphorylation activation downstream akt mapk pathways reactivation kit kit dependent signaling secondary resistance two patients objective responses imatinib underwent tumor biopsy documented progression 18 32 months matched pretreatment biopsies available patients fig 2b table 2 patient 1 primary kit exon 11 deletion patient 14 two activating kit mutations multiple pretreatment specimens k642e n822k example double mutations series 1 000 imatinib na patients gists 17 25 speculate one mutation initiating mutation developed malignant progression imatinib resistant tumors contained abundant amounts activated kit notably ratio phosphorylated akt p akt akt markedly increased progression samples fig 1 imatinib inhibits kit phosphorylation activation kit dependent signaling responding gastrointestinal stromal tumors paired biopsy specimens obtained initiation imatinib baseline within first week therapy treatment samples marked decreases levels phosphorylated p forms kit akt mitogen activated protein kinase mapk mtor heinrich et al 4766 journal clinical oncology downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved compared matched pretreatment samples contrast mapk significantly activated one progression samples patient 14 secondary kinase mutations common secondary primary imatinib resistance determine whether imatinib resistance associated acquired secondary kinase mutations screened cytoplasmic domain exons kit pdgfra dna prepared 79 tumor samples obtained 43 patients primary secondary drug resistance tables 2 3 90 7 patients 39 43 pretreatment kinase genotype previously determined 10 patients primary resistance median time treatmentfailure 3 6 months range 0 8 5 8 table 3 secondary mutations kit pdgfra kinase domain found specimens obtained patients time progression patient 39 primary progression dose 400 mg imatinib secondary kinase mutations found time progression 400mg cross 600mg led disease stablilization twentythree months later patient tumor progression ultimately underwent surgical debulking specimens procedure multiple secondary kinase mutations table 3 fig 3 thirty three patients secondary resistance median time treatment failure 20 2 months range 7 2 52 7 sixtyone progression specimens obtained 33 patients judged contain gists originalmutation whichwas knownfor 29 33 patients based analysis preimatinib specimens confirmed progression biopsies 54 54 samples overall 22 67 33 patients secondary resistance one secondary kinase mutations 21 kit 1 pdgfra table 2 fig 3 frequency distinctly higher patients primary resistance 10 v 67 p 002 notably secondary kit kinase mutations found gists table 2 patients secondary imatinib resistance patient age sex initial imatinib dose mg primary site best response time diagnosis imatinib days ttf days primary mutation secondary mutation mutations samples status 1 64 f 400 si pr 664 930 kit exon 11 del wkvve557 561 kit exon 17 n822k 4 4 awd 2 53 f 400 stomach pr 2 921 883 kit exon 11 del wk557 558 kit exon 17 y823d 1 1 dod 3 50 400 si pr 348 717 kit exon 11 del pmye551 554 kit exon 13 v654a 3 3 awd 4 25 600 stomach sd 403 508 none none 0 1 awd 5 39 400 si pr 591 504 kit exon 11 v560d kit exon 13 insertion kit exon 13 insertion v654a 3 4 kit exon 17 d816h 1 4 dod 6 54 600 si pr 1 037 512 kit exon 11 w557g kit exon 17 y823d 1 3 dod 7 43 400 iab pr 1 252 497 pdgfra v561d pdgfra d842v 1 1 dod 8 49 600 stomach pr 2 120 1 580 kit exon 11 del pmye551 554 none 0 2 dod 9 58 f 600 stomach pr 2 329 379 kit exon 11 del 554 558evqwk none 0 1 awd 10 49 600 si pr 643 1 564 kit exon 11 del kv558 559n kit exon 17 n822k 1 1 awd 11 50 f 400 si pr 3 615 349 kit exon 11 del vv559 560 none 0 1 dod 12 62 f 600 si pr 1 609 584 kit exon 11 del wkvv557 560c kit exon 17 y823d 1 4 dod 13 46 600 si pr 58 1 297 kit exon 11 del myevqw552 557 kit exon 13 v654a 1 1 awd 14 63 600 stomach pr 380 500 kit exon 13 k642e exon 17 n822h kit exon 17 c809g 2 2 dod 15 67 400 iab sd 155 215 none none 0 1 awd 16 52 400 stomach pr 27 246 kit exon 11 del kpmyevqwk550 558 none 0 2 dod 17 54 f 400 iab pr 778 766 kit exon 11 v560g kit exon 13 v654a 3 3 awd 18 67 400 iab pr 104 498 kit exon 11 v559d none 0 2 dod 19 59 400 iab pr 125 587 kit exon 11 v560d kit exon 17 d820a 2 2 dod 20 59 f 600 si pr 613 417 kit exon 9 kit exon 17 d820g 1 2 dod 21 54 400 stomach pr 952 366 kit exon 11 del kpmyevqwk550 558 kit exon 13 v654a 1 1 dod 22 65 400 iab pr 164 701 kit exon 11 delyevqwk553 558 kit exon 13 v654a 2 2 dod 23 70 f 400 stomach pr 2 285 846 kit exon 11 del evqwk554 558 kit exon 17 y823d 1 1 awd 24 45 f 600 stomach pr 60 384 kit exon 11 delkpmyevqwk550 558 kit exon 14 t6701 2 3 dod 25 44 f 400 stomach pr 411 1 028 kit exon 11 del wkvv557 560f kit exon 17 y823d 1 1 awd 26 37 600 si pr 141 499 kit exon 11 del evqwkvveeingnnyvyi554 571 none 0 1 dod 27 53 f 600 si pr 1 267 766 kit exon 9 none 0 1 awd 28 63 600 si pr 167 462 kit exon 9 kit exon 13 v654a 2 2 dod 29 60 400 stomach pr 122 237 kit exon 11 insertion k558np none 0 1 dod 30 63 400 stomach pr 138 505 kit exon 11 del kv558 559 none 0 1 dod 31 71 600 iab pr 507 747 kit exon 11 l576p kit exon 13 v654a 1 1 dod 32 38 400 si sd 33 416 kit exon 11 del kpmyevqw550 557fl kit exon 17 d820g 1 1 dod 33 60 400 stomach pr 275 252 kit exon 13 k642e kit exon 17 d816h 2 4 dod 34 77 400 stomach pr 448 881 kit exon 11 v560d np awd abbreviations ttf time treatment failure f female si small intestine pr partial response del deletion awd alive disease dod dead disease male sd stable disease iab intra abdominal np performed imatinib resistance gists www jco org 4767 downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved underlying primary kit mutation secondary mutations present pretreatment specimens secondary mutations identified one gist lacking primary kit pdgfra mutation addition secondary pdgfra mutation identified gist primary pdgfra v561d mutation secondary kit kinase mutations nonrandom involved either adenosine triphosphate atp binding pocket kinase domain v654a t670i kinase activation loop c809g d816h d820a e g n822k y823d figure 3 summarizes spectrum frequency secondary kit mutations series published reports contrast primary resistance two threekitexon 9mutant gists werefound additionalkitkinasemutations biopsies obtained development secondary imatinib resistance however frequency secondary kit mutations kit exon 9 mutant gists obviously different cases primary resistance versus secondary resistance 1 7 v 2 3 p 18 evidence clonal evolution polyclonal secondary resistance mutations individual patients multiple biopsy resection specimens available 16 patients progressive disease imatinib three fig 2 kit phosphorylation activation kit dependent signaling pathways gastrointestinal stromal tumors primary b secondary imatinib resistance two parts minimally progressing metastasis evaluated month 3 patient 40 lanes 1 2 two different progressing metastases evaluated month 6 patient 42 lanes 3 4 paired tumor specimens obtained treatment baseline time secondary imatinib resistance p phosphorylated mapk mitogenactivated protein kinase table 3 patients primary imatinib resistance patient age sex initial imatinib dose mg primary site best response time diagnosis imatinib days ttf days primary mutation secondary mutation status 35 52 600 li ne 829 ne pdgfra d842v none 0 2 dod 36 48 f 600 iab sd 294 156 kit exon 9 none 0 1 dod 37 45 400 si pd 620 23 kit exon 9 none 0 1 dod 38 33 600 si sd 2 415 166 kit exon 9 none 0 1 dod 39 36 f 400 si pd 638 74 kit exon 9 n822k 1 6 awd n822y 1 6 d820e 2 6 40 41 f 600 si sd 849 107 kit exon 9 none 0 3 awd 41 65 400 iab pd 1 609 84 kit n822k none 0 1 dod 42 52 600 si sd 1 378 173 kit exon 9 none 0 1 dod 43 51 400 si sd 1 145 173 kit exon 9 none 0 1 awd 44 54 600 stomach pd 1 590 89 pdgfra d842v none 0 1 dod abbreviations ttf time treatment failure male li large intestine ne assessable dod dead disease f female iab intra abdominal sd stable disease si small intestine pd progressive disease awd alive disease ttf could defined due use different imaging modalities staging exams patient detectable secondary mutation time initial progression however secondary kinase mutations found specimens obtained 1 year initial progression heinrich et al 4768 journal clinical oncology downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved patients 3 16 18 8 found evidence clonal evolution polyclonal secondary kinase mutations tables 2 4 patient 5 baselinekitv560d substitution one allele single adenine base insertion kit exon 13 allele resulting frameshift protein truncation thus patient tumorwasfunctionally homozygous due loss thewild type allele specimens obtained time imatinib resistance v654a mutation present two biopsy specimens d816h third specimen three specimens original preimatinib kit exon 11 13 mutations allelic distribution secondary kinase mutations kit mutant gists 90 heterozygous original activating mutation therefore one wild type one mutant kit allele without exception secondary mutations allele primary mutation table 4 noted patient 14 tables 2 4 gist two monoallelic preimatinib kit mutations k642e n822h acquired c809g substitutionfound time ofimatinib resistancewas allele k642e n822h mutations secondary kinase mutations lead imatinib resistance vitro determine secondary kinase mutations cause clinically observed imatinib resistance tested vitro sensitivity selected single double triplemutant kinases toimatinib gists manifest wide range mutations exon 11 table 2 chose representative kit exon 11 point mutation v560d standard comparing imatinib sensitivity single double mutant kit kinases 25 secondary mutations interest example v654a engineered v560d isoform well wild type cdna allowing comparative analyses effects secondary mutations imatinib sensitivity several secondary kit mutations identified series previously identified human cancers d816h d820a n822h k y823d 4 17 18 26 30 contrast reports v654a t670i c809g mutations occurring primary mutation isolated kit v560d mutation sensitive imatinib ic50 100 nmol l fig 4 v654a t670i resulted strong imatinib resistance ic50s 5 10 mol l respectively either isolation coexpressed kit v560d similar results obtained usingkitexon 9 ay primary mutation fig 4 number amino acids located near kit activation loopweremutatedinimatinib resistant tumors fig 3 table 2 vitro profiling isolated activation loop mutations revealed spectrum imatinib sensitivity rangingfrom relatively sensitive ic50 100 200 nmol l n882k y823d highly resistant ic50 5 000 nmol l d816h coexpressed v560d double mutant kinases extremely resistant imatinib exception v560d n822k moderately imatinib resistant fig 5a similar results obtained using kit exon 9 ay insertion k642e primary mutation fig 5b notably primary mutations intrinsically imatinib sensitive fig 3 secondary kit mutations imatinib resistant gastrointestinal stromal tumors data current report 21 patients published reports 43 patients indicated red black symbols respectively 6 14 patients different secondary mutations geographically separate metastases indicated numbered symbols kit codons listed shaded boxes table 4 alleleic distribution studies patient primary mutation secondary mutation allelic distribution mutations 5 kit v560d v654a cis 5 kit v560d d816h cis 5 kit v560d exon 13 insertion leading premature stop codon trans 14 kit k642e n822h c809 cis 19 kit v560d d820a cis 23 kit del554 558evqwk y823d cis 28 kit exon 9 v654a cis 33 kit k642e d816h cis abbreviations cis allele primary mutation trans opposite allele primary mutation imatinib resistance gists www jco org 4769 downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved coexpression ofd820gwithkitexon 9 resultedinmoderateimatinib resistance coexpression imatinib sensitive k642e mutation d816h resulted extreme imatinib resistance doubly mutant k642e n822h kinase modeled patient 14wasmoderately resistant toimatinib ic50of 2 mol l notably patient 14 received dose 600 mg daily imatinib experienced objective partial response lasted 500 days c809g mutation found time progression proved resistant vitro ic50 5 mol l isolation coexpressed kit k642e n822h patient 7 primary pdgfra v561d mutation known sensitive imatinib accompanied pdgfra d842v mutation onset progression doubly mutant v561d d842v kinase proved imatinib resistant ic50 similar isolated d842v isoform fig 6 17 gist cell lines secondary kinase mutations vitro imatinib resistance validate biochemical studies generated cell lines two patients secondary imatinib resistance neither patient part phase ii study gist48 cell line combination mutations homozygous v560d heterozygous d820a similar documented samples patient 19 concentrations imatinib higher 1 mol l required complete inhibition kit activation cell line fig 7a concentration 10 fold greater necessary block exon 11 mutant isoforms kit gist cells 8 21 31 activation akt partially completely inhibited imatinib doses 0 1 5 mol l gist430 cell line heterozygous kit exon 11 deletion mutation v654a substitution allele eight samples imatinib resistant tumors study similar combination kit exon 11 v654a mutations doubly mutantkitisoform expressed gist430 cellswasimatinib resistant ic50of 2 5 mol l ic90of 5 mol l akt activationin cell line partially inhibited imatinib doses higher 2 5 mol l fig 7a used rnai determine whether kit expression still required activation signaling pathways imatinibresistant gist cells shrna knockdown total kit expression previously described imatinib sensitive gist882 cell line homozygous kit k642e mutation 31 resulted parallel decreases phospho kit phospho akt phospho p70s6k proliferation marker cyclin expression fig 7b contrast change expression p70s6k akt pi3k comparable findings obtained kit shrna knockdown gist430 cells demonstrating activation proliferation survival signaling pathways remains kit dependent imatinib resistant cell line kitknockdown cell lines also induced flow cytometric evidence g1 block decreased phase markedly increased apoptosis data shown discussion analyzed genomic mechanisms imatinib resistance cohort patients part randomized phase ii study imatinib treatment advanced unresectable gist forty three 46 7 92 patients disease related treatment failures consented studies tumor samples resulting collection 83 fig 4 biochemical characterization secondary mutations involving adenosine triphosphate atp binding pocket kit secondary mutations involving atp binding pocket tested vitro sensitivity imatinib expressed isolation coexpressed kit exon 11 v560d kit exon 9 mutations p phosphorylated heinrich et al 4770 journal clinical oncology downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved specimens well monitored patients receiving standardized imatinib treatment largest examined date molecular mechanisms related drug resistance imatinib resistant gists show activation signaling pathways operative untreated gists interestingly primary imatinib resistance infrequently associated secondary kinase mutations 1 10 10 pdgfra d842v kit exon 9 mutations significantly represented group compared secondary imatinib resistance gists genotypes less responsive imatinib therapy gists associated kit exon 11 mutation 4 5 pdgfra d842v mutation moderate high level vitro resistance imatinib explaining mutation responsible primary imatinib resistance two patients primary pdgfra d842v mutation 4 8 32 conclusion strengthened fig 5 biochemical characterization secondary mutations involving kit activation loop secondary mutations involving kit activation loop tested vitro sensitivity imatinib expressed isolation coexpressed kit exon 11 v560d b isolation coexpressed kit exon 13 k642e kit exon 9 mutations p phosphorylated fig 6 platelet derived growth factor alpha pdgfra d842v mutation associated primary secondary imatinib resistance imatinib potently inhibits phosphorylation primary pdgfra exon 12 v561d mutation patient 17 pretreatment pdgfra d842v either expressed isolation patients 1 10 coexpressed v561d patient 17 progression specimen imatinib resistance gists www jco org 4771 downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved observation pdgfra d842v mutation associated secondary imatinib resistance gist harboring primary pdgfra v561d mutation debiec rychter et al8 also reported case imatinib resistant gist associated acquisition pdgfra d842v mutation gist primary kit exon 11 mutation themolecularmechanisms underlying primary resistancein gists kit exon 9 mutations yet unidentified may related differencesin underlying biology subset ofgists speculate kit exon 9 mutant gists alternative mechanism kit activation require enzymatic activity kit notably imatinib binds inactive form kit 33 34 therefore fig 7 biochemical characterization secondary kit mutations expressed imatinib resistant gastrointestinal stromal tumor gist cell lines gist48 imatinib resistant cell line homozygous kit v560d heterozygous d820a mutation gist430 imatinib resistant cell line heterozygous kit exon 11 deletion mutation v654a mutation allele gist882 imatinib sensitive gist cell line expressing homozygous k642e mutation effects imatinib treatment b effects rnai inhibition kit p kit phosphorylated kit p akt phosphorylated akt shrna short hairpin rna heinrich et al 4772 journal clinical oncology downloaded ascopubs org memorial sloan kettering cancer center may 25 2017 140 163 000 005 copyright 2017 american society clinical oncology rights reserved kit independent mechanisms kit phosphorylation could stabilize protein active imatinib resistant conformation gists kit exon 11 mutations commonly become imatinib resistant due acquisition secondary kinase mutations located allele original gain function kit mutation secondary kinase mutations appear predominant mechanism late imatinib resistance kit exon 11 mutant gists fluorescence situ hybridization assays data shown revealed low level kit amplification 3 4 copies per cell accompanied kit protein overexpression two patients suggesting genomic kit amplification minor role gist imatinib resistance believe differencein thefrequency secondary mutations gists underlying kit exon 11 versus kit exon 9 mutations reflective underlying biology tumors namely kit exon 11 mutant gists addicted kit signaling thankitexon 9 mutant gists model kitexon 9 mutant gists able use alternative signaling mechanisms drive cellular proliferation contrast kitexon 11mutant gists restricted signaling requirements themost common mechanism resistance expansion clone imatinibresistant kinase mutation using rnai kit knockdown demonstrated imatinibresistant gist cell lines secondary kinase mutations remain dependent onkit signalingfor growth survival findings implications future approaches growing problem imatinib resistance patients advanced gists activating mutation kit pdgfr platelet derived growth factor receptor considered main molecular driver gi stromal tumors gists almost kit mutations occur exon 9 11 13 17 pdgfra mutations exons 12 14 18 1 ten percent 15 gists lack kit pdgfra mutations often referred wild type gists though may mutations genes 2 3 gists localized detected approximately 40 eventually recur despite macroscopically complete surgery 4 adjuvant imatinib reduces risk provided gist harbors imatinib sensitive mutation 5 7 general kit exon 11 mutations sensitive imatinib whereas pdgfra exon 18 mutation leading asp842val d842v considered imatinib resistant 8 9 administering adjuvant imatinib least 3 years surgery recommended risk recurrence considered high 10 11 tumor mutation analysis important identify imatinib resistant mutations kit exon 9 mutations may require high dose imatinib 400 mg day 12 whether mutation analysis confers prognostic information beyond standard prognostic factors identified controversial 13 15 commonly used risk stratification schemes armed forces institute pathology afip scheme national institutes health nih scheme modification based tumor size site mitotic count include tumor mutation status risk stratification 16 18 yet certain mutations repeatedly linked survival gist clinical features example asp842val d842v associated gastric location favorable survival 14 19 whereas kit exon 11 mutations affecting codons 557 558 associated unfavorable outcome 13 20 gists kit exon 9 mutation usually arise small bowel associated frequent recurrence 21 22 outcome data related single mutations likely presence large number different mutations particularly kit exon 11 rarity gist moreover adjuvant imatinib modifies risk recurrence owing varying sensitivity different mutations drug 23 investigated influence kit pdgfra mutations recurrence free survival rfs gist patient population treated macroscopically complete surgery obtain large relatively unselected series study influence common single mutations survival first identified population based series literature pooled individual patients data 4 patients methods patients identified population based gist patient series published january 1 2000 january 1 2010 electronic medical literature databases reduce risk selection bias required series consist virtually gists diagnosed within defined geographical area time period include least 50 patients provide data main prognostic factors identified 10 series24 32 included total 2 560 patients 1 625 rfs information available study approved institutional review committee 4 patients operable gist detectable metastases time diagnosis 24 32 patients multiple recurrent gists excluded patients underwent macroscopically complete surgery none received adjuvant neoadjuvant systemic therapy tumor required morphology compatible gist stain kit immunohistochemistry median follow time patients alive 4 0 years range 0 01 45 8 years rfs 0 01 26 9 years mutation analysis done 1 091 42 6 2 560 gists data mutated gene exon available 144 patients 13 2 1 091 patients mutational data available similar distributions age sex tumor size mitotic count site patients mutational data available n 1 469 p 05 comparison appendix table a1 online mutation analysis usually done kit exons 9 11 13 17 pdgfra exons 12 18 numbers exons analyzed case available molecgist series molecgist series prospectively collected population based series consisting gist patients diagnosed france may 2006 june 2008 33 series included global pooled series study fully reported january 1 2010 patients received adjuvant neoadjuvant imatinib molecgist mutational data included current study survival data excluded avoid potential bias related selected use adjuvant imatinib higher risk patients fig 1 figure fig 1 consort diagram gist gi stromal tumors rfs recurrence free survival clinical records tumor histology reviewed diagnosis gist based histology immunostaining kit dog 1 33 596 patients series 507 patients local gist excised surgically 414 patients 81 7 mutation analysis kit exons 9 11 13 17 pdgfra exons 12 14 18 performed 93 tumors analyzed mutations similar features ones analyzed appendix table a2 study approved ethics committee 33 statistical analysis rfs calculated date surgery date making histologic diagnosis date first tumor recurrence patients alive censored time data collection patients died censored dates death survival groups compared using kaplan meier life table method log rank test hazard ratios cis calculated using univariable cox model prognostic factors compared using cox proportional hazards model frequency tables analyzed using 2 test continuous distributions groups compared mann whitney u test kruskall wallis test p values two sided adjusted multiple testing data analyzed using statistical package social sciences version 17 0 spss chicago il results patients mutation analysis data available 1 505 49 1 3 067 patients fig 1 median age patients mutation analyses available 64 years time gist detection range 9 106 years 778 patients 51 7 male six patients 0 4 younger 20 years pdgfra patients pdgfra mutations wild type gist smaller risk recurrence patients kit mutations pdgfra hazard ratio hr 0 34 95 ci 0 16 0 73 p 004 wild type gist 0 52 95 ci 0 35 0 76 p 001 fig 2a significant difference patients pdgfra mutations kit mutations age diagnosis sex tumor size p 10 tumors pdgfra mutations lower mitotic counts median two v four per 50 high power fields p 001 often gastric 88 0 v 54 7 p 001 appendix table a3 patients pdgfra exon 12 exon 18 mutations similar rfs fig 2b total 33 different mutations identified pdgfra frequent changes asp842val d842v val561asp v561d made 84 54 9 14 9 2 153 defined pdgfra alterations respectively figure fig 2 associations kit platelet derived growth factor alpha gene pdgfra mutations recurrence free survival surgery gi stromal tumors gists gists kit mutation pdgfra mutation mutation either gene one patient mutations kit pdgfra b gists either pdgfra exon 12 18 mutation c gists kit exon 9 11 13 17 mutation wt wild type kit exons mutations different kit exons associated similar rfs though patients exon 9 mutations tended inferior rfs compared patients exon 11 mutations p 074 fig 2c approximately 80 exon 9 mutated gists arose small bowel appendix table a3 gists kit exon 13 exon 17 mutations also nongastric exon 13 14 16 exon 17 six 10 whereas 583 59 4 982 exon 11 mutated gists site origin available gastric p 001 significant difference gists exon 9 13 17 mutations exon 11 mutations tumor size mitotic count p 05 pair wise comparison exon 13 mutations tended associated smaller mitotic counts p 064 kit mutation types total 301 different single kit mutations identified n 293 found exon 11 whereas 97 6 83 85 exon 9 mutations led ala502 tyr503 duplication 93 3 14 15 exon 13 mutations lys642glu k642e 66 7 six nine exon 17 mutations asn822lys n822k frequent 981 kit exon 11 mutations mutation type available deletions 43 8 followed substitution mutations 30 6 insertion deletion indel mutations 15 8 table 1 deletion single codon found 71 tumors 7 2 occurred nine codons 553 557 560 568 575 576 579 n 101 78 3 129 deletions two codons deletions 557 558 exon 11 deletion mutations involved codon 557 558 242 419 57 8 317 mutations 75 7 involved one codons 557 560 duplication mutations occurred 3 codons 557 558 cases except one tumor dna segment located 5 codon 557 duplicated median length ten codons range two 22 table table 1 associations type kit exon 11 mutation gist site size mitotic count table 1 associations type kit exon 11 mutation gist site size mitotic count type kit exon 11 mutation total patients different mutations gist site tumor size cm mitotic count per 50 hpfs esophagus stomach small intestine colon rectum non gi tract site na patients patients patients patients patients patients median range median range deletion 430 43 8 111 37 9 5 55 6 254 44 2 132 42 7 18 45 0 15 42 9 6 6 0 0 1 33 0 5 0 250 1 codon 71 7 2 9 3 1 1 11 1 45 7 8 21 6 8 2 5 0 2 5 7 0 5 2 0 9 22 0 3 0 60 2 codons 359 36 6 102 34 8 4 44 4 209 36 3 111 35 9 16 40 0 13 37 1 6 7 0 0 1 33 0 5 0 250 insertion deletion 155 15 8 96 32 7 3 33 3 73 12 7 60 19 4 15 37 5 2 5 7 2 6 4 0 7 26 0 5 0 250 duplication 52 5 3 37 12 6 0 0 45 7 8 7 2 3 0 0 0 0 0 4 5 1 5 19 0 3 0 45 insertion 27 2 8 13 4 4 1 11 1 16 2 8 4 1 3 0 0 3 8 6 3 8 0 0 5 45 0 4 0 34 substitution 300 30 6 19 6 5 0 0 176 30 6 103 33 3 7 17 5 12 34 3 2 5 5 0 3 30 0 3 0 150 17 1 7 17 5 8 0 0 11 1 9 3 1 0 0 0 3 8 6 0 4 5 0 6 20 0 2 5 0 13 total 981 100 0 293 100 0 9 100 0 575 100 0 309 100 0 40 100 0 35 100 0 13 6 0 0 1 45 0 4 0 250 note type mutation available 15 1 5 996 patients kit exon 11 mutation 981 patients type mutation available data available tumor site 13 patients 1 3 exact noncategorized tumor size 32 patients 3 3 exact noncategorized mitotic count 192 patients 19 6 abbreviations gist gi stromal tumor hpf high power field microscope na available includes mesentery n 8 peritoneum n 3 retroperitoneum n 8 n 10 intra abdominal sites 6 patients classified non gi tract gist includes six tumors two substitution mutations six duplication mutation insertion mutation four deletion mutation substitution mutation one insertion deletion mutation substitution mutation insertion deletion mutations frequent nongastric gastric gists 20 4 v 12 7 respectively p 001 whereas exon 11 duplication mutations common gastric nongastric gists 7 8 v 1 8 respectively p 001 table 1 gists exon 11 duplication mutation substitution mutation smaller gists exon 11 deletion mutation p 012 p 017 respectively gists substitution mutation also lower mitotic counts p 004 gists deletions one codon smaller deletions two codons p 045 lower mitotic counts p 004 one 2 9 35 patients kit exon 11 duplication mutations rfs data available gist recurrence whereas gists deletion mutation insertion deletion mutation recurred often fig 3a patients one deleted exon 11 codon favorable rfs longer deletion insertion deletion mutation fig 3b mutations involved codons 557 558 associated unfavorable rfs fig 3c figure fig 3 associations kit pdgfra mutations recurrence free survival surgery gi stromal tumors kit exon 11 mutation types b kit exon 11 deletion types c mutations involving kit exon 11 codons 557 558 three common kit alterations e common kit substitution mutations f pdgfra mutations common single mutations deletion trp557 lys558 kit exon 11 substitution asp842val pdgfra exon 18 duplication ala502 tyr503 kit exon 9 common single alterations table 2 besides gists ala502 tyr503dup gists lys642glu also frequently arose small intestine gists trp557 lys558del usually large detected tumors asp842val leu576pro trp557arg lys642glu val560del small median mitotic counts table table 2 frequent kit pdgfra alterations table 2 frequent kit pdgfra alterations protein alteration gene exon gists n 1 505 gist site tumor size cm mitotic count per 50 hpfs stomach small intestine colon rectum non gi tract site na tumors patients patients patients patients median range median range trp557 lys558del kit exon 11 101 6 7 65 65 6 24 24 2 7 7 1 3 3 0 2 8 0 0 6 30 0 6 0 200 asp842val d842v pdgfra exon 18 84 5 6 69 88 5 2 2 6 2 2 6 5 6 4 6 5 0 0 6 30 0 2 0 200 ala502 tyr503dup kit exon 9 83 5 5 9 10 8 68 81 9 3 3 6 3 3 6 0 6 0 0 8 32 0 3 0 115 val559asp v559d kit exon 11 70 4 7 50 72 5 18 26 1 0 0 1 1 4 1 5 3 0 8 30 0 4 0 150 val560asp v560d kit exon 11 51 3 4 35 70 0 13 26 0 2 4 0 0 2 0 1 6 0 0 7 25 0 4 0 33 trp557arg w557r kit exon 11 50 3 3 32 64 0 13 26 0 1 2 0 4 8 0 0 4 5 0 4 13 0 2 0 70 val560del equal val559del kit exon 11 41 2 7 26 63 4 13 31 7 1 2 4 1 2 4 0 5 0 1 0 22 0 1 0 60 leu576pro l576p kit exon 11 28 1 9 12 42 9 14 50 0 0 0 2 7 1 0 4 5 0 3 19 0 0 0 14 val559ala v559a kit exon 11 20 1 3 10 50 0 9 45 0 0 0 1 5 0 0 6 0 1 5 25 0 4 1 21 trp557gly w557g kit exon 11 19 1 3 9 47 4 8 42 1 1 5 3 1 5 3 0 7 3 0 5 22 0 5 0 150 lys642glu k642e kit exon 13 14 0 9 2 14 3 12 85 7 0 0 0 0 0 5 25 1 2 15 0 1 0 15 abbreviations gist gi stromal tumor hpf high power field microscope na available pdgfra platelet derived growth factor alpha gene none gists presented esophagus five gists rectal five colonic gists either trp557 lys558del asp842val trp557arg val560del trp557gly two additional patients asp842tyr fifteen additional patients val559gly nine val560gly nine val560glu one val560gln three additional patients trp557lys three common single alterations trp557 lys558del ala502 tyr503dup val559asp associated similar rfs fig 3d exon 11 substitution alterations trp557arg leu576pro val559ala tended best rfs whereas val559asp trp557gly worst fig 3e mutations causing substitution alterations asp842 rfs similar pdgfra mutations fig 3f different substitution mutations codon necessarily associated similar clinical features gists val559asp often gastric val559ala val559gly 72 5 v 42 9 p 003 patients val559ala tended favorable rfs val559asp p 098 patients trp557arg lower mitotic counts trp557gly median two v five respectively p 016 better rfs p 016 fig 3e table 2 multivariable survival analyses mutated gene kit v pdgfra wild type included covariable cox multivariable hazards model tumor size continuous variable site stomach v another site gi tract v non gi tract location mitotic count continuous variable presence kit mutations independent adverse influence rfs p 011 hr 1 89 95 ci 1 15 3 08 mitotic count size gastric site stronger prognostic factors mitotic count hr 1 018 95 ci 1 015 1 022 size hr 1 10 95 ci 1 07 1 12 gastric site hr 0 45 95 ci 0 33 0 62 p 001 analysis restricted kit mutated gists mutated exon 9 11 13 17 entered together tumor size site mitotic count model mutated exon independently associated rfs appendix table a4 patients identical mutation greatly variable outcomes depending standard prognostic factors notably tumor mitotic count fig 4 tumor size mitotic count entered continuous covariables cox multivariable model subset patients kit exon 11 mutation leading trp557 lys558del independently associated rfs hr 1 10 95 ci 1 04 1 17 p 001 hr 1 02 95 ci 1 003 1 03 respectively p 012 similarly tumor size independent prognostic factor rfs subset patients val559asp mitotic count subset ala502 tyr503dup data shown figure fig 4 influence tumor mitotic count recurrence free survival subgroups defined tumor mutation analysis gi stromal tumors kit exon 11 deletion mutation b trp557 lys558del c kit exon 11 substitution mutation val559asp e ala502 tyr503dup f pdgfra mutation afip nih schemes finally investigated afip modified nih schemes stratify risk recurrence subsets patients mutation type associated favorable rfs table 3 approximately 30 patients stratified high risk category additional 20 intermediate risk category suggesting risk stratification schemes may overestimate risk gist recurrence mutational type favorable presence favorable mutation table 3 entered covariable cox multivariable hazards model favorable v mutation either afip scheme modified nih scheme mutation type afip scheme appendix table a5 mutation type modified nih scheme appendix table a6 independently influenced rfs stratifying gists using prognostic heat maps 4 8 3 23 3 patients one favorable mutations 21 40 estimated risk recurrence within first 10 years surgery 8 3 22 4 greater 40 risk appendix table a7 table table 3 risk groups patents prognostically favorable kit pdgfra mutation table 3 risk groups patents prognostically favorable kit pdgfra mutation risk group kit exon 11 duplication mutation 5 year rfs 96 9 kit exon 11 deletion 1 codon 5 year rfs 82 2 kit exon 11 mutation leading trp557arg val559ala leu576pro 5 year rfs 82 5 95 2 pdgfra mutation 5 year rfs 82 2 total patients patients patients patients patients afip risk group 1 2 low risk 20 54 1 20 46 5 31 56 4 35 45 5 106 50 1 3a intermediate risk 9 24 3 12 27 9 11 20 0 18 23 4 50 23 6 3b 4 5 6a 6b high risk 8 21 6 11 25 6 13 23 6 24 31 1 56 26 4 modified nih risk group low low 20 54 1 20 44 4 31 47 0 36 46 7 107 49 5 intermediate 9 24 3 6 13 3 7 10 6 17 21 8 39 18 1 high 8 21 6 19 42 2 18 27 3 25 32 1 70 32 4 abbreviations afip armed forces institute pathology nih national institutes health pdgfra platelet derived growth factor alpha gene rfs recurrence free survival discussion mutation analysis kit pdgfra provided potentially important prognostic information suggested others 34 patients pdgfra mutations substantially better rfs kit mutations tumor kit mutations associated increased risk gist recurrence multivariable analysis besides pdgfra mutations identified mutational subgroups favorable rfs patients exon 11 duplication mutations low risk recurrence suggesting patients generally candidates adjuvant treatments mutational subgroups favorable rfs identified earlier included patients one codon deletion mutation kit exon 11 substitution mutations leading trp557arg val559ala leu576pro commonly used risk stratification schemes overestimated risk gist recurrence patients one generally favorable mutation types presence favorable mutation independent influence rfs beyond provided afip modified nih schemes multivariable analyses standard prognostic factors mitotic count particular generally predict risk recurrence accurately gist mutation status 7 demonstrate standard factors mitotic count great impact outcome despite identical kit pdgfra mutation note patients generally unfavorable mutation kit exon 9 mutation leading ala502 tyr503 duplication exon 11 deletion mutation causing try557 lys558 deletion still low risk gist recurrence provided mitotic count low contrast patients pdgfra mutations high risk recurrence tumor mitotic count high therefore tumor mutation status interpreted isolation prognostic factors observation well line studies micro gists common clinically benign consist less 10 millimeter lesions mitotic figures yet often contain kit pdgfra mutations 35 36 relative influence common asp842val alteration pdgfra alterations rfs unknown current data suggest risk recurrence generally low regardless type pdgfra mutation similarly prior data risk associated kit exon 13 17 mutations keeping joint analysis 54 patients mutations 37 gists kit exon 13 17 mutation intestinal despite outcome patients inferior patients kit mutations approximately 70 kit exon 17 mutations lead asn822lys 37 considered moderately imatinib sensitive 38 may longer reasonable group kit exon 11 mutations together studies prognosis since exon 11 mutations associated variable outcomes clinical features present findings suggest even mutations single codon leading substitutions different amino acid residues may associated different risks recurrence study limitations collected population based data attempt avoid selection bias heterogeneity prognostic factors series 4 tumor mutation analysis available 49 1 3 067 patients found evidence selection bias main prognostic factors cannot exclude mutations may missed exons mutations occur rarely sequenced may inflated numbers gists mutations found tumor rupture data frequently unavailable assessed mitotic counts available per one microscope high power field vary size depending microscope knowledge current series largest mutation analyses gists reported despite numbers patients remained relatively small subgroup analyses often lack rfs information conclude gist mutation analyses provide clinically useful prognostic information patient population treated surgery alone standard factors generally stronger prognosis associated rfs even subsets patients identical kit pdgfra mutation yet mutational data important consider prognostically favorable mutation found duplication mutation kit exon 11 gastrointestinal stromal tumors gist activating mutation either kitor pdgfra imatinib selective tyrosine kinase inhibitor achieves partial response stable disease 80 patients metastatic gist clear patients gistdevelop resistance imatinib chronic therapy identify mechanism resistance studied 31patients gistwho treated withimatinib underwent surgical resection were13 patients whowere nonresistant toimatinib 3 withprimary resistance and15 acquired resistance afterinitial benefit drug secondary mutations kitor pdgfra nonresistant primary resistance groups contrast secondary mutations found 7 of15 46 patients acquired resistance whomhad primary mutationinkitexon11 secondary mutations located kitexon 17 kit phosphorylation heterogeneous correlate clinical response imatinib mutation status acquired resistance imatinib gistcommonly occurs via secondary gene mutation kit kinase domain implications strategies delay prevent imatinib resistance employ newer targeted therapies gastrointestinal stromal tumor gist common mesenchymal tumor intestinal tract surgical removal primary gist tumor recurrence frequent previously 50 patients died within 5 years initial diagnosis 1 1998 kitamura colleagues reported gist commonly associated activating mutations kit receptor tyrosine kinase 2 recently showed knock kit activation mutation kitv558d sufficient induce gist mice 3 imatinib mesylate sti571 gleevec novartis pharmaceuticals basel switzerland oral agent specifically inhibits bcr abl abl tyrosine kinases well kit pdgfr receptor tyrosine kinases 4 initially imatinib applied treatment chronic myelogenous leukemia cml achieves 90 complete response rate 5 6 subsequently imatinib tested metastatic unresectable gist found induce partial response stable disease 80 patients 7 8 primary resistance imatinib gist continues progress despite institution therapy occurs 15 patients remarkably 2 year survival patients metastatic gist f70 9 90 cases gists activating mutations either kit pdgfra receptor tyrosine kinases 10 13 common site kit mutation 5v end exon 11 encodes juxtamembrane domain usually deletions substitutions codons 550 560 occur kit exon 9 mutation occurs 10 15 patients defines distinct subset gists often located small bowel aggressive clinical behavior 11 14 infrequently mutation identified kit exon 13 17 15 16 another 5 patients gist pdgfra mutation typically involves exon 12 18 12 13 10 patients detectable mutation either kit pdgfra particular gists occur pediatric patients nearly always wild type genes 11a recent data suggest possible correlation imatinib response type mutation tumors exon 9 mutation wild type kit less likely respond imatinib 17 18 although patients advanced gist benefit imatinib treatment clear many patients subsequently develop resistance agent median time progression 24 months 9 mechanism acquired resistance imatinib gist well defined cml second site mutation bcr abl predominant mechanism imatinib resistance 19 20 therefore postulated acquired resistance imatinib gist due secondary site mutation kit pdgfra genes molecular analysis 65 tumor nodules 31 patients treated imatinib clin cancer res 2005 11 11 june 1 2005 4182 www aacrjournals org authors affiliations departments 1 pathology 2 surgery 3 medicine 4 memorial sloan kettering cancer center developmental biology 5 cell biology 6 structural biology programs sloan kettering institute new york newyork received 11 4 04 revised 1 16 05 accepted 3 4 05 grant support acs mrsg cce 106841 c r antonescu ca102774 hl dk55748 p besmer po1 ca 47179 f brennan ca94503 ca102613 r p dematteo costs publication article defrayed part payment page charges article must therefore hereby marked advertisement accordance 18 u c section 1734 solely indicate fact requests reprints cristina r antonescu department pathology memorial sloan kettering cancer center 1275 york avenue new york ny 10021 phone 212 639 5721 fax 212 717 3203 e mail antonesc mskcc org f2005 american association cancer research cancer therapy preclinical research downloaded clincancerres aacrjournals org may 29 2017 2005 american association cancer three patients showed primary resistance imatinib whereas 15 patients acquired resistance imatinib therapy materials methods clinicopathologic analysis patients diagnosis gist treated imatinib underwent surgical resection tumor memorial sloan kettering cancer center identified prospective sarcoma database 31 patients adequate tumor tissue molecular analyses following treatment patient tumor treatment information obtained database reviewing medical charts primary resistance imatinib defined continued growth tumor despite institution imatinib therapy acquired resistance defined new tumor growth occurred subsequent initial period 2 3 months imatinib patient stable responding disease tumor size determined computed tomography scan magnetic resonance imaging study approved institutional review board pathologic material examined diagnosis confirmed using standard h e staining cd117 immunohistochemistry formalin fixed paraffin embedded tissue previously described 11 histologic response imatinib based gross microscopic findings necrosis fibrosis scored tumor nodule minimal 10 response low 10 50 response moderate 50 90 response high z90 response degree response kit pdgfra genotyping mutation analysis done described previously 11 genomic dna isolated snap frozen tumor tissue samples stored 70jc using standard phenol chloroform organic extraction protocol adequate dna mutational analysis obtained 65 tumor nodules 31 patients cases tested known sites kit exons 9 11 13 14 17 pdgfra exons 12 18 mutations one microgram genomic dna subjected pcr using platinum taqdna polymerase high fidelity life technologies inc gaithersburg md primer sequences annealing temperatures described 11 addition primers kit exon 14 gtctgatccactgaagctg 50jc accccatgaactgcctgtc 51jc pdgfra exon 12 tccagtcactgtgctgcttc gcaagggaaaagggagtctt 54jc pdgfra exon 18 accatggatcagccagtctt tgaaggaggatgagcctgacc 55jc direct sequencing pcr products done exons tested abi sequence compared national center biotechnology information human kit pdgfra gene sequences four patients including three acquired resistance adequate tumor tissue available mutation analysis surgical resection occurred prior initiation imatinib case including one patient developed second site mutation confirmed primary mutation recurrence cdna sequencing cloning tumors secondary kit mutations genomic dna analysis amplified sequenced kit cdna exons 10 18 way confirmed initial genotype results also determined whether secondary mutation occurred opposite allele primary mutation adequate rna obtained 32 tumor nodules 18 patients using rna wiz reagent ambion inc austin tx guanidinium isothiocynate phenol chloroform method five micrograms rna transcribed using reverse transcriptase superscript ii invitrogen carlsbad ca cdna subjected pcr using primers kit exon 10 gccggatcctttcgtaatcgtagct containing bamhi linker exon 18 ggcgaattctgtatacacagttgaaaatg ecori linker amplification kit insert done separately platinum taq invitrogen pfu ultra high fidelity dna polymerase stratagene la jolla ca order exclude false positive mutations due errors polymerase proofreading cloning kit insert chloramphenicolresistant pbc ks vector stratagene digested bamhi ecori ligated using rapid dna ligation kit roche applied science penzburg germany plasmid transformation xl 1 blue supercompetent cells stratagene colony selection chloramphenicol luria bertani agar plates done using standard methods sample 10 distinct clones picked expanded overnight shaking incubator colony pcr done using taq polymerase kit exon 10 18 linker primers clones containing insert selected dna isolated dna miniprep kit qiagen inc valencia ca directly sequenced western blotting adequate tissue protein extraction available 24 patients 43 tumor nodules also included comparison 8 untreated gists kit exon 11 mutations 4 frame deletions 4 substitutions cd117 negative desmin positive high grade gastric leiomyosarcoma included negative control preparation whole lysates 1 g snapfrozen tumor ground powder liquid nitrogen using powergen 700 homogenizer omni international marietta ga resuspended radioimmunoprecipitation assay lysis buffer upstate lake placid ny containing cocktail protease phosphatase inhibitors sigma st louis mo protein concentrations determined bio rad protein assay bio rad laboratories hercules ca protein extracts subjected electrophoresis immunoblotting using standard protocol antibodies included mouse anti phospho tyr20 tyr99 santa cruz biotechnology santa cruz ca rabbit polyclonal anti phospho kit y721 zymed lab inc san francisco ca rabbit anti kit oncogene science boston mouse anti actin santa cruz donkey anti mouse secondary antibody santa cruz anti rabbit secondary antibody calbiochem la jolla ca blots incubated immun star horseradish peroxidase luminol enhancer bio rad exposed kodak biomax mr film eastman kodak company rochester ny results phospho kit scored 1 strong 2 weak 3 negative fluorescence situ hybridization analysis touch preps frozen tumor tissue done nine nodules eight patients slides fixed 3 1 methanol acetic acid stored 20jc fluorescence situ hybridization fish done according standard procedures briefly slides pretreated pepsinhcl 0 007 mol l hcl 8 ag ml pepsin 37jc 3 5 minutes rinsed pbs fixated 1 formaldehyde 10 minutes rinsed dehydrated air dried slides denatured 70 formamide 68jc 2 4 minutes quenched dehydrated airdried kit probes used two overlapping bac clones ctd 3180g20 rp11 722f21 invitrogen labeled nick translation spectrum green vysis abbott laboratories il chromosome 4 centromeric probe labeled spectrum orange cep 4 vysis used reference probe mix 50 80 ng kit bac 2 al cot 1 dna invitrogen ethanol precipitated resuspended hybridization buffer kit probe mix denatured 70jc 10 minutes followed pre annealing 37jc 30 minutes kit probe combined denatured cep 4 probe slide coverslipped incubated overnight 37jc standard posthybridization washes slides stained 4v 6 diamidino 2 phenylindole mounted antifade vectashield vector laboratories analysis done using nikon e800 epifluorescence microscope metasystems isis 3 imaging software minimum 100 cells scanned separate regions slide results imatinib resistance may develop chronic imatinib therapy identified 31 patients gist treated imatinib subsequently underwent surgical resection 8 females 23 males median www aacrjournals org clin cancer res 2005 11 11 june 1 2005 4183 acquired imatinib resistance gist research downloaded clincancerres aacrjournals org may 29 2017 2005 american association cancer age initial diagnosis 56 range 31 84 years extent disease original diagnosis gist localized 19 patients metastatic 12 disease start imatinib therapy consisted primary tumor alone 6 patients primary gist liver peritoneal metastasis 7 patients recurrent liver peritoneal disease remaining 18 patients imatinib administered starting dose 400 600 mg per day based responsiveness disease imatinib time surgery categorized patients nonresistant primary resistance acquired resistance elective removal residual gist following either partial response stable disease imatinib done 14 nonresistant patients table 1 median duration imatinib therapy 8 1 22 months surgery done 3 patients primary resistance imatinib progression disease detected computed tomography within 3 months starting therapy table 2 15 patients developed acquired resistance either partial response n 10 stable disease n 5 response imatinib subsequently acquired resistance radiologic progression disease median length imatinib treatment patients acquired resistance 18 8 32 months table 2 one patients 14 first operation remove stable disease later underwent another resection remove resistant disease two patients required surgery complication imatinib therapy involving either bleeding perforation histologic response correlates clinical response imatinib primary tumor located stomach 8 patients small bowel 17 rectum 5 intra abdominal 1 patient median size primary gist 9 5 2 6 24 cm 13 tumors larger 10 cm morphology primary tumor spindle shaped 25 cases epithelioid 4 mixed 2 tumors histologic response graded based combined extent tumor necrosis fibrosis usually absent untreated gist although central necrosis may present large tumors patients primary resistance uniformly lacked histologic evidence response imatinib patients 29 31 table 2 six 14 patients nonresistant group histologic response 90 moderate histologic responses 50 90 occurred four others three nonresistant patients minimal histologic response despite clinical partial response stable disease contrast acquired resistant group two patients least one nodule lacking response table 2 secondary kit mutation occurs acquired imatinib resistance nonresistant group 11 patients kit mutation 3 exon 9 7 exon 11 1 exon 13 1 patient pdgfra mutation exon 18 2 wildtype identifiable secondary mutations nonresistant group however one patient 2 homozygous exon 11 deletion 6 15 nodules tested significance finding unclear patient stable disease currently evaluable tumor expected 3 patients primary resistance imatinib also single mutation involved kit exon 9 11 pdgfra exon 18 15 patients acquired resistance imatinib 14 common primary kit mutation 11 exon 11 3 exon 9 1 patient wild type secondary mutations clin cancer res 2005 11 11 june 1 2005 4184 www aacrjournals org table 1 patients nonresistant gist primary tumor location histology disease imatinib start time imatinib months best clinical response histologic response primary mutation secondary mutation phosphokit y721 1 sb p l22 pr 2 3 ex 11vyidptql569 576del absent 0 2 2 3 2ac sb p pe l15 pr 2 3 ex 11p551hmy552 554del absent 0 6 loh 3 6 1 2b l20 pr 1 2 ex 11p551hmy552 554del absent 0 3 loh 3 3 3 3a sb pe 12 sd 1 ex 9 ins502ay absent 0 1 na 3bc pe 18 sd 1 ex 9 ins502ay absent 0 3 1 3 4c sb p l3 sd 1 ex 9 ins502ay absent 0 3 1 5 rectum p 9 pr 4 wt absent 0 1 na 6 sb p 4 sd 2 ex 9 ins502ay absent 0 1 1 7 rectum p 6 pr 4 ex 11vqwkv555 559del absent 0 1 3 8 sb p 2 pr 1 ex 11k558n ins559q absent 0 1 na 9 sb pe l1 pr 3 4 wt absent 0 1 na 10 stomach p 8 pr 4 ex 11wk557 558del absent 0 1 na 11 stomach e pe 5 sd 2 3 pdgfra d842v absent 0 2 1 12 sb p pe l9 pr 4 ex 11ingnnyvyidptql563 576del absent 0 2 3 13 rectum p 8 pr 3 ex 13 k642e absent 0 1 2 14a sb pe 3 pr 4 ex 11vyidptql569 576del absent 0 1 2 abbreviations sb small bowel spindle e epithelioid p primary tumor pe peritoneal l liver pr partial response sd stable disease del deletion loh loss heterozygosity ins insertion na available time start imatinib surgical resection c tested kitamplification fish cancer therapy preclinical research downloaded clincancerres aacrjournals org may 29 2017 2005 american association cancer detected 10 tumor nodules samples seven 46 patients involved five different residues majority kit exon 17 mutations occurred six seven patients secondary mutations involved substitutions three n822k two d820y one y823d table 2 fig 1 two secondary mutations included substitutions exon 13 v654a exon 14 t670i identified loss wild type allele loss heterozygosity wk557 8del one patient 28 one three nodules tested one nonresistant patient 2 10 15 patients acquired resistance one nodule available molecular analysis two identical secondary mutation identified nodules tested three patients secondary mutation present one nodules interestingly one patient 24 nodule showed exon 13 v654a mutation whereas subsequent nodule removed 7 months later showed exon 17 www aacrjournals org clin cancer res 2005 11 11 june 1 2005 4185 table 2 patients either acquired primary resistance imatinib primary tumor location histology disease imatinib start time imatinib months best clinical response histologic response primary mutation secondary mutation phosphokit y721 14bc sb pe 29 pr 1 ex 11vyidptql569 576del d820y 2 2 2 15 rectum e cw 28 sd 1 ex 11kvvee558 562del loh y823d 1of 1 3 16c sb pe 13 pr 1 ex 11l576p absent 0 1 1 17c rectum pe l15 pr 1 ex 9 ins502ay absent 0 1 1 18 sb pe 26 sd 1 2 ex 9 ins502ay absent 0 4 1 2 19 stomach pe l27 pr 1 4 ex 11wk557 8del t670i 2 2 1 2 20 abd pe l13 pr 1 ex 11v559d absent 0 2 2 21 stomach e pe 8 sd 1 2 wt absent 0 1 2 22c stomach e pe 25 pr 2 4 ex 11k558n ins559p n822k 1of 3 2 23 sb l14 sd 1 ex 9 ins502ay absent 0 1 na 24a sb pe 9 pr 1 2 ex 11v560del v654a 1of 2 2 24b pe 16 pr 1 ex 11v560del n822k 1of 2 1 25 sb l32 sd 2 4 ex 11v559d d820y 1of 2 3 26 stomach e l20 pr 1 ex 11ins574tqlpyd absent 0 2 na 27 sb p l16 pr 1 3 ex 11ngnnyvyidptqlpy564 578del n822k 1of 2 1 28 stomach p pe l25 pr 1 4 ex 11wk557 558del absent 0 3 loh 1of 3 2 29 stomach e l12 primr 1 pdgfra d842v absent 0 1 na 30 sb p l3 primr 1 ex 11py577 558del absent 0 1 1 31c sb pe l8 primr 1 ex 9 ins502ay absent 0 2 1 abbreviations sb small bowel spindle e epithelioid p primary tumor pe peritoneal l liver pr partial response sd stable disease del deletion loh loss heterozygosity ins insertion na available abd intra abdominal primr primary resistance cw chest wall time start imatinib surgical resection ctested kitamplification fish fig 1 primary secondary kit mutations seven patients acquired resistance imatinib primary mutations occurred exon 11 tended deletions secondary mutations detected predominantly exon 17 also exons 13 14 wild type sequence shown comparison numbers first column refer individual patients acquired imatinib resistance gist research downloaded clincancerres aacrjournals org may 29 2017 2005 american association cancer n822k substitution remaining four patients primary mutation identified confirm secondary mutations found genomic level exclude possibility polyclonal resistance presence additional kit mutations missed amplification kit cdna exons 10 18 done followed subcloning direct sequencing total number 32 samples 18 patients analyzed including 2 preimatinib samples 13 nonresistant nodules 6 patients 14 samples 9 patients acquired resistance 3 nodules 2 primary resistance patients analysis included 10 nodules 5 patients acquired resistance second site mutations secondary mutations confirmed 7 7 nodules absence mutation 3 nodules additional kit mutations detected notably primary secondary mutations located allele kit activation variable regardless response imatinib kit activation measured phosphorylation heterogeneous consistently correlate histologic response response imatinib table 1 fig 2a surprisingly 11 operations 10 nonresistant patients tested three specimens lacked evidence kit activation meanwhile five nonresistant patients strong kit activation fig 2a three moderate expression 14 operations 13 patients acquired resistance group 6 patients strong kit activation 5 moderate expression 2 none notably kit activation also variable subset patients second mutation strong activation three moderate expression three detectable phospho kit two instances fact three patients 22 24 27 identical n822k secondary mutation either weak n 1 strong n 2 phosphokit expression also intraindividual variability patient 19 carrying identical t670i secondary mutation two nodules tested phospho kit strong weak fig 2b furthermore patient 24 distinct secondary kit mutations two metachronous nodules strong kit activation one weak activation fig 2b kit activation variability also noted within resistant nodules without secondary mutations fig 2c patients primary resistance strong phosphokit staining fig 2c table 2 inconsistent kit activation also observed untreated control group although phospho kit detected eight tumors tested weak pattern five including four gists kit exon 11 frame deletions strong three fig 2d three untreated gists identical 557 8 wk deletion showed weak expression mature isoform kit amplification frequent event imatinib resistance tissue fish analysis available nine nodules eight patients three nonresistant 2 3 4 four acquired resistance three secondary mutations 16 17 22 one carrying d820y 14b one patient primary resistance two nodules 31 copy number ratio kit cep 4 nine nodules tested differ normal control liver sample evidence kit amplification identified tumors tested fig 3 discussion attempted determine mechanism acquired resistance imatinib patients gist order identify strategies prevent delay development molecular analysis kit pdgfra mutations cohort patients treated imatinib underwent surgical resection found second site kit mutations clin cancer res 2005 11 11 june 1 2005 4186 www aacrjournals org fig 2 western blot analysis activated total kit nonresistant group b acquired resistance patients secondary mutations c acquired resistance patients without secondary mutations 16 17 primary resistance untreated gists kit exon11mutations compared negative control leiomyosarcoma cancer therapy preclinical research downloaded clincancerres aacrjournals org may 29 2017 2005 american association cancer common acquired resistance imatinib 15 patients responsive stable disease experienced disease progression 7 identifiable second mutation case first second mutation located kit gene fact located allele contrast identified two simultaneous kit pdgfra mutations 300 untreated gist tumors analyzed resistant tumors second mutation primary kit exon 11 mutation primary mutations mainly frame deletions one insertion one substitution mutation fig 1 eight second site mutations substitutions involved five different residues comparison incidence point mutations nontreated gist 15 five residues derived either first exons 13 14 second exon 17 kit kinase domain contrast incidence exons 13 14 17 mutations untreated gist 1 11 15 six secondary mutations located exon 17 amino acids 820 823 common n822k seen three cases although n822k d820y previously reported primary sporadic mutation 10 germ line mutation familial gist syndrome 21 respectively believe resistant nodules n822k d820y substitutions represent secondary mutation associated exon 11 mutation one multiple nodules analyzed patient carried mutation similar kit kinase mutations n822k y823d described recently subset seminomas 22 date primary mutations reported kit exon 13 k642e substitution 10 15 17 however identified v654a mutation exon 13 interestingly recent study chen et al 23 reported v654a mutation six resistant nodules taken five gist patients similarly second mutation study also found allele primary mutation however several differences findings chen et al 23 including incidence second mutations acquired resistance 100 versus 46 b location second mutations exon 13 versus different areas kit kinase domain c type primary mutation tumors developing second mutation exon 9 11 versus exon 11 mutations discrepancies might due limited number cases series although primary mutations previously described kit exon 14 recent report shown t670i substitution patient acquired resistance 24 one seven patients second mutation similar t670i mutation new activating kinase mutations also reported seven nine 78 gist patients unique pattern resistance defined radiologically nodule within mass thought represent progression clone resistant imatinib 25 recently crystal structures autoinhibited inactive active conformations kit kinase determined well structure kit kinase complex imatinib 26 structures provide insight mechanism normal regulation kit kinase function moreover provide explanations basis constitutive activation mutations found neoplasms gist critical feature kit kinase function role juxtamembrane domain kit receptor regulating kinase activity inactive autoinhibited state juxtamembrane domain kit inserts kinase active site thus disrupts formation active conformation critical residues interactions wk557 558 vv559 560 mutation residues disrupts inhibitory interactions destabilizes inactive autoinhibited conformation kit kinase thus diversity juxtamembrane domain mutations represents majority primary oncogenic mutations found gist activation loop loop kit kinase includes y823 pseudosubstrate kit kinase another major site oncogenic mutation kit kinase activation y823 becomes phosphorylated seems stabilize open active conformation loop presumably strong negative electrostatic interactions phosphate residue oncogenic activation loop mutations y823d mimic y823 phosphorylation thus stabilizing active conformation loop structure kit imatinib complex revealed similar bcr abl imatinib binds inactive conformation kinase although kitimatinib complex deviates somewhat autoinhibited inactive kit kinase conformation therefore surprising loop mutations generally inhibited imatinib although seem exceptions therefore two possible mechanisms resistance imatinib therapy may develop first second site mutations may stabilize active conformation kit kinase prevents imatinib binding alternatively second site mutations may specifically interfere imatinib binding without affecting overall kit kinase conformation findings agreement predictions first five second site mutations study located loop 26 tyr823 found substituted aspartic acid one resistant tumors mutations included d820y n822k although y823d mutation introduces tyrosine phosphate mimic fig 4 others may destabilize inactive conformation introduction positively charged side chain positively charged pocket formed cooh terminal lobe kinase second one mutation seems block imatinib binding kit t670i gatekeeper residue thr670 replaced isoleucine residue mutation disrupts important h bond imatinib kinase www aacrjournals org 4187 clin cancer res 2005 11 11 june 1 2005 fig 3 fish analysis patient acquired resistance patient 18 secondary mutation showed normal chromosome copy number kit green reference centromeric chromosome 4 red acquired imatinib resistance gist research downloaded clincancerres aacrjournals org may 29 2017 2005 american association cancer isoleucine methyl group protrudes imatinib binding site precluding proper imatinib binding kit kinase fig 4 investigation cml revealed predominant mechanism acquired resistance imatinib second site mutation bcr abl 19 20 bcr abl mutations described date resistant cml either alter imatinib binding site residues responsible unique conformational change permits imatinib binding prevents inactive conformation necessary imatinib binding 20 bcr abl kinase domain substitutions occurred 29 32 90 resistant tumors whereas genomic amplification bcr abl transcript infrequent similarly found secondary kit mutation patients acquired resistance however several distinct features acquired resistance cml gist table 3 first preexisting bcr abl kinase mutations found 2 4 patients 20 5 24 patients 27 respond imatinib furthermore bcr abl kinase mutations also found 4 13 patients chronic phase cml stable disease imatinib predicted subsequent clinical relapse 20 find secondary mutations 14 gist patients nonresistant disease shah et al 20 found polyclonal resistance two four clones harbored distinct mutant bcr abl kinase domains mutations 12 32 cml patients detect polyclonal resistance 10 samples five patients acquired resistance cdna subclones one tumor nodule revealed single type secondary mutation although find coexisting multiple second mutations patients acquired resistance one patient two different secondary mutations resistant nodules two different time points thus tumor nodules within patient may develop independent means imatinib resistance mechanism development second mutation gist unclear long duration imatinib therapy median 27 months patients acquired resistance developed second mutation makes less likely preexisting clone responsible acquired resistance fig 5 cml presence bcr abl kinase domain mutations pre imatinib samples patients primary resistance relapse patients blast phase within 1 3 months finding multiple mutations resistant patients suggest clonal selection preexisting mutations confer imatinib resistance gist possibility preexisting mutations remains evaluated carefully inability find secondary mutations eight resistant gist patients suggests gene mutations altogether clin cancer res 2005 11 11 june 1 2005 4188 www aacrjournals org fig 4 interaction imatinib kit kinase activation loop loop 809 831 highlighted red autoinhibitory domain 547 579 highlighted purple sites mutation thr670 tyr823 shown space filling representation yellow active form adp bound adp shown ball stick representation b inactive form imatinib bound imatinib shown ball stick representation figures generated programs molscript 35 raster3d 36 coordinates correspond reported mol et al 26 table 3 comparison imatinib resistance cml gist cml gistc acquired resistance kinase domain mutations 29 32 7 15 median duration imatinib therapy 20 monthsb 27 months gene amplification acquired resistance 3 11 0 4 activation transcribed protein 11of 11 heterogeneous primary resistance kinase domain mutations pre imatinib 7 28 0 4 nonresistant disease kinase domain mutations 4 13 0 14 note results listed number patients feature number tested based refs 19 20 27 ccurrent report bchronic phase cml cancer therapy preclinical research downloaded clincancerres aacrjournals org may 29 2017 2005 american association cancer different mechanisms may involved fletcher et al 28 suggested additional mechanisms resistance besides secondary kit mutations gist including kit genomic amplification activation alternative receptor tyrosine kinase protein absence kit expression found evidence kit genomic amplification five resistant patients based experimental data possibilities increased serum acid glycoprotein levels increased multidrug resistance gene expression 29 30 recently van oosterom colleagues 17 31 found preliminary evidence bioavailability imatinib may decrease chronic therapy possibly due regulation hepatic enzymes responsible drug clearance well documented cessation imatinib stable measurable disease results tumor proliferation 32 therefore subtherapeutic imatinib levels may also result tumor progression analysis phosphorylated kit protein yielded heterogeneous results signifying complexity kit activation treated gist strikingly majority nonresistant patients still activated kit patients acquired resistance group also phosphokit almost half strong expression finding suggests gist progression may still depend kit alternative inhibitors kit downstream pathway may therapeutic benefit nevertheless three patients acquired resistance whose tumors lacked detectable kit activation discernible pattern kit activation based histologic response type primary mutation presence second kit mutation even correlation phosphorylated kit total kit findings consistent duensing et al 33 found phosphorylated total kit varied substantially among tumors even within identical kit genotype data kit phosphorylation may confounded fact stopped imatinib generally 2 5 days prior surgery currently established guidelines perioperative use imatinib approach seems safe secondary mutations chronic imatinib use several implications clinical management patients advanced gist recommendations based supposition risk developing resistance proportional amount residual viable tumor patients imatinib therefore consider extirpation gross disease whenever possible indeed rationale operating patients report nonresistant fact several nonresistant patients minor histologic responses along activated kit underscores potential viable tumor progress argument may even made debulking removing much gross disease possible patients extensive residual disease imatinib therapy alternatively optimal management may include combined use imatinib newer molecular agents although detect second mutations nonresistant patients shah et al 20 found second mutations 4 13 patients stable cml predicted subsequent relapse patients stable gist may found second mutation biopsy certainly considered additional surgery molecular agents due likelihood impending progression course serial tumor assessment practical gist cml variety molecular inhibitors currently investigation advanced gist refractory imatinib furthest along su11248 pfizer inc new york ny inhibitor multiple tyrosine kinases seems effective imatinib resistant gist especially patients exon 9 mutations 34 summary report clinical molecular study acquired resistance imatinib gist found secondary mutations common imatinib resistance mutations tend single amino acid substitutions kit kinase domains occur particularly exon 17 secondary mutations seen pre imatinib nonresistant primary resistant tumors considerable heterogeneity kit activation responsiveness gist imatinib findings implications strategies treat avert imatinib resistance might useful design secondgeneration kinase inhibitors purpose resistance commonly acquired patients metastatic gastrointestinal stromal tumor treated imatinib mesylate often due development secondary mutations kit kinase domain sought investigate efficacy second line tyrosine kinase inhibitors sorafenib dasatinib nilotinib commonly observed imatinib resistant kit mutations kitv654a kitt670i kitd820y kitn822k expressed ba f3 cellular system experimental design vitro drug screening stable ba f3 kit mutants recapitulating genotype imatinib resistant patients harboring primary secondary kit mutations investigated comparison made imatinib sensitive ba f3 kit mutant cells well ba f3 cells expressing secondary kit mutations efficacy drug treatment evaluated proliferation apoptosis assays addition biochemical inhibition kit activation results sorafenib potent imatinib resistant ba f3 kit double mutants tested including gatekeeper secondary mutation kitwk557 8del t670i resistant kinase inhibitors although three drugs tested decreased cell proliferation inhibited kit activation exon 13 kitv560del v654a exon 17 kitv559d d820y double mutants nilotinib lower concentrations conclusions results emphasize need tailored salvage therapy imatinib refractory gastrointestinal stromal tumors according individual molecular mechanisms resistance ba f3 kitwk557 8del t670i cells sensitive sorafenib inhibition whereas nilotinib potent imatinib resistant kitv560del v654a kitv559d d820y mutant cells dasatinib sorafenib gastrointestinal stromal tumor gist common mesenchymal tumor gastrointestinal tract constitutive activation oncogenic mutations kit platelet derived growth factor receptor pdgfra receptor tyrosine kinase plays important role pathogenesis gists 1 2 imatinib mesylate gleevec novartis pharmaceuticals potently inhibits bcr abl pdgfra pdgfrb kit frontline therapy chronic myelogenous leukemia advanced gists since regulatory approval imatinib mesylate achieves clinical response 80 gist patients 3 4 directly dependent presence genomic location kit pdgfra mutations patients kit exon 11 mutations partial response rate 84 whereas tumors exon 9 mutation wild type kit less likely respond 5 6 although imatinib improved survival gist complete response rare furthermore clear majority patients initially benefit tyrosine kinase inhibitors eventually become resistant median time progression imatinib mesylate 2 years 7 initial response 46 67 patients develop imatinib resistance acquisition secondary mutation kit kinase domain 8 10 secondary mutations tend single amino acid substitutions exon 17 often also occur exons 13 14 8 mechanism development second site kit mutations remains unclear resistant patients identifiable secondary mutations treated imatinib longer resistant patients lacking secondary mutations median 27 versus 14 5 months ref 8 findings suggest clonal selection existing mutations imatinib therapy unlikely explain acquired resistance frequency secondary mutations also determined location primary kit mutations gists harboring kit exon 11 mutations commonly becoming imatinib resistant due acquisition secondary mutations compared kit exon 9 mutated gist one possible explanation difference kit exon 9 mutant gists able use alternative mechanisms whereas kit exon 11 gists dependent kit signaling 10 polyclonal acquired resistance result multiple secondary kit mutations geographically separate metastases reported 18 10 12 consistent secondary clonal evolution primary mutation always detectable tumor never one new mutation sample findings suggest long term imatinib therapy leads clonal selection resistant tumor subclones rna interference kit knockdown experiments revealed imatinib resistant cell lines carrying secondary kit mutations remain dependent kit signaling growth survival 10 alternative mechanisms low level genomic kit amplifications play minor role imatinib resistance gist 8 11 initial success imatinib targeted therapeutic drug better understanding drug kinase interaction based crystallographic structure led development several novel small molecule compounds variable spectrum kinase inhibition median survival 15 months imatinib resistance develops 13 important validate novel therapeutic strategies within specific kit pdgfra genotype confer imatinib resistance alternative agents imatinib tested clinical trials include nilotinib sorafenib dasatinib initial results trials still analyzed efficacy drugs might clear due heterogeneity imatinib resistant patients enrolled thus systematic vitro approach validating individual drug efficacy vis vis specific resistant genotypes might better guide therapy clinical trial design small molecules function kit kinase inhibitors addition inhibit downstream targets nilotinib amn107 designed based crystal structures imatinib abl kinase complex multitargeted kinase inhibitor features higher affinity atp binding site abl kinase override imatinib resistance caused bcr abl mutation 14 addition nilotinib selectively inhibits kit pdgfr tyrosine kinases 14 another small molecule showed great promise chronic myelogenous leukemia patients resistant intolerant imatinib dasatinib bms 354825 dasatinib atp competitor dual src abl activity inhibits bcr abl mutants binding active form kinase 15 furthermore dasatinib potently inhibits kit wild type juxtamembrane domain mutant kit 16 sorafenib bay 43 9006 nexavar novel biaryl urea compound initially developed specific inhibitor serine threonine kinase raf addition sorafenib activity receptor tyrosine kinases including vascular endothelial growth factor receptor 2 vascular endothelial growth factor receptor 3 pdgfrb kit 17 sorafenib shown broad spectrum dose dependent antitumor activity xenograft models human colon lung breast ovarian pancreatic cancers melanoma 18 based hypothesis individual kit mutant oncoproteins affect drug sensitivity investigated efficacy sorafenib nilotinib dasatinib ba f3 cells expressing various kit mutants recapitulate imatinib resistant gist genotypes materials methods ba f3 kit mutant transformants kit mutations recapitulating genotype found four imatinib resistant gist patients introduced site directed mutagenesis pcr using quikchange ii xl site directed mutagenesis kit qiagen inc retroviral expression vector containing wild type human kit cdna gnnk isoform pmscv wtkit ires gfp generous gift dr gary gilliland harvard medical school boston kit double mutant isoforms generated hosting primary kit exon 11 mutation secondary mutation kit exon 13 14 17 follows kitv560del v654a kitwk557 8del t670i kitv559d d820y kitv560del n822k single kit mutants carrying secondary mutation alone also generated comparison including kitt670i kitd820y kitn822k imatinib sensitive exon 11 single mutants kitv559d kitwk557 8del well imatinib less sensitive exon 9 kitay502 3ins used controls parental cell line ba f3 german collection microorganisms cell cultures human animal cell cultures collection interleukin 3 il 3 dependent murine pro b cell line lacking intrinsic kit expression retroviral vector plasmids containing mutant kit cdna isoforms 25 g construct hygromycin selection marker cotransfected ba f3 cells ratio 25 1 via electroporation using genepulser bio rad stable transfectants established triple selection electroporated cells first selected hygromycin 7 10 days remaining cells sorted according green fluorescent protein fluorescence selected green fluorescent protein positive cells grown 2 weeks presence hygromycin finally cells grown absence il 3 cell surface kit expression confirmed flow cytometry using pe conjugated anti cd117 bd biosciences antibody total kit expression confirmed western blotting using anti kit oncogene antibody cell proliferation assays determine growth inhibition drug effects ba f3 kit mutant cells starved growth factor 4 h fifteen minutes drug administration growth factors added 10 ng ml il 3 il 3 dependent ba f3 kit mutants kitd820y kitv560del n822k 20 ng ml kl remaining kl dependent ba f3 kit mutant isoforms cells incubated 10 10 000 nmol l sorafenib dasatinib nilotinib imatinib 37 c 48 h concentrations drug used studies line concentrations achievable human plasma based pharmacokinetic studies kinase inhibitors dasatinib nilotinib synthesized kindly provided dr bayard clarkson sloan kettering institute new york ny imatinib sorafenib purchased commercially bromodeoxyuridine added incubated 6 h harvesting cells fixed stained using standard protocol brdu apc kit pharmingen bromodeoxyuridine incorporation determined flow cytometry experiments done triplicate minimum 2 105 events acquired data analyzed using flowjo software version 5 7 2 cell growth inhibition curves ic50 values plotted calculated using graphpad prism software version 4 03 apoptosis assays induction apoptosis ba f3 kit transformants evaluated flow cytometry using annexin v pe apoptosis detection kit pharmingen duplicate experiments cells density 1 106 activated kl cultured 24 well plates starved growth factors 4 h drug delivery fifteen minutes adding drug growth factors added drug incubation 48 h using concentrations indicated cells harvested stained anti annexin v pe antibody using provider protocol minimum 10 000 events analyzed facscan becton dickinson within hour staining data results analyzed flowjo immunoprecipitation western blotting ba f3 cells expressing single double kit mutants treated indicated doses drug cells starved serum growth factors 2 h growth factors added back medium 15 min drug administration drugs incubated 37 c absence serum 90 min treatment cells harvested subjected protein extraction whole cell lysate 200 g incubated 2 g anti kit antibody assay designs inc hour mixture incubated magna beads pierce biotechnology overnight 4 c beads washed isolated using magnetic column resuspended lds sample buffer invitrogen electrophoresis immunoblotting done protein extracts using standard protocol phosphorylated kit detected anti phosphotyrosine antibodies py20 py99 santa cruz biotechnology total kit detected mouse monoclonal anti c kit santa cruz biotechnology secondary antibodies used donkey anti mouse secondary santa cruz biotechnology results test existing second line inhibitors imatinib resistant kit oncoproteins developed stable ba f3 transfectants expressing single double mutant isoforms selected il 3 independent growth eight 10 kit mutant forms induced transformation conferring factor independent growth ba f3 cells kitd820y kitv560del n822k ba f3 mutants achieve il 3 independence ba f3 transfectants expressing double kit mutants kitv560del v654a kitwk557 8del t670i kitv559d d820y kitv560del n822k single kit mutant isoforms kitt670i kitn822k tested sorafenib dasatinib nilotinib imatinib efficacy drug compared imatinib sensitive kit juxtamembrane extracellular domain mutations ba f3 kitv559d kitwk557 8del kitay502 3ins dasatinib potent kinase inhibitors imatinib sensitive ba f3 kit mutants vitro cell proliferation assay showed dasatinib inhibited cell growth imatinib sensitive ba f3 kitv559d ic50 27 nmol l whereas imatinib nilotinib sorafenib ic50 63 44 66 nmol l respectively dasatinib results summarized table 1 dasatinib induced apoptosis 20 cells 10 nmol l whereas drugs induced programmed cell death 100 nmol l biochemical assays showed consistent kitv559d phosphorylation inhibition four inhibitors tested dasatinib potent kit kinase activity distinctly inhibited using 10 nmol l dasatinib whereas imatinib nilotinib sorafenib showed results 100 nmol l view inlineview popup table 1 comparison vitro cell proliferation inhibition imatinib resistant kit single double mutants expressed ba f3 cell system similarly growth kitwk557 8del cells inhibited dasatinib ic50 58 nmol l whereas imatinib nilotinib sorafenib ic50 460 83 211 nmol l respectively fig 1 furthermore significant induction apoptosis noted 100 nmol l dasatinib nilotinib fig 1 download figureopen new tabdownload powerpoint fig 1 imatinib sensitive juxtamembrane mutation kitwk557 8del showing sensitivity drugs tested however dasatinib nilotinib inhibited cell growth ic50 58 83 nmol l respectively induced significant apoptosis 100 nmol l b 7 aad 7 aminoactinomycin dasatinib inhibited imatinib less sensitive ba f3 kitay502 3ins ic50 74 nmol l whereas imatinib nilotinib sorafenib ic50 509 671 400 nmol l respectively dasatinib induced significant 90 programmed cell death kitay502 3ins cells 100 nmol l whereas nilotinib induced 65 sorafenib induced 35 programmed cell death concentration imatinib induce overt apoptosis 1 000 nmol l keeping cell proliferation inhibition apoptosis results imatinib inhibit kit phosphorylation 100 nmol l showed distinct inhibition kit activity 1 000 nmol l imatinib resistant gatekeeper single kitt670i double kitwk557 8del t670i mutants responsive sorafenib resistant dasatinib nilotinib sorafenib drug inhibited cell proliferation ba f3 kitwk557 8del t670i transfectant ic50 1 063 nmol l induction significant programmed cell death 1 000 5 000 nmol l fig 2 sorafenib also showed distinct inhibition kinase activity kl dependent double mutant kitwk557 8del t670i isoform 1 000 nmol l contrast imatinib dasatinib nilotinib inhibit phosphorylation even 10 000 nmol l similar experiments done ba f3 cells expressing gatekeeper single mutant kitt670i showed comparable drug sensitivity profile sorafenib significantly inhibited kinase activity single mutant 1 000 nmol l inhibited cell growth ic50 918 nmol l induced significant apoptosis 1 000 5 000 nmol l whereas imatinib nilotinib dasatinib show effects biochemical biological assays fig 2 fig 2 download figureopen new tabdownload powerpoint fig 2 sorafenib potently inhibited cell proliferation kinase activity induced overt apoptosis ba f3 transfectant expressing kitwk557 8del t670i mutant imatinib dasatinib nilotinib showed effect cell growth kit kinase activity ba f3 cells stably expressing kitwk557 8del t670i treated escalating doses sorafenib presence 20 ng ml kl 48 h cellular proliferation determined bromodeoxyuridine brdu incorporation assays inhibited sorafenib ic50 1 063 nmol l drugs show growth inhibition data shown 10 000 nmol l b apoptosis assays measured annexin v 7 aminoactinomycin staining showed sorafenib induced significant apoptosis 1 000 nmol l imatinib nilotinib dasatinib induce overt apoptosis 5 000 nmol l c immunoblotting phosphorylated py20 py99 total forms kit done evaluate inhibitory effect drugs kit activation sorafenib showed marked kit activation inhibition 1 000 nmol l whereas three drugs show inhibitory effect 10 000 nmol l nilotinib inhibited cell growth induced apoptosis imatinib resistant kitv560del v654a mutant cells lower doses dasatinib sorafenib nilotinib potently inhibited proliferation ba f3 kitv560del v654a cells ic50 192 nmol l double mutant also sensitive higher concentrations dasatinib sorafenib inhibition ic50 585 1 074 nmol l respectively fig 3a apoptosis assays showed similar results three inhibitors interestingly imatinib showed induction apoptosis 50 cells 1 000 nmol l whereas 5 000 nmol l 90 cells entered programmed cell death fig 3b biochemically nilotinib dasatinib sorafenib inhibited kit phosphorylation 1 000 nmol l imatinib partially inhibited phosphorylation kitv560del v654a oncoprotein 1 000 nmol l completely inhibited 5 000 nmol l fig 3c fig 3 download figureopen new tabdownload powerpoint fig 3 sorafenib dasatinib nilotinib showed significant inhibition cell growth ba f3 cells expressing kitv560del v654a 1 000 nmol l lower imatinib showed moderate effects 1 000 5 000 nmol l b nilotinib induced significant apoptosis 100 1 000 nmol l whereas dasatinib sorafenib imatinib 1 000 5 000 nmol l c nilotinib dasatinib sorafenib showed significant inhibition kit activation 1 000 nmol l whereas imatinib 5 000 nmol l imatinib resistant kitv559d d820y sensitive nilotinib dasatinib sorafenib inhibition three inhibitors tested decreased cell proliferation imatinib resistant ba f3 kitv559d d820y cells see table 1 nilotinib inhibited cell growth ic50 297 nmol l whereas sorafenib dasatinib similar effects slightly higher dose ic50 944 432 nmol l respectively dasatinib induced significant apoptosis 1 000 nmol l whereas nilotinib sorafenib showed moderate effect induction apoptosis 1 000 5 000 nmol l biochemically three drugs showed distinct inhibition kit phosphorylation 1 000 nmol l phosphorylation imatinib resistant kitv560del n822k inhibited nilotinib lower concentrations dasatinib sorafenib ba f3 kitv560del n822k cells achieve il 3 independence biochemically nilotinib completely inhibited kit phosphorylation 100 nmol l concentration whereas dasatinib sorafenib 1 000 nmol l data shown proliferation apoptosis studies done il 3 dependent cells show effect drugs tested result might related il 3 dependent signaling pathways targeted drugs tested discussion although gist patients respond imatinib initially often develop progressing lesions multiple metastatic sites within 2 5 years starting therapy imatinib resistant gists kit reactivated downstream signaling pathways remain kit dependent 8 11 half imatinib resistant patients development second site kit mutation cause kit reactivation 8 11 12 whereas remaining cases largely unknown second site mutations occur without exception allele primary mutation affect either first second kinase domains exons 13 14 17 mostly single amino acid substitutions 8 10 crystallographic studies showed imatinib binds inactive form kit 19 secondary kit mutations kinase domains stabilize active form kit thus weaken prevent drug binding leading clinical resistance imatinib alternatives presently tested clinical trials show limited survival benefit 13 urgent need validate novel therapeutic strategies tailored mechanism resistance individual patients shown several groups location primary kit pdgfra mutation directly predicts clinical response imatinib gist patients tumors underlying primary kit mutation exon 11 represent subgroup better response rates mutational subtypes 5 6 furthermore group also higher risk develop secondary mutations compared kit exon 9 mutated gists 8 11 12 observation supports probability developing secondary mutation increases duration imatinib treatment often longer gists harboring exon 11 mutation exon 9 mutation wild type gists however presence secondary kit mutations per se within acquired imatinib resistance setting indicator worse prognosis compared imatinib refractory tumors without secondary mutations latter group escapes inhibitory efficacy imatinib yet unresolved molecular mechanisms 12 another level complexity relies fact long term imatinib therapy leads clonal selection distinct resistant tumor subclones called polyclonal acquired resistance 10 12 thus tumor nodule progression may undergo individual clonal evolution resulting multiple secondary mutations developed different metastatic sites within patient designing salvage strategies imatinib resistant settings address inhibition known genomic activating mutations oncoprotein prior vitro drug screening studies imatinib resistant kit mutants investigate one drug time 9 20 see table 2 thus comparing results different laboratories using different cell systems primary gist cells ba f3 hek etc quite difficult study undertakes detailed investigation three imatinib alternative small molecules large panel ba f3 kit single double mutant cells using biological biochemical methods assessing drug response view inlineview popup table 2 imatinib dasatinib nilotinib drug sensitivity previously reported imatinib sensitive imatinib resistant kit mutant oncoproteins previous reports 12 suggested gists secondary kit exon 14 gatekeeper mutations t670i show aggressive behavior earlier metastasis shorter progression free survival interest kitt670i mutation identified primary mutation gist occurs selective pressure imatinib therapy gatekeeper amino acid t670 kit identified one key hydrogen bonds imatinib binding 19 21 thus mutations position profound effect drug binding bulky isoleucine substituting threonine residue abolishes affinity imatinib also dasatinib 22 23 similarly gatekeeper mutation bcr abl t315i resistant nilotinib 10 000 nmol l 14 keeping findings ba f3 kit cells expressing t670i single double mutant insensitive drugs except sorafenib sorafenib initially designed inhibit cell proliferation targeting raf mitogen activated protein kinase extracellular signal regulated kinase kinase extracellular signal regulated kinase pathway tumor angiogenesis targeting vascular endothelial growth factor receptor 2 pdgfrb 17 recently approved food drug administration treatment advanced renal cell carcinoma sorafenib also shown interact kit inhibiting kit kinase activity vitro kinase assays 18 data support ongoing phase ii national cancer institute sponsored multicenter investigation sorafenib imatinib sunitinib resistant gist clinicaltrials gov identifier nct00265798 however biochemical biological consequences sorafenib inhibition context kit mutant cells yet investigated importantly results show sorafenib inhibits kitt670i gatekeeper mutation either expressed single double mutant oncoprotein ba f3 cellular system thus sorafenib inhibited kit kinase activity cell growth well induced apoptosis 1 000 nmol l concentration ba f3 kitwk557 8del t670i double mutant ba f3 kitt670i single mutant cells fact sorafenib drug potent different secondary mutations tested including kit exons 13 17 dasatinib potent orally bioavailable inhibitor several kinases including bcr abl src family kinases 24 recently approved food drug administration treatment chronic myelogenous leukemia philadelphia chromosome positive resistance intolerance previous therapy including imatinib 25 dasatinib previously shown potent imatinib inhibiting kitv559d 23 kitv560g 16 wild type kit 15 results also indicate dasatinib effective kinase inhibitors imatinib sensitive exon 11 mutant ba f3 kitv559d kitwk557 8del cells schittenhelm et al 16 reported dasatinib inhibits kinase activity kitv560g d816v double mutant cells dose dependent manner screening sensitivity different activating tyrosine kinase oncoproteins carrying gatekeeper mutation carter et al 23 showed dasatinib effect kitv559d t670i double mutant results also indicate similar gatekeeper mutations bcr abl t315i egfr t670m oncoproteins resistant dasatinib therapy data support kitt670i expressed either single double mutant oncoprotein ba f3 cells confers resistance dasatinib contrast ba f3 cells expressing kit exon 13 exon 17 single double mutants sensitive dasatinib inhibition dasatinib recently examined phase study gist patients enrolled clinicaltrials gov identifier nct00099606 final data yet published nilotinib specifically designed potent inhibitor mutant wild type bcr abl compared imatinib 14 addition weisberg et al 14 showed nilotinib effective imatinib inhibiting exon 11 kitv560del exon 13 kitk642e mutant ba f3 cells different study nilotinib showed potent inhibition kitv560g mutant cells 26 previously reported results indicate nilotinib efficacy kit double mutants involving exon 11 exons 13 17 specifically nilotinib induced cell growth inhibition imatinib resistant kitv560del v654a kitv559d d820y mutant cells lower concentrations dasatinib sorafenib contrast nilotinib effective gatekeeper kit exon 14 single kitt670i double kitwk557 8del t670i mutant similarly weisberg et al 14 showed although nilotinib inhibited secondary mutations bcr abl effect inhibiting t315i gatekeeper mutation analogous t670i mutation kit nilotinib investigated combination imatinib patients imatinib resistant clinicaltrials gov identifier nct00135005 final results study pending previous preclinical data using ba f3 cellular system showed imatinib resistant kit v654a t670i mutants sensitive sunitinib su11248 ref 20 growth ba f3 cells expressing kitwk557 8del v654a kitwk557 8del t670i completely inhibited sunitinib concentrations 500 nmol l ic50 50 100 nmol l table 2 although imatinib resistant kit mutants common second kinase domain kit mutations investigated ba f3 cells expressing pdgfra d842v less sensitive sunitinib growth proliferation inhibited 5 mol l anecdotal evidence based mainly molecular modeling mutated kit receptors suggests secondary mutations acquired imatinib resistance setting might weaker others 27 although second site kit mutations affect imatinib binding site extent structural changes conferred varies individual amino acid substitution thus t670i substantially modifies binding pocket whereas v654a induces relatively confined changes 8 27 compared biochemical activity kit kinase tamborini et al 27 showed substitutions phosphorylation receptor persisted presence 3 mmol l imatinib keeping activated status kit receptors however single double mutant kitv654a oncoprotein expressed cos cells seemed unphosphorylated 6 mmol l imatinib whereas single double mutant kitt670i results suggest effects v654a mutation might subverted dose escalation imatinib would expected clinical response would restored results support findings showing imatinib partially inhibited phosphorylation kitv560del v654a oncoprotein 1 000 nmol l completely inhibited 5 000 nmol l similarly induction apoptosis imatinib evident 1 000 nmol l 50 cells whereas 5 000 nmol l cells entered programmed cell death alternative salvage options inhibiting kit signaling pathway presently explored strategies promise therapeutic solution challenge heterogeneous imatinib resistance bauer et al 28 investigated activity heat shock protein 90 inhibitor 17 allylamino 18 demethoxy geldanamycin 17 aag enhancing cellular degradation constitutively activated kit oncoprotein thus 17 aag effective inhibiting kit dependent imatinib sensitive imatinib resistant gist cell lines kit independent gist cells suggesting effects result mainly inactivation kit oncoprotein taking different approach sambol et al 29 showed efficacy flavopiridol cyclin dependent kinase inhibitor primary gist cell line carrying imatinib sensitive kitk642e exon 13 mutation induction apoptosis transcriptional regulation kit possible combination targeted kinase inhibitors drugs flavopiridol 17 aag may effective setting polyclonal acquired resistance conclusion second line tyrosine kinase inhibitors selectively potent imatinib resistant kit mutations might provide effective alternative imatinib management patients specifically sorafenib found effective kit double mutants tested including secondary mutation gatekeeper residue kitt670i resistant kinase inhibitors contrast imatinib resistant kitv560del v654a kitv559d d820y mutant cells inhibited three inhibitors tested nilotinib showing lower ic50 dasatinib sorafenib vitro data set stage designing future clinical trials selection second generation kinase inhibitor tailored based individual mechanism imatinib resistance presence kitt670i gatekeeper mutation example resistant disease may indicate sorafenib sunitinib next therapeutic option whereas secondary mutations may susceptible nilotinib gastrointestinal stromal tumors gists common mesenchymal tumors human digestive tract molecular etiology cellular origin unknown sequencing c kit complementary dna encodes proto oncogenic receptor tyrosine kinase kit five gists revealed mutations region transmembrane tyrosine kinase domains corresponding mutant kit proteins constitutively activated without kit ligand stem cell factor scf stable transfection mutant c kitcomplementary dnas induced malignant transformation ba f3 murine lymphoid cells suggesting mutations contribute tumor development gists may originate interstitial cells cajal iccs development iccs dependent scf kit interaction like gists cells express kit cd34 c kit proto oncogene encodes type iii receptor tyrosine kinase kit 1 ligand scf 2 scf kit interaction essential development melanocytes erythrocytes germ cells mast cells iccs 3 4 gain function mutations c kit gene found several tumor mast cell lines rodents humans 5 6 mast cell tumors humans 7 investigate mutational status c kit mesenchymal tumors human gastrointestinal gi tract collected 58 mesenchymal tumors developed gi wall 4 esophagus 36 stomach 14 small intestine 4 large intestine kit expression examined immunohistochemistry 8 eight authentic leiomyomas authentic schwannoma express kit remaining 49 mesenchymal tumors diagnosed gastrointestinal stromal tumors gists 94 46 49 expressed kit examination tumors expression cd34 reliable marker gists 9 revealed 82 40 49 cd34 positive 78 38 49 positive kit cd34 fig 1 three five kit negative gists also cd34 negative download high res image open new tab download powerpoint figure 1 coexpression kit cd34 human gists c serial sections tumor developed muscle layers stomach tof higher magnification c stained hematoxylin eosin boundary tumor indicated arrows b e stained anti kit c f anti cd34 8 arrowheads c indicate endothelial cells blood vessels express cd34 kit lm longitudinal muscle layer gto coexpression kit cd34 another gist also stomach demonstrated confocal laser scanning microscopy 8 g binding rabbit anti kit tumor cells demonstrated fitc labeled anti rabbit igg green h binding mouse anti cd34 tumor cells demonstrated rpe labeled anti mouse igg red merged confocal image g h showing coexpression kit cd34 tumor cells yellow bars c 50 100 compared immunohistochemical characteristics gists iccs cells regulate autonomous contraction gi tract 4 iccs located near circular muscle layer stomach 10 small intestine 11 large intestine 12 iccs small intestine surrounding myenteric ganglion cells easily identified specific localization examined immunohistochemical characteristics cells found double positive kit cd34 fig 2 g iccs circular muscle layer stomach small intestine iccs myenteric plexus region circular muscle layer large intestine also double positive kit cd34 download high res image open new tab download powerpoint figure 2 coexpression kit cd34 iccs surrounding myenteric ganglion cells normal human small intestine b serial sections normal human small intestine c higher magnification b respectively c stained anti kit b stained anti cd34 8 localization kit cd34 double positive cells consistent iccs present circular muscle layer cm longitudinal muscle layer lm surrounding myenteric ganglion cells gc 11 e g demonstration confocal laser scanning microscopy 8 e binding rabbit anti kit f binding mouse anti cd34 g merged confocal image e f arrows f g show endothelial cells fibroblast like cells express cd34 kit bars g 50 obtained complete coding region c kit cdna six gists control tissues using reverse transcriptase polymerase chain reaction rt pcr 13 ten independent c kit clones obtained sample 5 6 gists gist 1 5 4 6 clones 10 examined showed mutations region transmembrane tyrosine kinase domains hereafter called juxtamembrane domain fig 3 mutations located within 11 amino acid stretch lys 550 val 560 nonidentical sites mutations detectable domains c kit cdna 50 cdna clones gist show mutations conclude one two c kit alleles mutated case direct sequencing pcr products confirmed mutations juxtamembrane domain c kit cdna five gists download high res image open new tab download powerpoint figure 3 mutations c kitin gists gist 1 showed frame deletion 6 base pairs bp gist 2 showed frame deletion 15 bp gist 3 showed frame deletion observed gist 2 additional point mutation codon 550 aaa ata resulted lys550 ile substitution gist 4 showed point mutation codon 559 gtt gat resulted val559 asp substitution gist 5 showed frame deletion 27 bp c kit mutations juxtamembrane domain hmc 1 human mast cell leukemia cell line 5 fma3 murine mastocytoma cell line 6 shown comparison deleted amino acids shown dashes mutated amino acids boxes murine human kit different lengths 1 amino acid numbering fma3 kit different abbreviations used follows sp signal peptide ec extracellular domain tm transmembrane domain jm juxtamembrane domain tk1 tk2 tyrosine kinase domains ki kinase insert abbreviations amino acid residues follows asp e glu g gly h ile k lys l leu met n asn p pro q gln thr v val w trp tyr next examined whether c kit mutations found gists resulted constitutive activation c kit receptor tyrosine kinase transient introduction mutant c kit cdnas 293t human embryonic kidney hek cell line 5 14 15 wild type c kit cdna introduced negative control tyrosine kinase domain mutant juxtamembrane domain mutant found hmc 1 human mast cell leukemia cell line 5 introduced positive controls wild type kit phosphorylated tyrosine recombinant human rh scf added culture medium fig 4a contrast gain function kit mutants found hmc 1 cells phosphorylated tyrosine without addition rhscf reported previously 5 magnitude constitutive tyrosine phosphorylation greater tyrosine kinase domain mutant juxtamembrane domain mutant c kitmutants found gists also showed constitutive tyrosine phosphorylation 293t cells without rhscf fig 4a constitutive tyrosine phosphorylation juxtamembrane mutant hmc 1 cells similar magnitude juxtamembrane mutants gists vitro kinase assays 16 c kit mutants found gists exhibited constitutive kinase activation similar magnitude juxtamembrane domain mutant hmc 1 cells 5 fig 4b download figure open new tab download powerpoint download figure open new tab download powerpoint figure 4 constitutive activation mutant kit found gists mutant c kit cdnas found five gists transfected 293t cells human wild type c kit cdna mutant c kit cdnas found hmc 1 cells 6 21 also transfected controls tyrosine phosphorylation examined without rhscf stimulation 15 b immune complex kinase assay carried without rhscf stimulation 16 investigate biological consequences mutant c kit introduced c kit mutations found gists mouse c kit cdna 17 stably transfected cdna interleukin 3 il 3 dependent ba f3 murine lymphoid cell line 18 control mouse wild type c kit cdna also transfected ba f3 cells estimated ba f3 cell proliferation using mtt colorimetric assay 19 ba f3 cells wild type murine c kitgrew presence either recombinant mouse rm il 3 rmscf ba f3 cells mutated murine c kit grew autonomously without rmil 3 rmscf fig 5 b ba f3 cells mutated murine c kit also grew autonomously nude mice fig 5c 20 constitutive kinase activation kit mutants found five gists confirmed ba f3 cells fig 5d 5 21 download figure open new tab download powerpoint download figure open new tab download powerpoint figure 5 autonomous proliferation ba f3 cells transfected mutated murine c kit culture nude mice b mtt colorimetric assay 19 presence rmil 3 rmscf b data expressed mean four wells untransfected ba f3 cells ba f3 cells murine wild type c kit ba f3 cells murine mutated c kit gist 1 gist 2 gist 3 gist 4 gist 5 x c development tumors nude mice 20 injection ba f3 cells mutated c kit original ba f3 cells ba f3 cells murine wild type c kit form tumors data expressed mean results five mice immune complex kinase assay 16 ba f3 cells various c kit cdnas cultured without rmscf although various cells including hematopoietic stem cells express kit cd34 22 iccs cells double positive kit cd34 normal gi wall humans strongly suggests kit cd34 double positive gists might originate iccs although cannot exclude possibility iccs gists simply show common undifferentiated characteristics observed multipotential hematopoietic stem cells mechanism kit becomes constitutively activated appears different tyrosine kinase domain mutant juxtamembrane domain mutant 6 21 former constitutively activated without forming dimers 21 whereas latter constitutively dimerizes without binding scf 6 21 tyrosine kinase domain mutation kit found mast cell neoplasms 7 juxtamembrane domain mutation gists mechanisms different mutations cause malignant transformation different cell types remain investigated\n"],"name":"stdout"}]},{"metadata":{"id":"QYdfuykTcK1V","colab_type":"text"},"cell_type":"markdown","source":["##**Preprocessed Data**"]},{"metadata":{"id":"c1cdd_TTe0o2","colab_type":"code","outputId":"54a6e32b-9f67-4978-dfd0-36188f2437a7","executionInfo":{"status":"ok","timestamp":1543845941018,"user_tz":-330,"elapsed":767,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":206}},"cell_type":"code","source":["result = pd.merge(data, data_text,on='ID', how='left')\n","\n","result.Gene      = result.Gene.str.replace('\\s+', '_')\n","result.Variation = result.Variation.str.replace('\\s+', '_')\n","\n","result.head()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>Gene</th>\n","      <th>Variation</th>\n","      <th>Class</th>\n","      <th>TEXT</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>0</td>\n","      <td>FAM58A</td>\n","      <td>Truncating_Mutations</td>\n","      <td>1</td>\n","      <td>cyclin dependent kinases cdks regulate variety...</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>1</td>\n","      <td>CBL</td>\n","      <td>W802*</td>\n","      <td>2</td>\n","      <td>abstract background non small cell lung cancer...</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>2</td>\n","      <td>CBL</td>\n","      <td>Q249E</td>\n","      <td>2</td>\n","      <td>abstract background non small cell lung cancer...</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>3</td>\n","      <td>CBL</td>\n","      <td>N454D</td>\n","      <td>3</td>\n","      <td>recent evidence demonstrated acquired uniparen...</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>4</td>\n","      <td>CBL</td>\n","      <td>L399V</td>\n","      <td>4</td>\n","      <td>oncogenic mutations monomeric casitas b lineag...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["   ID    Gene             Variation  Class  \\\n","0   0  FAM58A  Truncating_Mutations      1   \n","1   1     CBL                 W802*      2   \n","2   2     CBL                 Q249E      2   \n","3   3     CBL                 N454D      3   \n","4   4     CBL                 L399V      4   \n","\n","                                                TEXT  \n","0  cyclin dependent kinases cdks regulate variety...  \n","1  abstract background non small cell lung cancer...  \n","2  abstract background non small cell lung cancer...  \n","3  recent evidence demonstrated acquired uniparen...  \n","4  oncogenic mutations monomeric casitas b lineag...  "]},"metadata":{"tags":[]},"execution_count":18}]},{"metadata":{"id":"aiNOJdnVJpVH","colab_type":"code","outputId":"c3758cea-2638-47ae-e5a1-b7c891150bb8","executionInfo":{"status":"ok","timestamp":1543845944821,"user_tz":-330,"elapsed":769,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":34}},"cell_type":"code","source":["result['Gene'].nunique()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/plain":["264"]},"metadata":{"tags":[]},"execution_count":19}]},{"metadata":{"id":"8islvoZWKKBb","colab_type":"code","outputId":"8c40485c-e892-4423-d57b-b56627a35eee","executionInfo":{"status":"ok","timestamp":1543845947477,"user_tz":-330,"elapsed":727,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":34}},"cell_type":"code","source":["result['Variation'].nunique()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/plain":["2996"]},"metadata":{"tags":[]},"execution_count":20}]},{"metadata":{"id":"9cD2CSvZZQBj","colab_type":"code","outputId":"92dfe572-5685-4e10-f136-9449db3053b3","executionInfo":{"status":"ok","timestamp":1543846048581,"user_tz":-330,"elapsed":1204,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":34}},"cell_type":"code","source":["result['Class'].nunique()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/plain":["9"]},"metadata":{"tags":[]},"execution_count":23}]},{"metadata":{"id":"7NHFkSppoig3","colab_type":"code","outputId":"5b9fcdd5-0fd4-488e-9e9c-6b8352fde304","executionInfo":{"status":"ok","timestamp":1543824809295,"user_tz":-330,"elapsed":2730,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":206}},"cell_type":"code","source":["result.loc[:, 'Text_count']  = result[\"TEXT\"].apply(lambda x: len(x.split()))\n","result.head()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>Gene</th>\n","      <th>Variation</th>\n","      <th>Class</th>\n","      <th>TEXT</th>\n","      <th>Text_count</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>0</td>\n","      <td>FAM58A</td>\n","      <td>Truncating_Mutations</td>\n","      <td>1</td>\n","      <td>cyclin dependent kinases cdks regulate variety...</td>\n","      <td>4370</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>1</td>\n","      <td>CBL</td>\n","      <td>W802*</td>\n","      <td>2</td>\n","      <td>abstract background non small cell lung cancer...</td>\n","      <td>4139</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>2</td>\n","      <td>CBL</td>\n","      <td>Q249E</td>\n","      <td>2</td>\n","      <td>abstract background non small cell lung cancer...</td>\n","      <td>4139</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>3</td>\n","      <td>CBL</td>\n","      <td>N454D</td>\n","      <td>3</td>\n","      <td>recent evidence demonstrated acquired uniparen...</td>\n","      <td>3841</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>4</td>\n","      <td>CBL</td>\n","      <td>L399V</td>\n","      <td>4</td>\n","      <td>oncogenic mutations monomeric casitas b lineag...</td>\n","      <td>4254</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["   ID    Gene             Variation  Class  \\\n","0   0  FAM58A  Truncating_Mutations      1   \n","1   1     CBL                 W802*      2   \n","2   2     CBL                 Q249E      2   \n","3   3     CBL                 N454D      3   \n","4   4     CBL                 L399V      4   \n","\n","                                                TEXT  Text_count  \n","0  cyclin dependent kinases cdks regulate variety...        4370  \n","1  abstract background non small cell lung cancer...        4139  \n","2  abstract background non small cell lung cancer...        4139  \n","3  recent evidence demonstrated acquired uniparen...        3841  \n","4  oncogenic mutations monomeric casitas b lineag...        4254  "]},"metadata":{"tags":[]},"execution_count":28}]},{"metadata":{"id":"rH3NX_locnIE","colab_type":"text"},"cell_type":"markdown","source":["###**Therir are 5 entries in the data having no text available but provided with Labels**"]},{"metadata":{"id":"msZQqHq-D6uA","colab_type":"code","outputId":"09671cc0-2e1d-443e-a82c-371360b6759f","executionInfo":{"status":"ok","timestamp":1543824813236,"user_tz":-330,"elapsed":891,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":206}},"cell_type":"code","source":["result[ result['Text_count'] == 1]"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>Gene</th>\n","      <th>Variation</th>\n","      <th>Class</th>\n","      <th>TEXT</th>\n","      <th>Text_count</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>1109</th>\n","      <td>1109</td>\n","      <td>FANCA</td>\n","      <td>S1088F</td>\n","      <td>1</td>\n","      <td>nan</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>1277</th>\n","      <td>1277</td>\n","      <td>ARID5B</td>\n","      <td>Truncating_Mutations</td>\n","      <td>1</td>\n","      <td>nan</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>1407</th>\n","      <td>1407</td>\n","      <td>FGFR3</td>\n","      <td>K508M</td>\n","      <td>6</td>\n","      <td>nan</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>1639</th>\n","      <td>1639</td>\n","      <td>FLT1</td>\n","      <td>Amplification</td>\n","      <td>6</td>\n","      <td>nan</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>2755</th>\n","      <td>2755</td>\n","      <td>BRAF</td>\n","      <td>G596C</td>\n","      <td>7</td>\n","      <td>nan</td>\n","      <td>1</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["        ID    Gene             Variation  Class TEXT  Text_count\n","1109  1109   FANCA                S1088F      1  nan           1\n","1277  1277  ARID5B  Truncating_Mutations      1  nan           1\n","1407  1407   FGFR3                 K508M      6  nan           1\n","1639  1639    FLT1         Amplification      6  nan           1\n","2755  2755    BRAF                 G596C      7  nan           1"]},"metadata":{"tags":[]},"execution_count":29}]},{"metadata":{"id":"DhXgSaYUHpEW","colab_type":"code","colab":{}},"cell_type":"code","source":["result.__delitem__('Text_count')"],"execution_count":0,"outputs":[]},{"metadata":{"id":"oIYS-5WuOs6n","colab_type":"code","outputId":"9a336bc9-40a8-41c0-abe8-1db0a5042f4a","executionInfo":{"status":"ok","timestamp":1543824816575,"user_tz":-330,"elapsed":700,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":206}},"cell_type":"code","source":["result.head()"],"execution_count":0,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>Gene</th>\n","      <th>Variation</th>\n","      <th>Class</th>\n","      <th>TEXT</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>0</td>\n","      <td>FAM58A</td>\n","      <td>Truncating_Mutations</td>\n","      <td>1</td>\n","      <td>cyclin dependent kinases cdks regulate variety...</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>1</td>\n","      <td>CBL</td>\n","      <td>W802*</td>\n","      <td>2</td>\n","      <td>abstract background non small cell lung cancer...</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>2</td>\n","      <td>CBL</td>\n","      <td>Q249E</td>\n","      <td>2</td>\n","      <td>abstract background non small cell lung cancer...</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>3</td>\n","      <td>CBL</td>\n","      <td>N454D</td>\n","      <td>3</td>\n","      <td>recent evidence demonstrated acquired uniparen...</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>4</td>\n","      <td>CBL</td>\n","      <td>L399V</td>\n","      <td>4</td>\n","      <td>oncogenic mutations monomeric casitas b lineag...</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["   ID    Gene             Variation  Class  \\\n","0   0  FAM58A  Truncating_Mutations      1   \n","1   1     CBL                 W802*      2   \n","2   2     CBL                 Q249E      2   \n","3   3     CBL                 N454D      3   \n","4   4     CBL                 L399V      4   \n","\n","                                                TEXT  \n","0  cyclin dependent kinases cdks regulate variety...  \n","1  abstract background non small cell lung cancer...  \n","2  abstract background non small cell lung cancer...  \n","3  recent evidence demonstrated acquired uniparen...  \n","4  oncogenic mutations monomeric casitas b lineag...  "]},"metadata":{"tags":[]},"execution_count":31}]},{"metadata":{"id":"dvMhT5ASc-0K","colab_type":"text"},"cell_type":"markdown","source":["## Splitting the data into Train , Test and Cross Validation Sets"]},{"metadata":{"id":"2_-d4x8BF9aU","colab_type":"code","colab":{}},"cell_type":"code","source":["X, x_test,Y, y_test = train_test_split(result, result['Class'], stratify=result['Class'], test_size=0.2)\n","x_train, x_cv, y_train, y_cv = train_test_split(X, Y,  stratify=Y, test_size=0.2)"],"execution_count":0,"outputs":[]},{"metadata":{"id":"UFnu2l8WHyh7","colab_type":"code","colab":{}},"cell_type":"code","source":["##Distrubutaion of data\n","train_class_dist= x_train['Class'].value_counts().sort_index()\n","test_class_dist = x_cv['Class'].value_counts().sort_index()\n","cv_class_dist   = x_test['Class'].value_counts().sort_index()"],"execution_count":0,"outputs":[]},{"metadata":{"id":"ta1EAaQ5dJ1E","colab_type":"text"},"cell_type":"markdown","source":["##Cheking the distributaion of Classes[1-9] in different Sets of data"]},{"metadata":{"id":"hWr3jB_dIKZ7","colab_type":"code","outputId":"5cfab2ff-126a-464a-e235-1d445520893b","executionInfo":{"status":"ok","timestamp":1543824827581,"user_tz":-330,"elapsed":1313,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":1082}},"cell_type":"code","source":["def print_distribution( VAL, name):\n","  VAL.plot(kind='bar')\n","  plt.xlabel('Class')\n","  plt.ylabel('Data points per Class')\n","  plt.title('Distribution of yi in '+name+' data')\n","  plt.grid()\n","  plt.show()\n","  \n","print_distribution( train_class_dist, 'Training' )\n","print_distribution( cv_class_dist, 'Cross Validation'  )\n","print_distribution( test_class_dist, 'Testing' )"],"execution_count":0,"outputs":[{"output_type":"display_data","data":{"image/png":"iVBORw0KGgoAAAANSUhEUgAAAfIAAAFjCAYAAADGq4k2AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4yLCBo\ndHRwOi8vbWF0cGxvdGxpYi5vcmcvNQv5yAAAIABJREFUeJzt3X98zfXj///72S8LMzY7qy1KhMr8\nWJLl91iGVSs/X+stsn7Ir0hYIhXKj4RpUgp7+VHeJpr8mExKb7NivSWK1Puj0H6xGfthtp3vH307\nb3sz51Bnx3O7XS+XLped53k+n4/7c+h+nj/O82myWCwWAQAAQ3JxdgAAAHD9KHIAAAyMIgcAwMAo\ncgAADIwiBwDAwChyAAAMjCJHldWsWTOFhYWpZ8+e6ty5s5599ll999131vfnzZunjz766Krr2L17\nt06dOnXF91atWqUFCxZIkkJDQ7Vv375rypedna3k5GRJ0vfff6/o6OhrWv56vfjii+rSpYt2795t\n1/yXbqc90yvy888/Kzw8XOHh4erYsaNat25tff3ee+/ZvR57x87IyFBERMQ1rfdaDB06VJ988onN\n+f7zP//TYRkASZIFqKKaNm1q+eOPPywWi8VSVlZm2bJli6V9+/aWb775xu51DBs2zPLtt9/anK9b\nt252zXepzz77zDJ58uRrWuaf0Lx5c8vx48crfdxLrV+/3jJkyBCnZvi7hgwZYlm/fv1V58nMzLSE\nhYVVUiJUV27O/iABVAaTyaRevXrp/Pnzmjdvnj7++GPFxMSoYcOGGjFihFatWqXVq1fLYrGodu3a\nevPNN7V582bt3btXv/76qyZMmKBffvlFGRkZ+umnnxQREaFz584pPT1dM2fOlCTt3btXM2bMUE5O\njiIjIzVu3DilpqZqypQp+vzzzyXJ+nrBggV6/fXXVVpaqoKCAg0aNMg634ULFzRz5kylpqbKxcVF\nXbp00YQJE+Tq6qrQ0FA988wzSkhIUHp6uiIiIhQTE3PZ9p46dUpTp07ViRMn5O7urqeeekqRkZEa\nPHiwysrKFB0drSlTpqhLly6SpLNnz6pz585KTk5W/fr1JUmzZ89WSUmJ6tSpU247/7Jo0SLr9MGD\nBys0NFTbt2/XiRMndN9992nevHkymUx2/xmlpqZq/vz58vf3l5ubm+bNm6d169Zp2bJlKi0tlZ+f\nn+bMmaPAwEC7xj558qQefPBBHT58WJ988ol27dql2rVra//+/XJ1ddXChQt155136sSJExo1apTy\n8vLUsWNHZWRkqGfPnnrsscfK5fv999/1wgsvKCcnR61atVJpaan1veTkZC1YsEDFxcWqVauWZs6c\nqbvuukuDBg1SRkaGwsPDlZiYqEOHDmn69OkqKCiQi4uLpkyZogceeMDu3xFwJRxaR7USGhqqAwcO\nqKioyDrt/PnzWrhwodatW6dt27YpOjpau3bt0tixY+Xv76+5c+eqd+/ekqQvv/xS77//voYOHXrZ\nug8dOqT169frk08+0UcffaSffvqpwhz33HOP/uM//kM9e/bU/Pnzy70XHx+v9PR0bd68WRs2bNC+\nffv02WefWd//9ttvtXbtWq1fv16rVq1Senr6ZeufOnWq2rVrp6SkJL333nuaMWOGTpw4oZUrV0qS\nVq5caS1xSfL29lZISIi2bNlinfb555+rT58+Nn6j/2vnzp1avny5kpKStHfvXqWlpdm97F8OHz6s\nQYMGad68eTp9+rRef/11LV++XNu3b1fDhg21ePHi6x77q6++UlRUlJKSknT//fcrPj5ekjRnzhx1\n6NBBO3fuVOfOnbVnz54rjvHWW28pJCREO3bs0JAhQ6xjlJSUKCYmRtOnT1dSUpJCQ0M1e/ZsSdIb\nb7yhW265Rdu2bZOHh4deeeUVRUdHa9u2bXrmmWc0bdq0a/4dAf8XRY5qpXbt2iorK1N+fr51Wo0a\nNWQymZSQkKDs7Gz16tVLTz/99BWXb9WqlXx8fK743kMPPSRXV1f5+vrqvvvuK3c+/lrs2rVLAwYM\nkJubmzw9PfXQQw/pv/7rvy4bx9/fX76+vvrjjz/KLX/x4kXt2bNHUVFRkqTAwEDdf//92rt371XH\njYiI0ObNmyVJP/30k8rKytS6dWu7c4eHh8vT01M1a9bU7bffflkue3h6eiokJESS5Ovrq/379+vm\nm2+WJLVt21a///77dY/duHFjtWjRQpJ09913W+fZt2+f9Vx6jx49ZDabrzjGvn37rB/oWrZsqTvu\nuEOS5Obmpj179lh/V1fLuXHjRvXq1UuSdO+991Y4H3AtOLSOauWvQ81eXl7Wae7u7lqxYoWWLFmi\nRYsWqVmzZpo2bZqaNWt22fLe3t4VrvvSgvfy8lJeXt51ZTxz5ky5cby9vXX69Gnr69q1a1t/dnV1\nLXeIV5Jyc3NlsVjKbWOdOnV05syZq44bGhqqqVOn6vfff9eOHTsUHh5+Tblt5bLHpdtdWlqq2NhY\n7dy5U6WlpcrPz1ejRo2ue+xLfx+XzpOXl1duXH9//yuOcfbs2XLj1KlTx/rzypUrtWHDBhUXF6u4\nuLjCUwqbNm3Sv//9b+Xn56usrEwWHnWBfwB75KhWkpKS1K5dO3l4eJSbfvfddys2NlYpKSnq2LHj\ndR3yPHv2bLmfvb29LysVe8q9fv36ys3Ntb7Ozc21nre2R7169eTi4lIuT25urnx9fa+6XM2aNdWt\nWzdt27ZNSUlJ1r1PZ9myZYt27typVatWKSkpSWPGjHHIOLVq1VJBQYH1dVZW1hXnq1Onjs6fP299\n/dcHo7S0NC1dulTvvvuukpKSNGPGjCsun5GRoSlTpmjmzJlKSkrS0qVL/8GtQHVGkaNasFgs2rZt\nm+Lj4zVu3Lhy7x05ckRjxoxRcXGxPDw81KJFC+selZubm86dO2fXGJs3b1ZZWZlOnz6t/fv3q23b\ntvLz81NWVpZOnz6t0tJSbdq0yTp/Revu2rWrEhISrBfCffrpp+XOZ9vi5uamjh07au3atZKk3377\nTfv27bProqqIiAh99NFHKioqsh6GdpbTp08rMDBQPj4+ysnJ0datW8udEvmntGzZUlu3bpUkffHF\nF8rMzLzifK1bt7ZetJiWlqbffvtN0p+F7uvrq4CAABUWFmrDhg0qKCiQxWKRm5ubCgoKVFJSojNn\nzqhmzZq64447VFJSYv3zccQ2oXqhyFGlDR48WOHh4erUqZM++ugjvf/++woKCio3T9OmTXXrrbcq\nIiJCffr00TvvvKOXX35ZktSzZ0+98MILWr58uc2xgoKC1K9fP/Xt21dDhgxRkyZNdNttt6lv376K\njIxUVFSU2rdvb52/Q4cO2rt3r/r27XtZ5ptvvll9+vRR37591bVrV+t5VXu99tprSk1NVXh4uEaO\nHKkZM2bolltusblcx44ddf78eafvjUt/fqjIzc1VWFiYxo8fr7Fjxyo9PV2zZs36R8eZMGGCtm/f\nrvDwcKWkpKh169ZXPDQ+YcIEffHFF+rRo4dWr15t/WDUqVMnmc1m9ejRQ8OGDdOQIUPk5eWlMWPG\nqFmzZvL29laHDh1Up04dde7cWT179tTAgQMVGhqq1q1ba/Dgwf/o9qD6MVk4SQPgEn369NHChQvV\npEkTZ0epNBaLxVreffv21XPPPacePXo4ORVgH/bIAVht3rxZfn5+1arEZ8+erddee02S9Msvv+jX\nX391+mkF4FqwRw5AkvTkk08qJydHsbGxatiwobPjVJrMzExNnDhRJ0+elIuLi4YPH65HH33U2bEA\nu1HkAAAYGIfWAQAwMIocAAADM+Sd3bKy7PteLwAAVYGfn1eF77FHDgCAgVHkAAAYGEUOAICBUeQA\nABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgRny6WcA\nUNUMm7XT4WMsiwl1+BiofA4t8sTERH3wwQdyc3PTmDFj1KxZM02cOFGlpaXy8/PT3Llz5eHhocTE\nRMXHx8vFxUUDBgxQ//79HRkLAIAqw2FFnpOTo7i4OK1fv14FBQVatGiRkpKSFBUVpV69euntt99W\nQkKCIiMjFRcXp4SEBLm7u6tfv34KCwtT3bp1HRUNAIAqw2HnyFNSUhQSEqLatWvLbDZr+vTpSk1N\nVffu3SVJ3bp1U0pKig4cOKCgoCB5eXnJ09NTwcHBSktLc1QsAACqFIftkZ84cUJFRUUaPny48vLy\nNHr0aBUWFsrDw0OS5Ovrq6ysLGVnZ8vHx8e6nI+Pj7KyshwVCwCAKsWh58hzc3P1zjvv6NSpU3ri\niSdksVis713686Uqmg4AAC7nsEPrvr6+atOmjdzc3NSwYUPVqlVLtWrVUlFRkSQpIyNDZrNZZrNZ\n2dnZ1uUyMzNlNpsdFQsAgCrFYUXesWNH7d27V2VlZcrJyVFBQYEeeOABJSUlSZK2b9+uTp06qVWr\nVjp48KDy8vKUn5+vtLQ0tW3b1lGxAACoUhx2aN3f3189e/bUgAEDJElTpkxRUFCQJk2apLVr1yog\nIECRkZFyd3fX+PHjFR0dLZPJpJEjR8rLy8tRsQAAqFJMFgOelM7KOufsCADwj+KGMLgaP7+Kd3C5\nRSsAAAZGkQMAYGAUOQAABkaRAwBgYBQ5AAAGRpEDAGBgFDkAAAZGkQMAYGAUOQAABkaRAwBgYBQ5\nAAAGRpEDAGBgFDkAAAZGkQMAYGAUOQAABkaRAwBgYBQ5AAAGRpEDAGBgFDkAAAZGkQMAYGAUOQAA\nBkaRAwBgYBQ5AAAGRpEDAGBgFDkAAAZGkQMAYGAUOQAABkaRAwBgYBQ5AAAGRpEDAGBgFDkAAAZG\nkQMAYGAUOQAABkaRAwBgYBQ5AAAGRpEDAGBgFDkAAAbm5qgVp6am6vnnn9edd94pSWratKmeeuop\nTZw4UaWlpfLz89PcuXPl4eGhxMRExcfHy8XFRQMGDFD//v0dFQsAgCrFYUUuSe3atVNsbKz19Usv\nvaSoqCj16tVLb7/9thISEhQZGam4uDglJCTI3d1d/fr1U1hYmOrWrevIaAAAVAmVemg9NTVV3bt3\nlyR169ZNKSkpOnDggIKCguTl5SVPT08FBwcrLS2tMmMBAGBYDt0jP3bsmIYPH66zZ89q1KhRKiws\nlIeHhyTJ19dXWVlZys7Olo+Pj3UZHx8fZWVlOTIWAABVhsOK/Pbbb9eoUaPUq1cv/f7773riiSdU\nWlpqfd9isVxxuYqmAwCAyzns0Lq/v7969+4tk8mkhg0bqn79+jp79qyKiookSRkZGTKbzTKbzcrO\nzrYul5mZKbPZ7KhYAABUKQ4r8sTERH344YeSpKysLJ0+fVqPPfaYkpKSJEnbt29Xp06d1KpVKx08\neFB5eXnKz89XWlqa2rZt66hYAABUKQ47tB4aGqoXX3xRycnJunjxol599VXdddddmjRpktauXauA\ngABFRkbK3d1d48ePV3R0tEwmk0aOHCkvLy9HxQIAoEoxWQx4Ujor65yzIwDAP2rYrJ0OH2NZTKjD\nx4Bj+PlVvIPLnd0AADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEA\nMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAw\nihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADc3N2AMCI3p21y+FjPBfT1eFjADA+9sgBADAwihwA\nAAOjyAEAMDCbRf7DDz/oiy++kCTNnz9fQ4YM0b59+xweDAAA2GazyGfMmKFGjRpp3759OnjwoKZO\nnarY2NjKyAYAAGywWeQ1atTQ7bffruTkZA0YMEBNmjSRiwtH5AEAuBHYbOTCwkJt3bpVO3bsUMeO\nHZWbm6u8vLzKyAYAAGywWeQvvPCCNm3apHHjxql27dpauXKlhg4dWgnRAACALTZvCNO+fXu1aNFC\ntWvXVnZ2tkJCQhQcHFwZ2QAAgA0298inT5+urVu3Kjc3V4MGDdKqVav06quvVkI0AABgi80iP3z4\nsPr376+tW7fq0Ucf1YIFC3T8+PHKyAYAAGywWeQWi0WStGvXLoWGhkqSiouLHZsKAADYxWaRN2rU\nSL1791Z+fr7uuusubdy4Ud7e3pWRDQAA2GDzYrcZM2bo6NGjaty4sSSpSZMmGjFihF0rLyoqUkRE\nhEaMGKGQkBBNnDhRpaWl8vPz09y5c+Xh4aHExETFx8fLxcVFAwYMUP/+/f/eFgEAUI3YdWeXzMxM\nbdmyRRs3btThw4c1duxYu1b+7rvvWvfeY2NjFRUVpTVr1ui2225TQkKCCgoKFBcXpxUrVmjlypWK\nj49Xbm7u9W8NAADVjM098gkTJujs2bM6cuSIgoODdeDAAY0ePdrmin/55RcdO3ZMXbt2lSSlpqbq\ntddekyR169ZNy5YtU6NGjRQUFCQvLy9JUnBwsNLS0qzn4gEAwNXZ3CNPT0/Xhx9+qEaNGik2NlZr\n1qzRwYMHba549uzZiomJsb4uLCyUh4eHJMnX11dZWVnKzs6Wj4+PdR4fHx9lZWVdz3YAAFAt2X3T\n9JKSEl24cEGBgYE6duzYVefduHGjWrdurQYNGlzx/b+uhLd3OgAAuDK77uy2dOlS9ejRQ48++qhu\nvfVWlZWVXXWZXbt26ffff9euXbuUnp4uDw8P1axZU0VFRfL09FRGRobMZrPMZrOys7Oty2VmZqp1\n69Z/f6sAAKgmbBb5mDFjVFpaKldXV7Vp00anT59Whw4drrrMggULrD8vWrRIgYGB+u6775SUlKRH\nHnlE27dvV6dOndSqVStNmTJFeXl5cnV1VVpamiZPnvz3twoAgGqiwiJPSEiocKEtW7aoX79+1zTQ\n6NGjNWnSJK1du1YBAQGKjIyUu7u7xo8fr+joaJlMJo0cOdJ64RsAALCtwiLfv3//VRe0t8gvvcJ9\n+fLll70fHh6u8PBwu9YFAADKq7DI33zzTZWVlcnFpfz1cBcvXpS7u7vDgwEAANsqvGr9xIkT6t27\nt86dO2ed9v333+uxxx7TmTNnKiUcAAC4ugqL/M0339SoUaPKnbNu2bKlnnvuOc2aNatSwgEAgKur\nsMizs7MVERFx2fTevXvr5MmTDg0FAADsU2GRl5SUVLhQYWGhQ8IAAIBrU2GR16lTR99///1l07/5\n5hvVq1fPoaEAAIB9Krxqfdy4cRo9erQeeeQRBQUFqbS0VPv371dSUpJWrVpVmRkBAEAFKtwjb9my\npdavXy8XFxd9+umn2rJli7y9vfXpp5+qYcOGlZkRAABU4Kq3aK1fv77dzx4HAACVz+6nnwEAgBsP\nRQ4AgIHZLPL169dXRg4AAHAdbD7G9PPPP9eDDz5o6KeSDZu10+FjLIsJdfgYAAD8XzaLvKioSKGh\noWrUqFG5h6WsXr3aocEAAIBtNot8xIgRlZEDAABcB5vnyNu1a6eCggIdPXpU7dq1080336z77ruv\nMrIBAAAbbBb53LlzlZCQoE8++USStGnTJs2YMcPhwQAAgG02i/zbb7/VO++8o1q1akmSRo4cqUOH\nDjk8GAAAsM1mkdeoUUOSZDKZJEmlpaUqLS11bCoAAGAXmxe7BQcHKyYmRpmZmVq+fLm2b9+udu3a\nVUY2AABgg80iHzdunLZt26abbrpJ6enpevLJJ/Xggw9WRjYAAGCDzSKXpDvuuENlZWUymUxq0qSJ\nozMBAAA72Szy2bNnKzk5WUFBQSorK9O8efMUERHBU9EAALgB2Czy1NRUbd682XpXt+LiYg0aNIgi\nBwDgBmDzqvX69evLze1/+97d3V2BgYEODQUAAOxjc4+8Xr166tu3r9q3by+LxaJvv/1WDRo00MKF\nCyVJzz//vMNDAgCAK7NZ5A0aNFCDBg2sr7t27erIPAAA4BrYLPJRo0ZVRg4AAHAdbJ4jBwAANy6K\nHAAAA7OryM+fPy9Jys7O1r59+1RWVubQUAAAwD42i3z69OnaunWrcnNzNWjQIK1cuVKvvvpqJUQD\nAAC22Czyw4cPq3///tq6daseffRRLVy4UMePH6+MbAAAwAabRW6xWCRJu3btUmhoqKQ/7+4GAACc\nz2aR33777erTp4/y8/N11113aePGjfL29q6MbAAAwAab3yOfOHGiMjIy1LhxY0lSkyZNNGbMGIcH\nAwAAtl11j7ysrEzPP/+8mjdvLjc3N5WVlenOO+/UpEmTKisfAAC4igr3yD/77DMtWrRIx48f1913\n3209V+7i4qKOHTtWWkAAAFCxCos8IiJCERERWrRokUaPHn3NKy4sLFRMTIxOnz6tCxcuaMSIEWre\nvLkmTpyo0tJS+fn5ae7cufLw8FBiYqLi4+Pl4uKiAQMGqH///n9rowAAqC5sniN/5plntGPHDp09\ne9a6Vy5J/fr1u+pyX3zxhVq0aKGnn35aJ0+e1LBhwxQcHKyoqCj16tVLb7/9thISEhQZGam4uDgl\nJCTI3d1d/fr1U1hYmOrWrfv3tw4AgCrOZpE/9dRTMplMlz2D3FaR9+7d2/rzH3/8IX9/f6Wmpuq1\n116TJHXr1k3Lli1To0aNFBQUJC8vL0lScHCw0tLSrF91AwAAFbNZ5BcvXtTHH3983QMMGjRI6enp\nWrJkiZ588kl5eHhIknx9fZWVlaXs7Gz5+PhY5/fx8VFWVtZ1jwcAQHVis8ibNGminJwc1atX77oG\n+Pjjj/Xjjz9qwoQJ5Q7NX/rzpSqaDgAALmezyNPT0/Xggw+qcePGcnV1tU5fvXr1VZf74Ycf5Ovr\nq1tuuUV33XWXSktLVatWLRUVFcnT01MZGRkym80ym83Kzs62LpeZmanWrVv/jU0CAKD6sOtit+ux\nb98+nTx5Ui+//LKys7NVUFCgTp06KSkpSY888oi2b9+uTp06qVWrVpoyZYry8vLk6uqqtLQ0TZ48\n+brGBACguqmwyA8fPqy7775bpaWl17XiQYMG6eWXX1ZUVJSKior0yiuvqEWLFpo0aZLWrl2rgIAA\nRUZGyt3dXePHj1d0dLRMJpNGjhxpvfANAABcXYVF/umnn+ruu+/W4sWLL3vPZDIpJCTkqiv29PTU\nvHnzLpu+fPnyy6aFh4crPDzcnrwAAOASFRb5Sy+9JElauXJlpYXB1Y3cOdHhY8SFznH4GACAf47N\np5/98ssveuKJJxQcHKx7771X0dHR+u233yojGwAAsMFmkU+fPl3Dhg3T119/ra+++kqDBg3StGnT\nKiMbAACwwWaRWywWde3aVTVr1lStWrUUFhZ23RfAAQCAf5bNIr948aIOHTpkff39999T5AAA3CBs\nfo980qRJGj9+vE6fPi1JMpvNmj17tsODAQAA22wWeatWrbRt2zadO3dOJpNJtWvXroxcAADADjaL\n/NixY4qNjdWxY8dkMpnUrFkzjR49Wo0aNaqMfAAA4CpsniOPiYlR586d9c477yg2Nlbt27fXpEmT\nKiMbAACwweYe+U033VTu2eONGzdWUlKSQ0MBAAD72Nwjb9++vXbs2KHCwkLl5+crOTlZbdq0kcVi\nUVlZWWVkBAAAFbC5R7548eIrft3snXfekclk0o8//uiQYAAAwDabRX7pd8gBAMCNxeahdQAAcOOi\nyAEAMLDrKvLjx4//0zkAAMB1sHmOvLS0VF9//bVycnIkScXFxVqyZIl27tzp8HAAAODqbBb5hAkT\ndPbsWR05ckTBwcE6cOCARo8eXRnZAACADTYPraenp+vDDz9Uo0aNFBsbqzVr1ujgwYOVkQ0AANhg\n9znykpISXbhwQYGBgTp27JgjMwEAADvZPLTevn17LV26VD169NBjjz2mwMBA7ugGAMANwmaRjxkz\nRqWlpXJ1dVXr1q115swZhYSEVEY2AABgg81D69HR0XJ1dZUk3XvvvQoLC9OQIUMcHgwAANhW4R55\nYmKi4uLidOrUKXXt2tU6vaSkRL6+vpWRDQAA2FBhkT/88MPq06ePXn755XJfN3NxcZG/v3+lhAMA\nAFd31UPrrq6umjVrlurWrSuTySSTyaQLFy5owIABlZUPAABchc2L3T744AMtWbJExcXFqlmzpi5c\nuKCHHnqoMrIBAAAbbF7stm3bNu3Zs0etWrXS3r179dZbb+nOO++sjGwAAMAGm0Veq1YteXh46OLF\ni5Kk7t27Kzk52eHBAACAbTYPrXt7eysxMVFNmzbVSy+9pMaNGyszM7MysgEAABtsFvns2bN1+vRp\nhYWFKT4+Xunp6Xr77bcrIxsAALDBZpGfOnVKR48elaurqyIiInTrrbdWRi4AAGCHCou8qKhI48eP\n148//qgWLVro/Pnz+vHHH9WxY0fNnDlTHh4elZkTAABcQYUXuy1evFj+/v7avn27YmNjtWzZMu3c\nuVM1atTQ/PnzKzMjAACoQIVFvm/fPsXExMjN7X932m+66SZNmzZNX3/9daWEAwAAV1dhkbu6ul7x\n8Lm7u7vq1Knj0FAAAMA+FRa5yWSqcKG/noYGAACcq8KL3b777rtyTz37i8ViUU5OjiMzAQAAO1VY\n5Nu2bfvbK58zZ47279+vkpISPfvsswoKCtLEiRNVWloqPz8/zZ07Vx4eHkpMTFR8fLxcXFw0YMAA\n9e/f/2+PDQBAdVBhkQcGBv6tFe/du1c///yz1q5dq5ycHD366KMKCQlRVFSUevXqpbffflsJCQmK\njIxUXFycEhIS5O7urn79+iksLEx169b9W+MDAFAd2LzX+vW67777tHDhQklSnTp1VFhYqNTUVHXv\n3l2S1K1bN6WkpOjAgQMKCgqSl5eXPD09FRwcrLS0NEfFAgCgSnFYkbu6uqpmzZqSpISEBHXu3FmF\nhYXWK+F9fX2VlZWl7Oxs+fj4WJfz8fFRVlaWo2IBAFClOKzI/7Jjxw4lJCTolVdeKTfdYrFccf6K\npgMAgMs5tMh3796tJUuWaOnSpfLy8lLNmjVVVFQkScrIyJDZbJbZbFZ2drZ1mczMTJnNZkfGAgCg\nynBYkZ87d05z5szRe++9Z71w7YEHHlBSUpIkafv27erUqZNatWqlgwcPKi8vT/n5+UpLS1Pbtm0d\nFQsAgCrF5tPPrteWLVuUk5OexTS7AAAMf0lEQVSjsWPHWqfNmjVLU6ZM0dq1axUQEKDIyEi5u7tr\n/Pjxio6Olslk0siRI+Xl5eWoWAAAVCkOK/KBAwdq4MCBl01fvnz5ZdPCw8MVHh7uqCgAAFRZDr/Y\nDQAAOA5FDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIE57OtnAFAZJn/7s8PHeOO+Ox0+BnC92CMH\nAMDAKHIAAAyMIgcAwMAocgAADIwiBwDAwChyAAAMjCIHAMDAKHIAAAyMIgcAwMC4sxtQjf323esO\nH6Nhm1ccPgZQnbFHDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUO\nAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCA\ngVHkAAAYmEOL/OjRo+rRo4dWrVolSfrjjz80ePBgRUVF6fnnn1dxcbEkKTExUX379lX//v21bt06\nR0YCAKBKcViRFxQUaPr06QoJCbFOi42NVVRUlNasWaPbbrtNCQkJKigoUFxcnFasWKGVK1cqPj5e\nubm5jooFAECV4rAi9/Dw0NKlS2U2m63TUlNT1b17d0lSt27dlJKSogMHDigoKEheXl7y9PRUcHCw\n0tLSHBULAIAqxc1hK3Zzk5tb+dUXFhbKw8NDkuTr66usrCxlZ2fLx8fHOo+Pj4+ysrIcFQsAgCrF\naRe7WSyWa5oOAAAuV6lFXrNmTRUVFUmSMjIyZDabZTablZ2dbZ0nMzOz3OF4AABQsUot8gceeEBJ\nSUmSpO3bt6tTp05q1aqVDh48qLy8POXn5ystLU1t27atzFgAABiWw86R//DDD5o9e7ZOnjwpNzc3\nJSUl6a233lJMTIzWrl2rgIAARUZGyt3dXePHj1d0dLRMJpNGjhwpLy8vR8UCAKBKcViRt2jRQitX\nrrxs+vLlyy+bFh4ervDwcEdFAQCgyuLObgAAGBhFDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR\n5AAAGBhFDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICBuTk7\nAKqfo08Ndej6m36wwqHrB4AbCXvkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAY\nGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICB8RhTAAAuMXLnRIePERc65x9b\nF3vkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhXrQMA/jHDZu106PqXxYQ6dP1GdMMU+RtvvKED\nBw7IZDJp8uTJatmypbMjAQBww7shivybb77R8ePHtXbtWv3yyy+aPHmy1q5d6+xYAADc8G6Ic+Qp\nKSnq0aOHJKlx48Y6e/aszp8/7+RUAADc+G6IIs/Ozla9evWsr318fJSVleXERAAAGIPJYrFYnB1i\n6tSp6tKli3Wv/F//+pfeeOMNNWrUyMnJAAC4sd0Qe+Rms1nZ2dnW15mZmfLz83NiIgAAjOGGKPIO\nHTooKSlJknTo0CGZzWbVrl3byakAALjx3RBXrQcHB+uee+7RoEGDZDKZNG3aNGdHAgDAEG6Ic+QA\nAOD63BCH1gEAwPWhyAEAMDCKHAAAA6PIbcjLy3N2hGtypUse0tPTnZDkn3HmzBlnR/hHpKSkODvC\n31ZSUqKTJ0+qpKTE2VH+lqryd6oqMPIlWhaLRWfOnNHp06edHYUit2XUqFHOjmCXzz//XN26dVNI\nSIgmTZpU7ha3EydOdGIy++3atUs9e/bU0KFDdfToUT388MMaPHiwQkND9eWXXzo7nt02btxY7r8N\nGzZo2rRp1tdGMWPGDOvPe/bsUVhYmMaOHasHH3xQu3fvdmIy+3355Zd65ZVXJP35Yapbt2564okn\nFBoaql27djk33DUIDg7W9OnTb4jSuF5ff/21evXqpccff1zff/+9+vbtq86dOys8PFzffPONs+PZ\n7X/+5380fPhwPfzww+revbueffZZ67+NjIwMp2S6Ib5+5myrV6+u8D1n/cFcq/fff18bNmxQnTp1\ntG7dOkVHR+uDDz6Ql5eXYT71vvvuu1q+fLlOnTql4cOHa/HixWrevLmys7M1fPhwdenSxdkR7RIX\nF6e6deuWy3vhwgWdOHHCiamu3ZEjR6w/x8XF6d///rcaNGigrKwsjRo1Sp06dXJiOvvExsbqvffe\nk1R+G3JycvTss8+qa9euzg1op3vuuUfh4eEaP368brnlFj322GNq06aN3NyM87/wuLg4xcfH6+zZ\nsxo8eLBWrFih5s2b6+TJk5owYYLWrFnj7Ih2mTZtmmbOnKkGDRro119/1cqVKzVt2jR99dVXevHF\nF7Vy5cpKz2ScvwUOtGLFCoWEhMhsNl/2nlEOI7q6uqpu3bqSpIEDB8rX11fR0dFasmSJTCaTk9PZ\nx8PDQwEBAQoICJDZbFbz5s0lSfXr11eNGjWcnM5+n332mRYvXqwjR44oJiZGgYGB2r17t2GO7vzl\n0r833t7eatCggSTJz8/PMAVSUlKiWrVqSZK8vLx06623SpLq1q1rmA+40p9/Fvfdd59WrFihgwcP\nat26dZo6dapq1aolX19fvf/++86OaJO7u7vMZrPMZrPq1Klj/fcdGBgoV1dXJ6ezX3FxsfXfwu23\n3279wNu5c2ctWrTIKZmM8a/RweLi4jRjxgxNmTJFHh4e5d5LTU11UqprExwcrGeffVYLFy6Up6en\nevTooRo1amjo0KHKzc11djy7+Pr66sMPP1R0dLQ+/vhjSX+e31+2bJluvvlmJ6ezX40aNTRu3Dj9\n+uuvev3119WmTRuVlZU5O9Y1+/nnn/X888/LYrHo+PHj2rp1q3r16qVly5bJy8vL2fHsEh0drcjI\nSHXo0EF169bViBEj1KZNG6Wmpqp///7Ojme3Sz90BAUFKSgoSNKft7M2ygOmvL29NX/+fOXk5Khh\nw4Z65ZVX1KlTJ/33f/+3fH19nR3Pbk2bNtULL7ygli1bavfu3br//vslSZMnT1aTJk2ckokbwvz/\nCgsLVaNGDbm4lL9s4NChQ7rnnnuclOrapKamql27duX2pM6fP68tW7ZowIABTkxmn6KiIu3cuVO9\ne/e2Tjt06JC+/fZb/etf/zLUXvmlNm7cqC+//FLz5893dpRr8n/PW952223y9/fXpk2bFBoaat3T\nvdHl5uZqz549OnnypCwWi+rXr68OHTrI39/f2dHslpCQoH79+jk7xt9SUFCgDRs2qF69eurdu7cS\nExOVlpam2267TQMHDlTNmjWdHdEuFotFycnJ+n//7/+padOm6ty5syTpp59+UrNmzZxyBJQiBwDA\nwLhqHQAAA6PIAQAwMC52A6qxzMxMzZkzR0ePHrWe8x49erTS09O1Z88evfXWW05OCMAWihyopiwW\ni0aOHKnIyEhrYR85ckTDhg3T2LFjnZwOgL0ocqCaSklJkclk0uOPP26d1qxZM23ZskXJycnWaZ9/\n/rk++OADeXh4qLS0VHPmzNGtt96q+Ph4JSYm6qabbpKnp6fmzp2r4uJivfjii5L+/BbCwIEDDX+1\nNXCjo8iBaurnn3+2fh/5Ut7e3uVe5+Xlaf78+QoICNB7772n1atXa9KkSYqNjVVSUpLq16+v3bt3\nKzMzUykpKbrjjjv02muv6cKFC1q3bl1lbQ5QbVHkQDXl6uqq0tJSm/PVr19fkyZNksViUVZWltq0\naSNJ6tevn5566in17NlT4eHhatSokdzc3LRmzRrFxMSoS5cuGjhwoKM3A6j2uGodqKaaNm2q7777\n7rLpR44cUWFhoSTp4sWLGjt2rKZPn65Vq1Zp8ODB1vleeuklxcXFydvbWyNHjtSXX36pxo0ba/Pm\nzXr44YeVkpJSbn4AjkGRA9VUu3btVKtWrXL36f7555/13HPPWe99nZ+fLxcXFwUGBurChQtKTk5W\ncXGxzp49q0WLFumWW25RVFSUHn/8cR08eFCbNm3SwYMH9cADD2jatGn6448/DPO8AsCoOLQOVGPv\nv/++3nzzTUVERKhu3bqqUaOGFixYoGPHjkn68+EiERER6tevnwICAhQdHa2JEydqz549ys/PV79+\n/VSnTh25ublp5syZOnPmjKZNmyYPDw9ZLBY9/fTThnnACmBU3KIVAAAD49A6AAAGRpEDAGBgFDkA\nAAZGkQMAYGAUOQAABkaRAwBgYBQ5AAAGRpEDAGBg/x+tI+Is3KqR0AAAAABJRU5ErkJggg==\n","text/plain":["<matplotlib.figure.Figure at 0x7fbc5abdf710>"]},"metadata":{"tags":[]}},{"output_type":"display_data","data":{"image/png":"iVBORw0KGgoAAAANSUhEUgAAAfIAAAFjCAYAAADGq4k2AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4yLCBo\ndHRwOi8vbWF0cGxvdGxpYi5vcmcvNQv5yAAAIABJREFUeJzt3Xl4jPfi/vF7sokQCSHUWkstrTVV\nTdQaSZPYl1iqttP0VE/ttJYuoqW1tUWIY2kttbS+0iJqGcXh8EVqO4S2UnXqWIrEEsQSmczvD7/O\nVw4xoZ2ZPLxf1+W6zGeeeT73xMg9zzLPmKxWq1UAAMCQ3FwdAAAAPDyKHAAAA6PIAQAwMIocAAAD\no8gBADAwihwAAAOjyOFy1apVU3h4uCIiItSkSRP17dtX+/fvt93/ySef6Msvv7zvOrZt26bTp0/f\n877Fixdr6tSpkqTQ0FDt2bPngfKlpaVp06ZNkqSDBw8qJibmgR7/sN588001bdpU27Zty9Pydz7P\nvIzb88svv6hfv34KCwtTeHi4unbtqn/84x8PvJ4/YsqUKRo0aNBd4//+979Vt25dXb16NdfHJiUl\nKTw8XNL9X0NPP/20Tp48ed8cznwNhIeHKykp6b7LZGZmauXKlQ7LAIOxAi5WtWpV62+//Wa1Wq3W\n7Oxs69q1a63BwcHW77//Ps/reOWVV6y7d++2u1zz5s3ztNydvv32W+vbb7/9QI/5M1SvXt16/Phx\np89rtVqtZ86csQYHB1u//PJLa3Z2ttVqtVr37dtnff75563btm1zWo5ff/3VWrt2beuVK1dyjE+Z\nMsU6bNiw+z52165d1rCwMLtz1KhRw3rixIn7LuPM10BYWJh1165d911m//791t69ezslD/I/tsiR\nr5hMJkVFRWno0KH65JNPJEkjR47UzJkzJd3euoyKilJkZKSio6P1888/a+rUqdq1a5feeustrV27\nVtOnT9e7776r6OhoLViwQNOnT9c777xjm2PXrl1q3769mjZtqilTpkjKufV25+3Dhw/rgw8+kNls\n1pAhQ3Isd/PmTY0ePVoRERGKiorShAkTZLFYJN3e8v/qq68UHR2tRo0aacKECfd8vqdPn1ZMTIwi\nIiLUunVr21ZWz549lZ2drZiYGG3dutW2fHp6uurUqaO0tDTb2MSJE/Xhhx/e9Tx/d+d4z549NX/+\nfL300ktq3Lixhg4dKus9rgm1YMECNWzYUN26dZPJZJIk1atXTzNnzlSlSpUk3d6TMnv2bEVERMhi\nseinn35St27dFBkZqXbt2tn2JGRkZKhfv36KiopSixYt9O677+rWrVu5jt+pQoUKevrpp7V+/foc\n46tXr1bHjh0lSfv371fHjh0VGRmpli1baseOHXc9nztfQ1u3blV4eLiioqL02Wef5VguPj5eERER\nCgsLU9++fXX58mWHvwYOHTqkVq1aKSIiQh999FGO+5YvX66oqCi9+OKLevnll3Xq1CmlpaWpf//+\n+te//qXu3btLkjZt2qQ2bdooIiJCHTt21I8//njPufBoosiRL4WGhurAgQO6ceOGbezq1auaNm2a\nli9frvXr1ysmJkZbtmzR4MGDVbJkSU2ePFktW7aUdPuX9Zw5c9SnT5+71n348GF9/fXX+uabb/Tl\nl1/qp59+yjXHM888ox49eigiIsJW+r9buHChzpw5ozVr1mjFihXas2ePvv32W9v9u3fv1rJly/T1\n119r8eLFOnPmzF3rf++999SgQQOZzWbNnj1b48aN08mTJ7Vo0SJJ0qJFi9S0aVPb8n5+fgoJCdHa\ntWttY999951atWpl5yf6fzZv3qz58+fLbDZr165d2rdv313L7N69O8e8vwsKClLp0qVtt61Wq8xm\ns0wmk4YOHaoePXpo/fr1GjdunIYNG6arV69q5cqVKlKkiNatWyez2Sx3d3cdPXo01/H/1rFjRyUm\nJtpu7927VxaLRcHBwZKk0aNHKyYmRuvXr9drr72m2NjYXJ+7xWLRO++8o9jYWK1bt05ubm624j10\n6JCWLFmir7/+Whs2bFBmZqYWL17s8NfAmDFj1KtXL5nNZtWrV8+2m//8+fP64IMPNH/+fG3YsEHl\ny5fXzJkzVbx4cQ0dOlR169bV0qVLlZWVpZEjR2rs2LEym80KDQ3VxIkTc/0Z4NFDkSNfKly4sLKz\ns5WRkWEbK1CggEwmkxISEpSWlqaoqCj99a9/vefj69Spo2LFit3zvjZt2sjd3V0BAQF67rnnchyP\nfxBbtmxRly5d5OHhIW9vb7Vp00b/+7//e9c8JUuWVEBAgH777bccj79165Z27Nhh26oqU6aMnn/+\nee3ateu+87Zu3Vpr1qyRJP3000/Kzs5W3bp185w7MjJS3t7e8vHx0ZNPPnlXLun2ln/x4sXtrqtZ\ns2aSpJMnTyotLc32hqJWrVoqXbq0kpOTVaxYMe3fv1/bt29Xdna23n//fdWoUSPX8f8WFRWl5ORk\nWwmuWrVK7dq1k5vb7V9fK1euVFRUlCTp2Wef1YkTJ3LN++uvvyozM1ONGjWSJHXo0MF2X82aNbVl\nyxYVLlxYbm5uqlev3n3XJf3x18DNmzeVnJxsewMaGRmpggULSpICAgK0d+9elSpVSpJUv379e+bx\n8PDQjh07bK+B3JbDo8vD1QGAezl58qQ8PT3l6+trG/P09NSCBQs0a9YsTZ8+XdWqVVNsbKyqVat2\n1+P9/PxyXfedBe/r66vLly8/VMYLFy7kmMfPz0/nz5+33S5cuLDt7+7u7rYtv99dunRJVqs1x3Ms\nUqSILly4cN95Q0ND9d577+nEiRPauHGjIiMjHyi3vVySVLRoUZ09e9buuvz9/SXd/ln4+vradsNL\n//dcWrVqpfT0dE2bNk3Hjh1T27ZtNWrUKEVFRd1z3MvL6668LVq0UGJiovr06SOz2az/+Z//sd2/\nevVqffHFF8rIyFB2dvY9DxX8Lj09Pcfzv/Pf7/r16xo/frztRLP09HTbG5Xc/BmvgTuXM5lMKlKk\niKTbew/i4uK0efNmWSwWZWRkqGLFivfMsWjRIq1YsUKZmZnKzMzM8e+ARx9b5MiXzGazGjRocNcv\n9aefflpxcXHauXOnGjVqdN/dqLlJT0/P8Xc/P7+7fsnmpdyLFy9u+0Us3f6lnJet2N8VLVpUbm5u\nOfJcunRJAQEB932cj4+PmjdvrvXr18tsNtu25v5Mzz//vMxm813jmzZt0vbt2+8aDwgIUHp6eo4S\nvfO5dOvWTcuXL9fatWt1+PBh27kAuY3/t44dO2rNmjXavn27KlWqpAoVKkiSzp49q3fffVcffvih\nzGaz5s6de9/n5efnl+NM9zvfNC1cuFC//vqrvvnmG5nNZnXt2vW+65L++Gvg9zcBv2fKzs62vR7W\nrl2rzZs3a/HixTKbzRo4cOA917Fv3z7NnTtXf//732U2mzVu3Lg8z49HA0WOfMVqtWr9+vVauHCh\nhgwZkuO+I0eOaODAgcrMzJSXl5dq1qxp2/Lw8PDQlStX8jTHmjVrlJ2drfPnz2vv3r2qX7++SpQo\nodTUVJ0/f14Wi0WrV6+2LZ/bups1a6aEhARZLBZdu3ZNq1atuudx5dx4eHioUaNGWrZsmSTpP//5\nj/bs2aOGDRvafWzr1q315Zdf6saNG6pZs2ae58yr3r17Kzk5WXPmzFF2drak28emY2Nj5e3tfdfy\nZcuWValSpWzH7vft26e0tDTVrl1b8fHxSkhIkCSVLFlSZcuWlclkynX8XoKDg3X58mUtXLjQdpKb\ndLuIfXx8VKlSJWVlZdl+lncekrlT+fLl5e7ubtvq/uabb2xznj9/XpUqVVKhQoV06tQpbd26Vdeu\nXZPkuNeAt7e3qlevru+++07S7dfmzZs3bXnKlCmjYsWK6eLFi1q3bp3teXl4eOjq1auyWq26cOGC\nAgICVLp0aV2/fl0rVqzQtWvX7rtnAo8Wihz5Qs+ePRUZGanGjRvryy+/1Jw5c1SrVq0cy1StWlVl\ny5ZV69at1apVK82YMcN2NnZERISGDh2q+fPn252rVq1aio6OVqdOndS7d29VqVJFFSpUUKdOndS+\nfXt1797ddiKVJL3wwgvatWuXOnXqdFfmUqVKqVWrVurUqZOaNWtmO1abV++//76SkpIUGRmpfv36\nady4cXriiSfsPq5Ro0a6evWqQ7bGpdtbmkuXLtWBAwcUFhamqKgoTZs2TVOnTlX9+vXvWt5kMunT\nTz+1fapg3LhxmjZtmnx8fNSuXTutWrVKERERioyMlKenp9q1a5fr+L24ubmpbdu2OnDgQI6fcfXq\n1dWkSRNFRESoa9euCg0NVd26ddWzZ897rsfT01Njx47V22+/raioKJlMJvn4+Ei6vXdg9+7dioiI\n0MSJEzVy5Ejt3LlTCxYscOhrYMyYMZo7d64iIiJ08OBBVa5cWdLtN2uXLl1SeHi4hg0bpsGDB+vM\nmTOaMGGCnn32WZ07d06NGzdWw4YNFRgYqLCwML3yyivq3bu3fH19c92Cx6PHZOVtG2BIrVq10rRp\n01SlShVXRwHgQmyRAwa0Zs0alShRghIHwFnrgNH85S9/0cWLFxUXF+fqKADyAYfuWp80aZL27t2r\nrKws9e3bV7Vq1dLw4cNlsVhUokQJTZ48WV5eXkpMTNTChQvl5uamLl26qHPnzo6KBADAI8VhRb5r\n1y59/vnnmjt3ri5evKgOHTooJCRETZo0UVRUlD799FOVKlVK7du3V4cOHZSQkCBPT09FR0dr8eLF\nts+nAgCA3DnsGPlzzz2nadOmSbp9YYjr168rKSlJLVq0kCQ1b95cO3fu1IEDB1SrVi35+vrK29tb\nQUFB97xkJAAAuJvDjpG7u7vbPtaRkJCgJk2aaPv27bYLfAQEBCg1NVVpaWk5rrRVrFgxpaam3nfd\nqal5+7wwAACPghIlfHO9z+FnrW/cuFEJCQkaPXp0jvHc9ujzaTgAAPLOoUW+bds2zZo1S3PnzpWv\nr698fHxs32Z19uxZBQYGKjAwMMdXMp47d06BgYGOjAUAwCPDYUV+5coVTZo0SbNnz7aduNawYUPb\n9Zs3bNigxo0bq06dOkpOTtbly5eVkZGhffv23fPKUQAA4G4OO0a+du1aXbx4UYMHD7aNTZgwQe++\n+66WLVum0qVLq3379vL09NSwYcMUExMjk8mkfv365fg2KAAAkDtDXqKVk90AAI8Tl57sBgAAHIci\nBwDAwChyAAAMjCIHAMDAKHIAAAyMIgcAwMAocgAADIwiBwDAwBx2ZTcAQN69MmGzw+eYNzLU4XPA\n+dgiBwDAwChyAAAMjCIHAMDAKHIAAAyMIgcAwMAocgAADIwiBwDAwChyAAAMjCIHAMDAKHIAAAyM\nIgcAwMAocgAADIwiBwDAwChyAAAMjCIHAMDAKHIAAAyMIgcAwMAocgAADIwiBwDAwDwcufKUlBS9\n8cYb6tOnj3r06KGBAwfq4sWLkqRLly6pbt266tu3r9q0aaOaNWtKkooWLaq4uDhHxgIA4JHhsCK/\ndu2axo4dq5CQENvYnQU9atQode7cWZJUsWJFLVq0yFFRAAB4ZDls17qXl5fmzp2rwMDAu+47duyY\nrly5otq1aztqegAAHgsOK3IPDw95e3vf874vvvhCPXr0sN1OS0vTwIED1a1bNyUmJjoqEgAAjxyH\nHiO/l8zMTO3du1djxoyRJPn7+2vQoEFq27atrly5os6dOys4OPieW/IAACAnp5+1vnv37hy71AsX\nLqxOnTrJ09NTxYoVU82aNXXs2DFnxwIAwJCcXuTJycmqXr267fauXbs0fvx4SbdPkPvpp59UsWJF\nZ8cCAMCQHLZr/dChQ5o4caJOnTolDw8Pmc1mTZ8+XampqSpfvrxtufr162vlypXq2rWrLBaLXnvt\nNZUsWdJRsQAAeKSYrFar1dUhHlRq6hVXRwCAP9UrEzY7fI55I0MdPgcco0QJ31zv48puAAAYGEUO\nAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCA\ngVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR\n5AAAGBhFDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGJhDizwlJUVhYWFavHixJGnkyJFq\n06aNevbsqZ49e2rLli2SpMTERHXq1EmdO3fW8uXLHRkJAIBHioejVnzt2jWNHTtWISEhOcaHDh2q\n5s2b51guPj5eCQkJ8vT0VHR0tMLDw+Xv7++oaAAAPDIctkXu5eWluXPnKjAw8L7LHThwQLVq1ZKv\nr6+8vb0VFBSkffv2OSoWAACPFIcVuYeHh7y9ve8aX7x4sXr16qUhQ4bowoULSktLU7FixWz3FytW\nTKmpqY6KBQDAI8Vhu9bvpV27dvL391eNGjU0Z84czZgxQ/Xq1cuxjNVqdWYkAAAMzalnrYeEhKhG\njRqSpNDQUKWkpCgwMFBpaWm2Zc6dO2d3dzwAALjNqUU+YMAAnThxQpKUlJSkp556SnXq1FFycrIu\nX76sjIwM7du3T/Xr13dmLAAADMthu9YPHTqkiRMn6tSpU/Lw8JDZbFaPHj00ePBgFSxYUD4+Pho/\nfry8vb01bNgwxcTEyGQyqV+/fvL19XVULAAAHikmqwEPSqemXnF1BAD4U70yYbPD55g3MtThc8Ax\nSpTIfQOXK7sBAGBgFDkAAAZGkQMAYGAUOQAABkaRAwBgYBQ5AAAGRpEDAGBgFDkAAAZGkQMAYGAU\nOQAABkaRAwBgYBQ5AAAGRpEDAGBgFDkAAAbmsO8jBx5lf5+wxeFz/G1kM4fPAcD42CIHAMDAKHIA\nAAzMbpEfOnRI//jHPyRJU6ZMUe/evbVnzx6HBwMAAPbZLfJx48apYsWK2rNnj5KTk/Xee+8pLi7O\nGdkAAIAddou8QIECevLJJ7Vp0yZ16dJFVapUkZsbe+QBAMgP7Dby9evXtW7dOm3cuFGNGjXSpUuX\ndPnyZWdkAwAAdtgt8qFDh2r16tUaMmSIChcurEWLFqlPnz5OiAYAAOyx+zny4OBg1axZU4ULF1Za\nWppCQkIUFBTkjGwAAMAOu1vkY8eO1bp163Tp0iV169ZNixcv1pgxY5wQDQAA2GO3yH/44Qd17txZ\n69atU4cOHTR16lQdP37cGdkAAIAddovcarVKkrZs2aLQ0FBJUmZmpmNTAQCAPLFb5BUrVlTLli2V\nkZGhGjVqaOXKlfLz83NGNgAAYIfdk93GjRunlJQUVa5cWZJUpUoVvfHGGw4PBgAA7MvTt5+dO3dO\nR44ckXR7t/qsWbO0efNmu49LSUnRG2+8oT59+qhHjx767bffNGrUKGVlZcnDw0OTJ09WiRIl9Mwz\nz+Q4E37BggVyd3d/yKcEAMDjw26Rv/XWW0pPT9eRI0cUFBSkAwcOaMCAAXZXfO3aNY0dO1YhISG2\nsalTp6pLly5q2bKllixZovnz52v48OG2z6cDAIAHY/cY+ZkzZ/T555+rYsWKiouL09KlS5WcnGx3\nxV5eXpo7d64CAwNtY7GxsYqIiJAkFS1aVJcuXfoD0QEAQJ4vmp6VlaWbN2+qTJkyOnr0qN3lPTw8\n5O3tnWPMx8dH7u7uslgsWrp0qdq0aSPp9u76YcOGqVu3bpo/f/4DPgUAAB5febqy29y5cxUWFqYO\nHTqobNmyys7OfugJLRaLhg8fruDgYNtu9+HDh6tt27YymUzq0aOH6tevr1q1aj30HAAAPC7sFvnA\ngQNlsVjk7u6uevXq6fz583rhhRceesJRo0apQoUK6t+/v23spZdesv09ODhYKSkpFDkAAHmQa5En\nJCTk+qC1a9cqOjr6gSdLTEyUp6enBg4caBs7duyY4uPj9fHHH8tisWjfvn2KjIx84HUDAPA4yrXI\n9+7de98H2ivyQ4cOaeLEiTp16pQ8PDxkNpt1/vx5FShQQD179pQkVa5cWWPGjFGpUqUUHR0tNzc3\nhYaGqnbt2g/xVAAAePzkWuTjx49Xdna23Nxyng9369YteXp62l1xzZo18/yRsrfeeitPywEAgJxy\nPWv95MmTatmypa5cuWIbO3jwoDp27KgLFy44JRwAALi/XIt8/Pjx6t+/v3x9fW1jtWvX1t/+9jdN\nmDDBKeEAAMD95VrkaWlpat269V3jLVu21KlTpxwaCgAA5E2uRZ6VlZXrg65fv+6QMAAA4MHkWuRF\nihTRwYMH7xr//vvvVbRoUYeGAgAAeZPrWetDhgzRgAED1K5dO9WqVUsWi0V79+6V2WzW4sWLnZkR\nAADkItct8tq1a+vrr7+Wm5ubVq1apbVr18rPz0+rVq1S+fLlnZkRAADk4r6XaC1evLgGDx7srCwA\nAOAB5fnbzwAAQP5DkQMAYGB2i/zrr792Rg4AAPAQ7H6N6XfffacXX3wxxxXejOaVCZsdPse8kaEO\nnwMAgP9mt8hv3Lih0NBQVaxYMceXpSxZssShwQAAgH12i/yNN95wRg4AAPAQ7B4jb9Cgga5du6aU\nlBQ1aNBApUqV0nPPPeeMbAAAwA67RT558mQlJCTom2++kSStXr1a48aNc3gwAABgn90i3717t2bM\nmKFChQpJkvr166fDhw87PBgAALDPbpEXKFBAkmQymSRJFotFFovFsakAAECe2D3ZLSgoSCNHjtS5\nc+c0f/58bdiwQQ0aNHBGNgAAYIfdIh8yZIjWr1+vggUL6syZM/rLX/6iF1980RnZAACAHXaLXJIq\nVaqk7OxsmUwmValSxdGZAABAHtkt8okTJ2rTpk2qVauWsrOz9cknn6h169Z8KxoAAPmA3SJPSkrS\nmjVrbFd1y8zMVLdu3ShyAADyAbtnrRcvXlweHv/X956enipTpoxDQwEAgLyxu0VetGhRderUScHB\nwbJardq9e7fKlSunadOmSZIGDRrk8JAAAODe7BZ5uXLlVK5cOdvtZs2aOTIPAAB4AHaLvH///s7I\nAQAAHoLdY+QAACD/cmiRp6SkKCwsTIsXL5Yk/fbbb+rZs6e6d++uQYMGKTMzU5KUmJioTp06qXPn\nzlq+fLkjIwEA8EjJU5FfvXpVkpSWlqY9e/YoOzvb7mOuXbumsWPHKiQkxDYWFxen7t27a+nSpapQ\noYISEhJ07do1xcfHa8GCBVq0aJEWLlyoS5cuPeTTAQDg8WK3yMeOHat169bp0qVL6tatmxYtWqQx\nY8bYXbGXl5fmzp2rwMBA21hSUpJatGghSWrevLl27typAwcOqFatWvL19ZW3t7eCgoK0b9++h39G\nAAA8RuwW+Q8//KDOnTtr3bp16tChg6ZNm6bjx4/bXbGHh4e8vb1zjF2/fl1eXl6SpICAAKWmpiot\nLU3FihWzLVOsWDGlpqY+6PMAAOCxZLfIrVarJGnLli0KDQ2VJNux7T/i9/XmdRwAANzNbpE/+eST\natWqlTIyMlSjRg2tXLlSfn5+DzWZj4+Pbty4IUk6e/asAgMDFRgYqLS0NNsy586dy7E7HgAA5M5u\nkQ8fPlwff/yx5s2bJ0mqUqWKBg4c+FCTNWzYUGazWZK0YcMGNW7cWHXq1FFycrIuX76sjIwM7du3\nT/Xr13+o9QMA8Li57wVhsrOzNWjQIH3xxReyWq3Kzs7WU089pejoaK1evfq+Kz506JAmTpyoU6dO\nycPDQ2azWR9//LFGjhypZcuWqXTp0mrfvr08PT01bNgwxcTEyGQyqV+/fvL19f1TnyQAAI+qXIv8\n22+/1fTp03X8+HE9/fTTtmPXbm5uatSokd0V16xZU4sWLbprfP78+XeNRUZGKjIy8kFyAwAA3afI\nW7durdatW2v69OkaMGCAMzMBAIA8snut9ddee00bN25Uenp6jjPKo6OjHRoMAADYZ7fIX331VZlM\npru+g5wiBwDA9ewW+a1bt/TVV185IwsAAHhAdj9+VqVKFV28eNEZWQAAwAOyu0V+5swZvfjii6pc\nubLc3d1t40uWLHFoMAAAYF+eTnYDAAD5U6671n/44QdJksViuecfAADgerluka9atUpPP/20Zs6c\nedd9JpMpx/eMAwAA18i1yEeNGiVJ97w6G1yj3+bhDp8jPnSSw+cAAPx57J61/ssvv6hXr14KCgrS\ns88+q5iYGP3nP/9xRjYAAGCH3SIfO3asXnnlFW3fvl3//Oc/1a1bN8XGxjojGwAAsMNukVutVjVr\n1kw+Pj4qVKiQwsPDOdkNAIB8wm6R37p1S4cPH7bdPnjwIEUOAEA+Yfdz5CNGjNCwYcN0/vx5SVJg\nYKAmTpzo8GAAAMA+u0Vep04drV+/XleuXJHJZFLhwoWdkQsAAOSB3SI/evSo4uLidPToUZlMJlWr\nVk0DBgxQxYoVnZEPAADch91j5CNHjlSTJk00Y8YMxcXFKTg4WCNGjHBGNgAAYIfdLfKCBQvm+O7x\nypUry2w2OzQUAADIG7tb5MHBwdq4caOuX7+ujIwMbdq0SfXq1ZPValV2drYzMgIAgFzY3SKfOXPm\nPT9uNmPGDJlMJv34448OCQYAAOyzW+R3foYcAADkL3Z3rQMAgPyLIgcAwMAeqsiPHz/+Z+cAAAAP\nwe4xcovFou3bt+vixYuSpMzMTM2aNUubN292eDgAAHB/dov8rbfeUnp6uo4cOaKgoCAdOHBAAwYM\ncEY2AABgh91d62fOnNHnn3+uihUrKi4uTkuXLlVycrIzsgEAADvsbpH/LisrSzdv3lSZMmV09OjR\nh5ps+fLlSkxMtN0+dOiQatasqWvXrsnHx0fS7W9bq1mz5kOtHwCAx43dIg8ODtbcuXMVFhamjh07\nqkyZMg99RbfOnTurc+fOkqTvv/9e69at09GjRzV+/HhVrVr1odYJAMDjzG6RDxw4UBaLRe7u7qpb\nt64uXLigkJCQPzxxfHy8Pv74Yw0dOvQPrwsAgMeV3WPkMTExcnd3lyQ9++yzCg8PV+/evf/QpAcP\nHtQTTzyhEiVKSJLi4uL08ssva/To0bpx48YfWjcAAI+TXLfIExMTFR8fr9OnT6tZs2a28aysLAUE\nBPyhSRMSEtShQwdJUq9evVT6QPj6AAAPbElEQVStWjWVL19esbGxWrJkiWJiYv7Q+gEAeFzkWuRt\n27ZVq1at9M477+T4uJmbm5tKliz5hyZNSkrSu+++K0kKDw+3jYeGhmrt2rV/aN0AADxO7rtr3d3d\nXRMmTJC/v79MJpNMJpNu3rypLl26PPSEZ8+eVaFCheTl5SWr1ao+ffro8uXLkm4X/FNPPfXQ6wYA\n4HFj92S3zz77TLNmzVJmZqZ8fHx08+ZNtWnT5qEnTE1NVbFixSRJJpNJXbp0UZ8+fVSwYEGVLFmS\ni80AAPAA7Bb5+vXrtWPHDsXExGjRokXatGmTTp8+/dAT1qxZU5999pntdsuWLdWyZcuHXh8AAI8z\nu2et/74b/NatW5KkFi1aaNOmTQ4PBgAA7LO7Re7n56fExERVrVpVo0aNUuXKlXXu3DlnZAMAAHbY\nLfKJEyfq/PnzCg8P18KFC3XmzBl9+umnzsgGAADssFvkp0+fVkpKitzd3dW6dWuVLVvWGbkAAEAe\n5FrkN27c0LBhw/Tjjz+qZs2aunr1qn788Uc1atRIH374oby8vJyZEwAA3EOuJ7vNnDlTJUuW1IYN\nGxQXF6d58+Zp8+bNKlCggKZMmeLMjAAAIBe5FvmePXs0cuRIeXj830Z7wYIFFRsbq+3btzslHAAA\nuL9ci9zd3f2eu889PT1VpEgRh4YCAAB5k2uRm0ymXB/0+7ehAQAA18r1ZLf9+/fn+Naz31mtVl28\neNGRmQAAQB7lWuTr1693Zg4AAPAQci3yMmXKODMHAAB4CHavtQ4AAPIvihwAAAOjyAEAMDCKHAAA\nA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADCwXL/9DACM\n4O3dPzt8jo+ee8rhcwAPiy1yAAAMjCIHAMDAKHIAAAzMqcfIk5KSNGjQID311O3jTVWrVtWrr76q\n4cOHy2KxqESJEpo8ebK8vLycGQsAAMNy+sluDRo0UFxcnO32qFGj1L17d0VFRenTTz9VQkKCunfv\n7uxYAAAYkst3rSclJalFixaSpObNm2vnzp0uTgQAgHE4fYv86NGjev3115Wenq7+/fvr+vXrtl3p\nAQEBSk1NdXYkAAAMy6lF/uSTT6p///6KiorSiRMn1KtXL1ksFtv9VqvVmXEAADA8p+5aL1mypFq2\nbCmTyaTy5curePHiSk9P140bNyRJZ8+eVWBgoDMjAQBgaE4t8sTERH3++eeSpNTUVJ0/f14dO3aU\n2WyWJG3YsEGNGzd2ZiQAAAzNqbvWQ0ND9eabb2rTpk26deuWxowZoxo1amjEiBFatmyZSpcurfbt\n2zszEvBY+8/+Dxw+R/l6ox0+B/A4c2qRFy5cWLNmzbprfP78+c6MAQDAI8PlHz8DAAAPjyIHAMDA\nKHIAAAyMIgcAwMAocgAADIwiBwDAwChyAAAMjCIHAMDAKHIAAAyMIgcAwMAocgAADIwiBwDAwChy\nAAAMjCIHAMDAKHIAAAyMIgcAwMAocgAADIwiBwDAwChyAAAMjCIHAMDAKHIAAAyMIgcAwMAocgAA\nDIwiBwDAwChyAAAMjCIHAMDAKHIAAAyMIgcAwMA8nD3hpEmTtHfvXmVlZalv377avHmzDh8+LH9/\nf0lSTEyMmjVr5uxYAAAYklOLfNeuXfr555+1bNkyXbx4UR06dFBwcLCGDh2q5s2bOzMKAACPBKcW\n+XPPPafatWtLkooUKaLr16/LYrE4MwIAAI8Upx4jd3d3l4+PjyQpISFBTZo0kbu7uxYvXqxevXpp\nyJAhunDhgjMjAQBgaE4/Ri5JGzduVEJCgubNm6dDhw7J399fNWrU0Jw5czRjxgyNHj3aFbEAADAc\np5+1vm3bNs2aNUtz586Vr6+vQkJCVKNGDUlSaGioUlJSnB0JAADDcmqRX7lyRZMmTdLs2bNtZ6kP\nGDBAJ06ckCQlJSXpqaeecmYkAAAMzam71teuXauLFy9q8ODBtrGOHTtq8ODBKliwoHx8fDR+/Hhn\nRgIAwNCcWuRdu3ZV165d7xrv0KGDM2MAAPDI4MpuAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGJhL\nruyGx1vKq30cuv6qny1w6PoBID9hixwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwA\nAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMA9XBwAA\nID/pt3m4w+eID530p62LLXIAAAyMIgcAwMAocgAADIxj5ACAP80rEzY7dP3zRoY6dP1GxBY5AAAG\nlm+2yD/66CMdOHBAJpNJb7/9tmrXru3qSAAA5Hv5osi///57HT9+XMuWLdMvv/yit99+W8uWLXN1\nLAAA8r18sWt9586dCgsLkyRVrlxZ6enpunr1qotTAQCQ/+WLIk9LS1PRokVtt4sVK6bU1FQXJgIA\nwBhMVqvV6uoQ7733npo2bWrbKn/ppZf00UcfqWLFii5OBgBA/pYvtsgDAwOVlpZmu33u3DmVKFHC\nhYkAADCGfFHkL7zwgsxmsyTp8OHDCgwMVOHChV2cCgCA/C9fnLUeFBSkZ555Rt26dZPJZFJsbKyr\nIwEAYAj54hg5AAB4OPli1zoAAHg4FDkAAAZGkQMAYGAUuR2XL192dYQHcq9THs6cOeOCJH+OCxcu\nuDrCn2Lnzp2ujvCHZWVl6dSpU8rKynJ1lD/kUXlNPQqMfIqW1WrVhQsXdP78eVdHocjt6d+/v6sj\n5Ml3332n5s2bKyQkRCNGjMhxidvhw4e7MFnebdmyRREREerTp49SUlLUtm1b9ezZU6Ghodq6daur\n4+XZypUrc/xZsWKFYmNjbbeNYty4cba/79ixQ+Hh4Ro8eLBefPFFbdu2zYXJ8m7r1q0aPXq0pNtv\nppo3b65evXopNDRUW7ZscW24BxAUFKSxY8fmi9J4WNu3b1dUVJRefvllHTx4UJ06dVKTJk0UGRmp\n77//3tXx8uzf//63Xn/9dbVt21YtWrRQ3759bf83zp4965JM+eLjZ662ZMmSXO9z1T/Mg5ozZ45W\nrFihIkWKaPny5YqJidFnn30mX19fw7zr/fvf/6758+fr9OnTev311zVz5kxVr15daWlpev3119W0\naVNXR8yT+Ph4+fv758h78+ZNnTx50oWpHtyRI0dsf4+Pj9cXX3yhcuXKKTU1Vf3791fjxo1dmC5v\n4uLiNHv2bEk5n8PFixfVt29fNWvWzLUB8+iZZ55RZGSkhg0bpieeeEIdO3ZUvXr15OFhnF/h8fHx\nWrhwodLT09WzZ08tWLBA1atX16lTp/TWW29p6dKlro6YJ7Gxsfrwww9Vrlw5HTt2TIsWLVJsbKz+\n+c9/6s0339SiRYucnsk4rwIHWrBggUJCQhQYGHjXfUbZjeju7i5/f39JUteuXRUQEKCYmBjNmjVL\nJpPJxenyxsvLS6VLl1bp0qUVGBio6tWrS5KKFy+uAgUKuDhd3n377beaOXOmjhw5opEjR6pMmTLa\ntm2bYfbu/O7O142fn5/KlSsnSSpRooRhCiQrK0uFChWSJPn6+qps2bKSJH9/f8O8wZVu/1s899xz\nWrBggZKTk7V8+XK99957KlSokAICAjRnzhxXR7TL09NTgYGBCgwMVJEiRWz/v8uUKSN3d3cXp8u7\nzMxM2/+FJ5980vaGt0mTJpo+fbpLMhnjf6ODxcfHa9y4cXr33Xfl5eWV476kpCQXpXowQUFB6tu3\nr6ZNmyZvb2+FhYWpQIEC6tOnjy5duuTqeHkSEBCgzz//XDExMfrqq68k3T6+P2/ePJUqVcrF6fKu\nQIECGjJkiI4dO6YPPvhA9erVU3Z2tqtjPbCff/5ZgwYNktVq1fHjx7Vu3TpFRUVp3rx58vX1dXW8\nPImJiVH79u31wgsvyN/fX2+88Ybq1aunpKQkde7c2dXx8uzONx21atVSrVq1JN2+nLVRvmDKz89P\nU6ZM0cWLF1W+fHmNHj1ajRs31r/+9S8FBAS4Ol6eVa1aVUOHDlXt2rW1bds2Pf/885Kkt99+W1Wq\nVHFJJi4I8/9dv35dBQoUkJtbztMGDh8+rGeeecZFqR5MUlKSGjRokGNL6urVq1q7dq26dOniwmR5\nc+PGDW3evFktW7a0jR0+fFi7d+/WSy+9ZKit8jutXLlSW7du1ZQpU1wd5YH893HLChUqqGTJklq9\nerVCQ0NtW7r53aVLl7Rjxw6dOnVKVqtVxYsX1wsvvKCSJUu6OlqeJSQkKDo62tUx/pBr165pxYoV\nKlq0qFq2bKnExETt27dPFSpUUNeuXeXj4+PqiHlitVq1adMm/frrr6pataqaNGkiSfrpp59UrVo1\nl+wBpcgBADAwzloHAMDAKHIAAAyMk92Ax9i5c+c0adIkpaSk2I55DxgwQGfOnNGOHTv08ccfuzgh\nAHsocuAxZbVa1a9fP7Vv395W2EeOHNErr7yiwYMHuzgdgLyiyIHH1M6dO2UymfTyyy/bxqpVq6a1\na9dq06ZNtrHvvvtOn332mby8vGSxWDRp0iSVLVtWCxcuVGJiogoWLChvb29NnjxZmZmZevPNNyXd\n/hRC165dDX+2NZDfUeTAY+rnn3+2fR75Tn5+fjluX758WVOmTFHp0qU1e/ZsLVmyRCNGjFBcXJzM\nZrOKFy+ubdu26dy5c9q5c6cqVaqk999/Xzdv3tTy5cud9XSAxxZFDjym3N3dZbFY7C5XvHhxjRgx\nQlarVampqapXr54kKTo6Wq+++qoiIiIUGRmpihUrysPDQ0uXLtXIkSPVtGlTde3a1dFPA3jscdY6\n8JiqWrWq9u/ff9f4kSNHdP36dUnSrVu3NHjwYI0dO1aLFy9Wz549bcuNGjVK8fHx8vPzU79+/bR1\n61ZVrlxZa9asUdu2bbVz584cywNwDIoceEw1aNBAhQoVynGd7p9//ll/+9vfbNe+zsjIkJubm8qU\nKaObN29q06ZNyszMVHp6uqZPn64nnnhC3bt318svv6zk5GStXr1aycnJatiwoWJjY/Xbb78Z5vsK\nAKNi1zrwGJszZ47Gjx+v1q1by9/fXwUKFNDUqVN19OhRSbe/XKR169aKjo5W6dKlFRMTo+HDh2vH\njh3KyMhQdHS0ihQpIg8PD3344Ye6cOGCYmNj5eXlJavVqr/+9a+G+YIVwKi4RCsAAAbGrnUAAAyM\nIgcAwMAocgAADIwiBwDAwChyAAAMjCIHAMDAKHIAAAyMIgcAwMD+HyRSRRPsinLTAAAAAElFTkSu\nQmCC\n","text/plain":["<matplotlib.figure.Figure at 0x7fbc58e60278>"]},"metadata":{"tags":[]}},{"output_type":"display_data","data":{"image/png":"iVBORw0KGgoAAAANSUhEUgAAAfIAAAFjCAYAAADGq4k2AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMS4yLCBo\ndHRwOi8vbWF0cGxvdGxpYi5vcmcvNQv5yAAAIABJREFUeJzt3X18zfXj//Hnsct2YdiFsqhF9Knm\nYl98zLUhlxWZi5Z9qPX5VOQqzPhi1SgXFaYpJDQpXxO5mM219DUifYWKpHwzdoUZZth2fn/063za\nlznb6pzjbY/77dbttvM+7/N6Pc/Q87wvzvttMpvNZgEAAEOq4ugAAACg4ihyAAAMjCIHAMDAKHIA\nAAyMIgcAwMAocgAADIwixx2vQYMG6ty5s7p06aK2bdvqhRde0DfffGN5/u2339Ynn3xyyzF27dql\n06dP3/S5ZcuWafbs2ZKksLAw7d+/v1z5cnJytHXrVknSt99+q6ioqHK9vqLGjBmjdu3aadeuXWVa\n/4/vsyzLS/Pjjz+qa9eu6tq1q1q3bq3GjRtbHs+fP7/M4/zRTz/9ZPm9p6SkaOLEiRUapywaNGig\njIyMW66TlZWl7du32ywDUIIZuMPVr1/ffObMGbPZbDYXFxebk5OTzS1atDB/9dVXZR7jueeeM+/b\nt8/qeh06dCjTen+0fv1684QJE8r1mr/CQw89ZD558qTd5/2jVatWmQcNGvSnx0lISDDPnz//zwcq\ngz/+fSrNmjVrzJMmTbJLHsDZ0R8kAHsymUzq1q2bLl26pLfffluffvqpYmJiVKdOHQ0ZMkTLli3T\nxx9/LLPZLC8vL7355pvasGGD9uzZoxMnTmjs2LH66aeflJmZqR9++EE9e/bUxYsXlZGRoalTp0qS\n9uzZoylTpuj8+fPq1auXRo0apb1792rixInavHmzJFkez549W6+//rqKioqUn5+vAQMGWNa7evWq\npk6dqr1796pKlSpq166dxo4dKycnJ4WFhelf//qXkpKSlJGRoZ49eyomJuaG93v69GlNmjRJp06d\nkouLi55//nn16tVLkZGRKi4uVlRUlCZOnKh27dpJki5cuKC2bdtq69at8vPzkyRNnz5dhYWFqlq1\naon3+bu5c+dalkdGRiosLEybNm3SqVOn1KxZM7399tsymUzl+nNavny5PvroI127dk0hISGaOnWq\n3NzclJaWpunTp+vatWsym80aNWqUTCaTFi1aJFdXV+Xm5iooKEgpKSlatGiRxowZo/vvv1/79+/X\nL7/8orp16yohIUHu7u7asWOHJk+eLC8vLz377LOaMmWKUlNTdffdd5fIsn37dk2dOlWurq7q27fv\nDe99/fr1Kiws1IMPPqi33npLJ06c0BtvvKHi4mJdvnzZ8vdsyZIlKioq0t13360ZM2bonnvuKdfv\nBCgNu9ZRKYWFhengwYMqKCiwLLt06ZLmzJmjlStXKiUlRVFRUdqxY4dGjhypmjVraubMmerevbsk\naefOnVqwYIEGDx58w9hHjhzRqlWr9Nlnn+mTTz7RDz/8UGqORx55RAMHDlSXLl00a9asEs8tXbpU\nGRkZ2rBhg1avXq39+/dr/fr1luf37dunFStWaNWqVVq2bNlNd/dOmjRJzZs3V2pqqubPn68pU6bo\n1KlTSkxMlCQlJiZaSlySfHx8FBoaquTkZMuyzZs3q0ePHlZ+o/+2bds2LV68WKmpqdqzZ48OHDhQ\n5tdKv30QmjdvnhITE7V161a5ublp7ty5kn77UDFp0iQlJycrISFBmzZtUufOndWhQwc9++yzio6O\nvmG81NRUzZkzR5s3b1ZmZqa2bdum69eva/z48XrzzTeVnJys48eP6+rVqze89vr165o4caLi4uKU\nnJyswsJCy3MHDx60/P43bdqky5cva/ny5WrYsKEGDBigbt266e2331ZmZqbeeOMNLV26VJs3b9Y9\n99yj9957r1y/E+BWKHJUSl5eXpYtpt+5ubnJZDIpKSlJOTk56tatm/75z3/e9PWNGjVSjRo1bvrc\n448/LicnJ/n6+qpZs2YljseXx44dO9SvXz85OzvL3d1djz/+uP77v//7hnlq1qwpX19fnTlzpsTr\nr1+/rt27dysiIkKSFBgYqL///e/as2fPLeft2bOnNmzYIEn64YcfVFxcrMaNG5c5d9euXeXu7i4P\nDw/df//9N+SyZvv27erRo4f8/f1lMpk0YMAAy54MX19frV69WidOnNADDzygt956y+p47du3l4+P\nj1xcXFS/fn2dPn1aJ06ckNlsVqtWrSRJAwcOlPkmV6v++eefVVxcrNDQUElS7969Lc81atRI27Zt\nk5eXl5ycnBQSEqJff/31hjFq1qyp/fv3q2bNmpKkpk2b6tSpU+X6nQC3wq51VEq/72r29va2LHNx\ncdGSJUv0/vvva+7cuWrQoIFiY2PVoEGDG17v4+NT6th/LHhvb2/l5eVVKOO5c+dKzOPj46OzZ89a\nHnt5eVl+dnJyUlFRUYnX5+bmymw2l3iPVatW1blz5245b1hYmCZNmqRff/1VW7ZsUdeuXcuV21ou\na/Ly8rR9+3bt3LlTkmQ2m3X9+nVJ0rRp0/Tee+9p0KBB8vDw0JgxY9S5c+dbjvfH91+lShUVFxcr\nLy+vxO/295L9vy5cuFDi/VStWtXyc35+vt544w3t27dP0m+/706dOt0wRmFhoWbPnq3t27dbPjzW\nq1fP2q8BKDO2yFEppaamqnnz5nJ1dS2x/OGHH1Z8fLzS0tLUunVrxcbGlnvsCxculPjZx8fnhkIr\nS7n7+fkpNzfX8jg3N9dy3LosqlevripVqpTIk5ubK19f31u+zsPDQx06dFBKSopSU1MthxPsJSAg\nQOHh4UpJSbFk2LZtmyTJ399fkydP1q5duzRhwgRFR0frypUr5Z7Dy8urxN6YrKysm65XtWpVXbp0\nyfL4/Pnzlp8//PBDpaena/Xq1UpNTVV4ePhNx1i/fr127dql5cuXKzU1VUOGDCl3XuBWKHJUKmaz\nWSkpKVq6dKlGjRpV4rmjR49q+PDhunbtmlxdXfXoo49aTtJydnbWxYsXyzTHhg0bVFxcrLNnz+rr\nr79W06ZN5e/vr+zsbJ09e1ZFRUVat26dZf3Sxm7fvr2SkpIsJ8J9/vnnJY5nW+Ps7KzWrVtrxYoV\nkqT//d//1f79+9WyZUurr+3Zs6c++eQTFRQU6NFHHy3znH+Fjh07KjU11VKamzZt0qJFi3Tt2jVF\nRkYqOztbkhQcHCwnJyc5OTnJxcWlXHs+HnjgAV25csXylbXff0f/V1BQkIqLiy3rffbZZ5bnzp07\np7p168rDw0O//vqrvvjiC+Xn50v6be/O73+mZ8+eVWBgoKpXr65z584pJSXFsh7wV2DXOiqFyMhI\nOTk56dKlS6pbt64WLFig4ODgEuvUr19f9957r3r27CkXFxd5enpq8uTJkqQuXbrolVde0fDhw63O\nFRwcrPDwcJ07d06DBg2y7Ebt06ePevXqpVq1aunJJ5/U999/L0lq1aqVFi9erD59+pQ4WSsyMlK/\n/vqrevToIZPJpK5du6pbt27let+vvfaaJk6cqM8++0wuLi6aMmVKmc6Wbt26tS5duqSnn366XPP9\nFRo2bKioqCg988wzMpvN8vPz0+uvvy5XV1c99dRTGjRokMxms5ycnBQbGytXV1eFhYUpOjpa6enp\nZfqg4ubmptjYWEVHR6tq1ap67rnnJOmGs+tdXV31+uuvKzo6Wu7u7goPD5e7u7sk6emnn9aIESPU\ntWtXNWjQQBMmTNDw4cOVmJio1q1ba+nSperfv7/mzp2r5ORkde7cWXXq1NErr7yil156STNmzLjp\nyXlAeZnMNzvDA0Cl16NHD82ZM6dSHM+9ePGimjZtqm+++UYeHh6OjgOUC7vWAdxgw4YN8vf3v6NL\nvFevXkpNTZUkJScnq379+pQ4DIktcgAlPPvsszp//rzi4+NVp04dR8exma+++kpxcXG6du2avL29\n9eqrr9r9fADgr2DTIj927JiGDBmiwYMHa+DAgbp+/bpiYmJ08uRJeXp6Kj4+Xj4+Plq7dq2WLl2q\nKlWqqF+/fjdcPQkAANyczXat5+fnKy4uznIhBUn6r//6L1WvXl1JSUnq3r279u/fr/z8fCUkJGjJ\nkiVKTEzU0qVLS3zlBgAAlM5mRe7q6qqFCxcqICDAsmz79u164oknJEn9+/dXx44ddfDgQQUHB8vb\n21vu7u4KCQkp9yUdAQCorGz29TNnZ2c5O5ccPj09XV988YVmzpwpPz8/xcbGKicnp8SVsGrUqGH5\nnmhpsrPL9n1eAADuBP7+3qU+Z9ez1s1ms4KCgpSYmKgHH3zwpvce5tw7AADKzq5F7ufnp2bNmkn6\n7YITx48fV0BAgHJycizrZGVlldgdDwAASmfXIm/btq127dol6bdbPQYFBalRo0Y6dOiQ8vLydPny\nZR04cEBNmza1ZywAAAzLZl8/O3z4sKZPn6709HQ5OzurZs2aeuuttzR16lRlZ2fLw8ND06dPl5+f\nn1JSUrRo0SKZTCYNHDjQckJcaThGDgCoTG51jNyQF4ShyAEAlcltc7IbAAD4a1HkAAAYGEUOAICB\nUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYmM3ufgYAKLvnpm2z+RwfxoTZfA7YH1vkAAAYGEUO\nAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCA\ngVHkAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICB2bTIjx07pk6dOmnZ\nsmUllu/atUsNGjSwPF67dq369Omjvn37auXKlbaMBADAHcXZVgPn5+crLi5OoaGhJZZfvXpVCxYs\nkL+/v2W9hIQEJSUlycXFReHh4ercubOqVatmq2gAANwxbLZF7urqqoULFyogIKDE8vfff18RERFy\ndXWVJB08eFDBwcHy9vaWu7u7QkJCdODAAVvFAgDgjmKzInd2dpa7u3uJZT///LN++OEHdevWzbIs\nJydHNWrUsDyuUaOGsrOzbRULAIA7il1PdnvzzTc1fvz4W65jNpvtlAYAAOOzW5FnZmbqxIkTGjNm\njPr166esrCwNHDhQAQEBysnJsayXlZV1w+54AABwczY72e3/qlmzprZs2WJ5HBYWpmXLlqmgoEAT\nJ05UXl6enJycdODAAU2YMMFesQAAMDSbFfnhw4c1ffp0paeny9nZWampqZo7d+4NZ6O7u7tr9OjR\nioqKkslk0tChQ+Xt7W2rWAAA3FFMZgMelM7OvujoCADwl3pu2jabz/FhTJjN54Bt+PuXvoHLld0A\nADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEAMDCKHAAAA6PIAQAw\nMIocAAADo8gBADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEAMDCK\nHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEAMDCKHAAAA7NpkR87dkydOnXSsmXLJElnzpzR\n4MGDNXDgQA0ePFjZ2dmSpLVr16pPnz7q27evVq5cactIAADcUWxW5Pn5+YqLi1NoaKhl2ezZs9Wv\nXz8tW7ZMnTt31uLFi5Wfn6+EhAQtWbJEiYmJWrp0qXJzc20VCwCAO4rNitzV1VULFy5UQECAZVls\nbKy6dOkiSapevbpyc3N18OBBBQcHy9vbW+7u7goJCdGBAwdsFQsAgDuKzYrc2dlZ7u7uJZZ5eHjI\nyclJRUVFWr58uR5//HHl5OSoRo0alnVq1Khh2eUOAABuze4nuxUVFSk6OlotWrQosdv9d2az2d6R\nAAAwLLsX+fjx43Xffffp5ZdfliQFBAQoJyfH8nxWVlaJ3fEAAKB0di3ytWvXysXFRcOHD7csa9So\nkQ4dOqS8vDxdvnxZBw4cUNOmTe0ZCwAAw3K21cCHDx/W9OnTlZ6eLmdnZ6Wmpurs2bNyc3NTZGSk\nJKlu3bp69dVXNXr0aEVFRclkMmno0KHy9va2VSwAAO4oJrMBD0pnZ190dAQA+Es9N22bzef4MCbM\n5nPANvz9S9/A5cpuAAAYGEUOAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYGEUOAICBUeQA\nABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYmM1uYwrcyd6btsPmc7wU097mcwAwPrbIAQAwMIocAAAD\no8gBADAwq0V++PBhbd++XZI0a9YsDRo0SPv377d5MAAAYJ3VIp8yZYqCgoK0f/9+HTp0SJMmTVJ8\nfLw9sgEAACusFrmbm5vuv/9+bd26Vf369VO9evVUpQp75AEAuB1YbeQrV65o48aN2rJli1q3bq3c\n3Fzl5eXZIxsAALDCapG/8sorWrdunUaNGiUvLy8lJiZq8ODBdogGAACssXpBmBYtWujRRx+Vl5eX\ncnJyFBoaqpCQEHtkAwAAVljdIo+Li9PGjRuVm5urAQMGaNmyZXr11VftEA0AAFhjtci/++479e3b\nVxs3blTv3r01e/ZsnTx50h7ZAACAFVaL3Gw2S5J27NihsLAwSdK1a9dsmwoAAJSJ1SIPCgpS9+7d\ndfnyZf3tb3/TmjVr5OPjY49sAADACqsnu02ZMkXHjh1T3bp1JUn16tXTkCFDbB4MAABYV6Yru2Rl\nZSk5OVlr1qzRd999p5EjR5Zp8GPHjqlTp05atmyZJOnMmTOKjIxURESERowYYdlFv3btWvXp00d9\n+/bVypUrK/hWAACofKxukY8dO1YXLlzQ0aNHFRISooMHD2rYsGFWB87Pz1dcXJxCQ0Mty+Lj4xUR\nEaFu3brpnXfeUVJSknr16qWEhAQlJSXJxcVF4eHh6ty5s6pVq/bn3hkAAJWA1S3yjIwMLVq0SEFB\nQYqPj9fy5ct16NAhqwO7urpq4cKFCggIsCzbu3evOnbsKEnq0KGD0tLSdPDgQQUHB8vb21vu7u4K\nCQnRgQMH/sRbAgCg8rC6Rf67wsJCXb16VYGBgTp+/Lj1gZ2d5exccvgrV67I1dVVkuTr66vs7Gzl\n5OSoRo0alnVq1Kih7OzsssYCAKBSK9OV3RYuXKhOnTqpd+/euvfee1VcXPynJ/79a21lXQ4AAG5k\ntciHDx+uoqIiOTk5qUmTJjp79qxatWpVock8PDxUUFAgd3d3ZWZmKiAgQAEBAcrJybGsk5WVpcaN\nG1dofAAAKptSizwpKanUFyUnJys8PLzck7Vs2VKpqal68skntWnTJrVp00aNGjXSxIkTlZeXJycn\nJx04cEATJkwo99gAAFRGpRb5119/fcsXWivyw4cPa/r06UpPT5ezs7NSU1P11ltvKSYmRitWrFCt\nWrXUq1cvubi4aPTo0YqKipLJZNLQoUPl7e1dsXcDAEAlYzLf4qB0cXGxqlQpeWL79evX5eLiYvNg\nt5KdfdGh8wPvTdth8zleimlv8zlw+3hu2jabz/FhTJjN54Bt+PuXvoFb6tfPTp06pe7du+vixX+X\n5rfffqunnnpK586d+2sTAgCACim1yN988029/PLLJXZzN2zYUC+99JKmTZtml3AAAODWSi3ynJwc\n9ezZ84bl3bt3V3p6uk1DAQCAsim1yAsLC0t90ZUrV2wSBgAAlE+pRV61alV9++23Nyz/6quvVL16\ndZuGAgAAZVPq189GjRqlYcOG6cknn1RwcLCKior09ddfKzU11XI3MwAA4FilbpE3bNhQq1atUpUq\nVfT5558rOTlZPj4++vzzz1WnTh17ZgQAAKW45SVa/fz8ynzvcQAAYH9Wb2MKAABuXxQ5AAAGZrXI\nV61aZY8cAACgAqwW+ebNm0tcphUAANw+rN6PvKCgQGFhYQoKCipxs5SPP/7YpsH+StyMAABwp7Ja\n5EOGDLFHDgAAUAFWd603b95c+fn5OnbsmJo3b667775bzZo1s0c2AABghdUinzlzppKSkvTZZ59J\nktatW6cpU6bYPBgAALDOapHv27dP7777rjw9PSVJQ4cO1ZEjR2weDAAAWGe1yN3c3CRJJpNJklRU\nVKSioiLbpgIAAGVi9WS3kJAQxcTEKCsrS4sXL9amTZvUvHlze2QDAABWWC3yUaNGKSUlRXfddZcy\nMjL07LPP6rHHHrNHNgAAYIXVIpekBx54QMXFxTKZTKpXr56tMwEAgDKyWuTTp0/X1q1bFRwcrOLi\nYr399tvq2bMnd0UDAOA2YLXI9+7dqw0bNliu6nbt2jUNGDCAIgcA4DZg9ax1Pz8/OTv/u+9dXFwU\nGBho01AAAKBsrG6RV69eXX369FGLFi1kNpu1b98+1a5dW3PmzJEkjRgxwuYhAQDAzVkt8tq1a6t2\n7dqWx+3bt7dlHgAAUA5Wi/zll1+2Rw4AAFABVo+RAwCA21eZvkf+V7l8+bLGjRunCxcu6Pr16xo6\ndKj8/f316quvSpIaNGig1157zZ6RAAAwtDIV+aVLl+Tl5aWcnBz98ssvCgkJUZUq5d+YX716tYKC\ngjR69GhlZmZq0KBB8vf314QJE9SwYUONHj1aO3fuVLt27co9NgAAlZHVNo6Li9PGjRuVm5urAQMG\nKDEx0bIFXV7Vq1dXbm6uJCkvL0/VqlVTenq6GjZsKEnq0KGD0tLSKjQ2AACVkdUi/+6779S3b19t\n3LhRvXv31pw5c3Ty5MkKTdajRw+dPn1anTt31sCBAxUdHa2qVatanvf19VV2dnaFxgYAoDKyWuRm\ns1mStGPHDoWFhUn67epuFfH555+rVq1a2rx5s5YuXaqxY8fedC4AAFA2Vo+R33///erRo4eqV6+u\nv/3tb1qzZo18fHwqNNmBAwfUunVrSdJDDz2kq1evqrCw0PJ8ZmamAgICKjQ2AACVkdUij46OVmZm\npurWrStJqlevnoYPH16hye677z4dPHhQXbp0UXp6ujw9PRUYGKj9+/eradOm2rRpkyIjIys0NgAA\nldEti7y4uFgjRozQRx99JLPZrOLiYj344IMKDw/XunXryj1Z//79NWHCBA0cOFCFhYV69dVX5e/v\nr8mTJ6u4uFiNGjVSy5YtK/xmAACobEot8vXr12vu3Lk6efKkHn74Ycvx6ypVqlh2j5eXp6en5Rrt\nf7R8+fIKjQcAQGVXapH37NlTPXv21Ny5czVs2DB7ZgIAAGVk9Rj5v/71L23ZskUXLlwocVZ5eHi4\nTYMBAADrrBb5888/L5PJdMM9yClyAAAcz2qRX79+XZ9++qk9sgAAgHKyekGYevXq6fz58/bIAgAA\nysnqFnlGRoYee+wx1a1bV05OTpblH3/8sU2DAQAA68p0shsAALg9lbpr/bvvvpMkFRUV3fQ/AADg\neKVukX/++ed6+OGHNW/evBueM5lMCg0NtWkwAABgXalFPn78eElSYmKi3cIAAIDysXqM/KefftJr\nr72mw4cPy2QyqXHjxoqNjVWdOnXskQ9/MHRbtM3nSAibYfM5AAB/HatfP4uLi9Nzzz2nL7/8Ul98\n8YUGDBig2NhYe2QDAABWWC1ys9ms9u3by8PDQ56enurcuTMnuwEAcJuwWuTXr1/XkSNHLI+//fZb\nihwAgNuE1WPk48aN0+jRo3X27FlJUkBAgKZPn27zYAAAwDqrRd6oUSOlpKTo4sWLMplM8vLyskcu\nAABQBlaL/Pjx44qPj9fx48dlMpnUoEEDDRs2TEFBQfbIBwAAbsHqMfKYmBi1bdtW7777ruLj49Wi\nRQuNGzfOHtkAAIAVVrfI77rrrhL3Hq9bt65SU1NtGgoAAJSN1S3yFi1aaMuWLbpy5YouX76srVu3\nqkmTJjKbzSouLrZHRgAAUAqrW+Tz5s276dfN3n33XZlMJn3//fc2CQYAAKyzWuR//A45AAC4vVjd\ntQ4AAG5fFDkAAAZWoSI/efLkX50DAABUgNVj5EVFRfryyy91/vx5SdK1a9f0/vvva9u2bTYPBwAA\nbs1qkY8dO1YXLlzQ0aNHFRISooMHD2rYsGH2yAYAAKywums9IyNDixYtUlBQkOLj47V8+XIdOnTI\nHtkAAIAVZT5GXlhYqKtXryowMFDHjx+3ZSYAAFBGVnett2jRQgsXLlSnTp301FNPKTAw8E9d0W3t\n2rX64IMP5OzsrOHDh6tBgwaKjo5WUVGR/P39NXPmTLm6ulZ4fAAAKhOrRT58+HAVFRXJyclJjRs3\n1rlz5xQaGlqhyc6fP6+EhAStWrVK+fn5mjt3rlJTUxUREaFu3brpnXfeUVJSkiIiIio0PgAAlY3V\nXetRUVFycnKSJP3Hf/yHOnfurEGDBlVosrS0NIWGhsrLy0sBAQGKi4vT3r171bFjR0lShw4dlJaW\nVqGxAQCojErdIl+7dq0SEhJ0+vRptW/f3rK8sLBQvr6+FZrs1KlTKigo0Isvvqi8vDwNGzZMV65c\nsexK9/X1VXZ2doXGBgCgMiq1yJ944gn16NFD//mf/1ni62ZVqlRRzZo1Kzxhbm6u3n33XZ0+fVr/\n+Mc/ZDabLc/98WcAAGDdLXetOzk5adq0aapWrZpMJpNMJpOuXr2qfv36VWgyX19fNWnSRM7OzqpT\np448PT3l6empgoICSVJmZqYCAgIqNDYAAJWR1WPkH3zwgdq1a6euXbvqqaeeUu/evfXwww9XaLLW\nrVtrz549Ki4u1vnz55Wfn6+WLVsqNTVVkrRp0ya1adOmQmMDAFAZWT1rPSUlRbt371ZUVJQSExO1\ndetWnT59ukKT1axZU126dLFs0U+cOFHBwcEaN26cVqxYoVq1aqlXr14VGhsAgMrIapF7enrK1dVV\n169flyR17NhRgwcPVmRkZIUmHDBggAYMGFBi2eLFiys0FgAAlZ3VIvfx8dHatWtVv359jR8/XnXr\n1lVWVpY9sgEAACusFvn06dN19uxZde7cWUuXLlVGRobeeecde2QDAABWWC3y06dP69ixY3JyclLP\nnj1177332iMXAAAog1KLvKCgQKNHj9b333+vRx99VJcuXdL333+v1q1ba+rUqVwPHQCA20CpXz+b\nN2+eatasqU2bNik+Pl4ffvhk1moLAAANj0lEQVShtm3bJjc3N82aNcueGQEAQClKLfL9+/crJiZG\nzs7/3mi/6667FBsbqy+//NIu4QAAwK2VWuROTk433X3u4uKiqlWr2jQUAAAom1KL3GQylfqi3++G\nBgAAHKvUk92++eabEnc9+53ZbNb58+dtmQkAAJRRqUWekpJizxwAAKACSi3ywMBAe+YAAAAVYPXu\nZwAA4PZFkQMAYGAUOQAABkaRAwBgYBQ5AAAGRpEDAGBgFDkAAAZGkQMAYGAUOQAABkaRAwBgYBQ5\nAAAGRpEDAGBgFDkAAAZW6t3PAMAIJuz70eZzvNHsQZvPAVQUW+QAABgYRQ4AgIFR5AAAGJhDiryg\noECdOnXSZ599pjNnzigyMlIREREaMWKErl275ohIAAAYkkOK/L333pOPj48kKT4+XhEREVq+fLnu\nu+8+JSUlOSISAACGZPci/+mnn3T8+HG1b99ekrR371517NhRktShQwelpaXZOxIAAIZl9yKfPn26\nYmJiLI+vXLkiV1dXSZKvr6+ys7PtHQkAAMOya5GvWbNGjRs3Vu3atW/6vNlstmccAAAMz64XhNmx\nY4d+/fVX7dixQxkZGXJ1dZWHh4cKCgrk7u6uzMxMBQQE2DMSAACGZtcinz17tuXnuXPnKjAwUN98\n841SU1P15JNPatOmTWrTpo09IwGV2v9+87rN56jTZLLN5wAqM4d/j3zYsGFas2aNIiIilJubq169\nejk6EgAAhuGwa60PGzbM8vPixYsdFQMAAENz+BY5AACoOIocAAADo8gBADAwihwAAAOjyAEAMDCK\nHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwA\nAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAAD\no8gBADAwihwAAANztveEM2bM0Ndff63CwkK98MILCg4OVnR0tIqKiuTv76+ZM2fK1dXV3rEAADAk\nuxb5nj179OOPP2rFihU6f/68evfurdDQUEVERKhbt2565513lJSUpIiICHvGAgDAsOy6a71Zs2aa\nM2eOJKlq1aq6cuWK9u7dq44dO0qSOnTooLS0NHtGAgDA0Oxa5E5OTvLw8JAkJSUlqW3btrpy5Ypl\nV7qvr6+ys7PtGQkAAENzyMluW7ZsUVJSkiZPnlxiudlsdkQcAAAMy+5FvmvXLr3//vtauHChvL29\n5eHhoYKCAklSZmamAgIC7B0JAADDsmuRX7x4UTNmzND8+fNVrVo1SVLLli2VmpoqSdq0aZPatGlj\nz0gAABiaXc9aT05O1vnz5zVy5EjLsmnTpmnixIlasWKFatWqpV69etkzEgAAhmbXIu/fv7/69+9/\nw/LFixfbMwYAAHcMruwGAICBUeQAABgYRQ4AgIFR5AAAGBhFDgCAgVHkAAAYmN1vYwoce36wTcev\n/8ESm44PALcTtsgBADAwihwAAAOjyAEAMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAOjyAEA\nMDCKHAAAA6PIAQAwMIocAAADo8gBADAwihwAAAPjNqYAAPzB0G3RNp8jIWzGXzYWW+QAABgYRQ4A\ngIFR5AAAGBhFDgCAgVHkAAAYGGetAwD+Ms9N22bT8T+MCbPp+EZ02xT5G2+8oYMHD8pkMmnChAlq\n2LChoyMBAHDbuy2K/KuvvtLJkye1YsUK/fTTT5owYYJWrFjh6FgAANz2botj5GlpaerUqZMkqW7d\nurpw4YIuXbrk4FQAANz+bosiz8nJUfXq1S2Pa9SooezsbAcmAgDAGExms9ns6BCTJk1Su3btLFvl\nTz/9tN544w0FBQU5OBkAALe322KLPCAgQDk5OZbHWVlZ8vf3d2AiAACM4bYo8latWik1NVWSdOTI\nEQUEBMjLy8vBqQAAuP3dFmeth4SE6JFHHtGAAQNkMpkUGxvr6EgAABjCbXGMHAAAVMxtsWsdAABU\nDEUOAICBUeQAABgYRW5FXl6eoyOUy81OecjIyHBAkr/GuXPnHB3hL5GWluboCH9aYWGh0tPTVVhY\n6Ogof8qd8nfqTmDkU7TMZrPOnTuns2fPOjoKRW7Nyy+/7OgIZbJ582Z16NBBoaGhGjduXIlL3EZH\nRzswWdnt2LFDXbp00eDBg3Xs2DE98cQTioyMVFhYmHbu3OnoeGW2Zs2aEv+tXr1asbGxlsdGMWXK\nFMvPu3fvVufOnTVy5Eg99thj2rVrlwOTld3OnTs1efJkSb99mOrQoYP+8Y9/KCwsTDt27HBsuHII\nCQlRXFzcbVEaFfXll1+qW7dueuaZZ/Ttt9+qT58+atu2rbp27aqvvvrK0fHK7Oeff9aLL76oJ554\nQh07dtQLL7xg+beRmZnpkEy3xdfPHO3jjz8u9TlH/cGU14IFC7R69WpVrVpVK1euVFRUlD744AN5\ne3sb5lPve++9p8WLF+v06dN68cUXNW/ePD300EPKycnRiy++qHbt2jk6YpkkJCSoWrVqJfJevXpV\np06dcmCq8jt69Kjl54SEBH300UeqXbu2srOz9fLLL6tNmzYOTFc28fHxmj9/vqSS7+H8+fN64YUX\n1L59e8cGLKNHHnlEXbt21ejRo3XPPffoqaeeUpMmTeTsbJz/hSckJGjp0qW6cOGCIiMjtWTJEj30\n0ENKT0/X2LFjtXz5ckdHLJPY2FhNnTpVtWvX1okTJ5SYmKjY2Fh98cUXGjNmjBITE+2eyTh/C2xo\nyZIlCg0NVUBAwA3PGWU3opOTk6pVqyZJ6t+/v3x9fRUVFaX3339fJpPJwenKxtXVVbVq1VKtWrUU\nEBCghx56SJLk5+cnNzc3B6cru/Xr12vevHk6evSoYmJiFBgYqF27dhlm787v/vj3xsfHR7Vr15Yk\n+fv7G6ZACgsL5enpKUny9vbWvffeK0mqVq2aYT7gSr/9WTRr1kxLlizRoUOHtHLlSk2aNEmenp7y\n9fXVggULHB3RKhcXFwUEBCggIEBVq1a1/PsODAyUk5OTg9OV3bVr1yz/Fu6//37LB962bdtq7ty5\nDslkjH+NNpaQkKApU6Zo4sSJcnV1LfHc3r17HZSqfEJCQvTCCy9ozpw5cnd3V6dOneTm5qbBgwcr\nNzfX0fHKxNfXV4sWLVJUVJQ+/fRTSb8d3//www919913Ozhd2bm5uWnUqFE6ceKEXn/9dTVp0kTF\nxcWOjlVuP/74o0aMGCGz2ayTJ09q48aN6tatmz788EN5e3s7Ol6ZREVFqVevXmrVqpWqVaumIUOG\nqEmTJtq7d6/69u3r6Hhl9scPHcHBwQoODpb02+WsjXKDKR8fH82aNUvnz59XnTp1NHnyZLVp00b/\n8z//I19fX0fHK7P69evrlVdeUcOGDbVr1y79/e9/lyRNmDBB9erVc0gmLgjz/125ckVubm6qUqXk\naQNHjhzRI4884qBU5bN37141b968xJbUpUuXlJycrH79+jkwWdkUFBRo27Zt6t69u2XZkSNHtG/f\nPj399NOG2ir/ozVr1mjnzp2aNWuWo6OUy/89bnnfffepZs2aWrduncLCwixbure73Nxc7d69W+np\n6TKbzfLz81OrVq1Us2ZNR0crs6SkJIWHhzs6xp+Sn5+v1atXq3r16urevbvWrl2rAwcO6L777lP/\n/v3l4eHh6IhlYjabtXXrVv3yyy+qX7++2rZtK0n64Ycf1KBBA4fsAaXIAQAwMM5aBwDAwChyAAAM\njJPdgEosKytLM2bM0LFjxyzHvIcNG6aMjAzt3r1bb731loMTArCGIgcqKbPZrKFDh6pXr16Wwj56\n9Kiee+45jRw50sHpAJQVRQ5UUmlpaTKZTHrmmWcsyxo0aKDk5GRt3brVsmzz5s364IMP5OrqqqKi\nIs2YMUP33nuvli5dqrVr1+quu+6Su7u7Zs6cqWvXrmnMmDGSfvsWQv/+/Q1/tjVwu6PIgUrqxx9/\ntHwf+Y98fHxKPM7Ly9OsWbNUq1YtzZ8/Xx9//LHGjRun+Ph4paamys/PT7t27VJWVpbS0tL0wAMP\n6LXXXtPVq1e1cuVKe70doNKiyIFKysnJSUVFRVbX8/Pz07hx42Q2m5Wdna0mTZpIksLDw/X888+r\nS5cu6tq1q4KCguTs7Kzly5crJiZG7dq1U//+/W39NoBKj7PWgUqqfv36+uabb25YfvToUV25ckWS\ndP36dY0cOVJxcXFatmyZIiMjLeuNHz9eCQkJ8vHx0dChQ7Vz507VrVtXGzZs0BNPPKG0tLQS6wOw\nDYocqKSaN28uT0/PEtfp/vHHH/XSSy9Zrn19+fJlValSRYGBgbp69aq2bt2qa9eu6cKFC5o7d67u\nueceRURE6JlnntGhQ4e0bt06HTp0SC1btlRsbKzOnDljmPsVAEbFrnWgEluwYIHefPNN9ezZU9Wq\nVZObm5tmz56t48ePS/rt5iI9e/ZUeHi4atWqpaioKEVHR2v37t26fPmywsPDVbVqVTk7O2vq1Kk6\nd+6cYmNj5erqKrPZrH/+85+GucEKYFRcohUAAANj1zoAAAZGkQMAYGAUOQAABkaRAwBgYBQ5AAAG\nRpEDAGBgFDkAAAZGkQMAYGD/D8aMR7U95Z3mAAAAAElFTkSuQmCC\n","text/plain":["<matplotlib.figure.Figure at 0x7fbc56600160>"]},"metadata":{"tags":[]}}]},{"metadata":{"id":"EjFY9R7OdXXl","colab_type":"text"},"cell_type":"markdown","source":["#One hot encoding of the featurres( Gene,Variation,Text) of different data sets"]},{"metadata":{"id":"f5ra0mBfX8Nu","colab_type":"code","colab":{}},"cell_type":"code","source":["G_V = CountVectorizer()\n","train_gene_onehotCoding = G_V.fit_transform(x_train['Gene'])\n","test_gene_onehotCoding  = G_V.transform(x_test['Gene'])\n","cv_gene_onehotCoding    = G_V.transform(x_cv['Gene'])"],"execution_count":0,"outputs":[]},{"metadata":{"id":"2Ys96BhhIvWi","colab_type":"code","outputId":"0b7c9e45-4405-44e9-b095-084c5c9cdd75","executionInfo":{"status":"ok","timestamp":1543824833000,"user_tz":-330,"elapsed":614,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":69}},"cell_type":"code","source":["print(train_gene_onehotCoding.shape)\n","print(test_gene_onehotCoding.shape)\n","print(cv_gene_onehotCoding.shape)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["(2124, 232)\n","(665, 232)\n","(532, 232)\n"],"name":"stdout"}]},{"metadata":{"id":"owTrIIj3avwk","colab_type":"code","colab":{}},"cell_type":"code","source":["V_V = CountVectorizer()\n","train_variation_onehotCoding = V_V.fit_transform(x_train['Variation'])\n","test_variation_onehotCoding  = V_V.transform(x_test['Variation'])\n","cv_variation_onehotCoding    = V_V.transform(x_cv['Variation'])"],"execution_count":0,"outputs":[]},{"metadata":{"id":"BqGmm5MnaxAe","colab_type":"code","colab":{}},"cell_type":"code","source":["T_V = CountVectorizer()\n","train_text_onehotCoding = T_V.fit_transform(x_train['TEXT'])\n","train_text_onehotCoding = normalize(train_text_onehotCoding, axis=0)\n","\n","test_text_onehotCoding = T_V.transform(x_test['TEXT'])\n","test_text_onehotCoding = normalize(test_text_onehotCoding, axis=0)\n","\n","cv_text_onehotCoding = T_V.transform(x_cv['TEXT'])\n","cv_text_onehotCoding = normalize(cv_text_onehotCoding, axis=0)\n"],"execution_count":0,"outputs":[]},{"metadata":{"id":"anMfJG--KeQT","colab_type":"code","outputId":"8c7aac37-b3bd-4710-f7e3-16c6150e5e3e","executionInfo":{"status":"ok","timestamp":1543824860185,"user_tz":-330,"elapsed":947,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":69}},"cell_type":"code","source":["print(train_text_onehotCoding.shape)\n","print(test_text_onehotCoding.shape)\n","print(cv_text_onehotCoding.shape)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["(2124, 125465)\n","(665, 125465)\n","(532, 125465)\n"],"name":"stdout"}]},{"metadata":{"id":"JgdpV4MabZBj","colab_type":"code","colab":{}},"cell_type":"code","source":["# combining Featuers GENE VARAITION and TEXT\n","\n","train_gene_var_onehotCoding = hstack((train_gene_onehotCoding,train_variation_onehotCoding))\n","test_gene_var_onehotCoding = hstack((test_gene_onehotCoding,test_variation_onehotCoding))\n","cv_gene_var_onehotCoding = hstack((cv_gene_onehotCoding,cv_variation_onehotCoding))\n","\n","train_x_onehotCoding = hstack((train_gene_var_onehotCoding, train_text_onehotCoding)).tocsr()\n","train_y = np.array(list(x_train['Class']))\n","\n","test_x_onehotCoding = hstack((test_gene_var_onehotCoding, test_text_onehotCoding)).tocsr()\n","test_y = np.array(list(x_test['Class']))\n","\n","cv_x_onehotCoding = hstack((cv_gene_var_onehotCoding, cv_text_onehotCoding)).tocsr()\n","cv_y = np.array(list(x_cv['Class']))"],"execution_count":0,"outputs":[]},{"metadata":{"id":"bW9hr8jdvNtg","colab_type":"code","outputId":"0616dd6b-3c02-4128-8ce0-b9fb94dc18a0","executionInfo":{"status":"ok","timestamp":1543824865053,"user_tz":-330,"elapsed":614,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":69}},"cell_type":"code","source":["print(train_x_onehotCoding.shape)\n","print(cv_x_onehotCoding.shape)\n","print(test_x_onehotCoding.shape)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["(2124, 127657)\n","(532, 127657)\n","(665, 127657)\n"],"name":"stdout"}]},{"metadata":{"id":"9gze08SBd0Pb","colab_type":"text"},"cell_type":"markdown","source":["##Applying Machine Learning Model( Classifiers) on the Data to predict labels"]},{"metadata":{"id":"ME_6A2cNfi8q","colab_type":"code","colab":{}},"cell_type":"code","source":["def predict_and_plot_confusion_matrix(train_x, train_y,test_x, test_y, clf):\n","    clf.fit(train_x, train_y)\n","    \n","    sig_clf = CalibratedClassifierCV(clf, method=\"sigmoid\")\n","    sig_clf.fit(train_x, train_y)\n","    pred_y = sig_clf.predict(test_x)\n","\n","    # for calculating log_loss we will provide the array of probabilities belongs to each class\n","    print(\"Test Log loss :\",log_loss(test_y, sig_clf.predict_proba(test_x)))\n","    print(confusion_matrix(test_y, pred_y))\n","    print(\"Score :\",sig_clf.score(test_x, test_y))\n","    \n","    print(\"Train Log loss :\",log_loss(train_y, sig_clf.predict_proba(train_x_onehotCoding)))\n","    print(\"CV loss :\",log_loss(cv_y, sig_clf.predict_proba(cv_x_onehotCoding)))\n","    \n","    \n","    print(\"Actual :\", test_y[0])\n","    print(\"Probability :\",np.round(sig_clf.predict_proba(test_x[0]),4))\n","    print(\"Predicted :\", sig_clf.predict(test_x[0]))\n","    # calculating the number of data points that are misclassified\n","    # print(\"Number of mis-classified points :\", np.count_nonzero((pred_y- test_y))/test_y.shape[0], 1-np.count_nonzero((pred_y- test_y))/test_y.shape[0])\n","    \n","    "],"execution_count":0,"outputs":[]},{"metadata":{"id":"_lTKfg_Yd_9Z","colab_type":"text"},"cell_type":"markdown","source":["##Logistic Regression SGDClassifier "]},{"metadata":{"id":"2kCcVHsJgPUH","colab_type":"code","outputId":"9d3beb8d-30d6-4441-aea0-c94ac03f5f89","executionInfo":{"status":"ok","timestamp":1543825947649,"user_tz":-330,"elapsed":7098,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":295}},"cell_type":"code","source":["clf = SGDClassifier(class_weight='balanced', alpha=0.001, penalty='l2', loss='log', random_state=32 ,max_iter=1000, tol=0.0001)\n","predict_and_plot_confusion_matrix(train_x_onehotCoding, train_y, test_x_onehotCoding, test_y, clf)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Test Log loss : 1.1309001736390696\n","[[ 62   1   3  23   9   4  11   1   0]\n"," [  3  36   0   1   1   1  49   0   0]\n"," [  2   0   2   7   0   0   7   0   0]\n"," [ 19   0   1  97   5   1  11   2   1]\n"," [  9   1   2   7  21   1   7   0   0]\n"," [  7   5   0   4   6  23  10   0   0]\n"," [  0  14   0   1   5   1 169   1   0]\n"," [  0   2   0   0   0   0   2   0   0]\n"," [  1   0   0   0   0   0   2   0   4]]\n","Score : 0.6225563909774436\n","Train Log loss : 0.638062593866239\n","CV loss : 1.0950819534818401\n","Actual : 5\n","Probability : [[0.061  0.0404 0.0508 0.1754 0.5514 0.0839 0.0232 0.0079 0.0061]]\n","Predicted : [5]\n"],"name":"stdout"}]},{"metadata":{"id":"-b4r_screKvY","colab_type":"text"},"cell_type":"markdown","source":["##Naive Bayes"]},{"metadata":{"id":"m4TNmU7UeTBC","colab_type":"code","outputId":"35312f11-84b2-45d8-85d1-e919161e7241","executionInfo":{"status":"ok","timestamp":1543826072870,"user_tz":-330,"elapsed":2514,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":295}},"cell_type":"code","source":["from sklearn.naive_bayes import MultinomialNB\n","clf = MultinomialNB()\n","predict_and_plot_confusion_matrix(train_x_onehotCoding, train_y, test_x_onehotCoding, test_y, clf)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Test Log loss : 1.3373565893711143\n","[[ 53   1   0  27   1   6  26   0   0]\n"," [  3  23   0   2   0   0  63   0   0]\n"," [  0   0   1   5   0   2  10   0   0]\n"," [ 21   1   0  94   3   2  16   0   0]\n"," [ 11   0   0  10  15   1  11   0   0]\n"," [  4   4   0   7   2  28  10   0   0]\n"," [  0   8   0   1   3   1 178   0   0]\n"," [  1   0   0   1   0   0   2   0   0]\n"," [  0   0   0   0   0   0   2   0   5]]\n","Score : 0.5969924812030075\n","Train Log loss : 0.9709757614748049\n","CV loss : 1.3301023783534078\n","Actual : 5\n","Probability : [[0.1321 0.1266 0.0342 0.1129 0.4108 0.0492 0.1208 0.0082 0.0052]]\n","Predicted : [5]\n"],"name":"stdout"}]},{"metadata":{"id":"lymafRaUeRWC","colab_type":"text"},"cell_type":"markdown","source":["##Random Forest"]},{"metadata":{"id":"gWNUc4hiizr1","colab_type":"code","outputId":"172d278a-e33f-49e1-95f4-727fa3126333","executionInfo":{"status":"ok","timestamp":1543826151265,"user_tz":-330,"elapsed":52110,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":295}},"cell_type":"code","source":["from sklearn.ensemble import RandomForestClassifier\n","clf = RandomForestClassifier()\n","predict_and_plot_confusion_matrix(train_x_onehotCoding, train_y, test_x_onehotCoding, test_y, clf)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Test Log loss : 1.3439806511882268\n","[[ 57   3   0  33   1   3  17   0   0]\n"," [  4  29   0   6   0   0  52   0   0]\n"," [  0   0   0   7   0   0  11   0   0]\n"," [ 23   1   0 101   0   1  11   0   0]\n"," [ 10   2   0  24   1   1  10   0   0]\n"," [  9   7   0   3  12  12  12   0   0]\n"," [  2  10   0   4   0   0 175   0   0]\n"," [  1   0   0   1   0   0   2   0   0]\n"," [  1   0   0   1   0   0   4   0   1]]\n","Score : 0.5654135338345865\n","Train Log loss : 0.628238850284262\n","CV loss : 1.3364707372360793\n","Actual : 5\n","Probability : [[0.1436 0.0767 0.0459 0.2898 0.2213 0.1271 0.0794 0.008  0.0082]]\n","Predicted : [4]\n"],"name":"stdout"}]},{"metadata":{"id":"baDjIVTHeV9Z","colab_type":"text"},"cell_type":"markdown","source":["##AdaBoost Classifier"]},{"metadata":{"id":"PjCKVkl9URHO","colab_type":"code","outputId":"b4e950d4-1e2a-4224-c01e-e26fa75452b6","executionInfo":{"status":"ok","timestamp":1540439070499,"user_tz":-330,"elapsed":60598,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAADFE/ld4IJCnnhv8/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":52}},"cell_type":"code","source":["from sklearn.ensemble import AdaBoostClassifier\n","clf = AdaBoostClassifier()\n","predict_and_plot_confusion_matrix(train_x_onehotCoding, train_y, cv_x_onehotCoding, cv_y, clf)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Log loss : 1.7432862058505991\n","Number of mis-classified points : 0.6221804511278195\n"],"name":"stdout"}]},{"metadata":{"id":"06mg3eWgfCtu","colab_type":"text"},"cell_type":"markdown","source":["Very bad accuracy because data is not suitable for the classifers "]},{"metadata":{"id":"E6fjEjaOfc6H","colab_type":"text"},"cell_type":"markdown","source":["##SVM"]},{"metadata":{"id":"CbUSnqKYVqrS","colab_type":"code","outputId":"21556c67-a3bb-4f38-9c11-ae8777358259","executionInfo":{"status":"ok","timestamp":1543826989262,"user_tz":-330,"elapsed":736434,"user":{"displayName":"Atul Rana","photoUrl":"https://lh4.googleusercontent.com/-tLVUUtH-AZE/AAAAAAAAAAI/AAAAAAAAMt0/14Rk9-FZ5I4/s64/photo.jpg","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":312}},"cell_type":"code","source":["from sklearn.svm import SVC\n","clf =  SVC(kernel=\"linear\", C=1)\n","predict_and_plot_confusion_matrix(train_x_onehotCoding, train_y, test_x_onehotCoding, test_y, clf)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Test Log loss : 1.0954353202071974\n","[[ 59   2   0  34   6   2  11   0   0]\n"," [  5  45   0   1   0   1  39   0   0]\n"," [  0   0   1   9   0   0   8   0   0]\n"," [ 11   1   1 111   3   1   9   0   0]\n"," [ 12   1   1   9  14   2   8   0   1]\n"," [  4   5   0   4   1  31  10   0   0]\n"," [  1  18   0   4   4   1 161   0   2]\n"," [  0   1   0   1   0   0   2   0   0]\n"," [  0   0   0   0   0   0   2   0   5]]\n","Score : 0.6421052631578947\n","Train Log loss : 0.7192675642326717\n","CV loss : 1.0394969723930823\n","Actual : 5\n","Probability : [[3.580e-02 2.700e-03 6.330e-02 1.457e-01 3.280e-01 2.545e-01 2.000e-04\n","  1.658e-01 4.100e-03]]\n","Predicted : [5]\n"],"name":"stdout"}]},{"metadata":{"id":"Pt4WOPVDfyX2","colab_type":"text"},"cell_type":"markdown","source":["\n","#Working on other Encoding Incomplete"]},{"metadata":{"id":"NfT13MPAmhXs","colab_type":"code","colab":{}},"cell_type":"code","source":["def evaluate_features(X, y, clf=None):\n","    if clf is None:\n","        clf = LogisticRegression()\n","    \n","    probas = cross_val_predict(clf, X, y, cv=StratifiedKFold(random_state=8), \n","                              n_jobs=-1, method='predict_proba', verbose=2)\n","    pred_indices = np.argmax(probas, axis=1)\n","    classes = np.unique(y)\n","    preds = classes[pred_indices]\n","    print('Log loss: {}'.format(log_loss(y, probas)))\n","    print('Accuracy: {}'.format(accuracy_score(y, preds)))\n","    print(\"Number of mis-classified points :\", np.count_nonzero((preds- y))/y.shape[0])\n","    print(confusion_matrix(y, preds))"],"execution_count":0,"outputs":[]},{"metadata":{"id":"VmMDkLFhg2tR","colab_type":"code","colab":{}},"cell_type":"code","source":["svd = TruncatedSVD(n_components=25, n_iter=25, random_state=12)\n","truncated_bag_of_words = svd.fit_transform(train_x_onehotCoding)"],"execution_count":0,"outputs":[]},{"metadata":{"id":"OtEu8HU5k4nM","colab_type":"code","outputId":"cc193cb2-55bc-43b9-beab-6e802b32f2f4","executionInfo":{"status":"ok","timestamp":1538560213547,"user_tz":-330,"elapsed":2176,"user":{"displayName":"Atul Rana","photoUrl":"","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":243}},"cell_type":"code","source":["evaluate_features(truncated_bag_of_words, train_y)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Log loss: 1.4344208845031459\n","Accuracy: 0.4651600753295669\n","Number of mis-classified points : 0.5348399246704332\n","[[114   0   0  77   5   8 159   0   0]\n"," [ 15   2   0   2   0   0 270   0   0]\n"," [  4   0   0   6   6   0  41   0   0]\n"," [ 67   0   0 186  17   2 167   0   0]\n"," [  7   0   0  51  28   9  60   0   0]\n"," [ 15   0   0  44   6  52  59   0   0]\n"," [  6   8   0   4   0   0 591   0   0]\n"," [  2   0   0   0   0   0   7   3   0]\n"," [  3   0   0   0   0   0   9   0  12]]\n"],"name":"stdout"},{"output_type":"stream","text":["[Parallel(n_jobs=-1)]: Done   3 out of   3 | elapsed:    0.2s finished\n"],"name":"stderr"}]},{"metadata":{"id":"9530dFa0nbXH","colab_type":"code","colab":{}},"cell_type":"code","source":["count_vectorizer = TfidfVectorizer()    \n","train_tfidf = count_vectorizer.fit_transform(x_train['TEXT'])\n","train_tfidf = normalize(train_tfidf, axis=0)\n","\n","test_tfidf = count_vectorizer.fit_transform(x_test['TEXT'])\n","test_tfidf = normalize(test_tfidf, axis=0)\n","\n","cv_tfidf = count_vectorizer.fit_transform(x_cv['TEXT'])\n","cv_tfidf = normalize(cv_tfidf, axis=0)"],"execution_count":0,"outputs":[]},{"metadata":{"id":"TSOnIOMVsX_V","colab_type":"code","colab":{}},"cell_type":"code","source":["train_x_tfidf = hstack((train_gene_var_onehotCoding, train_tfidf)).tocsr()\n","test_x_tfidf = hstack((test_gene_var_onehotCoding, test_tfidf)).tocsr()\n","cv_x_tfidf = hstack((cv_gene_var_onehotCoding, cv_tfidf)).tocsr()"],"execution_count":0,"outputs":[]},{"metadata":{"id":"45fgJhHju8ry","colab_type":"code","outputId":"67d94f6e-9800-426a-f2f9-1ce2be58273b","executionInfo":{"status":"ok","timestamp":1538560240216,"user_tz":-330,"elapsed":1370,"user":{"displayName":"Atul Rana","photoUrl":"","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":52}},"cell_type":"code","source":["print(train_x_tfidf.shape)\n","print(cv_x_tfidf.shape)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["(2124, 127921)\n","(532, 70385)\n"],"name":"stdout"}]},{"metadata":{"id":"r6t0ngAyuWGf","colab_type":"code","colab":{}},"cell_type":"code","source":["count_vectorizer = TfidfVectorizer()\n","tfidf = count_vectorizer.fit_transform(result['TEXT'])"],"execution_count":0,"outputs":[]},{"metadata":{"id":"YR4fk9aKv-Vz","colab_type":"code","colab":{}},"cell_type":"code","source":["svd = TruncatedSVD(n_components=25, n_iter=25, random_state=12)\n","truncated_tfidf = svd.fit_transform(tfidf)"],"execution_count":0,"outputs":[]},{"metadata":{"id":"BdZ7uqtIwYeN","colab_type":"code","outputId":"0d1f1f97-f30c-4094-c4be-516b10524188","executionInfo":{"status":"ok","timestamp":1538560296205,"user_tz":-330,"elapsed":1683,"user":{"displayName":"Atul Rana","photoUrl":"","userId":"06185900944594624481"}},"colab":{"base_uri":"https://localhost:8080/","height":243}},"cell_type":"code","source":["evaluate_features(truncated_tfidf, result['Class'].values.ravel())"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Log loss: 1.7543998955064897\n","Accuracy: 0.34628124059018367\n","Number of mis-classified points : 0.6537187594098163\n","[[181   5  19  90  75  15 183   0   0]\n"," [ 20  84   0  21   8   0 319   0   0]\n"," [ 12  14   0   4  10   7  42   0   0]\n"," [200   2  15 100  47  18 304   0   0]\n"," [ 38   6   0  36  67  19  76   0   0]\n"," [ 44   6   0  56   6  85  78   0   0]\n"," [ 20 193   0  81  26   0 633   0   0]\n"," [  1   0   0   5   0   0  13   0   0]\n"," [  5   0   0   6   0   0  26   0   0]]\n"],"name":"stdout"},{"output_type":"stream","text":["[Parallel(n_jobs=-1)]: Done   3 out of   3 | elapsed:    0.2s finished\n"],"name":"stderr"}]}]}